--- src/voms-admin.py.orig	2011-07-23 08:19:07.000000000 -0500
+++ src/voms-admin.py	2011-07-23 08:19:28.000000000 -0500
@@ -24,6 +24,8 @@
 
 voms_admin_default_encoding = 'utf-8'
 
+os.environ.setdefault("VOMS_LOCATION", "/usr")
+
 class EncodedStreamWriter:
     def __init__(self, writer):
         self.writer = writer
@@ -320,4 +322,4 @@
 if __name__ == '__main__':
     main()
 
-    
\ No newline at end of file
+    
