--- resources/scripts/init-voms-admin.py.orig	2011-07-21 13:45:25.000000000 -0500
+++ resources/scripts/init-voms-admin.py	2011-07-21 13:47:12.000000000 -0500
@@ -25,6 +25,9 @@
 
 import getopt, sys, os, os.path, commands,urllib,xml.dom.minidom, getopt, shutil, time, re, glob
 
+os.environ.setdefault("CATALINA_HOME", "/usr/share/tomcat5")
+os.environ.setdefault("VOMS_ADMIN_LOCATION_VAR", "/")
+
 def voms_admin_config_dir_prefix():
     conf_dir_prefix = os.environ.get('VOMS_ADMIN_LOCATION_VAR')
     if conf_dir_prefix is None:
--- resources/scripts/voms.py.orig	2011-07-25 16:44:44.000000000 -0500
+++ resources/scripts/voms.py	2011-07-25 16:46:03.000000000 -0500
@@ -20,6 +20,13 @@
 
 import exceptions,re,os,os.path,commands,glob,socket,string,shutil,time,popen2
 
+os.environ.setdefault("VOMS_ADMIN_LOCATION_VAR", "/")
+os.environ.setdefault("VOMS_LOCATION_VAR", "/var/lib/voms")
+os.environ.setdefault("VOMS_LOCATION", "/usr")
+os.environ.setdefault("VOMS_ADMIN_LOCATION", "/usr/share/voms-admin")
+os.environ.setdefault("VOMS_LOCATION_CONF", "/")
+
+
 voms_admin_server_version = "${server-version}"
 
 def exit_status(status):
         
 
