--- src/srm.v2.2.h	2010-09-09 11:03:53.000000000 -0500
+++ src/srm.v2.2.h	2010-09-09 15:41:51.000000000 -0500
@@ -711,7 +711,7 @@
 /// Size of array of struct srm2__TSURLReturnStatus* is 1..unbounded
     int                                  __sizestatusArray             ;
 /// Array of length 1..unbounded
-    struct srm2__TSURLReturnStatus*      statusArray                   ;	///< Nullable pointer.
+    struct srm2__TSURLReturnStatus**      statusArray                   ;	///< Nullable pointer.
 };
 
 /// "http://srm.lbl.gov/StorageResourceManager":TSURLLifetimeReturnStatus is a complexType.
@@ -734,7 +733,7 @@
 /// Size of array of struct srm2__TSURLLifetimeReturnStatus* is 1..unbounded
     int                                  __sizestatusArray             ;
 /// Array of length 1..unbounded
-    struct srm2__TSURLLifetimeReturnStatus*  statusArray                   ;	///< Nullable pointer.
+    struct srm2__TSURLLifetimeReturnStatus**  statusArray                   ;	///< Nullable pointer.
 };
 
 /// "http://srm.lbl.gov/StorageResourceManager":TMetaDataPathDetail is a complexType.
@@ -784,7 +783,7 @@
 /// Size of array of struct srm2__TMetaDataPathDetail* is 1..unbounded
     int                                  __sizepathDetailArray         ;
 /// Array of length 1..unbounded
-    struct srm2__TMetaDataPathDetail*    pathDetailArray               ;	///< Nullable pointer.
+    struct srm2__TMetaDataPathDetail**    pathDetailArray               ;	///< Nullable pointer.
 };
 
 /// "http://srm.lbl.gov/StorageResourceManager":TMetaDataSpace is a complexType.
@@ -816,7 +815,7 @@
 /// Size of array of struct srm2__TMetaDataSpace* is 1..unbounded
     int                                  __sizespaceDataArray          ;
 /// Array of length 1..unbounded
-    struct srm2__TMetaDataSpace*         spaceDataArray                ;	///< Nullable pointer.
+    struct srm2__TMetaDataSpace**         spaceDataArray                ;	///< Nullable pointer.
 };
 
 /// "http://srm.lbl.gov/StorageResourceManager":TDirOption is a complexType.
@@ -885,7 +884,7 @@
 /// Size of array of struct srm2__TGetFileRequest* is 1..unbounded
     int                                  __sizerequestArray            ;
 /// Array of length 1..unbounded
-    struct srm2__TGetFileRequest*        requestArray                  ;	///< Nullable pointer.
+    struct srm2__TGetFileRequest**        requestArray                  ;	///< Nullable pointer.
 };
 
 /// "http://srm.lbl.gov/StorageResourceManager":TPutFileRequest is a complexType.
@@ -903,7 +902,7 @@
 /// Size of array of struct srm2__TPutFileRequest* is 1..unbounded
     int                                  __sizerequestArray            ;
 /// Array of length 1..unbounded
-    struct srm2__TPutFileRequest*        requestArray                  ;	///< Nullable pointer.
+    struct srm2__TPutFileRequest**        requestArray                  ;	///< Nullable pointer.
 };
 
 /// "http://srm.lbl.gov/StorageResourceManager":TCopyFileRequest is a complexType.
@@ -923,7 +922,7 @@
 /// Size of array of struct srm2__TCopyFileRequest* is 1..unbounded
     int                                  __sizerequestArray            ;
 /// Array of length 1..unbounded
-    struct srm2__TCopyFileRequest*       requestArray                  ;	///< Nullable pointer.
+    struct srm2__TCopyFileRequest**       requestArray                  ;	///< Nullable pointer.
 };
 
 /// "http://srm.lbl.gov/StorageResourceManager":TGetRequestFileStatus is a complexType.
@@ -951,7 +950,7 @@
 /// Size of array of struct srm2__TGetRequestFileStatus* is 1..unbounded
     int                                  __sizestatusArray             ;
 /// Array of length 1..unbounded
-    struct srm2__TGetRequestFileStatus*  statusArray                   ;	///< Nullable pointer.
+    struct srm2__TGetRequestFileStatus**  statusArray                   ;	///< Nullable pointer.
 };
 
 /// "http://srm.lbl.gov/StorageResourceManager":TBringOnlineRequestFileStatus is a complexType.
@@ -975,7 +974,7 @@
 /// Size of array of struct srm2__TBringOnlineRequestFileStatus* is 1..unbounded
     int                                  __sizestatusArray             ;
 /// Array of length 1..unbounded
-    struct srm2__TBringOnlineRequestFileStatus*  statusArray                   ;	///< Nullable pointer.
+    struct srm2__TBringOnlineRequestFileStatus**  statusArray                   ;	///< Nullable pointer.
 };
 
 /// "http://srm.lbl.gov/StorageResourceManager":TPutRequestFileStatus is a complexType.
@@ -1005,7 +1004,7 @@
 /// Size of array of struct srm2__TPutRequestFileStatus* is 1..unbounded
     int                                  __sizestatusArray             ;
 /// Array of length 1..unbounded
-    struct srm2__TPutRequestFileStatus*  statusArray                    1;
+    struct srm2__TPutRequestFileStatus**  statusArray                    1;
 };
 
 /// "http://srm.lbl.gov/StorageResourceManager":TCopyRequestFileStatus is a complexType.
@@ -1031,7 +1030,7 @@
 /// Size of array of struct srm2__TCopyRequestFileStatus* is 1..unbounded
     int                                  __sizestatusArray             ;
 /// Array of length 1..unbounded
-    struct srm2__TCopyRequestFileStatus*  statusArray                    1;
+    struct srm2__TCopyRequestFileStatus**  statusArray                    1;
 };
 
 /// "http://srm.lbl.gov/StorageResourceManager":TRequestSummary is a complexType.
@@ -1079,7 +1078,7 @@
 /// Size of array of struct srm2__TSURLPermissionReturn* is 1..unbounded
     int                                  __sizesurlPermissionArray     ;
 /// Array of length 1..unbounded
-    struct srm2__TSURLPermissionReturn*  surlPermissionArray            1;
+    struct srm2__TSURLPermissionReturn**  surlPermissionArray            1;
 };
 
 /// "http://srm.lbl.gov/StorageResourceManager":TPermissionReturn is a complexType.
