diff -u -r1.2 test_33641.c
--- src/tools/test_33641.c	2 Feb 2010 17:24:40 -0000	1.2
+++ src/tools/test_33641.c	5 Jul 2011 18:05:03 -0000
@@ -167,8 +167,8 @@
             //_fault_to_error(ctx, __func__);
             return -1;
         }
-        A_certreq = A_req_resp.delegation__NewProxyReq->proxyRequest;
-        A_sdelegationID = A_req_resp.delegation__NewProxyReq->delegationID;
+        A_certreq = A_req_resp.getNewProxyReqReturn->proxyRequest;
+        A_sdelegationID = A_req_resp.getNewProxyReqReturn->delegationID;
     }
 
     /* generating a certificate from the request */
@@ -219,8 +219,8 @@
             //_fault_to_error(ctx, __func__);
             return -1;
         }
-        B_certreq = B_req_resp.delegation__NewProxyReq->proxyRequest;
-        B_sdelegationID = B_req_resp.delegation__NewProxyReq->delegationID;
+        B_certreq = B_req_resp.getNewProxyReqReturn->proxyRequest;
+        B_sdelegationID = B_req_resp.getNewProxyReqReturn->delegationID;
     }
 
     /* generating a certificate from the request */
