--- trunk/glexec/src/main_util.c	2013/11/29 10:03:03	17182
+++ trunk/glexec/src/main_util.c	2014/01/14 11:24:35	17277
@@ -527,23 +527,28 @@
     if ((value = glexec_conf_value("glexec:syslog_facility")))
 	glexec_set_syslog_facility(value);
 
-    /* Find destination and open as soon as possible */
+    /* Find destination */
     if ((value = glexec_conf_value("glexec:log_destination"))) {
-        if ((strcasecmp(value, "file")==0)) {
+        if ((strcasecmp(value, "file")==0))
 	    glexec_set_log_to_file();
-	    glexec_open_log();
-	} else	{ /* In any case use syslog in this case */
+	else	{
 	    glexec_set_log_to_syslog();
-	    glexec_open_log();
+	    /* If value is not understood: log an error to syslog */
 	    if ((strcasecmp(value, "syslog")!=0)) {
+		/* Need to open log at this moment to log the error */
+		glexec_open_log();
 		glexec_log(GLEXEC_LOG_ERROR,
 		    "Value (%s) for key 'glexec:log_destination' "
 		    "not understood. Use either syslog or file\n", value);
 		glexec_exit(GLEXEC_CONF_VALUE_MALFORMED,
 			    GLEXEC_NO_CHILD_EXIT_CODE);
 	    }
-        }
-    }
+	}
+    } else /* Default: syslog */
+	glexec_set_log_to_syslog();
+   
+    /* Now open log (even though we haven't determined the level yet) */
+    glexec_open_log();
 
     /* Set correct log level */
     if ((value = glexec_conf_value("glexec:log_level")))    {
@@ -557,14 +562,14 @@
             glexec_exit (GLEXEC_CONF_VALUE_MALFORMED,GLEXEC_NO_CHILD_EXIT_CODE);
         }
 	if (level<GLEXEC_LOG_LEVEL_LOW)	{
-	    /* truncate a too large loglevel */
+	    /* a too small loglevel */
 	    glexec_log(GLEXEC_LOG_WARNING,
 		"Specified loglevel %ld is too small, "
 		"using min. log level %u.\n",
 		level,GLEXEC_LOG_LEVEL_LOW);
 	    level=GLEXEC_LOG_LEVEL_LOW;
 	} else if (level>=GLEXEC_LOG_LEVEL_HIGH)	{
-	    /* truncate a too large loglevel */
+	    /* a too large loglevel */
 	    glexec_log(GLEXEC_LOG_WARNING,
 		"Specified loglevel %ld is too large, "
 		"using max. log level %d.\n",
