diff -ru gridftp-hdfs-0.5.0-orig/conf/gridftp.conf gridftp-hdfs-0.5.0/conf/gridftp.conf
--- gridftp-hdfs-0.5.0-orig/conf/gridftp.conf   2011-09-24 18:59:11.000000000 -0700
+++ gridftp-hdfs-0.5.0/conf/gridftp.conf    2011-09-30 13:16:19.000000000 -0700
@@ -1,6 +1,8 @@
 daemon 1
 detach 1
-log_level ERROR,WARN,INFO,ALL
-log_module stdio
+log_level ERROR,WARN,INFO
+log_single /var/log/gridftp-auth.log
+log_transfer /var/log/gridftp.log
 port 2811
 blocksize 1048576
+load_dsi_module hdfs
diff -ru gridftp-hdfs-0.5.0-orig/conf/gridftp-hdfs-environment gridftp-hdfs-0.5.0/conf/gridftp-hdfs-environment
--- gridftp-hdfs-0.5.0-orig/conf/gridftp-hdfs-environment   2011-09-24 18:41:05.000000000 -0700
+++ gridftp-hdfs-0.5.0/conf/gridftp-hdfs-environment    2011-09-30 13:13:42.000000000 -0700
@@ -1,4 +1,8 @@
 
+#Use hadoop configuration
+export conf=/etc/gridftp-hdfs/gridftp.conf
+
+
 # This file is designed to be sourced in bash to configure the 
 # basic Hadoop environment.
 
@@ -67,22 +71,4 @@
 
 unset IFS
 
-# Configurations for the authz
-# Only applies to the root user due to ownership issues
-if [ $UID -eq 0 ]; then
-  # LCAS configuration
-  export LCAS_LOG_LEVEL=0
-  export LCAS_DEBUG_LEVEL=0
-  export LCAS_DB_FILE=/etc/lcas/lcas.db
-  export LCAS_DIR=/etc/lcas
-  export LCAS_MOD_HOME=/usr/lib64/modules
- 
-  # LCMAPS configuration
-  export LCMAPS_LOG_LEVEL=0
-  export LCMAPS_DEBUG_LEVEL=0
-  export LCMAPS_DB_FILE=/etc/lcmaps.db
-  export LCMAPS_DIR=/etc/lcmaps
-  export LCMAPS_MOD_HOME=/usr/lib64/modules
-  export LCMAPS_POLICY_NAME=globus_gridftp_mapping
-fi
-
+export LD_PRELOAD=libjsig.so
diff -ru gridftp-hdfs-0.5.0-orig/conf/Makefile.in gridftp-hdfs-0.5.0/conf/Makefile.in 
--- gridftp-hdfs-0.5.0-orig/conf/Makefile.in    2011-09-24 19:16:33.000000000 -0700
+++ gridftp-hdfs-0.5.0/conf/Makefile.in 2011-10-12 12:45:03.000000000 -0700
@@ -180,11 +180,11 @@
 target_alias = @target_alias@
 gridftpdir = $(sysconfdir)/gridftp-hdfs
 dist_gridftp_DATA = gridftp.conf gridftp-inetd.conf gridftp-debug.conf replica-map.conf
-sysconfigdir = $(sysconfdir)/sysconfig
+sysconfigdir = $(sysconfdir)/sysconfig/gridftp.conf.d
 dist_sysconfig_DATA = gridftp-hdfs
 logrotatedir = $(sysconfdir)/logrotate.d
 dist_logrotate_DATA = gridftp-hdfs.logrotate
-envdir = $(datadir)/gridftp-hdfs
+envdir = $(sysconfdir)/sysconfig/gridftp.conf.d
 dist_env_DATA = gridftp-hdfs-environment
 all: all-am

diff -ru gridftp-hdfs-0.5.0-orig/scripts/xinetd/gridftp-hdfs-inetd.in gridftp-hdfs-0.5.0/scripts/xinetd/gridftp-hdfs-inetd.in 
--- gridftp-hdfs-0.5.0-orig/scripts/xinetd/gridftp-hdfs-inetd.in    2011-09-24 14:53:08.000000000 -0700
+++ gridftp-hdfs-0.5.0/scripts/xinetd/gridftp-hdfs-inetd.in 2011-10-12 13:30:20.000000000 -0700
@@ -1,16 +1,6 @@
 #!/bin/sh
 
-if [ -e @datadir_resolved@/gridftp-hdfs/gridftp-hdfs-environment ] ; then
-    . @datadir_resolved@/gridftp-hdfs/gridftp-hdfs-environment
-fi
+[ -e @sysconfdir_resolved@/sysconfig/globus-gridftp-server ] && . @sysconfdir_resolved@/sysconfig/globus-gridftp-server
 
-if [ -e @sysconfdir_resolved@/gridftp-hdfs/gridftp-hdfs-local.conf ] ; then
-    . @sysconfdir_resolved@/gridftp-hdfs/gridftp-hdfs-local.conf
-fi
-
-if [ -e @sysconfdir_resolved@/sysconfig/gridftp-hdfs ] ; then
-    . @sysconfdir_resolved@/sysconfig/gridftp-hdfs
-fi
-
-LD_PRELOAD=libjsig.so exec /usr/sbin/globus-gridftp-server -c @sysconfdir_resolved@/gridftp-hdfs/gridftp-inetd.conf -dsi hdfs
+exec /usr/sbin/globus-gridftp-server -c @sysconfdir_resolved@/gridftp-hdfs/gridftp-inetd.conf -dsi hdfs
 
diff -ru gridftp-hdfs-0.5.0-orig/scripts/gridftp-hdfs-standalone.in gridftp-hdfs-0.5.0/scripts/gridftp-hdfs-standalone.in 
--- gridftp-hdfs-0.5.0-orig/scripts/gridftp-hdfs-standalone.in  2011-09-24 15:11:49.000000000 -0700
+++ gridftp-hdfs-0.5.0/scripts/gridftp-hdfs-standalone.in   2011-10-12 13:30:38.000000000 -0700
@@ -1,18 +1,8 @@
 #!/bin/sh                                                            
 
-if [ -e @datadir_resolved@/gridftp-hdfs/gridftp-hdfs-environment ] ; then
-    . @datadir_resolved@/gridftp-hdfs/gridftp-hdfs-environment
-fi
-
-if [ -e @sysconfdir_resolved@/gridftp-hdfs/gridftp-hdfs-local.conf ] ; then
-    . @sysconfdir_resolved@/gridftp-hdfs/gridftp-hdfs-local.conf
-fi
-
-if [ -e @sysconfdir_resolved@/sysconfig/gridftp-hdfs ] ; then
-    . @sysconfdir_resolved@/sysconfig/gridftp-hdfs
-fi
+[ -e @sysconfdir_resolved@/sysconfig/globus-gridftp-server ] && . @sysconfdir_resolved@/sysconfig/globus-gridftp-server
 
 export CLASSPATH=$CLASSPATH
                                                                       
-LD_PRELOAD=libjsig.so exec /usr/sbin/globus-gridftp-server -c @sysconfdir_resolved@/gridftp-hdfs/gridftp-debug.conf -dsi hdfs -no-fork
+exec /usr/sbin/globus-gridftp-server -c @sysconfdir_resolved@/gridftp-hdfs/gridftp-debug.conf -dsi hdfs -no-fork

diff -ru gridftp-hdfs-0.5.0-orig/scripts/Makefile.in gridftp-hdfs-0.5.0/scripts/Makefile.in
--- gridftp-hdfs-0.5.0-orig/scripts/Makefile.in	2011-09-24 19:16:33.000000000 -0700
+++ gridftp-hdfs-0.5.0/scripts/Makefile.in	2011-10-12 13:51:15.000000000 -0700
@@ -46,12 +46,11 @@
 	$(ACLOCAL_M4)
 mkinstalldirs = $(install_sh) -d
 CONFIG_CLEAN_FILES = gridftp-hdfs-standalone
-am__installdirs = "$(DESTDIR)$(bindir)" "$(DESTDIR)$(initddir)" \
-	"$(DESTDIR)$(sbindir)" "$(DESTDIR)$(xinetddir)"
+am__installdirs = "$(DESTDIR)$(bindir)" "$(DESTDIR)$(sbindir)" \
+	"$(DESTDIR)$(xinetddir)"
 binSCRIPT_INSTALL = $(INSTALL_SCRIPT)
-initdSCRIPT_INSTALL = $(INSTALL_SCRIPT)
 sbinSCRIPT_INSTALL = $(INSTALL_SCRIPT)
-SCRIPTS = $(bin_SCRIPTS) $(initd_SCRIPTS) $(sbin_SCRIPTS)
+SCRIPTS = $(bin_SCRIPTS) $(sbin_SCRIPTS)
 SOURCES =
 DIST_SOURCES =
 am__vpath_adj_setup = srcdirstrip=`echo "$(srcdir)" | sed 's|.|.|g'`;
@@ -179,8 +178,6 @@
 target_alias = @target_alias@
 bin_SCRIPTS = gridftp-hdfs-standalone
 sbin_SCRIPTS = xinetd/gridftp-hdfs-inetd
-initddir = $(sysconfdir)/init.d
-initd_SCRIPTS = init/gridftp-hdfs
 xinetddir = $(sysconfdir)/xinetd.d
 dist_xinetd_DATA = xinetd/gridftp-hdfs
 all: all-am
@@ -236,25 +233,6 @@
 	  echo " rm -f '$(DESTDIR)$(bindir)/$$f'"; \
 	  rm -f "$(DESTDIR)$(bindir)/$$f"; \
 	done
-install-initdSCRIPTS: $(initd_SCRIPTS)
-	@$(NORMAL_INSTALL)
-	test -z "$(initddir)" || $(mkdir_p) "$(DESTDIR)$(initddir)"
-	@list='$(initd_SCRIPTS)'; for p in $$list; do \
-	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
-	  if test -f $$d$$p; then \
-	    f=`echo "$$p" | sed 's|^.*/||;$(transform)'`; \
-	    echo " $(initdSCRIPT_INSTALL) '$$d$$p' '$(DESTDIR)$(initddir)/$$f'"; \
-	    $(initdSCRIPT_INSTALL) "$$d$$p" "$(DESTDIR)$(initddir)/$$f"; \
-	  else :; fi; \
-	done
-
-uninstall-initdSCRIPTS:
-	@$(NORMAL_UNINSTALL)
-	@list='$(initd_SCRIPTS)'; for p in $$list; do \
-	  f=`echo "$$p" | sed 's|^.*/||;$(transform)'`; \
-	  echo " rm -f '$(DESTDIR)$(initddir)/$$f'"; \
-	  rm -f "$(DESTDIR)$(initddir)/$$f"; \
-	done
 install-sbinSCRIPTS: $(sbin_SCRIPTS)
 	@$(NORMAL_INSTALL)
 	test -z "$(sbindir)" || $(mkdir_p) "$(DESTDIR)$(sbindir)"
@@ -340,7 +318,7 @@
 check: check-am
 all-am: Makefile $(SCRIPTS) $(DATA)
 installdirs:
-	for dir in "$(DESTDIR)$(bindir)" "$(DESTDIR)$(initddir)" "$(DESTDIR)$(sbindir)" "$(DESTDIR)$(xinetddir)"; do \
+	for dir in "$(DESTDIR)$(bindir)" "$(DESTDIR)$(sbindir)" "$(DESTDIR)$(xinetddir)"; do \
 	  test -z "$$dir" || $(mkdir_p) "$$dir"; \
 	done
 install: install-am
@@ -385,7 +363,7 @@
 
 info-am:
 
-install-data-am: install-dist_xinetdDATA install-initdSCRIPTS
+install-data-am: install-dist_xinetdDATA
 
 install-exec-am: install-binSCRIPTS install-sbinSCRIPTS
 
@@ -412,20 +390,20 @@
 ps-am:
 
 uninstall-am: uninstall-binSCRIPTS uninstall-dist_xinetdDATA \
-	uninstall-info-am uninstall-initdSCRIPTS uninstall-sbinSCRIPTS
+	uninstall-info-am uninstall-sbinSCRIPTS
 
 .PHONY: all all-am check check-am clean clean-generic clean-libtool \
 	distclean distclean-generic distclean-libtool distdir dvi \
 	dvi-am html html-am info info-am install install-am \
 	install-binSCRIPTS install-data install-data-am \
 	install-dist_xinetdDATA install-exec install-exec-am \
-	install-info install-info-am install-initdSCRIPTS install-man \
-	install-sbinSCRIPTS install-strip installcheck installcheck-am \
-	installdirs maintainer-clean maintainer-clean-generic \
-	mostlyclean mostlyclean-generic mostlyclean-libtool pdf pdf-am \
-	ps ps-am uninstall uninstall-am uninstall-binSCRIPTS \
+	install-info install-info-am install-man install-sbinSCRIPTS \
+	install-strip installcheck installcheck-am installdirs \
+	maintainer-clean maintainer-clean-generic mostlyclean \
+	mostlyclean-generic mostlyclean-libtool pdf pdf-am ps ps-am \
+	uninstall uninstall-am uninstall-binSCRIPTS \
 	uninstall-dist_xinetdDATA uninstall-info-am \
-	uninstall-initdSCRIPTS uninstall-sbinSCRIPTS
+	uninstall-sbinSCRIPTS
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
