From 4341fe7cdba8e264407bf58c6faff8bfb8ce5b40 Mon Sep 17 00:00:00 2001
From: Derek Weitzel <djw8605@gmail.com>
Date: Wed, 27 Jan 2021 16:18:21 -0600
Subject: [PATCH] Fixing pass by reference

---
 src/multiuser.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/multiuser.cpp b/src/multiuser.cpp
index a56d807..6ff15d3 100644
--- a/src/multiuser.cpp
+++ b/src/multiuser.cpp
@@ -152,7 +152,7 @@ class UserSentry {
         int retval;
         // get the username from the extra attributes in the client
         std::string username;
-        auto got_token = client->eaAPI->Get("scitokens.name", &username);
+        auto got_token = client->eaAPI->Get("scitokens.name", username);
 
         // If we fail to get the username from the scitokens, then get it from
         // the depreciated way, client->name
