commit cea69f241f701d556845fb84b24c83dd5edca56b
Author: Diego Davila <davila.foyo@gmail.com>
Date:   Mon Apr 6 14:54:17 2020 -0500

    compile library as -5 instead of -4

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 43adfe9..8b578a0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,7 +31,7 @@ include_directories(${XROOTD_INCLUDES})
 
 add_library(XrdMultiuser SHARED src/multiuser.cpp)
 target_link_libraries(XrdMultiuser -ldl ${CAP_LIB} ${XROOTD_UTILS_LIB} ${XROOTD_SERVER_LIB})
-set_target_properties(XrdMultiuser PROPERTIES OUTPUT_NAME "XrdMultiuser-4" SUFFIX ".so" LINK_FLAGS "-Wl,--version-script=${CMAKE_SOURCE_DIR}/configs/export-lib-symbols")
+set_target_properties(XrdMultiuser PROPERTIES OUTPUT_NAME "XrdMultiuser-5" SUFFIX ".so" LINK_FLAGS "-Wl,--version-script=${CMAKE_SOURCE_DIR}/configs/export-lib-symbols")
 
 SET(LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib" CACHE PATH "Install path for libraries")
 
