From 5edcf5279b897ed9f97a86a45d12ceaaa7713eaa Mon Sep 17 00:00:00 2001
From: ddavila0 <davila.foyo@gmail.com>
Date: Mon, 26 Oct 2020 17:57:23 -0700
Subject: [PATCH] prevent silent crash on start when ofs.authlib is defined el8

---
 src/multiuser.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/multiuser.cpp b/src/multiuser.cpp
index c4f62d8..e8154a6 100644
--- a/src/multiuser.cpp
+++ b/src/multiuser.cpp
@@ -461,7 +461,7 @@ class MultiuserFileSystem : public XrdSfsFileSystem {
                     return false;
                 }
                 authLib = val;
-                std::vector<char> rest; rest.reserve(2048);
+                std::vector<char> rest(2048);
                 if (!Config.GetRest(&rest[0], 2048)) {
                     m_log.Emsg("Config", "authlib parameters line too long");
                     Config.Close();
