Index: globus_gsi_proxy_core-6.1/library/globus_gsi_proxy_handle_attrs.c
===================================================================
--- globus_gsi_proxy_core-6.1.orig/library/globus_gsi_proxy_handle_attrs.c	2009-02-25 08:42:06.000000000 -0600
+++ globus_gsi_proxy_core-6.1/library/globus_gsi_proxy_handle_attrs.c	2014-01-08 10:21:53.693149995 -0600
@@ -27,7 +27,7 @@
 #include "globus_i_gsi_proxy.h"
 #include <errno.h>
 
-#define DEFAULT_KEY_BITS                512
+#define DEFAULT_KEY_BITS                1024
 #define DEFAULT_PUB_EXPONENT            RSA_F4  /* 65537 */
 #define DEFAULT_SIGNING_ALGORITHM       NULL
 #define DEFAULT_CLOCK_SKEW              (5*60)    /* actually in seconds */
