diff -ru xrootd-hdfs-1.7.0-orig/src/XrdHdfsBootstrap.cc xrootd-hdfs-1.7.0/src/XrdHdfsBootstrap.cc
--- xrootd-hdfs-1.7.0-orig/src/XrdHdfsBootstrap.cc	2012-03-23 16:59:12.000000000 -0500
+++ xrootd-hdfs-1.7.0/src/XrdHdfsBootstrap.cc	2012-03-23 16:59:48.000000000 -0500
@@ -87,7 +87,7 @@
 }
 
 static int CheckEnvVar(const char * var, const char * input) {
-   char * equal_sign = strchr(input, '=');
+   const char * equal_sign = strchr(input, '=');
    if (equal_sign && (strncmp(input, var, equal_sign-input) == 0)) {
       setenv(var, equal_sign+1, 1);
       return 1;
diff -ru xrootd-hdfs-1.7.0-orig/src/XrdHdfs.hh xrootd-hdfs-1.7.0/src/XrdHdfs.hh
--- xrootd-hdfs-1.7.0-orig/src/XrdHdfs.hh	2012-03-23 16:59:12.000000000 -0500
+++ xrootd-hdfs-1.7.0/src/XrdHdfs.hh	2012-03-30 11:24:00.000000000 -0500
@@ -150,6 +150,7 @@
 // Other Functions
 //
         int            Chmod(const char *, mode_t) {return -ENOTSUP;}
+        int            Chmod(const char *, mode_t, XrdOucEnv *) {return -ENOTSUP;}
 
         int            Create(const char *, const char *, mode_t, XrdOucEnv &, int opts=0) {return -ENOTSUP;}
 
@@ -160,18 +161,23 @@
 const   char          *getVersion();
 
         int            Mkdir(const char *, mode_t, int) {return -ENOTSUP;}
+        int            Mkdir(const char *, mode_t, int, XrdOucEnv *) {return -ENOTSUP;}
 
         int            Remdir(const char *) {return -ENOTSUP;}
         int            Remdir(const char *, int) {return -ENOTSUP;}
+        int            Remdir(const char *, int, XrdOucEnv *) {return -ENOTSUP;}
 
         int            Rename(const char *, const char *) {return -ENOTSUP;}
+        int            Rename(const char *, const char *,XrdOucEnv *eP1, XrdOucEnv *) {return -ENOTSUP;}
 
         int            Stat(const char *, struct stat *, int resonly=0, XrdOucEnv* env=0);
 
         int            Truncate(const char *, unsigned long long) {return -ENOTSUP;}
+        int            Truncate(const char *, unsigned long long,XrdOucEnv *) {return -ENOTSUP;}
 
         int            Unlink(const char *) {return -ENOTSUP;}
         int            Unlink(const char *, int) {return -ENOTSUP;}
+        int            Unlink(const char *, int, XrdOucEnv *) {return -ENOTSUP;}
 
 static  int            Emsg(const char *, XrdOucErrInfo&, int, const char *x,
                             const char *y="");
