--- configure.in.orig	2014-03-04 13:19:41.906892067 -0600
+++ configure.in	2014-03-04 13:21:53.233975389 -0600
@@ -44,8 +44,8 @@
         UDT_CFLAGS=""
     fi
    
-    PKG_CHECK_MODULES([NICE], [nice >= 0.1])
-    PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.32 gthread-2.0 gio-2.0 gobject-2.0])
+    PKG_CHECK_MODULES([NICE], [nice >= 0.0.9])
+    PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.26 gthread-2.0 gio-2.0 gobject-2.0])
 fi
 
 AC_SUBST(NICE_LIBS)
--- ice.c.orig	2014-03-04 13:27:57.399208533 -0600
+++ ice.c	2014-03-04 13:30:44.135314952 -0600
@@ -94,7 +94,7 @@
 int ice_init(struct icedata *ice_data,
              const char *stun_host, unsigned int stun_port,
              int controlling) {
-    gboolean ok;
+    //gboolean ok;
 
     if (!lib_initialized)
         return ICE_FAILURE;
@@ -128,7 +128,7 @@
     }
 
     ice_data->agent = nice_agent_new(ice_data->gcontext,
-                                     NICE_COMPATIBILITY_RFC5245);
+                                     NICE_COMPATIBILITY_DRAFT19);
     if (ice_data->agent == NULL) {
         goto error;
     }
@@ -159,10 +159,11 @@
     nice_agent_attach_recv(ice_data->agent, ice_data->stream_id, 1,
                            ice_data->gcontext, cb_nice_recv, ice_data);
 
-    ok = nice_agent_gather_candidates(ice_data->agent, ice_data->stream_id);
+    /*ok = nice_agent_gather_candidates(ice_data->agent, ice_data->stream_id);
     if (!ok) {
         goto error;
-    }
+    }*/
+    nice_agent_gather_candidates(ice_data->agent, ice_data->stream_id);
 
     g_debug("starting event thread");
     ice_data->gloopthread = g_thread_create(&thread_mainloop, ice_data->gloop,
@@ -387,7 +388,7 @@
     }
     if (ice_data->gloopthread) {
         g_thread_join(ice_data->gloopthread);
-        g_thread_unref(ice_data->gloopthread);
+        /*g_thread_unref(ice_data->gloopthread);*/
         ice_data->gloopthread = NULL;
     }
 
