--- src/server.c	25 Jan 2012 14:17:15 -0000	1.150
+++ src/server.c	14 Jun 2012 17:22:49 -0000
@@ -3457,6 +3482,12 @@
 			memcpy(result + j, CONVARG_DBLQUOTESC, CONVARG_DBLQUOTESC_LEN);
 			j += CONVARG_DBLQUOTESC_LEN;
 		}
+		/* Must escape a few meta-characters for wordexp */
+		else if ((original[i] == '(') || (original[i] == ')') || (original[i] == '&'))
+		{
+			result[j++] = '\\';
+			result[j++] = original[i];
+		}
 		else
 		{	/* plain copy from the original */
 			result[j++] = original[i];
