Index: org.glite.ce.blahp/config/glite-ce-blah-parser
===================================================================
--- org.glite.ce.blahp.orig/config/glite-ce-blah-parser	2012-01-26 06:00:58.000000003 -0600
+++ org.glite.ce.blahp/config/glite-ce-blah-parser	2012-07-25 14:27:38.000000003 -0500
@@ -19,7 +19,7 @@
 #
 
 # description: gLite CE blah parser
-# chkconfig: 345 94 15
+# chkconfig: - 94 15
 
 # Source function library
 . /etc/rc.d/init.d/functions
