Index: org.glite.ce.blahp/config/blah.config.template
===================================================================
--- org.glite.ce.blahp.orig/config/blah.config.template	2012-08-30 16:37:04.000000003 -0500
+++ org.glite.ce.blahp/config/blah.config.template	2012-08-30 19:08:49.000000003 -0500
@@ -1,5 +1,5 @@
 #Supported batch systems
-supported_lrms=pbs,lsf,sge
+supported_lrms=pbs,lsf,sge,condor
 
 #DGAS logfile 
 BLAHPD_ACCOUNTING_INFO_LOG=
@@ -8,6 +8,9 @@
 #or delegating proxies to the worker node where a job is running.
 blah_disable_wn_proxy_renewal=no
 
+# Uncomment if you want to use CREAM
+#blah_disable_proxy_user_copy=yes
+
 #Set to yes to enable delegation (instead of copy) of renewed proxies
 #to worker nodes. NOTE: limited *and* delegated proxes are not 
 #accepted for default GSI authentication as of VDT 1.2, so this should
@@ -15,7 +18,7 @@
 blah_delegate_renewed_proxies=no
 
 #Path where PBS executables are located 
-pbs_binpath=`which qsub|sed 's|/[^/]*$||'`
+pbs_binpath=`which qsub >/dev/null 2>&1 && which qsub|sed 's|/[^/]*$||'`
 
 #Path where the PBS logs are located ($pbs_spoolpath/server_logs)
 #pbs_spoolpath=/usr/spool/PBS
@@ -64,7 +67,7 @@
 ####
 
 #Path where LSF executables are located 
-lsf_binpath=`which bsub|sed 's|/[^/]*$||'`
+lsf_binpath=`which bsub >/dev/null 2>&1 && which bsub|sed 's|/[^/]*$||'`
 
 #Path where the LSF conf file is located ($lsf_confpath/lsf.conf)
 lsf_confpath=/etc
@@ -115,10 +118,10 @@
 #
 
 #Updater location
-bupdater_path=
+bupdater_path=/usr/libexec/blahp/BUpdaterCondor
 
 #Notifier location
-bnotifier_path=
+bnotifier_path=/usr/libexec/blahp/BNotifier
 
 #Updater pid file
 bupdater_pidfile=/var/tmp/cream_tomcat_bupdater.pid
@@ -127,19 +130,19 @@
 bnotifier_pidfile=/var/tmp/cream_tomcat_bnotifier.pid
 
 #condor bin location
-condor_binpath=/opt/condor-c/bin
+condor_binpath=/usr/bin
 
 #Registry file location
-job_registry=
+job_registry=/var/tmp/job_registry.db
 #Set the following variable to 'yes' to have multiple BLAHPD instances
 #share the job registry -index- via mmap:
 job_registry_use_mmap=no
 
 #host for asyncronous notification 
-async_notification_host=
+async_notification_host=localhost
 
 #port for asyncronous notification
-async_notification_port=
+async_notification_port=55553
 
 #bupdater debug level
 bupdater_debug_level=1
@@ -154,6 +157,7 @@
 bnotifier_debug_logfile=/var/tmp/bnotifier.log
 
 # purge interval
+# might want to increase this to 5000000 if using Condor with CREAM
 purge_interval=7200
 
 #after that interval a bhist with -n bhist_logs_to_read is tried (default:120)
@@ -163,10 +167,11 @@
 finalstate_query_interval=30
 
 #after that interval an unseen job is set as done (status == 4) and exitstatus == 999 (default:3600)
+# might want to increase this to 7200 if using Condor with CREAM
 alldone_interval=3600
 
 #path to condor_config
-export CONDOR_CONFIG="/opt/condor-c/etc/condor_config"
+export CONDOR_CONFIG="/etc/condor/condor_config"
 
 #max number of concurrent threads to serve commands (default = 500)
 #blah_max_threaded_cmds=100
