--- src/classad_c_helper.C.orig	2012-07-13 13:52:45.000000000 -0500
+++ src/classad_c_helper.C	2012-07-13 13:52:57.000000000 -0500
@@ -289,8 +289,8 @@
     ExprList *et_value;
     et_value = ExprList::MakeExprList(et_ads);
 
-    if (ad->Insert (name, et_value)) return C_CLASSAD_NO_ERROR; 
-    else                             return C_CLASSAD_INSERT_FAILED;
+    if (ad->Insert (name, (ExprTree* &)et_value)) return C_CLASSAD_NO_ERROR; 
+    else                                   return C_CLASSAD_INSERT_FAILED;
    }
 
   int
