Index: lsf.pm
===================================================================
RCS file: /home/globdev/CVS/globus-packages/gram/jobmanager/lrms/lsf/source/lsf.pm,v
retrieving revision 1.1.2.1
diff -u -r1.1.2.1 lsf.pm
--- lsf.pm	17 Aug 2012 18:25:14 -0000	1.1.2.1
+++ lsf.pm	23 Oct 2012 14:56:06 -0000
@@ -359,6 +359,59 @@
             Globus::GRAM::Error::TEMP_SCRIPT_FILE_FAILED());
     }
 
+    my @xcount, @host_xcount;
+    @xcount = $description->xcount();
+    @host_xcount = $description->host_xcount();
+
+    if (@host_xcount)
+    {
+        my $res = "";
+        $self->log("host_xcount = " . scalar(@host_xcount));
+        foreach my $this_hostxcount (@host_xcount)
+        {
+            $self->log("host_xcount = " . $this_hostxcount);
+            my $this_xcount = shift @xcount;
+            if (!$this_xcount)
+            {
+                $this_xcount = 1;
+            }
+            my $mult = int($this_xcount) * int($this_hostxcount);
+
+            if ($res ne '')
+            {
+                $res .= " + ";
+            }
+            $res .= "$mult * {span[ptile=$this_xcount]}";
+        }
+        $rc = print JOB "#BSUB -R \"$res\"\n";
+        if (!$rc)
+        {
+            return $self->respond_with_failure_extension(
+                "print: $lsf_job_script_name: $!",
+                Globus::GRAM::Error::TEMP_SCRIPT_FILE_FAILED());
+        }
+    }
+    elsif (@xcount)
+    {
+        my $res = "";
+        $self->log("xcount = " . scalar(@xcount));
+        foreach my $this_xcount (@xcount)
+        {
+            if ($res ne '')
+            {
+                $res .= " + ";
+            }
+            $res .= "$this_xcount * {span[ptile=$this_xcount]}";
+        }
+        $rc = print JOB "#BSUB -R \"$res\"\n";
+        if (!$rc)
+        {
+            return $self->respond_with_failure_extension(
+                "print: $lsf_job_script_name: $!",
+                Globus::GRAM::Error::TEMP_SCRIPT_FILE_FAILED());
+        }
+    }
+
     foreach my $tuple ($description->environment())
     {
 	if(!ref($tuple) || scalar(@$tuple) != 2)
@@ -491,6 +544,7 @@
                 Globus::GRAM::Error::TEMP_SCRIPT_FILE_FAILED());
         }
     }
+
     $rc = close(JOB);
     if (!$rc)
     {
Index: lsf.rvf
===================================================================
RCS file: /home/globdev/CVS/globus-packages/gram/jobmanager/lrms/lsf/source/lsf.rvf,v
retrieving revision 1.1.2.1
diff -u -r1.1.2.1 lsf.rvf
--- lsf.rvf	17 Aug 2012 18:25:14 -0000	1.1.2.1
+++ lsf.rvf	23 Oct 2012 14:56:06 -0000
@@ -1,3 +1,24 @@
 Attribute: name
 Description: "The name of job to submit to the LSF batch system"
 ValidWhen: GLOBUS_GRAM_JOB_SUBMIT
+
+Attribute: host_xcount
+Description: "List of numbers of nodes to request for this job."
+ValidWhen: GLOBUS_GRAM_JOB_SUBMIT
+
+Attribute: xcount
+Description: "List of numbers of processes to run on each of the nodes corresponding to the host_xcount."
+ValidWhen: GLOBUS_GRAM_JOB_SUBMIT
+
+# Disable RSL attributes that aren't supported by LSF
+Attribute: host_count
+ValidWhen: 
+
+Attribute: min_memory
+ValidWhen:
+
+Attribute: host_names
+ValidWhen: 
+
+Attribute: host_types
+ValidWhen: 
Index: pkgdata/pkg_data_src.gpt.in
===================================================================
RCS file: /home/globdev/CVS/globus-packages/gram/jobmanager/lrms/lsf/source/pkgdata/pkg_data_src.gpt.in,v
retrieving revision 1.1.2.1
diff -u -r1.1.2.1 pkg_data_src.gpt.in
--- pkgdata/pkg_data_src.gpt.in	17 Aug 2012 18:25:16 -0000	1.1.2.1
+++ pkgdata/pkg_data_src.gpt.in	23 Oct 2012 14:56:06 -0000
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE gpt_package_metadata SYSTEM "package.dtd">
 <gpt_package_metadata Format_Version="0.02" Name="globus_gram_job_manager_lsf">
-    <Aging_Version Major="1" Minor="0" Age="0"/>
+    <Aging_Version Major="1" Minor="1" Age="0"/>
     <Description>Grid Engine Job Manager Support</Description>
     <Functional_Group>GRAM</Functional_Group>
     <Version_Stability Release="Alpha"/>
