--- a/config/02-ce-condor-defaults.conf
+++ b/config/02-ce-condor-defaults.conf
@@ -18,5 +18,5 @@ JOB_ROUTER_ENTRIES = \
 
 JOB_ROUTER_SCHEDD2_SPOOL=/var/lib/condor/spool
 JOB_ROUTER_SCHEDD2_NAME=$(FULL_HOSTNAME)
-JOB_ROUTER_SCHEDD2_POOL=$(FULL_HOSTNAME)
+JOB_ROUTER_SCHEDD2_POOL=$(FULL_HOSTNAME):9618
 
	Modified   config/02-ce-condor.conf
diff --git a/config/02-ce-condor.conf b/config/02-ce-condor.conf
index 97ea67a..286eb62 100644
--- a/config/02-ce-condor.conf
+++ b/config/02-ce-condor.conf
@@ -15,5 +15,5 @@ JOB_ROUTER_ENTRIES = \
 
 JOB_ROUTER_SCHEDD2_SPOOL=/var/lib/condor/spool
 JOB_ROUTER_SCHEDD2_NAME=$(FULL_HOSTNAME)
-JOB_ROUTER_SCHEDD2_POOL=$(FULL_HOSTNAME)
+JOB_ROUTER_SCHEDD2_POOL=$(FULL_HOSTNAME):9618
 

