Index: htcondor-ce-1.7/src/condor-ce
===================================================================
--- htcondor-ce-1.7.orig/src/condor-ce	2014-10-28 15:48:15.000000000 -0500
+++ htcondor-ce-1.7/src/condor-ce	2014-10-31 16:23:19.155481251 -0500
@@ -1,21 +1,20 @@
 #!/bin/bash
 #
-# condor	This script allows for starting and stopping Condor.
+# condor-ce	This script allows for starting and stopping HTCondor-CE.
 #
 # chkconfig: 2345 98 10
-# description: Condor is a high throughput computing batch processing
-# 	       platform.
+# description: HTCondor-CE is a job gateway built on the HTCondor batch system's job router
 # processname: condor_master
 # config: /etc/condor-ce/condor_config
 # pidfile: /var/run/condor-ce/condor_master.pid
 
 ### BEGIN INIT INFO
-# Provides: condor
+# Provides: condor-ce
 # Required-Start: $local_fs $network
 # Required-Stop: $local_fs $network
 # Default-Stop: 1 2 3 4 5 6
-# Short-Description: start and stop Condor
-# Description: Condor HTC computing platform
+# Short-Description: start and stop HTCondor-CE
+# Description: HTCondor-CE job gateway
 ### END INIT INFO
 
 # The program being managed
Index: htcondor-ce-1.7/src/condor-ce-collector
===================================================================
--- htcondor-ce-1.7.orig/src/condor-ce-collector	2014-10-28 15:48:15.000000000 -0500
+++ htcondor-ce-1.7/src/condor-ce-collector	2014-10-31 16:24:32.949422034 -0500
@@ -1,6 +1,6 @@
 #!/bin/bash
 #
-# condor	This script allows for starting and stopping a HTCondor-CE collector.
+# condor-ce-collector	This script allows for starting and stopping an HTCondor-CE collector.
 #
 # chkconfig: 2345 98 10
 # description: The HTCondor-CE collector is a central component for collecting data
@@ -10,12 +10,12 @@
 # pidfile: /var/run/condor-ce/condor_master.pid
 
 ### BEGIN INIT INFO
-# Provides: condor
+# Provides: condor-ce-collector
 # Required-Start: $local_fs $network
 # Required-Stop: $local_fs $network
 # Default-Stop: 1 2 3 4 5 6
-# Short-Description: start and stop Condor
-# Description: Condor HTC computing platform
+# Short-Description: start and stop HTCondor-CE collector
+# Description: Component for collecting data about deployed HTCondor-CEs
 ### END INIT INFO
 
 # The program being managed
