--- a/config/03-ce-shared-port-defaults.conf
+++ b/config/03-ce-shared-port-defaults.conf
@@ -21,8 +21,8 @@ SHARED_PORT=/usr/libexec/condor/condor_shared_port
 # The colons are effectively treated as comments in < 8.2 according to TJ.
 
 :if version < 8.3.2
-   SHARED_PORT_ARGS= -p 9620
-:   COLLECTOR_USES_SHARED_PORT = False
+SHARED_PORT_ARGS= -p 9620
+:COLLECTOR_USES_SHARED_PORT = False
 :else
-:   SHARED_PORT_ARGS= -p 9619
-:endif
\ No newline at end of file
+:SHARED_PORT_ARGS= -p 9619
+:endif
