diff --git a/CMakeLists.txt b/CMakeLists.txt
index c1a340a..3f402c7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,7 +58,7 @@ install(TARGETS classad_ce LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}/condor")
 
 install(PROGRAMS src/condor-ce src/condor-ce-collector DESTINATION ${SYSCONF_INSTALL_DIR}/rc.d/init.d)
 install(PROGRAMS src/condor_ce_startup src/condor_ce_startup_internal src/condor_ce_env_bootstrap src/condor_ce_client_env_bootstrap src/condor_ce_router_defaults src/osg-wrapper DESTINATION ${SHARE_INSTALL_PREFIX}/condor-ce)
-install(PROGRAMS src/condor_ce_config_generator src/condor_ce_config_val src/condor_ce_history src/condor_ce_hold src/condor_ce_info_status src/condor_ce_q src/condor_ce_qedit src/condor_ce_release src/condor_ce_rm src/condor_ce_submit src/condor_ce_version src/condor_ce_reconfig src/condor_ce_router_q src/condor_ce_status src/condor_ce_reschedule src/condor_ce_run src/condor_ce_trace src/condor_ce_ping src/condor_ce_off src/condor_ce_on src/condor_ce_restart src/condor_ce_job_router_tool DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
+install(PROGRAMS src/condor_ce_config_generator src/condor_ce_config_val src/condor_ce_history src/condor_ce_hold src/condor_ce_info_status src/condor_ce_q src/condor_ce_qedit src/condor_ce_release src/condor_ce_rm src/condor_ce_submit src/condor_ce_version src/condor_ce_reconfig src/condor_ce_router_q src/condor_ce_status src/condor_ce_reschedule src/condor_ce_run src/condor_ce_trace src/condor_ce_ping src/condor_ce_off src/condor_ce_on src/condor_ce_restart src/condor_ce_job_router_info DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
 
 install(FILES config/condor_config config/condor_mapfile DESTINATION ${SYSCONF_INSTALL_DIR}/condor-ce)
 install(FILES config/01-ce-collector.conf config/02-ce-auth-generated.conf config/01-ce-auth.conf config/01-common-auth.conf config/01-ce-router.conf config/02-ce-condor.conf config/02-ce-pbs.conf config/02-ce-lsf.conf config/02-ce-sge.conf config/03-ce-shared-port.conf config/03-managed-fork.conf DESTINATION ${SYSCONF_INSTALL_DIR}/condor-ce/config.d)
diff --git a/config/htcondor-ce.spec b/config/htcondor-ce.spec
index be874a4..1325068 100644
--- a/config/htcondor-ce.spec
+++ b/config/htcondor-ce.spec
@@ -263,7 +263,7 @@ fi
 %{_bindir}/condor_ce_config_val
 %{_bindir}/condor_ce_hold
 %{_bindir}/condor_ce_info_status
-%{_bindir}/condor_ce_job_router_tool
+%{_bindir}/condor_ce_job_router_info
 %{_bindir}/condor_ce_off
 %{_bindir}/condor_ce_on
 %{_bindir}/condor_ce_q
diff --git a/src/condor_ce_job_router_info b/src/condor_ce_job_router_info
new file mode 100755
index 0000000..d5584bf
--- /dev/null
+++ b/src/condor_ce_job_router_info
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+missing_tool()
+{
+    echo 'Could not find Job Router diagnostic tool. Please verify your HTCondor installation.'
+    exit 1
+}
+
+. /usr/share/condor-ce/condor_ce_env_bootstrap
+. /usr/share/condor-ce/condor_ce_client_env_bootstrap
+export GSI_AUTHZ_CONF=/dev/null
+CONDOR_BIN_DIR=$(/usr/bin/dirname $(/usr/bin/which condor_version 2> /dev/null ) 2> /dev/null ) 
+if [ -z "$CONDOR_BIN_DIR" ]; then
+    missing_tool
+elif [ -x "$CONDOR_BIN_DIR/condor_job_router_info" ]; then
+    exec "$CONDOR_BIN_DIR/condor_job_router_info" "$@"
+elif [ -x "$CONDOR_BIN_DIR/condor_job_router_tool" ]; then
+    exec "$CONDOR_BIN_DIR/condor_job_router_tool" "$@"
+else
+    missing_tool
+fi 
+
+
diff --git a/src/condor_ce_job_router_tool b/src/condor_ce_job_router_tool
deleted file mode 100644
index 9d01c4a..0000000
--- a/src/condor_ce_job_router_tool
+++ /dev/null
@@ -1,6 +0,0 @@
-#!/bin/sh
-
-. /usr/share/condor-ce/condor_ce_env_bootstrap
-. /usr/share/condor-ce/condor_ce_client_env_bootstrap
-export GSI_AUTHZ_CONF=/dev/null
-exec condor_job_router_tool "$@"
