diff --git a/src/condor_ce_trace b/src/condor_ce_trace
index 50272cf..c62354c 100755
--- a/src/condor_ce_trace
+++ b/src/condor_ce_trace
@@ -32,6 +32,18 @@ condorJobStatus = { \
   6: "Transferring Output",
 }
 
+def verify_matching_condor_versions():
+    lib_ver = htcondor.version()
+    try:
+        args = ('condor_version',)
+        p = Popen(args, stdout=PIPE, stderr=STDOUT)
+        (stdout, stderr) = p.communicate()
+    except RuntimeError, e:
+        raise CondorRunException('Could not find version information using condor_version')
+    tool_ver = stdout.split('\n')[0]
+    if lib_ver != tool_ver:
+        raise RuntimeError('Found multiple, mismatched versions of HTCondor installed. Please ensure that you only have one version of HTCondor installed.')
+
 def configureAuth():
     try:
         auth_methods = re.split(",?\s*", htcondor.param['SEC_CLIENT_AUTHENTICATION_METHODS'])
@@ -306,6 +318,19 @@ def main():
         print "Usage: condor_ce_trace [options] <hostname>"
         sys.exit(1)
 
+    try:
+        verify_matching_condor_versions()
+    except RuntimeError, e:
+        print "*"*80
+        print e
+        print "*"*80
+        sys.exit(1)
+    except Exception, e:
+        print "*"*80
+        print 'Uncaught exception, please contact goc@opensciencegrid.org.\n%s: %s' % (e.__class__.__name__,e)
+        print "*"*80
+        sys.exit(1)
+
     job_info = {}
     job_info['attribute'] = opts.attribute
     collector_hostname = args[0].split(":")[0]

