--- src/c/Makefile.am.orig	2012-01-19 19:59:13.000000000 -0600
+++ src/c/Makefile.am	2012-01-19 20:00:19.000000000 -0600
@@ -17,8 +17,6 @@
 DELEGATION_INCLUDE_DIR = $(top_builddir)/interface/glite/data/delegation/c
 
 # Copy gsoap file
-stdsoap2.c:
-	cp -f $(top_builddir)/$@ $@
 
 # Common Files
 soapdefs.h:
@@ -58,7 +56,6 @@
 	delegationClientLib.c \
 	delegationServer.c \
 	delegationServerLib.c \
-        stdsoap2.c \
 	soapdefs.h \
         env.c 
 
@@ -92,26 +89,26 @@
 	sed -e s/@libname\@/$(basename $@)_`expr $(INTERFACE_LIBTOOL_CURRENT) - $(INTERFACE_LIBTOOL_AGE)`/ $< > $@
 
 libglite_data_delegation_api_c_la_SOURCES = \
-	stdsoap2.c \
 	delegationClientLib.c \
 	env.c
 
 libglite_data_delegation_api_c_la_LIBADD = \
 	$(CGSI_GSOAP_LIBS) \
-	$(GLOBUS_GSS_NOTHR_LIBS)
+	$(GLOBUS_GSS_NOTHR_LIBS) \
+	-lgsoap
 
 libglite_data_delegation_api_c_la_LDFLAGS = \
 	$(SYMVER_LDFLAGS) \
 	-version-info $(INTERFACE_LIBTOOL_CURRENT):$(INTERFACE_LIBTOOL_REVISION):$(INTERFACE_LIBTOOL_AGE)
 
 libglite_data_delegation_server_api_c_la_SOURCES = \
-	stdsoap2.c \
 	delegationServerLib.c \
 	env.c
 
 libglite_data_delegation_server_api_c_la_LIBADD = \
 	$(CGSI_GSOAP_LIBS) \
-	$(GLOBUS_GSS_NOTHR_LIBS)
+	$(GLOBUS_GSS_NOTHR_LIBS) \
+	-lgsoap
 
 libglite_data_delegation_server_api_c_la_LDFLAGS = \
 	$(SYMVER_LDFLAGS) \
