diff --git a/common/configuration/install-database b/common/configuration/install-database
index b047801..1be6393 100644
--- a/common/configuration/install-database
+++ b/common/configuration/install-database
@@ -15,6 +15,7 @@ CONFIGURATIONS=/etc/gratia/services/service-configuration.properties
 AUTHORIZATIONS=/etc/gratia/services/service-authorization.properties
 MYSQL_CONFIG=/etc/my.cnf
 DEFAULT_MYSQL_PORT=3306
+MYSQL_SERVICE=mysqld
 
 
 ########## TMP FILES
@@ -105,11 +106,11 @@ function get_mysql_attributes {
   DBREADER=`grep         ^service.reporting.user     $AUTHORIZATIONS | tr '=' ' ' | awk '{print $NF;}'`
   DBREADERPASSWD=`grep   ^service.reporting.password $AUTHORIZATIONS | tr '=' ' ' | awk '{print $NF;}'`
   DBROOTPASSWD=`grep     ^service.mysql.rootpassword $AUTHORIZATIONS | tr '=' ' ' | awk '{print $NF;}'`
-  if [ "x$DBROOTPASSWD" = "x" ]  || [ $DBROOTPASSWD = "service.mysql.rootpassword" ]; then
+  if [ "x$DBROOTPASSWD" = "x" ]  || [ "$DBROOTPASSWD" = "service.mysql.rootpassword" ]; then
     DBROOTPASSWD=""
   fi
   DBROOTUSER=`grep ^service.mysql.rootuser     $AUTHORIZATIONS | tr '=' ' ' | awk '{print $NF;}'`
-  if [ "x$DBROOTUSER" = "x" ]  || [ $DBROOTUSER = "service.mysql.rootuser" ]; then
+  if [ "x$DBROOTUSER" = "x" ]  || [ "$DBROOTUSER" = "service.mysql.rootuser" ]; then
     DBROOTUSER=root
   fi
 }
@@ -125,7 +126,11 @@ function verify_mysql_installed {
     echo "You must install and configure MySQL 5 before configuring Gratia."
     exit 1
   fi
-  if [ ! -e "/etc/init.d/mysqld" ]; then
+  if [ -e "/etc/init.d/mysqld" ]; then
+    : OK
+  elif [ -e /usr/lib/systemd/system/mariadb.service ]; then
+    MYSQL_SERVICE=mariadb
+  else
     echo "ERROR: Could not find initd service: /etc/init.d/mysqld"
     echo "Did you install of MySql work correctly?"
     exit 1
@@ -136,12 +141,12 @@ function verify_mysql_installed {
 #-------------------------------
 # Check the service status
 function verify_mysql_running {
-  [ -z "$QUIET" ] && echo "... verifying mysqld daemon is running"
-  cmd="service mysqld status"
+  [ -z "$QUIET" ] && echo "... verifying $MYSQL_SERVICE daemon is running"
+  cmd="service $MYSQL_SERVICE status"
   echo "$cmd"
   $cmd; rtn=$?
   if [ $rtn -ne 0 ]; then
-    echo "ERROR: The MySql mysqld daemon does not appear to be running"
+    echo "ERROR: The MySql $MYSQL_SERVICE daemon does not appear to be running"
     echo "You will need it running to complete this installation."
     exit 1
   fi
@@ -359,7 +364,7 @@ MySQL server: $MSHOST ($DBHOST)"
 if [ -n "$PRINT_ONLY" ]; then
   [ -z "$QUIET" ] && echo "Run these commands in the MySQL database."
 
-  if [ $GSHOST = $MSHOST ]; then
+  if [ "$GSHOST" = "$MSHOST" ]; then
     # check root password
     check_if_root_passwd_set
     if [ "ROOT_SET" = "0" ]; then
@@ -381,7 +386,7 @@ EOF
   exit
 fi
 
-if [ $GSHOST = $MSHOST ]; then
+if [ "$GSHOST" = "$MSHOST" ]; then
   [ -z "$QUIET" ] && echo "Gratia Server and MySQL are running on the same hosts"
   verify_mysql_installed 
   verify_mysql_running 
