diff --git a/common/configuration/configure_tomcat b/common/configuration/configure_tomcat
index 24d63b5..4050189 100755
--- a/common/configuration/configure_tomcat
+++ b/common/configuration/configure_tomcat
@@ -33,6 +33,8 @@ function find_tomcat_version {
     export TOMCAT_VERSION=tomcat5    
   elif [ -r "/etc/tomcat6/tomcat6.conf" ];then
     export TOMCAT_VERSION=tomcat6    
+  elif [ -r "/etc/tomcat/tomcat.conf" ];then
+    export TOMCAT_VERSION=tomcat
   else
     logerr "Unable to determine tomcat version"
   fi
@@ -65,6 +67,9 @@ function verify_jsvc_installed {
       "tomcat6" )  JAR_FILES=":/usr/share/java/jakarta-commons-daemon.jar:\$CATALINA_HOME/bin/tomcat-juli.jar" 
                    JSVC_FILE=/usr/bin/jsvc
                    ;;
+      "tomcat" )   JAR_FILES=":/usr/share/java/commons-daemon.jar"
+                   JSVC_FILE=/usr/bin/jsvc
+                   ;;
             * ) logerr "The tomcat version could not be detected" ;;
     esac
   [ ! -e "$JSVC_FILE" ] && logerr "Cannot find the jsvc executable for the $TOMCAT_VERSION version: $JSVC_FILE"
@@ -108,6 +113,8 @@ function configuring_tomcat {
         ;;
       "tomcat6" )
          ;;
+      "tomcat" )
+         ;;
         * ) logerr "This version of tomcat is not supported: $TOMCAT_VERSION" ;;
     esac
   logit
