--- globus-job-get-output.in.orig	2012-11-01 21:00:51.000000000 -0500
+++ globus-job-get-output.in	2012-11-01 21:01:11.000000000 -0500
@@ -154,7 +154,7 @@
 fi
 
 if [ -z "${resource}" ]; then
-    resource=`echo ${jobid} | ${GLOBUS_SH_AWK} -F: '{print $2}' | \
+    resource=`echo ${jobid} | ${GLOBUS_SH_AWK-awk} -F: '{print $2}' | \
 	${GLOBUS_SH_SED-sed} 's|/||g'`
 fi
 
--- globus-job-clean.in.orig	2012-11-02 08:14:42.000000000 -0500
+++ globus-job-clean.in	2012-11-02 08:16:34.000000000 -0500
@@ -149,7 +149,7 @@
 
 if [ ${action} = cleanup ]; then
    if [ -z "${resource}" ]; then
-      resource=`echo ${jobid} | ${GLOBUS_SH_AWK} -F: '{print $2}' | \
+      resource=`echo ${jobid} | ${GLOBUS_SH_AWK-awk} -F: '{print $2}' | \
           ${GLOBUS_SH_SED-sed} 's|/||g'`
    fi
 fi
