Index: glite-lbjp-common-gss-3.1.3/src/glite_gss.c
===================================================================
--- glite-lbjp-common-gss-3.1.3.orig/src/glite_gss.c	2012-04-02 02:38:31.000000003 -0500
+++ glite-lbjp-common-gss-3.1.3/src/glite_gss.c	2012-07-26 12:58:13.000000003 -0500
@@ -753,7 +753,7 @@
 	 ret = EDG_WLL_GSS_ERROR_ERRNO;
 	 goto end;
       }
-      buffer.length = strlen(proxy_file);
+      buffer.length = strlen(buffer.value);
 
       major_status = gss_import_cred(&minor_status, &gss_cred,
 				     GSS_C_NO_OID, 1,
