diff -ruB org.glite.data.dm-util.orig/configure.ac org.glite.data.dm-util/configure.ac
--- org.glite.data.dm-util.orig/configure.ac	2011-07-08 14:36:17.172936410 -0500
+++ org.glite.data.dm-util/configure.ac	2011-07-08 14:38:51.725077304 -0500
@@ -22,7 +22,7 @@
 
 AC_PREREQ(2.57)
 # version 0.0.0 will be replaced by AC_GLITE_VERSION
-AC_INIT([GLite Data Management Utilities], [0.0.0])
+AC_INIT([GLite Data Management Utilities], [1.11.14])
 # auxiliary build tools directory
 AC_CONFIG_AUX_DIR([./project])
 AC_CONFIG_SRCDIR([src/Makefile.am])
@@ -123,10 +123,10 @@
     [VOMS_LOCATION=${withval}],
     [VOMS_LOCATION="/opt/lcg"])
 AC_MSG_RESULT([VOMS_LOCATION is ${VOMS_LOCATION}])
-VOMS_CFLAGS="-I${VOMS_LOCATION}/include -I${GLOBUS_LOCATION}/include/${GLOBUS_FLAVOUR}"
+VOMS_CFLAGS="-I${VOMS_LOCATION}/include/voms"
 AC_MSG_RESULT([VOMS_CFLAGS is ${VOMS_CFLAGS}])
-VOMS_THR_LIBS="-L${VOMS_LOCATION}/lib -L${VOMS_LOCATION}/lib64 -lvomsc_${GLOBUS_THR_FLAVOR}"
-VOMS_NOTHR_LIBS="-L${VOMS_LOCATION}/lib -L${VOMS_LOCATION}/lib64 -lvomsc_${GLOBUS_NOTHR_FLAVOR}"
+VOMS_THR_LIBS="-L${VOMS_LOCATION}/lib -L${VOMS_LOCATION}/lib64 -lvomsapi"
+VOMS_NOTHR_LIBS="-L${VOMS_LOCATION}/lib -L${VOMS_LOCATION}/lib64 -lvomsapi"
 AC_MSG_RESULT([VOMS_THR_LIBS is ${VOMS_THR_LIBS}])
 AC_MSG_RESULT([VOMS_NOTHR_LIBS is ${VOMS_NOTHR_LIBS}])
 AC_SUBST(VOMS_LOCATION)
diff -ruB org.glite.data.dm-util.orig/src/Makefile.am org.glite.data.dm-util/src/Makefile.am
--- org.glite.data.dm-util.orig/src/Makefile.am	2011-07-08 14:36:17.174083447 -0500
+++ org.glite.data.dm-util/src/Makefile.am	2011-07-08 14:37:49.712974000 -0500
@@ -185,10 +185,10 @@
 liblcg_util_la_LIBADD = \
 	$(VOMS_NOTHR_LIBS) \
 	$(GLOBUS_FTP_CLIENT_NOTHR_LIBS) \
-	-lglobus_gass_copy_$(GLOBUS_NOTHR_FLAVOR) \
-	-lglobus_gass_transfer_$(GLOBUS_NOTHR_FLAVOR) \
+	-lglobus_gass_copy \
+	-lglobus_gass_transfer \
 	$(GFAL_LIBS) \
-	-lcgsi_plugin_gsoap_2.7 
+	-lcgsi_plugin
 liblcg_util_la_LDFLAGS = \
 	-version-info $(INTERFACE_LIBTOOL_CURRENT):$(INTERFACE_LIBTOOL_REVISION):$(INTERFACE_LIBTOOL_AGE)	
 
