--- a/frontend/gwms_renew_proxies.py
+++ b/frontend/gwms_renew_proxies.py
@@ -27,9 +27,10 @@ class Proxy(object):
         self.cert = cert
         self.key = key
         self.tmp_output_fd = tempfile.NamedTemporaryFile(dir=os.path.dirname(output), delete=False)
-        os.chown(self.tmp_output_fd.name, uid, gid)
         self.output = output
         self.lifetime = lifetime
+        self.uid = uid
+        self.gid = gid
 
     def _voms_proxy_info(self, *opts):
         """Run voms-proxy-info. Returns stdout, stderr, and return code of voms-proxy-info
@@ -43,6 +44,7 @@ class Proxy(object):
         self.tmp_output_fd.flush()
         os.fsync(self.tmp_output_fd)
         self.tmp_output_fd.close()
+        os.chown(self.tmp_output_fd.name, self.uid, self.gid)
         os.rename(self.tmp_output_fd.name, self.output)
 
     def timeleft(self):
@@ -169,11 +171,13 @@ def main():
         if proxy_section == 'FRONTEND':
             if has_time_left(proxy.timeleft()):
                 print 'Skipping renewal of %s: time remaining within the specified frequency' % proxy.output
+                os.remove(proxy.tmp_output_fd.name)
                 continue
             stdout, stderr, client_rc = voms_proxy_init(proxy)
         elif proxy_section.startswith('PILOT'):
             if has_time_left(proxy.timeleft()) and has_time_left(proxy.actimeleft()):
                 print 'Skipping renewal of %s: time remaining within the specified frequency' % proxy.output
+                os.remove(proxy.tmp_output_fd.name)
                 continue
 
             voms_info = vomses[proxy_config['vo'].lower()]
@@ -203,6 +207,7 @@ def main():
             retcode = 1
             # don't raise an exception here to continue renewing other proxies
             print "ERROR: Failed to renew proxy %s:\n%s%s" % (proxy.output, stdout, stderr)
+            os.remove(proxy.tmp_output_fd.name)
 
     return retcode
 
--- a/creation/templates/gwms-renew-proxies.cron
+++ b/creation/templates/gwms-renew-proxies.cron
@@ -1,1 +1,1 @@
-23 * * * * [ -f /var/lock/subsys/gwms-renew-proxies ] && sleep $((RANDOM % 600)) && /usr/libexec/gwms_renew_proxies
+23 * * * * root [ -f /var/lock/subsys/gwms-renew-proxies ] && sleep $((RANDOM \% 600)) && /usr/libexec/gwms_renew_proxies
