Index: Condor.py
===================================================================
--- Condor.py	(.../tags/3.10.2/rsv-core/lib/python/rsv/Condor.py)	(revision 21532)
+++ Condor.py	(.../trunk/rsv-core/lib/python/rsv/Condor.py)	(revision 21532)
@@ -23,7 +23,7 @@
 
         (ret, out) = self.commands_getstatusoutput("condor_cron_q")
 
-        if not ret and out.index("-- Submitter") != -1:
+        if not ret:
             self.rsv.log("DEBUG", "Condor is running.  Output of condor_cron_q:\n%s" % out)
             return True
 
