diff --git a/rsv-core/lib/python/rsv/Condor.py b/rsv-core/lib/python/rsv/Condor.py
index b649fc2..cf0ab61 100644
--- a/rsv-core/lib/python/rsv/Condor.py
+++ b/rsv-core/lib/python/rsv/Condor.py
@@ -23,7 +23,7 @@ class Condor:
 
         (ret, out) = self.commands_getstatusoutput("condor_cron_q")
 
-        if not ret and out.index("-- Submitter") != -1:
+        if not ret:
             self.rsv.log("DEBUG", "Condor is running.  Output of condor_cron_q:\n%s" % out)
             return True
 
