Index: org.gridsite.core/src/Makefile
===================================================================
--- org.gridsite.core.orig/src/Makefile	2012-07-24 07:58:56.000000000 -0500
+++ org.gridsite.core/src/Makefile	2013-01-31 15:29:44.431600904 -0600
@@ -418,15 +418,16 @@
                  $(prefix)/sbin \
                  $(prefix)/share/man/man1 \
                  $(prefix)/share/man/man8 \
-                 $(prefix)/share/doc/gridsite-$(MINOR_VERSION)
+                 $(prefix)/share/doc/gridsite-$(MINOR_VERSION) \
+                 $(prefix)/libexec/gridsite/cgi-bin
 	sed 's/^\(#define GRST_VERSION\).*$$/\1 $(DEFVERSION)/' \
 	      ../interface/gridsite.h > $(prefix)/include/gridsite.h
 	cp -f ../interface/gridsite-gacl.h $(prefix)/include
 	cp -f urlencode $(prefix)/bin
 	cp -f findproxyfile $(prefix)/bin
-	cp -f real-gridsite-admin.cgi $(prefix)/sbin
-	cp -f gridsite-copy.cgi $(prefix)/sbin
-	cp -f gridsite-storage.cgi $(prefix)/sbin
+	cp -f real-gridsite-admin.cgi $(prefix)/libexec/gridsite/cgi-bin
+	cp -f gridsite-copy.cgi $(prefix)/libexec/gridsite/cgi-bin
+	cp -f gridsite-storage.cgi $(prefix)/libexec/gridsite/cgi-bin
 	cp -f ../CHANGES ../README ../INSTALL ../LICENSE ../VERSION \
                $(prefix)/share/doc/gridsite-$(MINOR_VERSION)
 	cp -f ../doc/index.html ../doc/*.conf ../doc/*.sh ../doc/*.spec \
@@ -519,7 +520,7 @@
 	ln -sf htproxyput $(prefix)/bin/htproxyunixtime
 	ln -sf htproxyput $(prefix)/bin/htproxyrenew
 	ln -sf htproxyput $(prefix)/bin/htproxyinfo
-	cp -f gridsite-delegation.cgi $(prefix)/sbin
+	cp -f gridsite-delegation.cgi $(prefix)/libexec/gridsite/cgi-bin
 
 post-install-debian:
 	sed -i \
