diff -uNr org.gridsite.core.ORIG/src/Makefile org.gridsite.core/src/Makefile
--- org.gridsite.core.ORIG/src/Makefile	2011-06-27 18:07:42.000000000 +0200
+++ org.gridsite.core/src/Makefile	2011-06-30 19:50:07.960436142 +0200
@@ -405,15 +405,17 @@
                  $(prefix)/share/man/man1 \
                  $(prefix)/share/man/man8 \
                  $(prefix)/$(libdir)/httpd/modules \
-                 $(prefix)/share/doc/gridsite-$(MINOR_VERSION)
+                 $(prefix)/share/doc/gridsite-$(MINOR_VERSION) \
+                 $(prefix)/libexec/gridsite/cgi-bin
 	sed 's/^\(#define GRST_VERSION\).*$$/\1 $(DEFVERSION)/' \
 	      ../interface/gridsite.h > $(prefix)/include/gridsite.h
 	cp -f ../interface/gridsite-gacl.h $(prefix)/include
 	cp -f urlencode $(prefix)/bin
 	cp -f findproxyfile $(prefix)/bin
-	cp -f real-gridsite-admin.cgi $(prefix)/sbin
-	cp -f gridsite-copy.cgi $(prefix)/sbin
-	cp -f gridsite-storage.cgi $(prefix)/sbin
+	cp -f real-gridsite-admin.cgi $(prefix)/libexec/gridsite/cgi-bin
+	cp -f gridsite-copy.cgi $(prefix)/libexec/gridsite/cgi-bin
+	cp -f gridsite-storage.cgi $(prefix)/libexec/gridsite/cgi-bin
+
 	cp -f ../CHANGES ../README ../INSTALL ../LICENSE ../VERSION \
                $(prefix)/share/doc/gridsite-$(MINOR_VERSION)
 	cp -f ../doc/index.html ../doc/*.conf ../doc/*.sh ../doc/*.spec \
@@ -502,7 +504,7 @@
 	ln -sf htproxyput $(prefix)/bin/htproxyunixtime
 	ln -sf htproxyput $(prefix)/bin/htproxyrenew
 	ln -sf htproxyput $(prefix)/bin/htproxyinfo
-	cp -f gridsite-delegation.cgi $(prefix)/sbin
+	cp -f gridsite-delegation.cgi $(prefix)/libexec/gridsite/cgi-bin
 	   
 #
 # Distributions
