diff --git a/bigtop-detect-javahome b/bigtop-detect-javahome
index a2671eb..6eaa91b 100644
--- a/bigtop-detect-javahome
+++ b/bigtop-detect-javahome
@@ -70,6 +70,11 @@ MISCJAVA_HOME_CANDIDATES=(
     '/usr/lib/jvm/jre-openjdk'
 )
 
+ALTERNATIVES_JAVA_HOME_CANDIDATES=(
+    '/etc/alternatives/java_sdk'
+    '/etc/alternatives/jre'
+)
+
 # Note that the JDK versions recommended for production use in CDH
 # may not reflect the current recommendations for Apache Bigtop
 case ${BIGTOP_JAVA_MAJOR} in
@@ -79,11 +84,12 @@ case ${BIGTOP_JAVA_MAJOR} in
     ;;
     8) JAVA_HOME_CANDIDATES=(${JAVA8_HOME_CANDIDATES[@]} ${OPENJAVA8_HOME_CANDIDATES[@]})
     ;;
-    *) JAVA_HOME_CANDIDATES=(${JAVA7_HOME_CANDIDATES[@]}
-                             ${JAVA8_HOME_CANDIDATES[@]}
-                             ${MISCJAVA_HOME_CANDIDATES[@]}
+    *) JAVA_HOME_CANDIDATES=(${ALTERNATIVES_JAVA_HOME_CANDIDATES[@]}
                              ${OPENJAVA7_HOME_CANDIDATES[@]}
-                             ${OPENJAVA8_HOME_CANDIDATES[@]})
+                             ${OPENJAVA8_HOME_CANDIDATES[@]}
+                             ${JAVA7_HOME_CANDIDATES[@]}
+                             ${JAVA8_HOME_CANDIDATES[@]}
+                             ${MISCJAVA_HOME_CANDIDATES[@]})
     ;;
 esac
 
