diff --git a/tag2distrepo.py b/tag2distrepo.py
index 20c8da6..57e554d 100644
--- a/tag2distrepo.py
+++ b/tag2distrepo.py
@@ -22,7 +22,8 @@ def tag2distrepo(cbtype, tag, build, user, force=False):
 
     keys = tag['extra'].get("tag2distrepo.keys", '').split()
 
-    if keys:
+    # Signing is performed by a separate plugin in osg koji
+    if keys and False:
         logger.debug("Ensuring signed RPMs are written out")
         [rpms, _] = readTaggedRPMS(tag['id'], rpmsigs=True)
         for rpm in rpms:
@@ -31,15 +32,16 @@ def tag2distrepo(cbtype, tag, build, user, force=False):
                     write_signed_rpm(rpm['id'], key, False)
 
     task_opts = {
-        'arch': tag['arches'].split(),
+        'arch': tag['arches'].split() + ['src'],
         'comp': None,
         'delta': [],
         'event': None,
         'inherit': False,
-        'latest': False,
+        'latest': True,
         'multilib': False,
-        'split_debuginfo': False,
-        'skip_missing_signatures': False,
+        'split_debuginfo': True,
+        'with_src': True,
+        'skip_missing_signatures': True,
         'allow_missing_signatures': not keys,
     }
     logging.debug(
