Index: globus_xio_handle.c
===================================================================
--- globus_xio_handle.c.orig	2010-05-17 09:58:53.000000000 -0500
+++ globus_xio_handle.c	2011-09-07 16:27:26.000000000 -0500
@@ -726,11 +726,19 @@
         }
     }
 
+    /*
+     * Timeout the close calls
+     * We allow 60s per open handle
+     */
+    globus_abstime_t next_timeout;
     globus_mutex_lock(&globus_i_xio_mutex);
     {
         while(monitor.count != 0)
         {
-            globus_cond_wait(&globus_i_xio_cond, &globus_i_xio_mutex);
+            GlobusTimeAbstimeSet(next_timeout, 60, 0);
+            if (ETIMEDOUT == globus_cond_timedwait(&globus_i_xio_cond, &globus_i_xio_mutex, &next_timeout)) {
+                monitor.count--;
+            }
         }
     }
     globus_mutex_unlock(&globus_i_xio_mutex);
