diff --git a/externals/c-ares/src/configureHook.sh b/externals/c-ares/src/configureHook.sh
index d2ec560..49adaf9 100644
--- a/externals/c-ares/src/configureHook.sh
+++ b/externals/c-ares/src/configureHook.sh
@@ -4,4 +4,4 @@
 #              Please install XCode and run `xcode-select --install` afterwards.
 #              Also remember to start XCode at least once and agree to the EULA.
 
-sh configure LDFLAGS="$LDFLAGS -rdynamic" CPPFLAGS="$CPPFLAGS -D_FILE_OFFSET_BITS=64" CFLAGS="$CFLAGS -fno-strict-aliasing -fasynchronous-unwind-tables -fno-omit-frame-pointer -fno-optimize-sibling-calls -fvisibility=hidden -fPIC"
+sh configure --enable-shared=no LDFLAGS="$LDFLAGS -rdynamic" CPPFLAGS="$CPPFLAGS -D_FILE_OFFSET_BITS=64" CFLAGS="$CFLAGS -fno-strict-aliasing -fasynchronous-unwind-tables -fno-omit-frame-pointer -fno-optimize-sibling-calls -fvisibility=hidden -fPIC"
diff --git a/externals/c-ares/src/makeHook.sh b/externals/c-ares/src/makeHook.sh
index 3e65bd1..ea4ba8c 100644
--- a/externals/c-ares/src/makeHook.sh
+++ b/externals/c-ares/src/makeHook.sh
@@ -1,6 +1,6 @@
 #!/bin/sh
 
 make clean
-make
+make libcares.la
 strip -S .libs/libcares.a
 
