diff --git a/cvmfs/catalog_mgr_rw.cc b/cvmfs/catalog_mgr_rw.cc
index 66dad90..7d98e88 100644
--- a/cvmfs/catalog_mgr_rw.cc
+++ b/cvmfs/catalog_mgr_rw.cc
@@ -844,10 +844,11 @@ void WritableCatalogManager::FinalizeCatalog(WritableCatalog *catalog,
 
 void WritableCatalogManager::ScheduleCatalogProcessing(
                                                      WritableCatalog *catalog) {
-  MutexLockGuard guard(catalog_processing_lock_);
-
-  // register catalog object for WritableCatalogManager::CatalogUploadCallback
-  catalog_processing_map_[catalog->database_path()] = catalog;
+  {
+    MutexLockGuard guard(catalog_processing_lock_);
+    // register catalog object for WritableCatalogManager::CatalogUploadCallback
+    catalog_processing_map_[catalog->database_path()] = catalog;
+  }
   spooler_->ProcessCatalog(catalog->database_path());
 }
 
