Index: java_classad.2.4/find_java_home.sh
===================================================================
--- java_classad.2.4.orig/find_java_home.sh	2004-12-11 10:40:58.000000003 -0600
+++ java_classad.2.4/find_java_home.sh	2012-03-13 14:31:10.000000003 -0500
@@ -2,7 +2,8 @@
 
 # Quick hack to try to find JAVA_HOME (the root of your java installation).
 # Assumes that "java" is in your path, and the executable is in
-# $JAVA_HOME/bin/java.  In a valid installation, $JAVA_HOME/jre/lib/rt.jar
+# $JAVA_HOME/bin/java or $JAVA_HOME/jre/bin/java.
+# In a valid installation, $JAVA_HOME/jre/lib/rt.jar
 # should exist.
 
 # On success, prints the name of the directory to stdout.
@@ -15,12 +16,12 @@
     echo ERROR: java not found in your path
     exit 1
 fi
-if [ -h $JAVA ]
-then
-    JAVA=`ls -l $JAVA | sed -e 's/.*-> *//'`
-fi
+JAVA=`readlink -f $JAVA`
 JAVA_HOME=`dirname $JAVA`
 JAVA_HOME=`dirname $JAVA_HOME`
+if [ `basename $JAVA_HOME` == 'jre' ]; then
+    JAVA_HOME=`dirname $JAVA_HOME`
+fi
 if [ -r $JAVA_HOME/jre/lib/rt.jar ]
 then
     echo $JAVA_HOME
