From 539114a078b757fe90c45363d2c80b619216af7f Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Wed, 14 Feb 2018 15:48:23 -0600
Subject: [PATCH 107/108] voms-proxy-direct: drop newsubject, etc. from header
 (SOFTWARE-3123)

---
 src/utils/vomsdirect.h | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/src/utils/vomsdirect.h b/src/utils/vomsdirect.h
index f0dde73..d21b09d 100644
--- a/src/utils/vomsdirect.h
+++ b/src/utils/vomsdirect.h
@@ -105,12 +105,10 @@ class Direct {
   std::string voID;
 
   std::string hostcert, hostkey;
 
   bool newformat;
-  std::string newsubject;
-  std::string newissuer;
  public:
 
   Direct(int argc, char** argv);
   ~Direct();
   bool Run();
@@ -143,19 +141,15 @@ class Direct {
 
   X509_EXTENSION *create_extension(const std::string &string);
   STACK_OF(X509_EXTENSION) *create_and_add_extension(const std::string &string, STACK_OF(X509_EXTENSION) *exts);
 
   bool rfc;
-  std::string pastac;
-  std::string pastproxy;
   std::string keyusage;
   std::string netscape;
   std::string exkusage;
-  std::string newserial;
 
   std::vector<std::string> extensions;
   std::vector<std::string> acextensions;
-  bool selfsigned;
   void PrintProxyCreationError(int error, void *additional);
 };
 
 #endif
-- 
2.6.3

