Index: voms-2.0.8/src/sslutils/sslutils.c
===================================================================
--- voms-2.0.8.orig/src/sslutils/sslutils.c	2012-05-07 00:25:36.000000003 -0500
+++ voms-2.0.8/src/sslutils/sslutils.c	2012-08-13 14:15:38.000000003 -0500
@@ -3674,15 +3674,18 @@
     } 
   }
   else {
-    if (!proxy_load_user_cert_and_key_pkcs12(certname, cert, stack, key, callback))
-      goto err;
+      if (!proxy_load_user_cert_and_key_pkcs12(certname, cert, stack, key, callback)) {
+          *stack = NULL;
+          goto err;
+      }
   }    
 
   return 1;
 
 err:
-  if (chain)
+  if (chain) {
     sk_X509_free(chain);
+  }
   if (cert) {
     X509_free(*cert);
     *cert = NULL;
