--- voms-2.0.6/src/sslutils/sslutils.c	2011-07-22 05:17:52.000000000 -0500
+++ voms-2.0.6/src/sslutils/sslutils.c.new	2011-11-02 21:25:58.877000000 -0500
@@ -3661,15 +3661,18 @@
     } 
   }
   else {
-    if (!proxy_load_user_cert_and_key_pkcs12(certname, cert, stack, key, callback))
-      goto err;
+      if (!proxy_load_user_cert_and_key_pkcs12(certname, cert, stack, key, callback)) {
+          *stack = NULL;
+          goto err;
+      }
   }    
 
   return 1;
 
 err:
-  if (chain)
+  if (chain) {
     sk_X509_free(chain);
+  }
   if (cert) {
     X509_free(*cert);
     *cert = NULL;
