Index: voms-2.0.6/src/api/ccapi/voms_api.h
===================================================================
--- voms-2.0.6.orig/src/api/ccapi/voms_api.h	2011-07-22 05:17:52.000000003 -0500
+++ voms-2.0.6/src/api/ccapi/voms_api.h	2012-06-25 15:33:52.000000003 -0500
@@ -33,9 +33,15 @@
 #define NOGLOBUS
 extern "C" {
 #ifndef GSSAPI_H_
+/*
+ *  Also check against _GSSAPI_H_ as that is what the Kerberos 5 code defines and
+ *  what header files on some systems look for.
+ */
+#ifndef _GSSAPI_H_
 typedef void * gss_cred_id_t;
 typedef void * gss_ctx_id_t;
 #endif
+#endif
 
 #include <openssl/x509.h>
 #include <openssl/bio.h>
Index: voms-2.0.6/voms-1.9.19.2/src/api/ccapi/voms_api.h
===================================================================
--- voms-2.0.6.orig/voms-1.9.19.2/src/api/ccapi/voms_api.h	2010-03-30 07:55:20.000000003 -0500
+++ voms-2.0.6/voms-1.9.19.2/src/api/ccapi/voms_api.h	2012-06-25 15:34:22.000000003 -0500
@@ -34,9 +34,15 @@
 extern "C" {
 #ifdef NOGLOBUS
 #ifndef GSSAPI_H_
+/*
+ *  Also check against _GSSAPI_H_ as that is what the Kerberos 5 code defines and
+ *  what header files on some systems look for.
+ */
+#ifndef _GSSAPI_H_
 typedef void * gss_cred_id_t;
 typedef void * gss_ctx_id_t;
 #endif
+#endif
 #else
 #include "gssapi.h"
 #endif
