diff --git a/setup_mysql.py b/setup_mysql.py
index 3e17b84..53d407a 100755
--- a/setup_mysql.py
+++ b/setup_mysql.py
@@ -3,11 +3,13 @@
 # Setup prog for Panda Server
 #
 #
-from version import __version__, __provides__
-prefix = '/data/atlpan/srv'
+#from version import __version__, __provides__
+release_version='0.0.2'
+# prefix = '/data/atlpan/srv'
 # prefix = '/data/pansrv/srv'
 
 
+import os
 import re
 import sys
 import socket
@@ -52,7 +54,7 @@ class install_panda(install_org):
     def initialize_options (self):
         install_org.initialize_options(self)
 #        self.prefix = '/data/atlpan/srv'
-        self.prefix = prefix
+#        self.prefix = prefix
 
 
 # generates files using templates and install them
@@ -107,7 +109,7 @@ class install_data_panda (install_data_org):
                     raise RuntimeError,"%s doesn't have the .template extension" % srcFile
                 # dest filename
                 destFile = re.sub('(\.exe)*\.template$','',srcFile)
-                destFile = destFile.split('/')[-1]
+                destFile = re.sub(r'^templates/','',destFile)
                 destFile = '%s/%s' % (tmpDir,destFile)
                 # open src
                 inFile = open(srcFile)
@@ -121,6 +123,9 @@ class install_data_panda (install_data_org):
                         raise RuntimeError,'unknown pattern %s in %s' % (item,srcFile)
                     # get pattern
                     patt = getattr(self,item)
+                    # remove install root, if any
+                    if self.root is not None and patt.startswith(self.root):
+                            patt = patt[len(self.root):]
                     # remove build/*/dump for bdist
                     patt = re.sub('build/[^/]+/dumb','',patt)
                     # remove /var/tmp/*-buildroot for bdist_rpm
@@ -128,6 +133,10 @@ class install_data_panda (install_data_org):
                     # replace
                     filedata = filedata.replace('@@%s@@' % item, patt)
                 # write to dest
+                if '/' in destFile:
+                    destSubDir = os.path.dirname(destFile)
+                    if not os.path.exists(destSubDir):
+                        os.makedirs(destSubDir)
                 oFile = open(destFile,'w')
                 oFile.write(filedata)
                 oFile.close()
@@ -145,9 +154,10 @@ class install_data_panda (install_data_org):
         
 # setup for distutils
 setup(
-#    name="panda-server-mysql",
-    name=__provides__,
-    version=__version__,
+    name="panda-server-mysql",
+#   name=__provides__,
+#   version=__version__,
+    version=release_version,
     description='MySQL branch of the PanDA Server Package',
     long_description='''This package contains PanDA Server Components''',
     license='GPL',
@@ -166,41 +176,41 @@ setup(
               ],
     data_files=[
                 # config files 
-                ('etc/panda', ['templates/panda_server-httpd.conf.rpmnew.template',
-                               'templates/panda_server-httpd-FastCGI.conf.rpmnew.template',
-                               'templates/panda_server.cfg.rpmnew.template',
+                ('/etc/panda', ['templates/panda_server-httpd.conf.template',
+                               'templates/panda_server-httpd-FastCGI.conf.template',
+                               'templates/panda_server.cfg.template',
                                'templates/panda_server-grid-env.sh.template',
                                ]
                  ),
                 # sysconfig
-                ('etc/sysconfig', ['templates/panda_server-sysconfig.rpmnew.template',
+                ('/etc/sysconfig', ['templates/sysconfig/panda_server.template',
                                    ]
                  ),
                 # logrotate
-                ('etc/logrotate.d', ['templates/panda_server-logrotate.template',
+                ('/etc/logrotate.d', ['templates/logrotate.d/panda_server.template',
                                      ]
                  ),
                 # init script
-                ('etc/init.d', ['templates/panda_server-ctl.exe.template',
+                ('/etc/rc.d/init.d', ['templates/init.d/panda_server.exe.template',
                                    ]
                  ),
                 # crons
-                ('usr/bin', ['templates/panda_server-add.sh.exe.template',
-                             'templates/panda_server-priority.sh.exe.template',
-                             'templates/panda_server-copyArchive.sh.exe.template',
-                             'templates/panda_server-copyROOT.sh.exe.template',
-                             'templates/panda_server-vomsrenew.sh.exe.template',
-                             'templates/panda_server-archivelog.sh.exe.template',
-                             'templates/panda_server-tmpwatch.sh.exe.template',
-                             'templates/panda_server-backupJobArch.sh.exe.template',
-                             'templates/panda_server-deleteJobs.sh.exe.template',
-                             'templates/panda_server-merge.sh.exe.template',
-                             'templates/panda_server-datasetManager.sh.exe.template',
-                             'templates/panda_server-evpPD2P.sh.exe.template',
-                             'templates/panda_server-callback.sh.exe.template',
+                ('/usr/bin', ['templates/panda_server-add.exe.template',
+                             'templates/panda_server-priority.exe.template',
+                             'templates/panda_server-copyArchive.exe.template',
+                             'templates/panda_server-copyROOT.exe.template',
+                             'templates/panda_server-vomsrenew.exe.template',
+                             'templates/panda_server-archivelog.exe.template',
+                             'templates/panda_server-tmpwatch.exe.template',
+                             'templates/panda_server-backupJobArch.exe.template',
+                             'templates/panda_server-deleteJobs.exe.template',
+                             'templates/panda_server-merge.exe.template',
+                             'templates/panda_server-datasetManager.exe.template',
+                             'templates/panda_server-evpPD2P.exe.template',
+                             'templates/panda_server-callback.exe.template',
                              'templates/panda_server-makeSlsXml.exe.template',
-                             'templates/panda_server-boostUser.sh.exe.template',
-                             'templates/panda_server-runRebro.sh.exe.template',
+                             'templates/panda_server-boostUser.exe.template',
+                             'templates/panda_server-runRebro.exe.template',
                              ]
                  ),
                 # var dirs
