--- globus_gram_job_manager_setup_pbs-4.4/pbs.in	7 Aug 2007 16:23:01 -0000	1.18.6.10
+++ globus_gram_job_manager_setup_pbs-4.4/pbs.in	30 Jan 2008 21:44:06 -0000
@@ -62,6 +61,7 @@
     my $email_when = '';
     my $args;
     my $cache_pgm = "$Globus::Core::Paths::bindir/globus-gass-cache";
+    my @environment;
 
     $self->log("Entering pbs submit");
 
@@ -368,7 +357,36 @@
 
     $rsh_env = '';
 
-    foreach my $tuple ($description->environment())
+    @environment = $description->environment();
+
+    ##### OSG-Specific modification                 #####
+    ##### These should not affect any non-OSG site, #####
+    ##### unless you define $OSG_GRID               #####
+
+    # First, we figure out if this is an OSG installation, and if so, where 
+    # OSG is installed on the worker nodes
+    my $osg_grid = '';
+    my $use_osg_grid = 1;
+    my $use_dynamic_wn_tmp = 1;
+    map {
+        if ($_->[0] eq "OSG_GRID") {
+            $osg_grid =  $_->[1]; 
+        } elsif ($_->[0] eq "OSG_DONT_USE_OSG_GRID_FOR_GL") {
+            $use_osg_grid = 0;
+        }
+    } @environment;
+
+    # If this is an OSG installation, we set GLOBUS_LOCATION based on OSG_GRID.
+    if ($osg_grid ne '') {
+        map {
+            if ($use_osg_grid && $_->[0] eq "GLOBUS_LOCATION") { 
+                $_->[1] = $osg_grid . "/globus"; 
+            }
+        } @environment;
+    }
+    ##### End OSG-Specific modification             #####
+
+    foreach my $tuple (@environment())
     {
         if(!ref($tuple) || scalar(@$tuple) != 2)
         {
