--- src/c/Makefile.am.orig	2012-01-19 22:01:12.179774855 -0600
+++ src/c/Makefile.am	2012-01-19 22:11:53.146859539 -0600
@@ -38,7 +38,8 @@
 	-lglite-sd-c \
 	$(CGSI_GSOAP_LIBS) \
 	$(GLOBUS_GSS_NOTHR_LIBS) \
-	$(GLIB_LIBS)
+	$(GLIB_LIBS) \
+    -lgsoap
 
 libglite_data_transfer_api_simple_c_la_LDFLAGS = \
 	-version-info $(INTERFACE_LIBTOOL_CURRENT):$(INTERFACE_LIBTOOL_REVISION):$(INTERFACE_LIBTOOL_AGE)
@@ -51,7 +52,8 @@
 	libglite_data_transfer_api_simple_c.la \
 	libglite_data_transfer_channel_api_c.la \
 	-lglite_data_util \
-	$(GLIB_LIBS)
+	$(GLIB_LIBS) \
+    -lgsoap
 
 libglite_data_transfer_channel_api_simple_c_la_LDFLAGS = \
 	-version-info $(INTERFACE_LIBTOOL_CURRENT):$(INTERFACE_LIBTOOL_REVISION):$(INTERFACE_LIBTOOL_AGE)
@@ -65,7 +67,8 @@
 	libglite_data_transfer_fts_api_c.la \
 	-lglite_data_util \
 	$(GLIB_LIBS) \
-	-luuid
+	-luuid \
+    -lgsoap
 
 libglite_data_transfer_fts_api_simple_c_la_LDFLAGS = \
 	-version-info $(INTERFACE_LIBTOOL_CURRENT):$(INTERFACE_LIBTOOL_REVISION):$(INTERFACE_LIBTOOL_AGE)
@@ -78,16 +81,16 @@
 
 TRANSFER_INCLUDE_DIR=$(top_builddir)/interface/glite/data/transfer/c
 
-if USE_GSOAP_2_7_9
-STDSOAP2_C_FIX=patch -p0 -b stdsoap2.c $(top_srcdir)/src/c/stdsoap2.c.patch
-else
-STDSOAP2_C_FIX=@echo 'No patch needed for stdsoap2.c'
-endif
+#if USE_GSOAP_2_7_9
+#STDSOAP2_C_FIX=patch -p0 -b stdsoap2.c $(top_srcdir)/src/c/stdsoap2.c.patch
+#else
+#STDSOAP2_C_FIX=@echo 'No patch needed for stdsoap2.c'
+#endif
 
 # Copy gsoap file
-stdsoap2.c:
-	cp -f $(top_builddir)/$@ $@
-	$(STDSOAP2_C_FIX)
+#stdsoap2.c:
+#	cp -f $(top_builddir)/$@ $@
+#	$(STDSOAP2_C_FIX)
 
 # Common Files
 soapdefs.h:
@@ -171,7 +174,6 @@
 	ftsClientLib.c \
 	ftsServer.c \
 	ftsServerLib.c \
-	stdsoap2.c \
 	soapdefs.h \
 	env.c
 
@@ -197,7 +199,6 @@
 	sed -e s/@libname\@/$(basename $@)_`expr $(INTERFACE_LIBTOOL_CURRENT) - $(INTERFACE_LIBTOOL_AGE)`/ $< > $@
 
 libglite_data_transfer_channel_api_c_la_SOURCES = \
-	stdsoap2.c \
 	channelClientLib.c \
 	env.c 
 
@@ -206,13 +207,13 @@
 	-version-info $(INTERFACE_LIBTOOL_CURRENT):$(INTERFACE_LIBTOOL_REVISION):$(INTERFACE_LIBTOOL_AGE)
 
 libglite_data_transfer_fts_api_c_la_SOURCES = \
-	stdsoap2.c \
 	ftsClientLib.c \
 	env.c 
 
 libglite_data_transfer_fts_api_c_la_LDFLAGS = \
 	$(SYMVER_LDFLAGS) \
-	-version-info $(INTERFACE_LIBTOOL_CURRENT):$(INTERFACE_LIBTOOL_REVISION):$(INTERFACE_LIBTOOL_AGE)
+	-version-info $(INTERFACE_LIBTOOL_CURRENT):$(INTERFACE_LIBTOOL_REVISION):$(INTERFACE_LIBTOOL_AGE) \ 
+    -lgsoap
 
 AM_CFLAGS = -I$(top_builddir)/interface \
 	$(GLITE_CFLAGS) \
