From 988762b7728013c40521b2c79ba21a001b080821 Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Tue, 12 Jul 2016 16:58:55 -0500
Subject: [PATCH] Fix CMakeLists.txt to work with condor-classads

---
 src/CMakeLists.txt | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e5ebfed..dac2960 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -46,11 +46,10 @@ set(AUTODIR "autogen")
 find_package(PkgConfig REQUIRED)
 pkg_check_modules(GSOAP gsoapssl++)
 pkg_check_modules(LOG4CPP log4cpp)
 pkg_check_modules(VOMS voms-2.0)
 pkg_check_modules(GRIDSITE_OPENSSL gridsite-openssl)
-pkg_check_modules(CLASSAD emi-classad-ns)
 pkg_check_modules(GLOBUS_GSSAPI_GSI globus-gssapi-gsi)
 
 set(Boost_USE_MULTITHREADED ON)
 find_package( Boost COMPONENTS date_time filesystem regex thread)
 
@@ -71,11 +70,11 @@ add_definitions( -pipe -DWANT_NAMESPACES -DWANT_CLASSAD_NAMESPACE
 
 link_directories(${VOMS_LIBRARY_DIRS}
                  ${GSOAP_LIBRARY_DIRS}
                  ${GRIDSITE_OPENSSL_LIBRARY_DIRS})
 
-include_directories(${CLASSAD_INCLUDE_DIRS}
+include_directories(/usr/include/classad
                     ${VOMS_INCLUDE_DIRS}
                     ${GSOAP_INCLUDE_DIRS}
                     ${LOG4CPP_INCLUDE_DIRS}
                     ${Boost_INCLUDE_DIRS}
                     ${GLOBUS_GSSAPI_GSI_INCLUDE_DIRS}
@@ -131,11 +130,11 @@ add_library(glite_ce_cream_client_soap SHARED ${AUTODIR}/CREAM_CLIENTClient.cpp
                                               CreamProxy_AcceptNewJobSubmissions.cpp
                                               CreamProxy_QueryEvent.cpp)
 
 set_target_properties(glite_ce_cream_client_soap PROPERTIES VERSION 0.0.0 SOVERSION 0)
 
-target_link_libraries(glite_ce_cream_client_soap ${CLASSAD_LIBRARIES}
+target_link_libraries(glite_ce_cream_client_soap classad
                                                  ${VOMS_LIBRARIES}
                                                  ${GSOAP_LIBRARIES}
                                                  ${LOG4CPP_LIBRARIES}
                                                  ${GRIDSITE_OPENSSL_LIBRARIES}
                                                  ${Boost_LIBRARIES})
@@ -149,11 +148,11 @@ add_library(glite_ce_cream_client_util SHARED CEUrl.cpp
                                               scoped_timer.cpp
                                               VOMSWrapper.cpp)
 
 set_target_properties(glite_ce_cream_client_util PROPERTIES VERSION 0.0.0 SOVERSION 0)
 
-target_link_libraries(glite_ce_cream_client_util ${CLASSAD_LIBRARIES}
+target_link_libraries(glite_ce_cream_client_util classad
                                                  ${VOMS_LIBRARIES}
                                                  ${GSOAP_LIBRARIES}
                                                  ${LOG4CPP_LIBRARIES}
                                                  ${GRIDSITE_OPENSSL_LIBRARIES}
                                                  ${Boost_LIBRARIES})
-- 
2.6.3

