Index: xacml-1.0/Makefile.am
===================================================================
RCS file: /cvs/glite/org.glite.security.saml2-xacml2-c-lib/xacml-1.0/Makefile.am,v
retrieving revision 1.5
diff -u -r1.5 Makefile.am
--- xacml-1.0/Makefile.am	15 Oct 2008 18:00:01 -0000	1.5
+++ xacml-1.0/Makefile.am	3 Jan 2011 14:07:01 -0000
@@ -6,14 +6,14 @@
 SCHEMA=schema/xacml_authz_service.wsdl
 GSOAP_FILE = xacml_auth_service.gsoap 
 GSOAP_FILE_PATCHED = xacml_auth_service-patched.gsoap
-GSOAP_SOURCE_FILES= soapStub.h \
-                    soapH.h \
-                    soapC.cpp \
-                    soapClient.cpp \
-                    soapClientLib.cpp \
-                    soapServer.cpp \
-                    soapServerLib.cpp \
-                    XACMLAuthorizationPortTypeSOAPBinding.nsmap
+GSOAP_SOURCE_FILES= xacml_soapStub.h \
+                    xacml_soapH.h \
+                    xacml_soapC.cpp \
+                    xacml_soapClient.cpp \
+                    xacml_soapClientLib.cpp \
+                    xacml_soapServer.cpp \
+                    xacml_soapServerLib.cpp \
+                    xacml_soap.nsmap
 
 AM_CPPFLAGS=-DWITH_OPENSSL=1
 AM_CXXFLAGS=-g -Wall -O0
@@ -44,7 +44,7 @@
         xacml_datatypes.h \
         xacml_i.h \
         xacml_server.h
-nodist_libxacml_la_SOURCES = soapStub.h soapH.h soapC.cpp soapClient.cpp soapServer.cpp
+nodist_libxacml_la_SOURCES = xacml_soapStub.h xacml_soapH.h xacml_soapC.cpp xacml_soapClient.cpp xacml_soapServer.cpp
 libxacml_la_LIBADD = -lgsoap++ -lpthread -ldl
 
 xacml_server_SOURCES = xacml_server_example.c xacml_io_example.h xacml_io_example.c
@@ -58,7 +58,7 @@
 	$(WSDL2H) -a -x -f $(SCHEMA) -o $@
 
 $(GSOAP_SOURCE_FILES): $(GSOAP_FILE_PATCHED)
-	$(SOAPCPP2) -1 -w -x $(GSOAP_FILE_PATCHED) || true
+	$(SOAPCPP2) -p xacml_soap -n -1 -w -x $(GSOAP_FILE_PATCHED) || true
 
 $(GSOAP_FILE_PATCHED): $(GSOAP_FILE)
 	cp $(GSOAP_FILE) $(GSOAP_FILE_PATCHED)
Index: xacml-1.0/xacml_client.cpp
===================================================================
RCS file: /cvs/glite/org.glite.security.saml2-xacml2-c-lib/xacml-1.0/xacml_client.cpp,v
retrieving revision 1.8
diff -u -r1.8 xacml_client.cpp
--- xacml-1.0/xacml_client.cpp	29 Jun 2010 14:18:33 -0000	1.8
+++ xacml-1.0/xacml_client.cpp	3 Jan 2011 14:07:03 -0000
@@ -17,14 +17,16 @@
 #include "xacml_i.h"
 #include "xacml_client.h"
 #include "xacml_authz_interop_profile.h"
-#include "soapStub.h"
-#include "soapH.h"
+#include "xacml_soapStub.h"
+#include "xacml_soapH.h"
 
 #include <cassert>
 #include <ctime>
 #include <sstream>
 #include <iomanip>
 
+extern SOAP_NMAC struct Namespace xacml_soap_namespaces[];
+
 #ifndef DONT_DOCUMENT_INTERNAL
 namespace xacml
 {
@@ -449,6 +451,7 @@
     }
 
     soap_init(&soap);
+    soap.namespaces = xacml_soap_namespaces;
 
     query = soap_new_XACMLsamlp__XACMLAuthzDecisionQueryType(&soap, -1);
 
Index: xacml-1.0/xacml_i.h
===================================================================
RCS file: /cvs/glite/org.glite.security.saml2-xacml2-c-lib/xacml-1.0/xacml_i.h,v
retrieving revision 1.9
diff -u -r1.9 xacml_i.h
--- xacml-1.0/xacml_i.h	11 Aug 2009 08:36:49 -0000	1.9
+++ xacml-1.0/xacml_i.h	3 Jan 2011 14:07:03 -0000
@@ -19,7 +19,7 @@
 
 #ifndef DONT_DOCUMENT_INTERNAL
 #include "xacml.h"
-#include "soapStub.h"
+#include "xacml_soapStub.h"
 
 #include <pthread.h>
 
Index: xacml-1.0/xacml_io.cpp
===================================================================
RCS file: /cvs/glite/org.glite.security.saml2-xacml2-c-lib/xacml-1.0/xacml_io.cpp,v
retrieving revision 1.5
diff -u -r1.5 xacml_io.cpp
--- xacml-1.0/xacml_io.cpp	7 Nov 2008 09:45:10 -0000	1.5
+++ xacml-1.0/xacml_io.cpp	3 Jan 2011 14:07:03 -0000
@@ -15,12 +15,14 @@
  */
 
 #include "xacml_i.h"
-#include "soapH.h"
+#include "xacml_soapH.h"
 
 #include <iostream>
 #include <sstream>
 #include <dlfcn.h>
 
+extern SOAP_NMAC struct Namespace xacml_soap_namespaces[];
+
 /**
  * Load and use an I/O module from a shared object for a request
  * @ingroup xacml_io
@@ -145,10 +147,10 @@
      * the XACML message body
      */
     /* All namespaces */
-    for (int i = 0 ; namespaces[i].ns; i++)
+    for (int i = 0 ; xacml_soap_namespaces[i].ns; i++)
     {
-        namespace_prefixes << "xmlns:" << namespaces[i].id
-                           << "=\"" << namespaces[i].ns << "\" ";
+        namespace_prefixes << "xmlns:" << xacml_soap_namespaces[i].id
+                           << "=\"" << xacml_soap_namespaces[i].ns << "\" ";
     }
 
     /* Find the Content-Length HTTP header */
Index: xacml-1.0/xacml_ns.cpp
===================================================================
RCS file: /cvs/glite/org.glite.security.saml2-xacml2-c-lib/xacml-1.0/xacml_ns.cpp,v
retrieving revision 1.3
diff -u -r1.3 xacml_ns.cpp
--- xacml-1.0/xacml_ns.cpp	15 Oct 2008 18:00:01 -0000	1.3
+++ xacml-1.0/xacml_ns.cpp	3 Jan 2011 14:09:10 -0000
@@ -15,5 +15,15 @@
  */
 
 #ifndef DONT_DOCUMENT_INTERNAL
-#include "XACMLAuthorizationPortTypeSOAPBinding.nsmap"
+
+#include "xacml_soap.nsmap"
+
+// This doesn't actually do anything, it's just to avoid linking errors with
+// libgsoap++
+// The true namespaces map is auto-generated in xacml_soap.nsmap
+SOAP_NMAC struct Namespace namespaces[] =
+{
+        {NULL, NULL, NULL, NULL}
+};
+
 #endif
Index: xacml-1.0/xacml_server.cpp
===================================================================
RCS file: /cvs/glite/org.glite.security.saml2-xacml2-c-lib/xacml-1.0/xacml_server.cpp,v
retrieving revision 1.11
diff -u -r1.11 xacml_server.cpp
--- xacml-1.0/xacml_server.cpp	27 Aug 2010 10:43:27 -0000	1.11
+++ xacml-1.0/xacml_server.cpp	3 Jan 2011 14:07:03 -0000
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-#include "soapH.h"
+#include "xacml_soapH.h"
 
 #include "xacml_server.h"
 #include "xacml_i.h"
@@ -56,7 +56,7 @@
     client_soap -> send_timeout = 10; /* Time out of 10 seconds on send */
 
     /* printf ("Thread: soap_serve().\n"); */
-    soap_serve(client_soap);
+    xacml_soap_serve(client_soap);
 
     /* printf ("Thread: soap_destroy().\n"); */
     soap_destroy(client_soap);
