Index: OSGPKITools-1.1.0/osgpkitools/osg-cert-request
===================================================================
--- OSGPKITools-1.1.0.orig/osgpkitools/osg-cert-request	2013-02-06 13:15:48.000000000 -0600
+++ OSGPKITools-1.1.0/osgpkitools/osg-cert-request	2013-02-06 16:46:15.480460547 -0600
@@ -21,11 +21,11 @@
 from pprint import pprint
 
 from osgpkitools import OSGPKIUtils
-from OSGPKIUtils import CreateOIMConfig
-from OSGPKIUtils import check_response_500
-from OSGPKIUtils import charlimit_textwrap
-from OSGPKIUtils import check_failed_response
-from ExceptionDefinitions import *
+from osgpkitools.OSGPKIUtils import CreateOIMConfig
+from osgpkitools.OSGPKIUtils import check_response_500
+from osgpkitools.OSGPKIUtils import charlimit_textwrap
+from osgpkitools.OSGPKIUtils import check_failed_response
+from osgpkitools.ExceptionDefinitions import *
 
 
 # Set up Option Parser
Index: OSGPKITools-1.1.0/osgpkitools/osg-cert-retrieve
===================================================================
--- OSGPKITools-1.1.0.orig/osgpkitools/osg-cert-retrieve	2013-02-06 13:15:48.000000000 -0600
+++ OSGPKITools-1.1.0/osgpkitools/osg-cert-retrieve	2013-02-06 16:46:35.130461254 -0600
@@ -31,16 +31,16 @@
 import ssl
 import tempfile
 
-from OSGPKIUtils import charlimit_textwrap
 from osgpkitools import OSGPKIUtils
-from OSGPKIUtils import CreateOIMConfig
-from OSGPKIUtils import extractEEC
-from OSGPKIUtils import extractHostname
-from OSGPKIUtils import check_failed_response
-from OSGPKIUtils import check_response_500
-from OSGPKIUtils import print_failure_reason_exit
-from OSGPKIUtils import check_for_pending
-from ExceptionDefinitions import *
+from osgpkitools.OSGPKIUtils import charlimit_textwrap
+from osgpkitools.OSGPKIUtils import CreateOIMConfig
+from osgpkitools.OSGPKIUtils import extractEEC
+from osgpkitools.OSGPKIUtils import extractHostname
+from osgpkitools.OSGPKIUtils import check_failed_response
+from osgpkitools.OSGPKIUtils import check_response_500
+from osgpkitools.OSGPKIUtils import print_failure_reason_exit
+from osgpkitools.OSGPKIUtils import check_for_pending
+from osgpkitools.ExceptionDefinitions import *
 
 
 # Set up Option Parser
Index: OSGPKITools-1.1.0/osgpkitools/osg-gridadmin-cert-request
===================================================================
--- OSGPKITools-1.1.0.orig/osgpkitools/osg-gridadmin-cert-request	2013-02-06 13:15:48.000000000 -0600
+++ OSGPKITools-1.1.0/osgpkitools/osg-gridadmin-cert-request	2013-02-06 16:46:53.931018828 -0600
@@ -36,16 +36,16 @@
 import tempfile
 
 from osgpkitools import OSGPKIUtils
-from OSGPKIUtils import CreateOIMConfig
-from OSGPKIUtils import extractHostname
-from OSGPKIUtils import extractEEC
-from OSGPKIUtils import get_request_count
-from OSGPKIUtils import check_response_500
-from OSGPKIUtils import check_failed_response
-from OSGPKIUtils import print_failure_reason_exit
-from OSGPKIUtils import check_for_pending
-from OSGPKIUtils import charlimit_textwrap
-from ExceptionDefinitions import *
+from osgpkitools.OSGPKIUtils import CreateOIMConfig
+from osgpkitools.OSGPKIUtils import extractHostname
+from osgpkitools.OSGPKIUtils import extractEEC
+from osgpkitools.OSGPKIUtils import get_request_count
+from osgpkitools.OSGPKIUtils import check_response_500
+from osgpkitools.OSGPKIUtils import check_failed_response
+from osgpkitools.OSGPKIUtils import print_failure_reason_exit
+from osgpkitools.OSGPKIUtils import check_for_pending
+from osgpkitools.OSGPKIUtils import charlimit_textwrap
+from osgpkitools.ExceptionDefinitions import *
 
 
 # Set up Option Parser
