--- globus_gass_cache.c.~1~	2011-03-03 04:35:05.000000000 -0600
+++ globus_gass_cache.c	2011-09-02 16:11:24.000000000 -0500
@@ -1471,14 +1471,12 @@
 
     if ( GLOBUS_SUCCESS == rc )
     {
-        int skip = 0;
+        int skip = strlen(names->local_root) + strlen(names->separator);
 
         assert(mangle_prefix_end != NULL);
 
         names->local_mangle_root = globus_libc_malloc(
-                strlen(names->local_root) +
-                strlen(names->separator) +
-                mangle_prefix_len + 1);
+                skip + mangle_prefix_len + 1);
 
         if (names->local_mangle_root == NULL)
         {
@@ -1487,8 +1485,8 @@
         }
         else
         {
-            sprintf(names->local_mangle_root, "%s%s%n",
-                    names->local_root, names->separator, &skip);
+            sprintf(names->local_mangle_root, "%s%s",
+                    names->local_root, names->separator);
             memcpy(&names->local_mangle_root[skip],
                    mangle_prefix,
                    mangle_prefix_len);
@@ -1497,14 +1495,12 @@
     }
     if ( GLOBUS_SUCCESS == rc )
     {
-        int skip = 0;
+        int skip = strlen(names->global_root) + strlen(names->separator);
 
         assert(mangle_prefix_end != NULL);
 
         names->global_mangle_root = globus_libc_malloc(
-                strlen(names->global_root) +
-                strlen(names->separator) +
-                mangle_prefix_len + 1);
+                skip + mangle_prefix_len + 1);
 
         if (names->global_mangle_root == NULL)
         {
@@ -1513,8 +1509,8 @@
         }
         else
         {
-            sprintf(names->global_mangle_root, "%s%s%n",
-                    names->global_root, names->separator, &skip);
+            sprintf(names->global_mangle_root, "%s%s",
+                    names->global_root, names->separator);
             memcpy(&names->global_mangle_root[skip],
                    mangle_prefix,
                    mangle_prefix_len);
