Index: singularity-2.3.1/etc/singularity.conf
===================================================================
--- singularity-2.3.1.orig/etc/singularity.conf
+++ singularity-2.3.1/etc/singularity.conf
@@ -123,7 +123,7 @@ user bind control = yes
 # DEFAULT: yes
 # Enabling this option will make it possible to specify bind paths to locations
 # that do not currently exist within the container.
-enable overlay = yes
+enable overlay = no
 
 
 # MOUNT SLAVE: [BOOL]
