Index: osg-info-services
===================================================================
--- osg-info-services.orig	2013-10-09 17:48:42.000000002 -0500
+++ osg-info-services	2013-10-09 17:51:36.000000002 -0500
@@ -62,8 +62,14 @@
     if not dir_can_read_write(temp_dir, gip_uid, gip_gid):
         os.chown(temp_dir, gip_uid, gip_gid)
 
+# fix_ownership and osg_info_wrapper both call gip_common.config() which uses
+# OptionParser and eats command-line options. Therefore we need to save
+# sys.argv and restore it.
+save_argv = sys.argv
+sys.argv = []
 fix_ownership()
 import osg_info_wrapper
+sys.argv = save_argv
 import gip_common
 import gip_ldap
  
