Index: osg-info-services
===================================================================
--- osg-info-services.orig	2013-08-21 08:15:02.000000002 -0500
+++ osg-info-services	2013-10-07 16:05:23.000000002 -0500
@@ -435,7 +435,7 @@
                 if isinstance(val, types.IntType):
                     out.append('%s = %i;' % (key, val))
                 elif val in ("TRUE", "FALSE"):
-                    out.append("%s = %s" % (key, val))
+                    out.append("%s = %s;" % (key, val))
                 else:
                     out.append('%s = "%s";' % (key, str(val)))
             message = "[\n" + "\n        ".join(out) + "\n\n]"
