--- osg-info-services	(revision 4630)
+++ osg-info-services	(working copy)
@@ -1,5 +1,8 @@
 #!/usr/bin/python
 
+import warnings
+warnings.filterwarnings("ignore", category=DeprecationWarning)
+
 import os
 import re
 import pwd
@@ -432,7 +437,7 @@
                 if isinstance(val, types.IntType):
                     out.append('%s = %i;' % (key, val))
                 elif val in ("TRUE", "FALSE"):
-                    out.append("%s = %s" % (key, val))
+                    out.append("%s = %s;" % (key, val))
                 else:
                     out.append('%s = "%s";' % (key, str(val)))
             message = "[\n" + "\n        ".join(out) + "\n\n]"
