From 3ca190eaca768341ad084a2e81026f88b2e7c96c Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Tue, 4 Dec 2018 17:19:34 -0600
Subject: [PATCH] Fix last resort cache

---
 bin/stashcp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/stashcp b/bin/stashcp
index 939c1e7..4b34abf 100755
--- a/bin/stashcp
+++ b/bin/stashcp
@@ -434,7 +434,7 @@ def get_best_stashcache():
         
     if order_str == '':
         # Unable to find a geo_ip server to use, return random choice from caches!
-        minsite = random.choice(caches_list)
+        minsite = random.choice(caches_list)['name']
         logging.error("Unable to use Geoip to find closest cache!  Returning random cache %s", minsite)
         return minsite
     else:
-- 
2.6.3

