From 47c217c541e9c245660d3deef6bd5e4ae6e9bfe3 Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Tue, 23 Oct 2018 19:28:12 -0500
Subject: [PATCH] Make file size query failure not fatal

---
 bin/stashcp | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/bin/stashcp b/bin/stashcp
index 5c67814..939c1e7 100755
--- a/bin/stashcp
+++ b/bin/stashcp
@@ -177,16 +177,15 @@ def doStashCpSingle(sourceFile, destination, debug=False):
     xrdcp_version = subprocess.Popen(['echo $(xrdcp -V 2>&1)'], stdout=subprocess.PIPE, shell=True).communicate()[0][:-1]
     try:
         fileSize = int(re.findall(r"Size:   \d+", xrdfs_stdout)[0].split(":   ")[1])
-    except Exception as e:
-        sys.stderr.write("Unable to find size of file\n")
+        logging.debug("Size of the file %s is %s", sourceFile, fileSize)
+        payload['filesize'] = fileSize
+    except (ValueError, IndexError) as e:
+        sys.stderr.write("Unable to find size of file from the origin\n")
         print str(xrdfs_stdout)
         sys.stderr.write(str(xrdfs_stderr))
         sys.stderr.write("\n")
-        return 1
-    logging.debug("Size of the file %s is %s", sourceFile, str(fileSize))
     
     payload['xrdcp_version'] = xrdcp_version
-    payload['filesize'] = fileSize
     
     end1=int(time.time()*1000)
     payload['end1']=end1
-- 
2.6.3

