--- lcmaps_scas_client.c	2011-07-31 14:22:16.000000000 -0500
+++ src/scas-client/lcmaps_scas_client.c	2011-08-01 09:39:39.000000000 -0500
@@ -831,6 +831,11 @@
     lcmaps_log_debug (5, "%s: The number of certificates on the stack is: %d\n", logstr, depth);
     lcmaps_log_debug (5, "%s: The amount of CA certificates is: %d\n", logstr, amount_of_CAs);
 
+    if (amount_of_CAs == depth) {
+        lcmaps_log (1, "%s: No non-CA certificates in the chain.\n", logstr);
+        goto fail_scas_client;
+    }
+
     user_dn   = X509_NAME_oneline(X509_get_subject_name(sk_X509_value(myChain, depth - 1 - amount_of_CAs)), NULL, 0);
     lcmaps_log_debug (5,"%s: User DN is : %s\n", logstr, user_dn);
 
