Index: src/saml2-xacml2/io_handler/xacml_io_ssl.c
===================================================================
--- src/saml2-xacml2/io_handler/xacml_io_ssl.c	(revision 18528)
+++ src/saml2-xacml2/io_handler/xacml_io_ssl.c	(working copy)
@@ -706,6 +706,7 @@
 #ifdef OVERRIDE_API_HOST_CHECK
     char            *override_hostname         = NULL;
 #endif
+    SSL_CTX * ctx = NULL;
     ssl_io_state_t  *state = NULL;
 
     long             err     = 0;
@@ -833,12 +834,16 @@
         XACML_LOG_FUNC(LOG_INFO, "%s: Is connected to host '%s' on port '%d'\n", __func__, host, port);
 
         /* SSL */
-	state->ssl = xacml_io_SSL_client_connect(
-		state -> socket,
-                xacml_io_setup_client_ctx(NULL,
-					  xacml_io_certCAPath,
-					  xacml_io_certFilePub,
-					  xacml_io_certFilePriv));
+	ctx = xacml_io_setup_client_ctx(NULL,
+					xacml_io_certCAPath,
+					xacml_io_certFilePub,
+					xacml_io_certFilePriv);
+	state->ssl = xacml_io_SSL_client_connect( state -> socket, ctx);
+	/* Decrease reference counter for ctx again: either SSL_new added one,
+	 * or we should fail */
+	if (ctx)
+	    SSL_CTX_free(ctx);
+
         if (state->ssl == NULL)
         {
             XACML_LOG_FUNC(LOG_ERR, "%s: Failure in SSL layer setup and connection!\n", __func__);
