Index: condor-cron-1.0.9/etc/condor_config
===================================================================
--- condor-cron-1.0.9.orig/etc/condor_config	2012-07-05 10:32:12.000000002 -0500
+++ condor-cron-1.0.9/etc/condor_config	2013-07-16 18:18:44.000000002 -0500
@@ -2,6 +2,9 @@
 # If you wish to make changes to the condor-cron configuration, create files
 # in /etc/condor-cron/config.d containing your changes.
 
+# All settings defined here may be overridden by settings defined in files in
+# /etc/condor-cron/config.d
+
 # The CONDOR_HOST is intentionally left blank to prevent conflicts with a real
 # condor install that might be running on the machine.
 CONDOR_HOST =
@@ -43,7 +46,8 @@
 
 # The UID.GID of the user to run Condor-Cron.  We have to set this
 # at install time since we create the cndrcron user
-CONDOR_IDS=100.101
+# This value will be overridden by /etc/condor-cron/config.d/condor_ids
+#CONDOR_IDS=100.101
 
 # CA certificate directory for Condor-G
 GSI_DAEMON_TRUSTED_CA_DIR = /etc/grid-security/certificates
