--- gsissh.spec.orig	2011-07-07 13:33:57.000000000 -0500
+++ gsissh.spec	2011-07-07 13:34:50.000000000 -0500
@@ -98,7 +98,8 @@
 
 # This is the patch that adds GSI support
 # Based on http://grid.ncsa.illinois.edu/ssh/dl/patch/openssh-4.3p2.patch
-Patch99: openssh-4.3p2-gsissh.patch
+Patch98: openssh-4.3p2-gsissh.patch
+Patch99: sigchld.patch
 
 License: BSD
 Group: Applications/Internet
@@ -251,7 +252,8 @@
 %patch71 -p1 -b .engine
 %patch73 -p1 -b .biguid
 %patch74 -p1 -b .cryptoaudit
-%patch99 -p1 -b .gsi
+%patch98 -p1 -b .gsi
+%patch99 -p0
 
 sed 's/sshd.pid/gsisshd.pid/' -i pathnames.h
 sed 's!$(piddir)/sshd.pid!$(piddir)/gsisshd.pid!' -i Makefile.in
