From 7c303dabf248883954b8b32e5a85f0e303eaf561 Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Tue, 4 Aug 2015 14:29:47 -0500
Subject: [PATCH] Change default dvers of upcoming routes to el6 and el7

---
 data/promoter.ini | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/data/promoter.ini b/data/promoter.ini
index 3dbff70..67e6f55 100644
--- a/data/promoter.ini
+++ b/data/promoter.ini
@@ -70,19 +70,19 @@ extra_dvers=
 
 [route upcoming]
 from=osg-upcoming-%s-development
 to=osg-upcoming-%s-testing
 repotag=osgup
-dvers=el5 el6
-extra_dvers=el7
+dvers=el6 el7
+extra_dvers=el5
 
 [route upcoming-prerelease]
 from=osg-upcoming-%s-testing
 to=osg-upcoming-%s-prerelease
 repotag=osgup
-dvers=el5 el6
-extra_dvers=el7
+dvers=el6 el7
+extra_dvers=el5
 
 [route goc-el5-itb]
 from=osg-3.2-%s-development
 to=goc-%s-itb
 repotag=osg32
-- 
2.4.6

