--- configure.ac.orig	2011-09-05 21:22:39.000000000 -0500
+++ configure.ac	2011-09-05 22:23:15.000000000 -0500
@@ -1,7 +1,7 @@
 dnl
 dnl  University of Illinois/NCSA Open Source License
 dnl
-dnl  Copyright © 2003-2010, NCSA.  All rights reserved.
+dnl  Copyright Â© 2003-2008, NCSA.  All rights reserved.
 dnl
 dnl  Developed by:
 dnl
@@ -43,14 +43,14 @@
 # First [] is used as the prompt
 # Last [] is used for the version information
 AC_INIT([UberFTP], [2.6], [Jason Alt <jalt@ncsa.uiuc.edu>],[uberftp-client])
-#AC_INIT([ftp], [2.8], [Jason Alt <jalt@ncsa.uiuc.edu>],[mssftp-client])
+#AC_INIT([ftp], [2.7], [Jason Alt <jalt@ncsa.uiuc.edu>],[mssftp-client])
 AM_INIT_AUTOMAKE 
 
 GPT_MAJOR="2"
 GPT_MINOR="6"
 AC_SUBST(GPT_MAJOR)
 AC_SUBST(GPT_MINOR)
-AC_OUTPUT(pkg_data_src.gpt)
+AC_OUTPUT(pkgdata/pkg_data_src.gpt)
 
 # Check which c compiler we are using
 AC_PROG_CC
@@ -100,66 +100,6 @@
 	linktype="static"
 fi
 
-# Check whether the user has a Globus flavor type
-globus_flavor="no"
-AC_ARG_WITH(globus-flavor,
-	[  --with-globus-flavor=TYPE       Describe Globus flavor type (ex: gcc32dbg)],
-	[ globus_flavor="$withval" ]
-)
-
-if test "x$globus_flavor" = "xyes" ; then
-	AC_MSG_ERROR(--with-globus-flavor=TYPE must specify a flavor)
-fi
-if test "x$globus_flavor" = "xno" ; then
-	AC_MSG_ERROR(--with-globus-flavor=TYPE must specify a flavor)
-fi
-
-# Find GLOBUS_LOCATION
-globus_location="no"
-AC_ARG_WITH(globus,
-	[  --with-globus=PATH      Location of Globus Installation],
-	[
-		globus_location="$withval"
-	]
-)
-
-if test "x$globus_location" = "xyes" ; then
-	AC_MSG_ERROR(--with-globus=PATH must specify a path)
-elif test "x$globus_location" = "xno" ; then
-	if test "x$GLOBUS_LOCATION" = "x" ; then
-		AC_MSG_ERROR(--with-globus=PATH must specify a path)
-	else
-		globus_location="$GLOBUS_LOCATION"
-	fi
-fi
-
-if test ! -d "$globus_location" ; then
-	AC_MSG_ERROR(--with-globus=PATH must specify a valid path)
-fi
-
-
-# Find GPT_LOCATION
-gpt_location="no"
-AC_ARG_WITH(gpt,
-	[  --with-gpt=PATH         Location of GPT Installation],
-	[
-		gpt_location="$withval"
-	]
-)
-if test "x$gpt_location" = "xyes" ; then
-	AC_MSG_ERROR(--with-gpt=PATH must specify a path)
-elif test "x$gpt_location" = "xno" ; then
-	if test -x $globus_location/sbin/gpt-build ; then
-		gpt_location=$globus_location
-	elif test "x$GPT_LOCATION" != "x" ; then
-		gpt_location=$GPT_LOCATION
-	fi
-fi
-
-if test ! -d "$gpt_location" ; then
-	AC_MSG_ERROR(--with-gpt=PATH must specify a valid path $gpt_location)
-fi
-
 # Default TPC Window Sizing
 AC_ARG_WITH(tcp-window,
 	[  --with-tcp-window=value      Default value for TCP Windows ],
@@ -176,36 +116,8 @@
 )
 AC_DEFINE_UNQUOTED(DEFAULT_BLKSIZE, $default_blksize, [ Setting the default buffer size ])
 
-# Now determine the link flags
-AC_MSG_CHECKING(for GPT linkline helper)
-if test -x $gpt_location/sbin/gpt_build_config ; then
-	gpt_linkline_helper="$gpt_location/sbin/gpt_build_config"
-else
-	AC_MSG_ERROR(Cannot find gpt_build_config: GPT installation is incomplete)
-fi
-AC_MSG_RESULT(yes)
-
-#
-# Build Globus linkline
-#
-GLOBUS_LOCATION=$globus_location
-GLOBUS_FLAVOR_NAME=${globus_flavor}
-export GLOBUS_LOCATION
-export GLOBUS_FLAVOR_NAME
-${gpt_linkline_helper} -f ${globus_flavor} -link ${linktype} -src pkg_data_src.gpt
-
-. ./gpt_build_temp.sh
-rm -f ./gpt_build_temp.sh
-
-AC_MSG_CHECKING(for Globus linkline)
-if test -z "$GPT_CONFIG_PGM_LINKS" ; then
-	AC_MSG_ERROR(gpt_build_config failed)
-fi
-AC_MSG_RESULT(yes)
-
-
-GLOBUS_CFLAGS="$GPT_CONFIG_CFLAGS -I$globus_location/include/$globus_flavor"
-GLOBUS_LIBS="$GPT_CONFIG_PGM_LINKS $GPT_CONFIG_LIBS"
+GLOBUS_CFLAGS="-I/usr/lib64/globus/include -I/usr/lib/globus/include -I/usr/include/globus"
+GLOBUS_LIBS="-lglobus_gssapi_gsi"
 GLOBUS_LDFLAGS="-L$globus_location/lib"
 
 AC_SUBST(GLOBUS_CFLAGS)
