diff --git a/CMakeLists.txt b/CMakeLists.txt
index 55e607b..d795fe8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,9 +17,9 @@ if( CMAKE_COMPILER_IS_GNUCC )
   set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror" )
 endif()
 
-SET( CMAKE_SHARED_LINKER_FLAGS "--no-undefined")
-SET( CMAKE_MODULE_LINKER_FLAGS "--no-undefined")
-SET( CMAKE_EXE_LINKER_FLAGS "--no-undefined")
+SET( CMAKE_SHARED_LINKER_FLAGS "-Wl,--no-undefined")
+SET( CMAKE_MODULE_LINKER_FLAGS "-Wl,--no-undefined")
+SET( CMAKE_EXE_LINKER_FLAGS "-Wl,--no-undefined")
 
 include_directories("${HDFS_INCLUDES}" "${JVM_INCLUDES}" "${JVM_MD_INCLUDES}")
 
@@ -57,17 +57,17 @@ target_link_libraries(XrdHdfsReal ${HDFS_LIB} ${XROOTD_UTILS})
 
 add_executable(xrootd_hdfs_envcheck src/XrdHdfsEnvCheck.cc)
 
-if (NOT DEFINED CMAKE_INSTALL_LIBDIR)
-  SET(CMAKE_INSTALL_LIBDIR "lib")
+if (NOT DEFINED LIB_INSTALL_DIR)
+  SET(LIB_INSTALL_DIR "lib")
 endif()
 
 install(
   TARGETS XrdHdfs
-  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} )
+  LIBRARY DESTINATION ${LIB_INSTALL_DIR} )
 
 install(
   TARGETS XrdHdfsReal
-  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} )
+  LIBRARY DESTINATION ${LIB_INSTALL_DIR} )
 
 install(
   PROGRAMS xrootd_hdfs_envcheck
diff --git a/src/XrdHdfs.cc b/src/XrdHdfs.cc
index d311f6f..f5f49ff 100644
--- a/src/XrdHdfs.cc
+++ b/src/XrdHdfs.cc
@@ -338,7 +338,6 @@ int XrdHdfsFile::Open(const char               *path,      // In
    static const char *epname = "open";
 #endif
    //const int AMode = S_IRWXU|S_IRWXG|S_IROTH|S_IXOTH; // 775
-   char *opname;
    int open_flag = 0;
 
 // Verify that this object is not already associated with an open file
@@ -406,12 +405,9 @@ int XrdHdfsFile::Open(const char               *path,      // In
 */
 // Prepare to create or open the file, as needed
 //
-   if (openMode & O_CREAT) {
-       opname = (char *)"create";
-   } else if (openMode & O_TRUNC) {
-         open_flag  = O_TRUNC;
-                 opname = (char *)"truncate";
-      } else opname = (char *)"open";
+   if (openMode & O_TRUNC) {
+       open_flag = O_TRUNC;
+   }
 
 // Setup a new filesystem instance.
    if (!Connect(client))
