--- /usr/src/debug/globus_gass_transfer-4.3/library/globus_gass_transfer_server.c	2009-01-20 16:39:11.000000000 -0600
+++ library/globus_gass_transfer_server.c	2011-08-02 16:40:00.000000000 -0500
@@ -704,6 +704,7 @@
     globus_i_gass_transfer_lock();
     req = globus_handle_table_lookup(&globus_i_gass_transfer_request_handles,
 				     request);
+    globus_i_gass_transfer_unlock();
 
     if(req == GLOBUS_NULL)
     {
@@ -742,7 +743,9 @@
 	/* delete our reference to this request and proto */
 	req->proto->destroy(req->proto,
 			    request);
+        globus_i_gass_transfer_lock();
 	globus_i_gass_transfer_request_destroy(request);
+        globus_i_gass_transfer_unlock();
 	break;
 
       case GLOBUS_GASS_TRANSFER_REQUEST_INVALID:
@@ -768,11 +771,9 @@
 	goto error_exit;
     }
 
-    globus_i_gass_transfer_unlock();
     return GLOBUS_SUCCESS;
 
   error_exit:
-    globus_i_gass_transfer_unlock();
     return rc;
 }
 /* globus_gass_transfer_authorize() */
