--- gridftp/src/main/java/org/globus/ftp/FTPClient.java.orig	2013-02-22 10:35:23.053674988 -0600
+++ gridftp/src/main/java/org/globus/ftp/FTPClient.java	2013-02-22 10:42:12.345737579 -0600
@@ -2063,7 +2063,7 @@
      * {@link http://www.ogf.org/documents/GFD.47.pdf [GridFTP v2 Protocol Description]}
      * checksum feature has the following syntax:
      * <pre>
-     * CKSUM <algorithm>[, …]
+     * CKSUM <algorithm>[, ]
      * </pre>
      * getSupportedCksumAlgorithms parses checsum feauture parms and form a
      * list of checksum algorithms supported by the server
@@ -2132,7 +2132,7 @@
      * whole file existing on the server. The syntax is:
      * CKSM <algorithm> <offset> <length> <path> CRLF
      * Server executes this command by calculating specified type of checksum over
-     * portion of the file starting at the offset and of the specified length. If length is –1,
+     * portion of the file starting at the offset and of the specified length. If length is 1,
      * the checksum will be calculated through the end of the file. On success, the server
      * replies with
      * 2xx <checksum value>
