--- bestman.server.in	2012-01-13 14:52:52.000000000 -0600
+++ bestman.server.in	2012-04-30 14:04:43.000000000 -0500
@@ -170,6 +170,12 @@
 }
 JAVAEXE=$JAVA_HOME/bin/java
 JVR=`$JAVAEXE -version 2>&1`
+RTN=$?
+if [ $RTN -ne 0 ]; then
+    echo "Executing Java version check failed with error:"
+    echo $JVR
+    exit 1
+fi
 JVR=`echo $JVR | sed 's/[^0-9. ]//g'`
 JVR=`echo $JVR | sed 's/\./ /g'`
 Ver_separator $JVR
