--- bestman2/sources/server/src/gov/lbl/srm/server/TSupportedURLWithFILE.java	2011-12-16 11:27:13.000000000 -0600
+++ bestman2/sources/server/src/gov/lbl/srm/server/TSupportedURLWithFILE.java	2012-01-31 13:18:58.000000000 -0600
@@ -1309,10 +1309,11 @@
 
     public boolean exists(File f) {	
 	//return f.exists();
-	/*if (f.exists()) {
+	if ((_uid==null)&&(f.exists())) {
+	    TSRMLog.warning(this.getClass(), null, "event=sudoNoUidSpecified", "No uid for sudo, using current user");
 	    return true;
 	}
-	*/
+	
 	TSRMLog.debug(this.getClass(), null, "event=checkExistsSudoStarts", null);
 	String path  = f.getPath();
 
