diff --git a/bestman2/setup-osg/bestman.in/bestmanclient.conf.in b/bestman2/setup-osg/bestman.in/bestmanclient.conf.in
index 6c5325d..09fb1b7 100644
--- a/bestman2/setup-osg/bestman.in/bestmanclient.conf.in
+++ b/bestman2/setup-osg/bestman.in/bestmanclient.conf.in
@@ -49,7 +49,14 @@ ServiceHandle=/srm/v2/server
 ###########################################################
 #BufferSize=1048576
 #dcau=true
-#Parallelism=1
+Parallelism=0
+###########################################################
+#
+# Use external globus-url-copy by default.
+# Comment out to use internal implementation.
+#
+###########################################################
+GUCPath=/usr/bin/globus-url-copy
 ###########################################################
 #
 # User personal GSI related settings
