diff --git a/sources/server/build.xml b/sources/server/build.xml
index d5fcd07..0b3adef 100644
--- a/sources/server/build.xml
+++ b/sources/server/build.xml
@@ -89,6 +89,8 @@
     <fileset file="${log4j.path}"/>
     <fileset file="${bcprov-jdk15.path}"/>
     <fileset file="${vomsjapi.path}"/>
+    <fileset file="${voms-api-java.path}"/>
+    <fileset file="${canl.path}"/>
     <fileset file="${jcl-over-slf4j.path}"/>
     <fileset file="${je-4.1.10.path}"/>
     <fileset file="${servlet-api.path}"/>
diff --git a/sources/server/src/gov/lbl/srm/util/GUMSXACMLClient.java b/sources/server/src/gov/lbl/srm/util/GUMSXACMLClient.java
index 185bba1..5798953 100644
--- a/sources/server/src/gov/lbl/srm/util/GUMSXACMLClient.java
+++ b/sources/server/src/gov/lbl/srm/util/GUMSXACMLClient.java
@@ -41,7 +41,14 @@ package gov.lbl.srm.util;
 import java.util.*;
 import java.io.*;
 
-import org.glite.voms.VOMSValidator;
+import org.italiangrid.voms.VOMSAttribute;
+import org.italiangrid.voms.VOMSValidators;
+import org.italiangrid.voms.ac.VOMSACValidator;
+import org.italiangrid.voms.ac.impl.DefaultVOMSValidator;
+import org.italiangrid.voms.store.impl.DefaultUpdatingVOMSTrustStore;
+import org.italiangrid.voms.util.CertificateValidatorBuilder;
+
+import eu.emi.security.authn.x509.X509CertChainValidatorExt;
 
 //  import gov.bnl.gums.command.Configuration;
 //import gov.bnl.gums.admin.*;
@@ -56,11 +63,11 @@ import java.net.URL;
 import java.security.cert.X509Certificate;
 
 class ValidatorControl {
-    org.glite.voms.VOMSValidator _vv = null;
+    VOMSACValidator _vv = null;
     TSRMMutex _vvGuard = new TSRMMutex();
     
     public ValidatorControl() {
-	Class clazz = org.glite.voms.VOMSValidator.class;
+	Class clazz = VOMSACValidator.class;
 	String result0 =org.theshoemakers.which4j.Which4J.which(clazz);
 	System.out.println("which vomsvalidator?"+result0);
     }
@@ -73,45 +80,48 @@ class ValidatorControl {
 	List result = null;
 	try {
 	    if (_vv == null) {
-		//_vv = new org.glite.voms.VOMSValidator(null, null);
-		org.glite.voms.PKIStore vomsStore = null;
+		//_vv = VOMSValidators.newValidator();
+		DefaultUpdatingVOMSTrustStore vomsStore = null;
 		String vomsDir = System.getProperty( "VOMSDIR" );
-		vomsDir = (vomsDir == null ) ? org.glite.voms.PKIStore.DEFAULT_VOMSDIR : vomsDir;
+		vomsDir = (vomsDir == null ) ? DefaultUpdatingVOMSTrustStore.DEFAULT_VOMS_DIR : vomsDir;
 		TSRMLog.info(ValidatorControl.class, null, "vomsDir="+vomsDir, null);
 		
 		File theDir = new File(vomsDir);
 		if (theDir.exists() && theDir.isDirectory() && theDir.list().length > 0) {
-		    vomsStore = new org.glite.voms.PKIStore(vomsDir, org.glite.voms.PKIStore.TYPE_VOMSDIR, true);
-		    vomsStore.rescheduleRefresh(900000);
+		    List<String> vomsDirs = new ArrayList<String>();
+		    vomsDirs.add(vomsDir);
+		    vomsStore = new DefaultUpdatingVOMSTrustStore(vomsDirs, 900000);
 		}
 		
-		org.glite.voms.PKIStore caStore;
 		String caDir = System.getProperty( "CADIR" );
-		caDir = (caDir == null) ? org.glite.voms.PKIStore.DEFAULT_CADIR : caDir;
+		caDir = (caDir == null) ? DefaultVOMSValidator.DEFAULT_TRUST_ANCHORS_DIR : caDir;
 		TSRMLog.info(ValidatorControl.class, null, "caDir="+caDir, null);
-		caStore = new org.glite.voms.PKIStore( caDir, org.glite.voms.PKIStore.TYPE_CADIR, true );
-		caStore.rescheduleRefresh(900000);
 		
-		_vv = new org.glite.voms.VOMSValidator(null, new org.glite.voms.ac.ACValidator(new org.glite.voms.PKIVerifier(vomsStore,caStore)));
+                CertificateValidatorBuilder builder = new CertificateValidatorBuilder();
+                X509CertChainValidatorExt certChainValidator = builder.trustAnchorsDir(caDir).build();
+                        // .storeUpdateListener(storeUpdateListener).lazyAnchorsLoading(true)
+                        // .validationErrorListener(certChainValidationErrorListener).build();
+		_vv = VOMSValidators.newValidator(vomsStore, certChainValidator);
 	    }
-	    _vv.setClientChain(certChain);
 	    
-	    List vc = _vv.parse().getVOMSAttributes();
+	    List vc = _vv.parse(certChain);
             if ((vc == null) || (vc.size() == 0)) {
 		return null;
             } 
             int nVomsCerts = vc.size();
             TSRMLog.debug(ValidatorControl.class, null, "vomsCertSize="+nVomsCerts, null);
             for (int i=0; i<nVomsCerts; i++) {
-		org.glite.voms.VOMSAttribute curr = (org.glite.voms.VOMSAttribute)(vc.get(i));
+		VOMSAttribute curr = (VOMSAttribute)(vc.get(i));
                  TSRMLog.debug(ValidatorControl.class, null, "listVomsCert-"+i+"th="+curr.toString(), null);
             }
+            List vomsCerts;
             if (gov.lbl.srm.server.Config._doValidateVoms) {
-                _vv=_vv.validate();
+                vomsCerts = _vv.validate(certChain);
+            } else {
+                vomsCerts = _vv.parse(certChain);
             }
 	    TSRMLog.debug(ValidatorControl.class, null, "event=validated", null);
 	    
-            List vomsCerts = _vv.parse().getVOMSAttributes();
 	    if ((vomsCerts == null) || (vomsCerts.size() < nVomsCerts)) {
 		String detail = "null vomsCerts";
 		if (vomsCerts != null) {
@@ -248,10 +258,10 @@ class VOMSInfo {
 	    int total = vomsCerts.size();
 	    
 	    for (int i=0; i<total; i++) {
-	    org.glite.voms.VOMSAttribute curr = (org.glite.voms.VOMSAttribute)(vomsCerts.get(i));
+	    VOMSAttribute curr = (VOMSAttribute)(vomsCerts.get(i));
 	    //AttributeCertificate ac = curr.getAC();
 	    //TSRMLog.info(GUMSClient.class, "ac issuer="+ac.getIssuer().toString(), null, null);
-	    result.addAll(curr.getFullyQualifiedAttributes());
+	    result.addAll(curr.getFQANs());
 	    }
 	    return result;
 	} catch (Exception e) {
