From 3f35913c0fa13da6bdeaafaddc753aca49d2e2ef Mon Sep 17 00:00:00 2001
From: Yu Ming Zhu <yzhu@redhat.com>
Date: Jul 23 2019 12:03:35 +0000
Subject: cli:fix typo in mock-config


---

diff --git a/cli/koji_cli/commands.py b/cli/koji_cli/commands.py
index c11eed1..e53a7f5 100644
--- a/cli/koji_cli/commands.py
+++ b/cli/koji_cli/commands.py
@@ -1000,7 +1000,7 @@ def anon_handle_mock_config(goptions, session, args):
     if 'mock.package_manager' in buildcfg['extra']:
         opts['package_manager'] = buildcfg['extra']['mock.package_manager']
     if 'mock.yum.module_hotfixes' in buildcfg['extra']:
-        opts['module_hotfixes'] = self.config['extra']['mock.yum.module_hotfixes']
+        opts['module_hotfixes'] = buildcfg['extra']['mock.yum.module_hotfixes']
 
     output = koji.genMockConfig(name, arch, **opts)
     if options.ofile:

