diff -Nur koji-1.15.0.orig/builder/kojid koji-1.15.0/builder/kojid
--- koji-1.15.0.orig/builder/kojid	2018-03-16 11:56:22.542475550 -0700
+++ koji-1.15.0/builder/kojid	2018-03-16 11:58:57.281149897 -0700
@@ -3491,8 +3491,8 @@
             #IF specific
             'imgdir': os.path.join(self.workdir, 'scratch_images'),
             'tmpdir': os.path.join(self.workdir, 'oz-tmp'),
-            'verbose' : True,
-            'timeout': 7200,
+            'verbose': True,
+            'timeout': self.options.oz_install_timeout,
             'output': 'log',
             'raw': False,
             'debug': True,
@@ -5652,6 +5652,7 @@
                                      'resolver-status.properties *.lastUpdated',
                 'failed_buildroot_lifetime' : 3600 * 4,
                 'rpmbuild_timeout' : 3600 * 24,
+                'oz_install_timeout': None,
                 'cert': None,
                 'ca': '',  # FIXME: Unused, remove in next major release
                 'serverca': None}
@@ -5659,7 +5660,7 @@
         for name, value in config.items('kojid'):
             if name in ['sleeptime', 'maxjobs', 'minspace', 'retry_interval',
                         'max_retries', 'offline_retry_interval', 'failed_buildroot_lifetime',
-                        'timeout', 'rpmbuild_timeout',]:
+                        'timeout', 'rpmbuild_timeout', 'oz_install_timeout',]:
                 try:
                     defaults[name] = int(value)
                 except ValueError:
diff -Nur koji-1.15.0.orig/builder/kojid.conf koji-1.15.0/builder/kojid.conf
--- koji-1.15.0.orig/builder/kojid.conf	2017-12-18 14:10:22.000000000 -0800
+++ koji-1.15.0/builder/kojid.conf	2018-03-16 11:59:35.122314808 -0700
@@ -35,6 +35,10 @@
 ; Timeout for build duration (24 hours)
 ; rpmbuild_timeout=86400
 
+; Install timeout(seconds) for image build
+; if it's unset, use the number in /etc/oz/oz.cfg, supported since oz-0.16.0
+; oz_install_timeout=7200
+
 ; The URL for the xmlrpc server
 server=http://hub.example.com/kojihub
 
diff -Nur koji-1.15.0.orig/docs/source/image_build.rst koji-1.15.0/docs/source/image_build.rst
--- koji-1.15.0.orig/docs/source/image_build.rst	2017-12-18 14:10:22.000000000 -0800
+++ koji-1.15.0/docs/source/image_build.rst	2018-03-16 12:00:26.644539341 -0700
@@ -660,6 +660,11 @@
     #.  python-psphere => 0.5
     #.  VMDKStream => 0.2
     #.  pykickstart
+#.  Edit ``/etc/kojid/kojid.conf``, and set an second value, eg: 7200 for
+    ``oz_install_timeout``. It's a timeout waiting guest installing. If it's
+    not specified, oz will use its default value. Since ``oz-0.16.0`` it can be
+    configured in ``/etc/oz/oz.cfg`` as ``install`` in the ``[timeouts]``
+    section.
 #.  Edit ``/etc/oz/oz.cfg``, and set the memory value in the ``[libvirt]``
     section to at least 2048. Set ``safe_generation`` under ``[icicle]`` to yes.
 #.  Run: ``mkdir -p ~root/.psphere/templates``, and then copy the following
