From 56062420be5b561366a0124c9de7f9b42377e852 Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Fri, 6 Dec 2019 20:09:25 -0600
Subject: [PATCH] Don't warn on use_old_ssl - OSG still relies on it

---
 koji/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/koji/__init__.py b/koji/__init__.py
index f5e2374..53b8494 100644
--- a/koji/__init__.py
+++ b/koji/__init__.py
@@ -2070,7 +2070,7 @@ def __init__(self, baseurl, opts=None, sinfo=None):
         self.rsession = None
         self.new_session()
         self.opts.setdefault('timeout',  60 * 60 * 12)
-        self._old_ssl_warning = False
+        self._old_ssl_warning = True
 
 
     def new_session(self):
-- 
2.6.3

