From f4605434eae74335389bcfb88f557a59efb1c218 Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Thu, 8 Sep 2016 11:18:34 -0500
Subject: [PATCH] koji_proxy_cert

---
 koji/ssl/SSLCommon.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/koji/ssl/SSLCommon.py b/koji/ssl/SSLCommon.py
index 316ff6d..a71eeab 100644
--- a/koji/ssl/SSLCommon.py
+++ b/koji/ssl/SSLCommon.py
@@ -92,11 +92,11 @@ def CreateSSLContext(certs):
         if f and not os.access(f, os.R_OK):
             raise StandardError, "%s does not exist or is not readable" % f
 
     ctx = SSL.Context(SSL.SSLv23_METHOD)   # Use best possible TLS Method
     ctx.set_passwd_cb(passwd_callback, None)
-    ctx.use_certificate_file(key_and_cert)
+    ctx.use_certificate_chain_file(key_and_cert)
     retries_left = 2
     while True:
         try:
             ctx.use_privatekey_file(key_and_cert)
             break
-- 
2.6.3

