From c83d65f95338d33d2b6cb6390645269b82f51276 Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Thu, 8 Sep 2016 11:28:33 -0500
Subject: [PATCH] kojid_setup_dns

---
 builder/kojid | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/builder/kojid b/builder/kojid
index 446ab2e..6cc50e0 100755
--- a/builder/kojid
+++ b/builder/kojid
@@ -1117,11 +1117,12 @@ class BuildArchTask(BaseBuildTask):
         #    raise koji.BuildError, "the distribution tag is not set in the original srpm"
 
         self.updateWeight(name)
 
         rootopts = {
-            'repo_id': repo_id
+            'repo_id': repo_id,
+            'setup_dns': True,
             }
         br_arch = self.find_arch(arch, self.session.host.getHost(), self.session.getBuildConfig(root, event=event_id))
         broot = BuildRoot(self.session, self.options, root, br_arch, self.id, **rootopts)
         broot.workdir = self.workdir
 
-- 
2.6.3

