From b5b1cc0ce5cc648ea010f33063329add6feff745 Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Thu, 8 Sep 2016 11:28:33 -0500
Subject: [PATCH] kojid_setup_dns

---
 builder/kojid | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/builder/kojid b/builder/kojid
index 301df42..254f09d 100755
--- a/builder/kojid
+++ b/builder/kojid
@@ -1211,7 +1211,8 @@ class BuildArchTask(BaseBuildTask):
         self.updateWeight(name)
 
         rootopts = {
-            'repo_id': repo_id
+            'repo_id': repo_id,
+            'setup_dns': True,
             }
         br_arch = self.find_arch(arch, self.session.host.getHost(), self.session.getBuildConfig(root, event=event_id))
         broot = BuildRoot(self.session, self.options, root, br_arch, self.id, **rootopts)
-- 
2.6.3

