From 5f565a0cec5355e7983c761e7cb5411de025cf68 Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Fri, 15 Nov 2019 21:16:33 -0600
Subject: [PATCH] koji_passwd_retry

---
 koji/ssl/SSLCommon.py | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/koji/ssl/SSLCommon.py b/koji/ssl/SSLCommon.py
index aec1112..7c39e9a 100644
--- a/koji/ssl/SSLCommon.py
+++ b/koji/ssl/SSLCommon.py
@@ -59,7 +59,18 @@ def CreateSSLContext(certs):
     ctx = SSL.Context(SSL.SSLv23_METHOD)   # Use best possible TLS Method
     ctx.set_passwd_cb(passwd_callback, None)
     ctx.use_certificate_file(key_and_cert)
-    ctx.use_privatekey_file(key_and_cert)
+    retries_left = 2
+    while True:
+        try:
+            ctx.use_privatekey_file(key_and_cert)
+            break
+        except SSL.Error as err:
+            if retries_left > 0 and err[0][0][2] == 'bad decrypt':
+                print "Could not decrypt private key. Incorrect passphrase?"
+                globals()['cached_passwd'] = None
+                retries_left -= 1
+            else:
+                raise err
     ctx.load_verify_locations(peer_ca_cert)
     verify = SSL.VERIFY_PEER | SSL.VERIFY_FAIL_IF_NO_PEER_CERT
     ctx.set_verify(verify, our_verify)
-- 
2.6.3

