diff --git a/builder/kojid b/builder/kojid
index 767db9a..a4277a3 100755
--- a/builder/kojid
+++ b/builder/kojid
@@ -54,6 +54,7 @@ from fnmatch import fnmatch
 from gzip import GzipFile
 from optparse import OptionParser
 from StringIO import StringIO
+from yum import repoMDObject
 
 #imports for LiveCD and Appliance handler
 image_enabled = False
@@ -557,8 +558,18 @@ class BuildRoot(object):
         pathinfo = koji.PathInfo(topdir='')
         #XXX - cheap hack to get relative paths
         repodir = pathinfo.repo(self.repo_info['id'], self.repo_info['tag_name'])
-        relpath = os.path.join(repodir, self.br_arch, 'repodata', 'pkgorigins.gz')
+        repomdpath = os.path.join(repodir, self.br_arch, 'repodata', 'repomd.xml')
+
         opts = dict([(k, getattr(self.options, k)) for k in 'topurl','topdir'])
+        fo = koji.openRemoteFile(repomdpath, **opts)
+        try:
+            repodata = repoMDObject.RepoMD('ourrepo', fo)
+        except:
+            raise koji.BuildError, "Unable to parse repomd.xml file for %s" % os.path.join(repodir, self.br_arch)
+        data  = repodata.getData('origin')
+        pkgorigins  = data.location[1]
+
+        relpath = os.path.join(repodir, self.br_arch, pkgorigins)
         fo = koji.openRemoteFile(relpath, **opts)
         #at this point we know there were external repos at the create event,
         #so there should be an origins file.
