From c14b5410098d4767cb4fee80cbc170b5bfe6f0d8 Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Sat, 4 Apr 2020 15:44:09 -0500
Subject: [PATCH] Restore Python 2.6 compatibility to kojid

---
 builder/kojid | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/builder/kojid b/builder/kojid
index 32d5cfa..f2c8e3e 100755
--- a/builder/kojid
+++ b/builder/kojid
@@ -772,7 +772,8 @@ class BuildRoot(object):
             #at this point we know there were external repos at the create event,
             #so there should be an origins file.
             origin_idx = {}
-            with GzipFile(fileobj=fo, mode='r') as fo2:
+            fo2 = GzipFile(fileobj=fo, mode='r')
+            try:
                 if six.PY3:
                     fo2 = io.TextIOWrapper(fo2, encoding='utf-8')
                 for line in fo2:
@@ -782,6 +783,8 @@ class BuildRoot(object):
                     #first field is formated by yum as [e:]n-v-r.a
                     nvra = "%(name)s-%(version)s-%(release)s.%(arch)s" %  koji.parse_NVRA(parts[0])
                     origin_idx[nvra] = parts[1]
+            finally:
+                fo2.close()
         # mergerepo starts from a local repo in the task workdir, so internal
         # rpms have an odd-looking origin that we need to look for
         localtail = '/repo_%s_premerge/' % self.repo_info['id']
-- 
2.6.3

