Index: configs/lcmaps.cfg
===================================================================
--- configs.orig/lcmaps.cfg	2012-11-19 17:51:01.000000003 -0600
+++ configs/lcmaps.cfg	2013-02-12 11:55:36.000000003 -0600
@@ -9,7 +9,8 @@
              "-capath /etc/grid-security/certificates"
              "-cert   /etc/grid-security/xrd/xrdcert.pem"
              "-key    /etc/grid-security/xrd/xrdkey.pem"
-             "--endpoint https://red-auth.unl.edu:8443/gums/services/GUMSXACMLAuthorizationServicePort"
+# Change this URL to your GUMS server
+             "--endpoint https://yourgums.yourdomain:8443/gums/services/GUMSXACMLAuthorizationServicePort"
 
 verifyproxy = "lcmaps_verify_proxy.mod"
           "--allow-limited-proxy"
