diff -urN xrootd-lcmaps/cmake/FindXrootd.cmake xrootd-lcmaps.fix/cmake/FindXrootd.cmake
--- xrootd-lcmaps-1.7.5/cmake/FindXrootd.cmake	2020-02-24 12:46:38.887572526 -0600
+++ xrootd-lcmaps-1.7.5/cmake/FindXrootd.cmake	2020-02-24 12:47:42.535475741 -0600
@@ -19,7 +19,7 @@
   PATHS /opt/xrootd
 )
 
-FIND_LIBRARY(XROOTD_SECGSI XrdSecgsi-4
+FIND_LIBRARY(XROOTD_SECGSI XrdSecgsi-5
   HINTS
   ${XROOTD_DIR}
   $ENV{XROOTD_DIR}
@@ -28,7 +28,7 @@
   PATH_SUFFIXES lib
 )
 
-FIND_LIBRARY(XROOTD_CRYPTOSSL XrdCryptossl-4
+FIND_LIBRARY(XROOTD_CRYPTOSSL XrdCryptossl-5
   HINTS
   ${XROOTD_DIR}
   $ENV{XROOTD_DIR}
