Index: mock-1.0.36/py/mock.py
===================================================================
--- mock-1.0.36.orig/py/mock.py	2012-11-01 15:53:00.000000003 -0500
+++ mock-1.0.36/py/mock.py	2012-11-26 17:14:28.000000003 -0600
@@ -754,8 +754,9 @@
     os.environ["HOME"] = chroot.homedir
 
     # New namespace starting from here
-    base_unshare_flags = mockbuild.util.CLONE_NEWNS|mockbuild.util.CLONE_NEWUTS
-    extended_unshare_flags = base_unshare_flags|mockbuild.util.CLONE_NEWPID|mockbuild.util.CLONE_NEWIPC
+    base_unshare_flags = mockbuild.util.CLONE_NEWNS
+    # CLONE_NEWUTS not supported on el5 kernels so it's going into extended_unshare_flags
+    extended_unshare_flags = base_unshare_flags|mockbuild.util.CLONE_NEWPID|mockbuild.util.CLONE_NEWIPC|mockbuild.util.CLONE_NEWUTS
     try:
         mockbuild.util.unshare(extended_unshare_flags)
     except mockbuild.exception.UnshareFailed, e:
