From db05f802927274e509d6a88efbe81a18f4d71a6e Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Fri, 12 Aug 2016 14:19:56 -0500
Subject: [PATCH 1/5] level1

---
 globus_ftp_control_data.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/globus_ftp_control_data.c b/globus_ftp_control_data.c
index 7cdb090..c3c9a36 100644
--- a/globus_ftp_control_data.c
+++ b/globus_ftp_control_data.c
@@ -55,10 +55,11 @@
     data_conn = (globus_ftp_data_connection_t *)                      \
                      globus_malloc(                                   \
                           sizeof(globus_ftp_data_connection_t));      \
     data_conn->whos_my_daddy = stripe;                                \
     data_conn->offset = 0;                                            \
+    data_conn->smallest_offset = -1;                                  \
     data_conn->callback = cb;                                         \
     data_conn->user_arg = ua;                                         \
     data_conn->bytes_ready = 0;                                       \
     data_conn->eod = GLOBUS_FALSE;                                    \
     data_conn->close = GLOBUS_FALSE;                                  \
@@ -118,10 +119,11 @@ typedef struct globus_l_ftp_c_data_layout_s
  */
 typedef struct globus_ftp_data_connection_s
 {
     globus_io_handle_t                          io_handle;
     globus_off_t                                offset;
+    globus_off_t                                smallest_offset;
     struct globus_ftp_data_stripe_s *           whos_my_daddy;
     globus_ftp_control_data_connect_callback_t  callback;
     void *                                      user_arg;
     globus_size_t                               bytes_ready;
 
@@ -153,10 +155,13 @@ typedef struct globus_ftp_data_stripe_s
     globus_fifo_t                               command_q;
     globus_io_handle_t                          listener_handle;
     globus_ftp_control_parallelism_t            parallel;
     globus_ftp_control_host_port_t              host_port;
     struct globus_i_ftp_dc_transfer_handle_s *  whos_my_daddy;
+    globus_off_t                                largest_entry;            // The largest TCP read size - relevant for stream mode.
+    globus_off_t                                est_block_size;           // The largest block size (minimum observed diff in offsets)
+    int                                         largest_connection_count; // Maximum number of connections we observed
 
     int                                         connection_count;
     int                                         total_connection_count;
 
     globus_bool_t                               eof;
-- 
2.6.3

