From 94f16458a4e9edc94cf4215385b632334dc21267 Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Fri, 12 Aug 2016 14:20:43 -0500
Subject: [PATCH 2/5] level2

---
 globus_ftp_control_data.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/globus_ftp_control_data.c b/globus_ftp_control_data.c
index c3c9a36..81e8fe1 100644
--- a/globus_ftp_control_data.c
+++ b/globus_ftp_control_data.c
@@ -6355,10 +6355,16 @@ globus_l_ftp_data_stream_stripe_poll(
 
     globus_assert(stripe->whos_my_daddy->whos_my_daddy->mode == GLOBUS_FTP_CONTROL_MODE_STREAM);
     /*
      *  check to see that there is a connection
      */
+
+    globus_fifo_t remaining_command_q;
+    globus_fifo_init(&remaining_command_q);
+    globus_fifo_t remaining_conn_q;
+    globus_fifo_init(&remaining_conn_q);
+
     while(!globus_fifo_empty(&stripe->free_conn_q) &&
        !globus_fifo_empty(&stripe->command_q))
     {
         entry = (globus_l_ftp_handle_table_entry_t *)
             globus_fifo_peek(&stripe->command_q);
-- 
2.6.3

