Index: globus_ftp_control_data.c
===================================================================
--- globus_ftp_control_data.c.orig	2014-11-03 09:36:01.453844211 -0600
+++ globus_ftp_control_data.c	2015-02-10 16:14:50.750424249 -0600
@@ -155,6 +155,8 @@
     globus_ftp_control_parallelism_t            parallel;
     globus_ftp_control_host_port_t              host_port;
     struct globus_i_ftp_dc_transfer_handle_s *  whos_my_daddy;
+    globus_off_t                                largest_entry;
+    int                                         largest_connection_count;
 
     int                                         connection_count;
     int                                         total_connection_count;
@@ -6272,6 +6274,12 @@
     /*
      *  check to see that there is a connection
      */
+
+    globus_fifo_t remaining_command_q;
+    globus_fifo_init(&remaining_command_q);
+    globus_fifo_t remaining_conn_q;
+    globus_fifo_init(&remaining_conn_q);
+
     while(!globus_fifo_empty(&stripe->free_conn_q) &&
        !globus_fifo_empty(&stripe->command_q))
     {
@@ -6319,22 +6327,78 @@
             }
             else if(entry->direction == GLOBUS_FTP_DATA_STATE_CONNECT_READ)
             {
+
+                /* BB - Prevent a single stripe from getting too far ahead
+                 * The basic idea here is to not issue the read from the TCP
+                 * socket until all the data streams are at about the same
+                 * offset.  This will cause the data of the fastest streams
+                 * to sit in the OS socket buffer, then eventually packets to
+                 * be dropped - slowing down the fastest streams, and making
+                 * all TCP streams to be about the same speed.  This is beneficial
+                 * for writing to spinning disks, as the writes the OS will see
+                 * will be much closer to sequential, and prevent the file from
+                 * fragmenting on disk.  This is also beneficial for operations
+                 * that need to see a re-ordered bytestream before flushing
+                 * data from memory (HDFS DSI plugin, checksumming).
+                 */
+                globus_ftp_data_connection_t * tmp_conn;
+                globus_off_t min_offset = data_conn->offset;
+                globus_list_t * list;
+                for(list = stripe->all_conn_list;
+                    !globus_list_empty(list);
+                    list = globus_list_rest(list))
+                {
+                    tmp_conn = (globus_ftp_data_connection_t *)globus_list_first(list);
+                    if ((tmp_conn->offset < min_offset) && (!tmp_conn->eod)) {
+                        min_offset = tmp_conn->offset;
+                    }
+                }
+
+                /* Since we loop over all free connections, whenever the
+                 * slowest socket finally catches up, we'll register reads for
+                 * all free connections with a single call to the poll function.
+                 * This should prevent deadlocks.
+                 */
+
                 /* remove from queue */
                 globus_fifo_dequeue(&stripe->command_q);
                 globus_fifo_dequeue(&stripe->free_conn_q);
 
-                result = globus_io_register_read(
-                             &data_conn->io_handle,
-                             entry->buffer,
-                             entry->length,
-                             entry->length,
-                             globus_l_ftp_stream_read_callback,
-                             (void *)entry);
-                globus_assert(result == GLOBUS_SUCCESS);
+                stripe->largest_entry = (entry->length > stripe->largest_entry) ? entry->length : stripe->largest_entry;
+                stripe->largest_connection_count = (stripe->connection_count > stripe->largest_connection_count) ? stripe->connection_count : stripe->largest_connection_count;
+
+                if ((stripe->connection_count < stripe->largest_connection_count) || (data_conn->offset - min_offset <= 5*stripe->largest_entry)) {
+                    //printf("BB - Current offset: %llu, min_offset: %llu.\n", data_conn->offset, min_offset);
+
+
+                    result = globus_io_register_read(
+                                 &data_conn->io_handle,
+                                 entry->buffer,
+                                 entry->length,
+                                 entry->length,
+                                 globus_l_ftp_stream_read_callback,
+                                 (void *)entry);
+                    globus_assert(result == GLOBUS_SUCCESS);
+                }
+                else
+                {
+                    //printf("BB - Adding entry to remaining queue.\n");
+                    globus_fifo_enqueue(&remaining_command_q, entry);
+                    globus_fifo_enqueue(&remaining_conn_q, data_conn);
+                }
+
             }
         }
     }
 
+    /* If there are leftover commands, add them back to the stripe's queue. */
+    while (!globus_fifo_empty(&remaining_command_q)) {
+        globus_fifo_enqueue(&stripe->command_q, globus_fifo_dequeue(&remaining_command_q));
+        globus_fifo_enqueue(&stripe->free_conn_q, globus_fifo_dequeue(&remaining_conn_q));
+    }
+    globus_fifo_destroy(&remaining_command_q);
+    globus_fifo_destroy(&remaining_conn_q);
+
     return GLOBUS_SUCCESS;
 }
 
@@ -6616,6 +6680,10 @@
             /*
              *  in big buffer mode this should never be entered
              */
+            globus_fifo_t remaining_command_q;
+            globus_fifo_init(&remaining_command_q);
+            globus_fifo_t remaining_conn_q;
+            globus_fifo_init(&remaining_conn_q);
             while(!globus_fifo_empty(&stripe->command_q) && !done)
             {
                 globus_assert(transfer_handle->big_buffer == GLOBUS_NULL);
@@ -6654,30 +6722,60 @@
                     data_conn = (globus_ftp_data_connection_t *)
                         globus_fifo_dequeue(&stripe->free_conn_q);
 
-                    /*
-                     *  set the entries offset to the offset on the
-                     *  data_conn.
-                     *  If use is requesting more bytes than are available
-                     *  on this connection set the length to bytes_ready
-                     */
-                    entry->whos_my_daddy = data_conn;
-                    entry->offset = data_conn->offset;
-                    if(entry->length > data_conn->bytes_ready)
-                    {
-                        entry->length = data_conn->bytes_ready;
+                    /* BB - See commentary in stream_stripe_poll above. */
+                    globus_ftp_data_connection_t * tmp_conn;
+                    globus_off_t min_offset = data_conn->offset;
+                    globus_list_t * list;
+                    //unsigned short stripe_count = 0;
+                    for(list = stripe->all_conn_list;
+                        !globus_list_empty(list);
+                        list = globus_list_rest(list))
+                    {   
+                        tmp_conn = (globus_ftp_data_connection_t *)globus_list_first(list);
+                        if ((tmp_conn->offset < min_offset) && (!tmp_conn->eod)) {
+                            min_offset = tmp_conn->offset;
+                        }
+                        //printf("BB - Stripe %u, offset %u.\n", stripe_count, tmp_conn->offset);
+                        //stripe_count++;
                     }
 
-                    /*
-                     *  register a read
-                     */
-                    res = globus_io_register_read(
-                              &data_conn->io_handle,
-                              entry->buffer,
-                              entry->length,
-                              entry->length,
-                              globus_l_ftp_eb_read_callback,
-                              (void *)entry);
-                    globus_assert(res == GLOBUS_SUCCESS);
+                    stripe->largest_entry = (entry->length > stripe->largest_entry) ? entry->length : stripe->largest_entry;
+                    stripe->largest_connection_count = (stripe->connection_count > stripe->largest_connection_count) ? stripe->connection_count : stripe->largest_connection_count;
+                    //printf("BB - offset: %lu, min_offset: %lu, diff: %lu, max diff: %lu outstanding connections: %d.\n", data_conn->offset, min_offset, data_conn->offset - min_offset, 5*stripe->largest_entry, stripe->connection_count);
+
+                    if ((stripe->connection_count < stripe->largest_connection_count) || (data_conn->offset - min_offset <= 5*stripe->largest_entry)) {
+
+                        /*
+                         *  set the entries offset to the offset on the
+                         *  data_conn.
+                         *  If use is requesting more bytes than are available
+                         *  on this connection set the length to bytes_ready
+                         */
+                        entry->whos_my_daddy = data_conn;
+                        entry->offset = data_conn->offset;
+                        if(entry->length > data_conn->bytes_ready)
+                        {
+                            entry->length = data_conn->bytes_ready;
+                        }
+
+                        /*
+                         *  register a read
+                         */
+                        res = globus_io_register_read(
+                                  &data_conn->io_handle,
+                                  entry->buffer,
+                                  entry->length,
+                                  entry->length,
+                                  globus_l_ftp_eb_read_callback,
+                                  (void *)entry);
+                        globus_assert(res == GLOBUS_SUCCESS);
+                    }
+                    else
+                    {
+                        //printf("BB - Adding entry to remaining queue.\n");
+                        globus_fifo_enqueue(&remaining_command_q, entry);
+                        globus_fifo_enqueue(&remaining_conn_q, data_conn);
+                    }
                 }
                 /*
                  *  if we have not hit EOF and there are no available data
@@ -6688,6 +6786,13 @@
                     done = GLOBUS_TRUE;
                 }
             }/* end while */
+            /* If there are leftover commands, add them back to the stripe's queue. */
+            while (!globus_fifo_empty(&remaining_command_q)) {
+                globus_fifo_enqueue(&stripe->command_q, globus_fifo_dequeue(&remaining_command_q));
+                globus_fifo_enqueue(&stripe->free_conn_q, globus_fifo_dequeue(&remaining_conn_q));
+            }
+            globus_fifo_destroy(&remaining_command_q);
+            globus_fifo_destroy(&remaining_conn_q);
         }
     }
 
@@ -6976,6 +7081,8 @@
         stripe->listening = GLOBUS_FALSE;
         globus_fifo_init(&stripe->command_q);
 
+        stripe->largest_entry = 1024*1024;
+        stripe->largest_connection_count = 0;
         stripe->total_connection_count = 0;
         stripe->eods_received = 0;
         stripe->eof_sent = GLOBUS_FALSE;
