--- condor.pm	2011-12-19 21:22:34.000000000 -0600
+++ condor.pm	2012-06-04 16:51:00.000000000 -0500
@@ -553,6 +553,34 @@
                 Globus::GRAM::Error::TEMP_SCRIPT_FILE_FAILED());
         }
     }
+    my @xcount = $description->xcount();
+    if(@xcount)
+    {   
+        $self->log("xcount = " . scalar(@xcount));
+        foreach my $this_xcount (@xcount)
+        {   
+            $self->log("xcount = " . $this_xcount);
+            $rc = print SCRIPT_FILE "request_cpus=$this_xcount \n";
+            if (!$rc)
+                {
+                    return $self->respond_with_failure_extension(
+                    "print: $script_filename: $!",
+                    Globus::GRAM::Error::TEMP_SCRIPT_FILE_FAILED());
+                }
+        }   
+    }
+
+    if($description->min_memory() ne '')
+    {
+        my $memory_request = $description->min_memory();
+        $rc = print SCRIPT_FILE "request_memory=" . $memory_request . "\n";
+        if (!$rc)
+        {
+            return $self->respond_with_failure_extension(
+                "print: $script_filename: $!",
+                Globus::GRAM::Error::TEMP_SCRIPT_FILE_FAILED());
+        }
+    }
 
     # NFS Lite mode
     if ($isNFSLite && !$isManagedFork) {
--- condor.rvf	2011-12-19 21:22:34.000000000 -0600
+++ condor.rvf	2012-06-01 10:50:41.000000000 -0500
@@ -17,3 +17,8 @@
               if it's defined there, otherwise the condor system default."
 ValidWhen: GLOBUS_GRAM_JOB_SUBMIT
 
+
+Attribute: xcount
+Description: "To specify the number of cores for a job"
+ValidWhen: GLOBUS_GRAM_JOB_SUBMIT
+
