Index: condor.pm
===================================================================
--- condor.pm.orig	2015-02-11 13:52:23.834285581 -0600
+++ condor.pm	2015-02-11 13:52:25.526305693 -0600
@@ -545,10 +545,23 @@
                 "print: $script_filename: $!",
                 Globus::GRAM::Error::TEMP_SCRIPT_FILE_FAILED());
         }
     }
 
+    # default max_wall_time can be specified in globus-gram-job-manager.rvf
+    if($description->max_wall_time() ne '')
+    {
+        my $max_wall_time = $description->max_wall_time() ;
+        $rc = print SCRIPT_FILE "PeriodicRemove= (JobStatus == 2) && ( (time() - EnteredCurrentStatus) > (" . $max_wall_time . " * 60))\n";
+        if (!$rc)
+        {
+            return $self->respond_with_failure_extension(
+                "print: $script_filename: $!",
+                Globus::GRAM::Error::TEMP_SCRIPT_FILE_FAILED());
+        }
+    }
+
     # NFS Lite mode
     if ($isNFSLite && !$isManagedFork) {
         print SCRIPT_FILE "should_transfer_files = YES\n";
         print SCRIPT_FILE "when_to_transfer_output = ON_EXIT\n";
         print SCRIPT_FILE "transfer_output = true\n";
