Index: condor.pm
===================================================================
--- condor.pm.orig	2011-08-09 11:32:02.000000000 -0500
+++ condor.pm	2011-09-06 17:24:28.000000000 -0500
@@ -737,7 +737,7 @@
 
     if($num_abort > 0)
     {
-        $state = Globus::GRAM::JobState::FAILED;
+        return Globus::GRAM::Error::SYSTEM_CANCELLED();
     }
     elsif($num_done == $description->count())
     {
