Index: condor.pm
===================================================================
--- condor.pm.orig	2015-02-11 13:51:24.370578898 -0600
+++ condor.pm	2015-02-11 13:51:27.717618667 -0600
@@ -893,11 +893,11 @@
         return { JOB_STATE => Globus::GRAM::JobState::DONE };
     }
 
     if($num_abort > 0)
     {
-        $state = Globus::GRAM::JobState::FAILED;
+        return Globus::GRAM::Error::SYSTEM_CANCELLED();
     }
     elsif($num_done == $description->count())
     {
         $self->nfssync( $description->stdout(), 0 )
             if $description->stdout() ne '';
