diff -Nru ./globus_xio_gsi.c ../../../../branches/gt525/globus-xio-gsi-driver/_quilt/globus_xio_gsi_driver-2.4/globus_xio_gsi.c
--- ./globus_xio_gsi.c	2011-03-08 10:19:46.000000000 -0600
+++ ../../../../branches/gt525/globus-xio-gsi-driver/_quilt/globus_xio_gsi_driver-2.4/globus_xio_gsi.c	2013-10-28 16:12:28.000000000 -0500
@@ -764,7 +764,7 @@
 
     if(t[0] >= 20 &&
        t[0] <= 26 &&
-       t[1] == 3 && (t[2] == 0 || t[2] == 1))
+       t[1] == 3)
     {
         /* it's a SSL token */
         *length = (t[3] << 8) | t[4];
@@ -1532,7 +1532,7 @@
     {
         result = GlobusXIOErrorWrapFailedWithMessage(result,
             "The GSI XIO driver failed to establish a connection%s",
-            " via the underlying protocol (TCP).");
+            " via the underlying protocol.");
         goto error_destroy_handle;
     }
     

