Index: gridftp/server-lib/src/globus_gridftp_server_control.c
===================================================================
RCS file: /home/globdev/CVS/globus-packages/gridftp/server-lib/src/globus_gridftp_server_control.c,v
retrieving revision 1.78.2.4
diff -u -r1.78.2.4 globus_gridftp_server_control.c
--- globus_gridftp_server_control.c	6 Mar 2012 05:28:30 -0000	1.78.2.4
+++ globus_gridftp_server_control.c	17 May 2012 20:47:14 -0000
@@ -3764,7 +3764,7 @@
             globus_calloc(1, sizeof(globus_l_libc_cached_pwent_t));
         rc = globus_libc_getpwuid_r(
             uid, &pwent->pw, pwent->buffer, GSU_MAX_PW_LENGTH, &result_pw);
-        if(rc != 0)
+        if(rc != 0 || result_pw == NULL)
         {
             goto error_pwent;
         }
