Index: modules/srmclient/src/main/bin/srmping
===================================================================
--- modules/srmclient/src/main/bin/srmping.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srmping	2013-05-30 15:55:20.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srm-release-files
===================================================================
--- modules/srmclient/src/main/bin/srm-release-files.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srm-release-files	2013-05-30 15:56:20.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/command-line-client
===================================================================
--- modules/srmclient/src/main/bin/command-line-client.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/command-line-client	2013-05-30 15:35:16.000000002 -0500
@@ -4,13 +4,11 @@
 commanddir=`dirname $command`
 OUR_PATH=`dirname $commanddir`
 
-if [ "$SRM_PATH" != "" -a "$SRM_PATH" != "${OUR_PATH}" ]
-    then
-    echo "WARNING: SRM_PATH is defined, but does not point to the expected place " 2>&1
-    echo "WARNING: a wrong version of srm client may be executed " 2>&1
-    echo "WARNING: SRM_PATH=$SRM_PATH " 2>&1
-    echo "WARNING: expected path ${OUR_PATH} " 2>&1
-elif [ "$SRM_PATH" = "" ]
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+. /etc/sysconfig/dcache-srmclient
+fi
+
+if [ "$SRM_PATH" = "" ]
     then
     SRM_PATH=${OUR_PATH}
     export SRM_PATH
Index: modules/srmclient/src/main/bin/srmcp
===================================================================
--- modules/srmclient/src/main/bin/srmcp.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srmcp	2013-05-30 15:51:54.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srmstage
===================================================================
--- modules/srmclient/src/main/bin/srmstage.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srmstage	2013-05-30 15:55:38.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srm-get-space-metadata
===================================================================
--- modules/srmclient/src/main/bin/srm-get-space-metadata.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srm-get-space-metadata	2013-05-30 15:56:08.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srm-check-permissions
===================================================================
--- modules/srmclient/src/main/bin/srm-check-permissions.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srm-check-permissions	2013-05-30 15:55:58.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srm-get-request-status
===================================================================
--- modules/srmclient/src/main/bin/srm-get-request-status.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srm-get-request-status	2013-05-30 15:56:03.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srm-get-permissions
===================================================================
--- modules/srmclient/src/main/bin/srm-get-permissions.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srm-get-permissions	2013-05-30 15:56:02.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/gridftpcopy
===================================================================
--- modules/srmclient/src/main/bin/gridftpcopy.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/gridftpcopy	2013-05-30 15:35:16.000000002 -0500
@@ -4,13 +4,11 @@
 commanddir=`dirname $command`
 OUR_PATH=`dirname $commanddir`
 
-if [ "$SRM_PATH" != "" -a "$SRM_PATH" != "${OUR_PATH}" ]
-    then
-    echo "WARNING: SRM_PATH is defined, but does not point to the expected place " 2>&1
-    echo "WARNING: a wrong version of srm client may be executed " 2>&1
-    echo "WARNING: SRM_PATH=$SRM_PATH " 2>&1
-    echo "WARNING: expected path ${OUR_PATH} " 2>&1
-elif [ "$SRM_PATH" = "" ]
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+. /etc/sysconfig/dcache-srmclient
+fi
+
+if [ "$SRM_PATH" = "" ]
     then
     SRM_PATH=${OUR_PATH}
     export SRM_PATH
Index: modules/srmclient/src/main/bin/gridftplist
===================================================================
--- modules/srmclient/src/main/bin/gridftplist.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/gridftplist	2013-05-30 15:35:16.000000002 -0500
@@ -4,13 +4,7 @@
 commanddir=`dirname $command`
 OUR_PATH=`dirname $commanddir`
 
-if [ "$SRM_PATH" != "" -a "$SRM_PATH" != "${OUR_PATH}" ]
-    then
-    echo "WARNING: SRM_PATH is defined, but does not point to the expected place " 2>&1
-    echo "WARNING: a wrong version of srm client may be executed " 2>&1
-    echo "WARNING: SRM_PATH=$SRM_PATH " 2>&1
-    echo "WARNING: expected path ${OUR_PATH} " 2>&1
-elif [ "$SRM_PATH" = "" ]
+if [ "$SRM_PATH" = "" ]
     then
     SRM_PATH=${OUR_PATH}
     export SRM_PATH
Index: modules/srmclient/src/main/bin/srm-advisory-delete
===================================================================
--- modules/srmclient/src/main/bin/srm-advisory-delete.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srm-advisory-delete	2013-05-30 15:55:55.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srm-get-request-summary
===================================================================
--- modules/srmclient/src/main/bin/srm-get-request-summary.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srm-get-request-summary	2013-05-30 15:56:04.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srm-set-permissions
===================================================================
--- modules/srmclient/src/main/bin/srm-set-permissions.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srm-set-permissions	2013-05-30 15:57:15.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srmmv
===================================================================
--- modules/srmclient/src/main/bin/srmmv.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srmmv	2013-05-30 15:55:14.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srm-abort-files
===================================================================
--- modules/srmclient/src/main/bin/srm-abort-files.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srm-abort-files	2013-05-30 15:55:41.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srm_v1
===================================================================
--- modules/srmclient/src/main/bin/srm_v1.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srm_v1	2013-05-30 15:35:16.000000002 -0500
@@ -4,13 +4,11 @@
 commanddir=`dirname $command`
 OUR_PATH=`dirname $commanddir`
 
-if [ "$SRM_PATH" != "" -a "$SRM_PATH" != "${OUR_PATH}" ]
-    then
-    echo "WARNING: SRM_PATH is defined, but does not point to the expected place " 2>&1
-    echo "WARNING: a wrong version of srm client may be executed " 2>&1
-    echo "WARNING: SRM_PATH=$SRM_PATH " 2>&1
-    echo "WARNING: expected path ${OUR_PATH} " 2>&1
-elif [ "$SRM_PATH" = "" ]
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+. /etc/sysconfig/dcache-srmclient
+fi
+
+if [ "$SRM_PATH" = "" ]
     then
     SRM_PATH=${OUR_PATH}
     export SRM_PATH
Index: modules/srmclient/src/main/bin/srm-get-space-tokens
===================================================================
--- modules/srmclient/src/main/bin/srm-get-space-tokens.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srm-get-space-tokens	2013-05-30 15:56:10.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srm-release-space
===================================================================
--- modules/srmclient/src/main/bin/srm-release-space.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srm-release-space	2013-05-30 15:57:14.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srm-bring-online
===================================================================
--- modules/srmclient/src/main/bin/srm-bring-online.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srm-bring-online	2013-05-30 15:55:57.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srm-storage-element-info
===================================================================
--- modules/srmclient/src/main/bin/srm-storage-element-info.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srm-storage-element-info	2013-05-30 15:57:22.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srmmkdir
===================================================================
--- modules/srmclient/src/main/bin/srmmkdir.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srmmkdir	2013-05-30 15:55:10.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srm-get-metadata
===================================================================
--- modules/srmclient/src/main/bin/srm-get-metadata.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srm-get-metadata	2013-05-30 15:56:01.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srm-abort-request
===================================================================
--- modules/srmclient/src/main/bin/srm-abort-request.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srm-abort-request	2013-05-30 15:55:46.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/adler32
===================================================================
--- modules/srmclient/src/main/bin/adler32.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/adler32	2013-05-30 15:51:36.000000002 -0500
@@ -1,5 +1,9 @@
 #! /bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srm-extend-file-lifetime
===================================================================
--- modules/srmclient/src/main/bin/srm-extend-file-lifetime.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srm-extend-file-lifetime	2013-05-30 15:55:59.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srmrmdir
===================================================================
--- modules/srmclient/src/main/bin/srmrmdir.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srmrmdir	2013-05-30 15:55:30.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srmls
===================================================================
--- modules/srmclient/src/main/bin/srmls.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srmls	2013-05-30 15:52:01.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srmrm
===================================================================
--- modules/srmclient/src/main/bin/srmrm.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srmrm	2013-05-30 15:55:26.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
Index: modules/srmclient/src/main/bin/srm-get-request-tokens
===================================================================
--- modules/srmclient/src/main/bin/srm-get-request-tokens.orig	2013-05-23 10:28:44.000000002 -0500
+++ modules/srmclient/src/main/bin/srm-get-request-tokens	2013-05-30 15:56:06.000000002 -0500
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+if [ -e /etc/sysconfig/dcache-srmclient ]; then
+    . /etc/sysconfig/dcache-srmclient
+fi
+
 if [ -z "$SRM_PATH" ]; then
     SRM_PATH="/usr/share/srm"
 fi
