Index: lib/python/gip_common.py
===================================================================
--- lib/python/gip_common.py	(revision 4503)
+++ gip/lib/python/gip_common.py	(working copy)
@@ -243,7 +243,7 @@
         self.cp = cp
         self.map_location = cp_get(cp, "vo", "user_vo_map",
                                    vdtDir('$VDT_LOCATION/monitoring/osg-user-vo-map.txt',
-                                          '/etc/osg/osg-user-vo-map.txt'))
+                                          '/var/lib/osg/user-vo-map'))
 
         log.info("Using user-to-VO map location %s." % self.map_location)
         self.voi = []
Index: libexec/osg_info_wrapper.py
===================================================================
--- libexec/osg_info_wrapper.py	(revision 4503)
+++ gip/libexec/osg_info_wrapper.py	(working copy)
@@ -94,13 +94,13 @@
         cp = config()
 
     temp_dir = os.path.expandvars(cp_get(cp, "gip", "temp_dir", \
-        gipDir("$GIP_LOCATION/var/tmp", '/var/gip/tmp'))) 
+        gipDir("$GIP_LOCATION/var/tmp", '/var/cache/gip'))) 
     plugin_dir = os.path.expandvars(cp_get(cp, "gip", "plugin_dir", \
         gipDir("$GIP_LOCATION/plugins", '/usr/libexec/gip/plugins')))
     provider_dir = os.path.expandvars(cp_get(cp, "gip", "provider_dir", \
         gipDir("$GIP_LOCATION/providers", '/usr/libexec/gip/providers')))
     static_dir = os.path.expandvars(cp_get(cp, "gip", "static_dir", \
-        gipDir("$GIP_LOCATION/var/ldif", '/var/gip/ldif')))
+        gipDir("$GIP_LOCATION/var/ldif", '/etc/gip/ldif.d')))
 
     # Make sure that our directories exist.
     create_if_not_exist(temp_dir, plugin_dir, provider_dir, static_dir)
@@ -517,7 +517,7 @@
     log.debug("Set a %.2f second timeout." % timeout)
     t1 = -time.time()
     module_log_loc = os.path.expandvars(gipDir("$GIP_LOCATION/var/logs/module.log",
-                                               '/var/gip/log/module.log'))
+                                               '/var/log/gip/module.log'))
     try:
         sys.stderr = open(module_log_loc, 'a')
     except:
Index: libexec/osg-info-wrapper
===================================================================
--- libexec/osg-info-wrapper	(revision 4503)
+++ gip/libexec/osg-info-wrapper	(working copy)
@@ -1,3 +1,4 @@
 #!/bin/sh
 
-/usr/libexec/gip/osg_info_wrapper.py 2> /dev/null
+python -m osg_info_wrapper 2> /dev/null
+
Index: gip/lib/python/gip_osg.py
===================================================================
--- gip/lib/python/gip_osg.py	(revision 4503)
+++ gip/lib/python/gip_osg.py	(working copy)
@@ -113,9 +113,12 @@
         raise ValueError("osg-attributes.conf does not exists; we may be "
                          "running in an unconfigured OSG install!")
     # Check to see if the osg-user-vo-map.txt exists and that its size is > 0
+    if 'VDT_LOCATION' in os.environ:
+        defaultLoc = os.path.join(etcDir, "osg-user-vo-map.txt")
+    else:
+        defaultLoc = "/var/lib/osg/user-vo-map"
+    osg_user_vo_map = cp_get(cp, "vo", "user_vo_map", defaultLoc)
 
-    osg_user_vo_map = cp_get(cp, "vo", "user_vo_map", '%s/osg-user-vo-map.txt' % etcDir)
-
     if not os.path.exists(osg_user_vo_map):
         raise ValueError("osg-user-vo-map.txt does not exists; we may be "
                          "running in an unconfigured OSG install!")
