Index: globus_scheduler_event_generator-4.0/init/globus-scheduler-event-generator-lsb.in
===================================================================
--- globus_scheduler_event_generator-4.0.orig/init/globus-scheduler-event-generator-lsb.in	2011-09-22 18:39:30.000000000 -0500
+++ globus_scheduler_event_generator-4.0/init/globus-scheduler-event-generator-lsb.in	2011-09-22 18:40:33.000000000 -0500
@@ -70,7 +70,7 @@
         pidfile="$(printf "$GLOBUS_SEG_PIDFMT" "$lrm")"
         logdir="$(printf "$GLOBUS_SEG_LOGFMT" "$lrm")"
         if [ ! -d "$logdir" ]; then
-            (umask 022; mkdir "$logdir")
+            (umask 022; mkdir -p "$logdir")
         fi
         daemon ${GLOBUS_SEG_NICE_LEVEL} \
             --pidfile "$pidfile" "${prog}" \
