From c1b3e82ef26d8ea37c7898a8f9ddcd4787bd3dee Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <mselmeci@wisc.edu>
Date: Thu, 2 Jan 2025 18:14:46 -0600
Subject: [PATCH] Fix use of non-constant in format string

---
 src/CurlUtil.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CurlUtil.cc b/src/CurlUtil.cc
index 378f465..f715254 100644
--- a/src/CurlUtil.cc
+++ b/src/CurlUtil.cc
@@ -916,7 +916,7 @@ CurlWorker::Run() {
             std::string err;
             auto result = iter->second->WaitSocketCallback(err);
             if (result == -1) {
-                m_logger->Warning(kLogXrdClPelican, ("Error when invoking the broker callback: " + err).c_str());
+                m_logger->Warning(kLogXrdClPelican, "Error when invoking the broker callback: %s", err.c_str());
                 iter->second->Fail(XrdCl::errErrorResponse, 1, err);
                 m_op_map.erase(handle);
                 broker_reqs.erase(entry.fd);
-- 
2.43.5

