diff --git a/src/XrdClCurl/CurlFile.cc b/src/XrdClCurl/CurlFile.cc
index 8ca3f79..eb388fe 100644
--- a/src/XrdClCurl/CurlFile.cc
+++ b/src/XrdClCurl/CurlFile.cc
@@ -675,6 +675,7 @@ File::ReadPrefetch(uint64_t offset, uint64_t size, void *buffer, XrdCl::Response
                 GetConnCallout(), &m_default_header_callout
             )
         );
+        lock.unlock();
         try {
             m_queue->Produce(m_prefetch_op);
         } catch (...) {
diff --git a/src/XrdClCurl/CurlUtil.cc b/src/XrdClCurl/CurlUtil.cc
index 8fdbeed..5093bdf 100644
--- a/src/XrdClCurl/CurlUtil.cc
+++ b/src/XrdClCurl/CurlUtil.cc
@@ -674,6 +674,7 @@ HandlerQueue::Produce(std::shared_ptr<CurlOperation> handler)
         [&]{return m_ops.size() < m_max_pending_ops;}
     );
     if (std::chrono::steady_clock::now() > handler_expiry) {
+        lk.unlock();
         handler->Fail(XrdCl::errOperationExpired, 0, "Operation expired while waiting for worker");
         return;
     }
