Index: emi-trustmanager-3.0.3/src/org/glite/security/util/proxy/ProxyPolicy.java
===================================================================
--- emi-trustmanager-3.0.3.orig/src/org/glite/security/util/proxy/ProxyPolicy.java	2014-11-25 14:28:46.000000000 -0600
+++ emi-trustmanager-3.0.3/src/org/glite/security/util/proxy/ProxyPolicy.java	2014-11-25 16:43:09.905549620 -0600
@@ -18,6 +18,8 @@
 
 package org.glite.security.util.proxy;
 
+import java.io.IOException;
+
 import org.bouncycastle.asn1.ASN1Object;
 import org.bouncycastle.asn1.ASN1EncodableVector;
 import org.bouncycastle.asn1.ASN1OctetString;
@@ -142,7 +144,11 @@
         ASN1EncodableVector v = new ASN1EncodableVector();
         v.add(new ASN1ObjectIdentifier(m_oid));
         if (m_policy != null) {
-            v.add(new DEROctetString(m_policy));
+            try {
+                v.add(new DEROctetString(m_policy));
+            } catch (IOException e) {
+                // nothing here
+            }
         }
 
         return new DERSequence(v);
