Index: emi-trustmanager-3.0.3/src/org/glite/security/util/proxy/ProxyRestrictionData.java
===================================================================
--- emi-trustmanager-3.0.3.orig/src/org/glite/security/util/proxy/ProxyRestrictionData.java	2014-11-25 14:28:46.000000000 -0600
+++ emi-trustmanager-3.0.3/src/org/glite/security/util/proxy/ProxyRestrictionData.java	2014-11-25 16:05:17.241610021 -0600
@@ -168,7 +168,7 @@
         Enumeration<ASN1Primitive> subTreeEnum = subSeq.getObjects();
         while (subTreeEnum.hasMoreElements()) {
             ASN1Primitive object = subTreeEnum.nextElement();
-            vector.add(new GeneralSubtree((ASN1Sequence) object));
+            vector.add(new GeneralSubtree(GeneralName.getInstance(object)));
         }
     }
 
