--- src/org/glite/security/util/proxy/ProxyCertInfoExtension.java.orig	2012-09-17 14:55:48.000000000 -0500
+++ src/org/glite/security/util/proxy/ProxyCertInfoExtension.java	2012-09-17 14:55:57.000000000 -0500
@@ -47,7 +47,7 @@
     /** The oid of the proxy cert info extension, defined in the RFC 3820. */
     public static final String PROXY_CERT_INFO_EXTENSION_OID = "1.3.6.1.5.5.7.1.14";
     /** The oid of the rfc draft proxy cert extension. */
-    public static final String DRAFT_PROXY_CERT_INFO_EXTENSION_OID = "1.3.6.1.4.1.3536.1.1.222";
+    public static final String DRAFT_PROXY_CERT_INFO_EXTENSION_OID = "1.3.6.1.4.1.3536.1.222";
 
     /**
      * The sub proxy path lenght, default is not limited.
