commit 9a3eb04c011a324ec3e0e235d596d0a44d50a14c
Author: Carl Edquist <edquist@cs.wisc.edu>
Date:   Tue Feb 21 17:28:51 2017 -0600

    SSL_PROTOCOL_DEFAULT = TLSv1 -> TLSv1.2 (SOFTWARE-2523)

diff --git a/src/org/glite/security/trustmanager/ContextWrapper.java b/src/org/glite/security/trustmanager/ContextWrapper.java
index db7237d..e5be5b9 100644
--- a/src/org/glite/security/trustmanager/ContextWrapper.java
+++ b/src/org/glite/security/trustmanager/ContextWrapper.java
@@ -160,8 +160,8 @@ public class ContextWrapper implements SSLContextWrapper {
     /** CRL required default: true. */
     public static final String CRL_REQUIRED_DEFAULT = "true";
 
-    /** SSL protocol default: TLSv1. */
-    public static final String SSL_PROTOCOL_DEFAULT = "TLSv1";
+    /** SSL protocol default: TLSv1.2,TLSv1.1,TLSv1. */
+    public static final String SSL_PROTOCOL_DEFAULT = "TLSv1.2";
 
     /** CRL support enabled default: true. */
     public static final String CRL_ENABLED_DEFAULT = "true";
