From 4785adb1aea65fcc841e69757cf68dfc7f5cf15e Mon Sep 17 00:00:00 2001
From: Matevz Tadel <mtadel@ucsd.edu>
Date: Fri, 28 Feb 2025 09:21:39 -0800
Subject: [PATCH 10/10] Fix gstream configuration processing.

---
 src/XrdXrootd/XrdXrootdConfigMon.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/XrdXrootd/XrdXrootdConfigMon.cc b/src/XrdXrootd/XrdXrootdConfigMon.cc
index 46b07b435..51a790393 100644
--- a/src/XrdXrootd/XrdXrootdConfigMon.cc
+++ b/src/XrdXrootd/XrdXrootdConfigMon.cc
@@ -104,7 +104,7 @@ bool XrdXrootdProtocol::ConfigGStream(XrdOucEnv &myEnv, XrdOucEnv *urEnv)
    XrdXrootdGStream *gs;
    static const int numgs=sizeof(gsObj)/sizeof(struct XrdXrootdGSReal::GSParms);
    char vbuff[64];
-   bool aOK, gXrd[numgs] = {false, false, true, false, true};
+   bool aOK, gXrd[numgs] = {false, false, false, true, false, true};
 
 // For each enabled monitoring provider, allocate a g-stream and put
 // its address in our environment.
-- 
2.43.5

