From d28399667342c6d9df22b1490a19ca06bf395b2a Mon Sep 17 00:00:00 2001
From: Brian Bockelman <bbockelman@morgridge.org>
Date: Thu, 13 Mar 2025 08:54:59 -0500
Subject: [PATCH 12/12] Temporary workaround for segfault noted in upstream
 #2450

---
 src/XrdPfc/XrdPfcPathParseTools.hh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/XrdPfc/XrdPfcPathParseTools.hh b/src/XrdPfc/XrdPfcPathParseTools.hh
index 2d445c6eb..029cd51bb 100644
--- a/src/XrdPfc/XrdPfcPathParseTools.hh
+++ b/src/XrdPfc/XrdPfcPathParseTools.hh
@@ -95,7 +95,7 @@ struct PathTokenizer : private SplitParser
          if (t == 0) break;
          m_dirs.emplace_back(t);
       }
-      if (parse_as_lfn && *get_reminder() == 0 && ! m_dirs.empty())
+      if (parse_as_lfn && (get_reminder() == 0 || *get_reminder() == 0) && ! m_dirs.empty())
       {
          m_reminder = m_dirs.back();
          m_dirs.pop_back();
-- 
2.43.5

