From 1e9b480f28c7c59a3048dd8ee63a8dd39bafa443 Mon Sep 17 00:00:00 2001
From: Brian Lin <brianhlin@gmail.com>
Date: Fri, 11 Mar 2022 12:46:16 -0600
Subject: [PATCH 17/17] Bump the release number (SOFTWARE-5085)

---
 packaging/makesrpm.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xrootd/packaging/makesrpm.sh b/xrootd/packaging/makesrpm.sh
index f595c75da..e0e522df0 100755
--- a/xrootd/packaging/makesrpm.sh
+++ b/xrootd/packaging/makesrpm.sh
@@ -149,7 +149,7 @@ fi
 #-------------------------------------------------------------------------------
 # Deal with release candidates
 #-------------------------------------------------------------------------------
-RELEASE=1
+RELEASE=1.1
 if test x`echo $VERSION | egrep $RCEXP` != x; then
   RELEASE=0.`echo $VERSION | sed 's/.*-rc/rc/'`
   VERSION=`echo $VERSION | sed 's/-rc.*//'`
-- 
2.25.1

