commit 3cd4200f5f1608169581816dd922b0b00e0b8781 (HEAD, master)
Author: David Smith <david.smith@cern.ch>
Date:   Wed Jan 31 10:45:22 2024 +0100

    [XrdSecsss] Fix possible buffer overrun when serialising creds. Closes #2143

diff --git a/src/XrdSecsss/XrdSecsssEnt.cc b/src/XrdSecsss/XrdSecsssEnt.cc
index fdfa8657b..cfad10ace 100644
--- a/src/XrdSecsss/XrdSecsssEnt.cc
+++ b/src/XrdSecsss/XrdSecsssEnt.cc
@@ -235,7 +235,7 @@ bool XrdSecsssEnt::Serialize()
 // must be at the end because it can optionally be pruned when returned.
 //
    if (eP->credslen && eP->credslen <= XrdSecsssRR_Data::MaxCSz)
-      {tLen += eP->credslen;
+      {tLen += eP->credslen + 3;
        incCreds = true;
       }
