From eb4295e0a2fc17c2210186424f949001eb4af93f Mon Sep 17 00:00:00 2001
From: Andrew Hanushevsky <abh@slac.stanford.edu>
Date: Wed, 16 Oct 2024 04:41:33 -0700
Subject: [PATCH] [Server] Fix for improper error handling for current R5 also
 fixes #2357

---
 src/XrdOfs/XrdOfs.cc | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/XrdOfs/XrdOfs.cc b/src/XrdOfs/XrdOfs.cc
index cb8d89cdaf0..e91d728b487 100644
--- a/src/XrdOfs/XrdOfs.cc
+++ b/src/XrdOfs/XrdOfs.cc
@@ -2549,12 +2549,19 @@ int XrdOfs::Emsg(const char    *pfx,    // Message prefix value
 // If the error is EBUSY then we just need to stall the client. This is
 // a hack in order to provide for proxy support
 //
+// The hack unfotunately is now beinng triggered for reads and writes when
+// it was never so before (presumably due to client changes). So do not
+// apply the hack for these operations. This gets a better fix in R 6.0
+//
+if (strcmp("read", op) && strcmp("readv", op) && strcmp("pgRead", op) && 
+    strcmp("write",op) && strcmp("pgwrite",op)) {
     if (ecode < 0) ecode = -ecode;
     if (ecode == EBUSY) return 5;  // A hack for proxy support
 
 // Check for timeout conditions that require a client delay
 //
    if (ecode == ETIMEDOUT) return OSSDelay;
+   }
 
 // Format the error message
 //

