commit 0d6850050a6df922144ce6384be72baa4e48ae63
Author: Andrew Hanushevsky <abh@stanford.edu>
Date:   Mon Jul 20 11:49:22 2020 -0700

    [CMS] Use correct adminpath for creating Unix socket. Fixes #1255

diff --git a/src/XrdCms/XrdCmsConfig.cc b/src/XrdCms/XrdCmsConfig.cc
index 6618a99..164355a 100644
--- xrootd/src/XrdCms/XrdCmsConfig.cc
+++ xrootd/src/XrdCms/XrdCmsConfig.cc
@@ -926,7 +926,6 @@ int XrdCmsConfig::ConfigProc(int getrole)
                 ||  !strcmp(var, "oss.localroot")
                 ||  !strcmp(var, "oss.remoteroot")
                 ||  !strcmp(var, "oss.namelib")
-                ||  !strcmp(var, "all.adminpath")
                 ||  !strcmp(var, "all.export")
                 ||  !strcmp(var, "all.manager")
                 ||  !strcmp(var, "all.role")
@@ -1420,7 +1419,7 @@ int XrdCmsConfig::xapath(XrdSysError *eDest, XrdOucStream &CFile)
 // Record the path
 //
    if (AdminPath) free(AdminPath);
-   AdminPath = XrdOucUtils::genPath(pval,XrdOucUtils::InstName(myName,0));
+   AdminPath = XrdOucUtils::genPath(pval,XrdOucUtils::InstName(myInsName,0));
    free(pval);
    AdminMode = mode;
    return 0;
