From 187320500cbbefa284a6c27ee16157dea1c6b05a Mon Sep 17 00:00:00 2001
From: Andrew Hanushevsky <abh@stanford.edu>
Date: Wed, 25 Mar 2020 18:52:02 -0700
Subject: [PATCH] [Server] Add missing initializer to avoid SEGV on busy
 systems.

---
 src/XrdOfs/XrdOfsTPCJob.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/XrdOfs/XrdOfsTPCJob.cc b/src/XrdOfs/XrdOfsTPCJob.cc
index 897317e..f85a5dd 100644
--- a/xrootd/src/XrdOfs/XrdOfsTPCJob.cc
+++ b/xrootd/src/XrdOfs/XrdOfsTPCJob.cc
@@ -58,7 +58,7 @@ XrdOfsTPCJob::XrdOfsTPCJob(const char *Url, const char *Org,
                            const char *Cks, short lfnLoc[2],
                            const char *Spr, const char *Tpr)
                           : XrdOfsTPC(Url, Org, Lfn, Pfn, Cks, Spr, Tpr),
-                            myProg(0), Status(isWaiting)
+                            Next(0), myProg(0), eCode(0), Status(isWaiting)
 {  lfnPos[0] = lfnLoc[0]; lfnPos[1] = lfnLoc[1]; }
   
 /******************************************************************************/
-- 
1.8.3.1

