commit 7f2367037fc518624083a244a1144f362f0fcf36
Author: Andrew Hanushevsky <abh@stanford.edu>
Date:   Thu Apr 9 16:09:15 2020 -0700

    [Proxy] Correct length calculation when constructing login ID.

diff --git a/src/XrdPss/XrdPssUrlInfo.cc b/src/XrdPss/XrdPssUrlInfo.cc
index cd8ed32..a955eba 100644
--- a/xrootd/src/XrdPss/XrdPssUrlInfo.cc
+++ b/xrootd/src/XrdPss/XrdPssUrlInfo.cc
@@ -133,11 +133,11 @@ void XrdPssUrlInfo::setID(const char *tid)
 //
    if (tid == 0) tid = tident;
    if ((colon = index(tid, ':')) && (atP = index(colon+1, '@')))
-      {int n = atP - colon + 2;
+      {int n = atP - colon;
        if (n <= (int)sizeof(theID))
           {*theID = 'u';
            strncpy(theID+1, colon+1, n); // Include '@'
-           theID[n] = 0;
+           theID[n+1] = 0;
           } else *theID = 0;
        } else *theID = 0;
 }
