Index: xrootd-4.2.1/src/XrdFileCache/XrdFileCacheIOEntireFile.cc
===================================================================
--- xrootd-4.2.1.orig/src/XrdFileCache/XrdFileCacheIOEntireFile.cc
+++ xrootd-4.2.1/src/XrdFileCache/XrdFileCacheIOEntireFile.cc
@@ -86,6 +86,10 @@ int IOEntireFile::Read (char *buff, long
 {
    clLog()->Debug(XrdCl::AppMsg, "IO::Read() [%p]  %lld@%d %s", this, off, size, m_io.Path());
 
+   // protect from reads over the file size
+   if (off + size > m_io.FSize())
+       size =  m_io.FSize() - off;
+
    ssize_t bytes_read = 0;
    ssize_t retval = 0;
 
@@ -97,15 +101,10 @@ int IOEntireFile::Read (char *buff, long
       bytes_read += retval;
       buff += retval;
       size -= retval;
-   }
-
 
-   if ((size > 0))
-   {
-      clLog()->Debug(XrdCl::AppMsg, "IO::Read() missed %d bytes %s", size, m_io.Path());
-      if (retval > 0) bytes_read += retval;
+      if ((size > 0))
+         clLog()->Debug(XrdCl::AppMsg, "IO::Read() missed %d bytes %s", size, m_io.Path());
    }
-
    if (retval < 0)
    {
       clLog()->Error(XrdCl::AppMsg, "IO::Read(), origin bytes read %d %s", retval, m_io.Path());
Index: xrootd-4.2.1/src/XrdFileCache/XrdFileCacheIOFileBlock.cc
===================================================================
--- xrootd-4.2.1.orig/src/XrdFileCache/XrdFileCacheIOFileBlock.cc
+++ xrootd-4.2.1/src/XrdFileCache/XrdFileCacheIOFileBlock.cc
@@ -132,6 +132,10 @@ int IOFileBlock::Read (char *buff, long 
    int bytes_read = 0;
    clLog()->Debug(XrdCl::AppMsg, "IOFileBlock::Read() %lld@%d block range [%d-%d] \n %s", off, size, idx_first, idx_last, m_io.Path());
 
+   // protect from reads over the file size
+   if (off + size > m_io.FSize())
+       size =  m_io.FSize() - off;
+
    for (int blockIdx = idx_first; blockIdx <= idx_last; ++blockIdx )
    {
       // locate block
@@ -192,15 +196,10 @@ int IOFileBlock::Read (char *buff, long 
          bytes_read += retvalBlock;
          buff += retvalBlock;
          off += retvalBlock;
-         long long remain = readBlockSize - retvalBlock;
-
-         // cancel read if not successfull
-         if (remain > 0)
-         {
-            clLog()->Warning(XrdCl::AppMsg, "IOFileBlock::Read() Incomplete prefetch block[%d] readSize =%d  offset[%lld], remain = %d %s",
-                                    blockIdx, readBlockSize, off, remain, m_io.Path());
-            return bytes_read;
-         }
+      }
+      else if (retvalBlock > 0) {
+         clLog()->Warning(XrdCl::AppMsg, "IOFileBlock::Read() incomplete read, missing bytes %d %s", readBlockSize-retvalBlock, m_io.Path());
+         return bytes_read + retvalBlock;
       }
       else
       {
