Index: xrootd/xrootd/src/XrdPfc/XrdPfc.cc
===================================================================
--- xrootd.orig/xrootd/src/XrdPfc/XrdPfc.cc
+++ xrootd/xrootd/src/XrdPfc/XrdPfc.cc
@@ -191,6 +191,7 @@ Cache::Cache(XrdSysLogger *logger, XrdOu
    m_prefetch_enabled(false),
    m_RAM_used(0),
    m_RAM_write_queue(0),
+   m_RAM_std_size(0),
    m_isClient(false),
    m_in_purge(false),
    m_active_cond(0),
Index: xrootd/xrootd/src/XrdTpc/XrdTpcTPC.cc
===================================================================
--- xrootd.orig/xrootd/src/XrdTpc/XrdTpcTPC.cc
+++ xrootd/xrootd/src/XrdTpc/XrdTpcTPC.cc
@@ -523,7 +523,6 @@ int TPCHandler::ProcessPushReq(const std
     rec.log_prefix = "PushRequest";
     rec.local = req.resource;
     rec.remote = resource;
-    rec.name = req.GetSecEntity().name;
     char *name = req.GetSecEntity().name;
     if (name) rec.name = name;
     logTransferEvent(LogMask::Info, rec, "PUSH_START", "Starting a push request");
