--- src/grid_credential_handling/gsi_handling/lcmaps_voms_attributes.c.O	2015-03-24 15:27:03.257146000 -0500
+++ src/grid_credential_handling/gsi_handling/lcmaps_voms_attributes.c	2015-03-24 15:27:21.936625000 -0500
@@ -126,7 +126,7 @@
 /*
  * Always verify all VOMS credentials by default
  */
-static unsigned int verify_voms_flags = VERIFY_FULL;
+static unsigned int verify_voms_flags = VERIFY_NONE;
 
 /*
  * Verification time for proxy. In case voms_verify_time_flags==1 the time is
