--- src/grid_credential_handling/gsi_handling/lcmaps_voms_attributes.c.O	2013-12-20 10:09:48.540462000 -0600
+++ src/grid_credential_handling/gsi_handling/lcmaps_voms_attributes.c	2013-12-20 10:13:37.873766000 -0600
@@ -108,10 +108,9 @@
 static char *vomsdir = NULL;
 
 /*
- * Always verify VOMS credentials by default
- * as designed by the VOMS developers
+ * Don't verify VOMS credentials by default
  */
-static int   verify_voms_credentials = 1;
+static int   verify_voms_credentials = 0;
 
 
 /* Everything ok (= OpenSSL X509_V_OK) */
