--- src/grid_credential_handling/gsi_handling/lcmaps_voms_attributes.c.orig	2014-02-28 06:55:27.000000000 -0600
+++ src/grid_credential_handling/gsi_handling/lcmaps_voms_attributes.c	2014-12-10 16:56:48.000000000 -0600
@@ -110,10 +110,9 @@
 static char *vomsdir = NULL;
 
 /*
- * Always verify VOMS credentials by default
- * as designed by the VOMS developers
+ * Don't verify VOMS credentials by default
  */
-static int   verify_voms_credentials = 1;
+static int   verify_voms_credentials = 0;
 
 #ifdef GRST_ACTIVE
 
