diff -u --recursive ../lcmaps-1.4.28.orig/src/grid_credential_handling/lcmaps_credential.c ./src/grid_credential_handling/lcmaps_credential.c
--- ../lcmaps-1.4.28.orig/src/grid_credential_handling/lcmaps_credential.c	2012-02-03 14:17:55.945151255 -0600
+++ ./src/grid_credential_handling/lcmaps_credential.c	2012-02-03 14:18:20.830006588 -0600
@@ -192,6 +192,22 @@
             }
         }
     }
+ 
+    /*
+     * Retrieve a newly created X509 struct and X509 chain from gss credential (owned by lcmaps_credential)
+     */
+    if ((plcmaps_credential->px509_cred == NULL) && !(plcmaps_credential->px509_cred = lcmaps_cred_to_x509(gss_credential))) {
+        lcmaps_log(0, "%s: could not get X509 cred!\n", logstr);
+        retval |= LCMAPS_CRED_ERROR;
+    } else if ((plcmaps_credential->pem_string == NULL) && (lcmaps_x509_to_pem_string(&(plcmaps_credential->pem_string), plcmaps_credential->px509_cred))) {
+        lcmaps_log(0, "%s: could not extract X509 cred to PEM string!\n", logstr);
+        retval |= LCMAPS_CRED_ERROR;
+    }
+    if ((plcmaps_credential->px509_chain == NULL) && !(plcmaps_credential->px509_chain = lcmaps_cred_to_x509_chain(gss_credential))) {
+        lcmaps_log(0, "%s: could not get X509 chain!\n", logstr);
+        retval |= LCMAPS_CRED_ERROR;
+    } 
+
     return retval;
 }
 
Only in ./src/grid_credential_handling: lcmaps_credential.c.orig
diff -u --recursive ../lcmaps-1.4.28.orig/src/grid_credential_handling/x509_handling/lcmaps_x509_utils.c ./src/grid_credential_handling/x509_handling/lcmaps_x509_utils.c
--- ../lcmaps-1.4.28.orig/src/grid_credential_handling/x509_handling/lcmaps_x509_utils.c	2012-02-03 14:17:55.944151213 -0600
+++ ./src/grid_credential_handling/x509_handling/lcmaps_x509_utils.c	2012-02-03 14:18:20.829151620 -0600
@@ -223,6 +223,40 @@
     return dn;
 }
 
+
+/******************************************************************************
+Function:       lcmaps_x509_to_pem_string
+Description:    Reads a PEM String from a X509 object
+Parameters:
+                cert_string: PEM-encoded string 
+                px: pointer to a X509 certificate
+Returns:        0 on success, non-zero otherwise
+******************************************************************************/
+
+int lcmaps_x509_to_pem_string(char **certstring, X509 *px)
+{
+    if (px == NULL) return -1;
+    BUF_MEM *bptr;
+    BIO* mem = BIO_new(BIO_s_mem());
+    if (!mem) {
+        lcmaps_log (1, "OpenSSL memory allocation failed!");
+        return 1;
+    }
+    if (!PEM_write_bio_X509(mem, px)) {
+        lcmaps_log (1, "Unable to write PEM string into new memory object.");
+        return 2;
+    }
+    long i = BIO_get_mem_data(mem, &bptr);
+    *certstring = (char *)malloc(i+1);
+    if (!(*certstring)) {
+        lcmaps_log (1, "Unable to allocate memory for new cert string.");
+        return 3;
+    }
+    memcpy(*certstring, bptr, i); *((*certstring)+i) = '\0';
+    BIO_free_all(mem);
+    return 0;
+}
+
 /******************************************************************************
 Function:       lcmaps_pem_string_to_x509
 Description:    Reads a X509 certificate from a PEM-encoded string
