--- src/grid_credential_handling/gsi_handling/lcmaps_voms_attributes.c.orig	2011-10-23 16:18:55.000000000 -0500
+++ src/grid_credential_handling/gsi_handling/lcmaps_voms_attributes.c	2011-10-23 16:20:19.000000000 -0500
@@ -332,7 +332,7 @@
   int handle, attrSrcNum;
   int errNo = 0;
   int i, j, k;
-  struct attribute *tmp_list;
+  struct attribute *tmp_list, tmp_attr;
 
   char *logstr = "\t  get_attributes()";
 
@@ -379,7 +379,8 @@
 	tmp_list = NULL;
 	return 1;
       }
-      VOMS_GetAttribute(vo, handle, j, &(tmp_list[k++]), vd, &errNo);
+      tmp_attr.name = tmp_attr.qualifier = tmp_attr.value = NULL;
+      VOMS_GetAttribute(vo, handle, j, &tmp_attr, vd, &errNo);
       if (errNo) {
 	lcmaps_log(0, "%s Error: VOMS_GetAttribute, %s \n", logstr,
 		   retmsg[errNo]);
@@ -387,6 +388,10 @@
 	tmp_list = NULL;
 	return 1;
       }
+      if (tmp_attr.name) tmp_list[k].name = strdup(tmp_attr.name);
+      if (tmp_attr.value) tmp_list[k].value = strdup(tmp_attr.value);
+      if (tmp_attr.qualifier) tmp_list[k].qualifier = strdup(tmp_attr.qualifier);
+      k++;
     }
   }
   // XXX should the 'struct lcmaps_voms_generic_attr_t' differ from the voms_apic.h 
