Index: src/Makefile.am
===================================================================
--- src/Makefile.am	(revision 15292)
+++ src/Makefile.am	(revision 15293)
@@ -145,7 +145,7 @@
 	$(interfacedir)/lcmaps_verify_account_from_pem.h \
         lcmaps_verify_account_from_pem.c
 liblcmaps_verify_account_from_pem_la_LDFLAGS = $(VOMS_CPP_LIBS)
-liblcmaps_verify_account_from_pem_la_LIBADD = -L$(libdir) -llcmaps
+liblcmaps_verify_account_from_pem_la_LIBADD = ./liblcmaps.la
 
 
 
@@ -154,7 +154,7 @@
         $(interfacedir)/lcmaps_return_account_from_pem.h \
         lcmaps_return_account_from_pem.c
 liblcmaps_return_account_from_pem_la_LDFLAGS = $(VOMS_CPP_LIBS)
-liblcmaps_return_account_from_pem_la_LIBADD = -L$(libdir) -llcmaps
+liblcmaps_return_account_from_pem_la_LIBADD = ./liblcmaps.la
 
 
 
@@ -165,7 +165,7 @@
 liblcmaps_return_poolindex_la_LDFLAGS = \
     $(VOMS_CPP_LIBS)
 liblcmaps_return_poolindex_la_LIBADD = \
-    -L$(libdir) -llcmaps \
+    ./liblcmaps.la \
     $(GLOBUS_COMMON_LIBS) $(GLOBUS_COMMON_NOTHR_LIBS) \
     $(GLOBUS_GSS_ASSIST_LIBS) $(GLOBUS_GSS_ASSIST_NOTHR_LIBS) \
     $(GLOBUS_GSSAPI_GSI_LIBS) $(GLOBUS_GSSAPI_GSI_NOTHR_LIBS)
@@ -173,7 +173,7 @@
 liblcmaps_gss_assist_gridmap_la_SOURCES = \
         lcmaps_gss_assist_gridmap.c
 liblcmaps_gss_assist_gridmap_la_LDFLAGS = $(VOMS_CPP_LIBS)
-liblcmaps_gss_assist_gridmap_la_LIBADD = -L$(libdir) -llcmaps
+liblcmaps_gss_assist_gridmap_la_LIBADD = ./liblcmaps.la
 
 
 
@@ -241,12 +241,12 @@
 	$(interfacedir)/lcmaps_return_poolindex_without_gsi.h \
 	lcmaps_return_poolindex_without_gsi.c
 liblcmaps_return_poolindex_without_gsi_la_LDFLAGS = $(VOMS_CPP_LIBS)
-liblcmaps_return_poolindex_without_gsi_la_LIBADD = -L$(libdir) -llcmaps_without_gsi
+liblcmaps_return_poolindex_without_gsi_la_LIBADD = ./liblcmaps_without_gsi.la
 
 liblcmaps_gss_assist_gridmap_without_gsi_la_SOURCES = \
         lcmaps_gss_assist_gridmap.c
 liblcmaps_gss_assist_gridmap_without_gsi_la_LDFLAGS = $(VOMS_CPP_LIBS)
-liblcmaps_gss_assist_gridmap_without_gsi_la_LIBADD = -L$(libdir) -llcmaps_without_gsi
+liblcmaps_gss_assist_gridmap_without_gsi_la_LIBADD = ./liblcmaps_without_gsi.la
 
 
 
