Index: voms-mysql-plugin-3.1.5.1/src/mysqlwrap.cc
===================================================================
--- voms-mysql-plugin-3.1.5.1.orig/src/mysqlwrap.cc	2012-04-11 17:25:18.161224943 -0500
+++ voms-mysql-plugin-3.1.5.1/src/mysqlwrap.cc	2012-04-11 17:25:52.119110697 -0500
@@ -53,11 +53,16 @@
 
 sqliface::interface *myinterface::getSession()
 {
+  if (isconnected == false)
+      reconnect();
   return this;
 }
 
-  void myinterface::releaseSession(sqliface::interface *face)
+void myinterface::releaseSession(sqliface::interface *face)
 {
+    if (mysql) mysql_close(mysql);
+    mysql = NULL;
+    isconnected = false;
 }
 
 char *myinterface::errorMessage(void)
@@ -888,12 +893,16 @@
   std::string dn = std::string(dnname);
   std::string::size_type pos = 0;
 
+  // Why the hell is he doing this?? 
+
+  /*
   while((pos = ca.find_first_of("'", pos+3)) != std::string::npos)
     ca.insert(pos, "'");
 
   pos = 0;
   while((pos = dn.find_first_of("'", pos+3)) != std::string::npos)
     dn.insert(pos, "'");
+  */
 
   OPENSSL_free(caname);
   OPENSSL_free(dnname);
