--- netlogger/parsers/modules/gridftp_auth.py	2012-09-11 16:30:51.418552958 -0500
+++ netlogger/parsers/modules/gridftp_auth.py	2012-09-11 15:20:55.133288455 -0500
@@ -78,10 +78,9 @@
         elif message.startswith('Configuration'):
             event = dict(event=ns('config'))
         elif message.startswith('New'):
-            m = re.search('from:\s*(\S+)', message)
-            host, port =  m.group(1).split(':')
-            event = dict(event=ns('conn.start'), host=host, 
-                         port=int(port))
+            self.set_hostport(message)
+            event = dict(event=ns('conn.start'), host=self.host, 
+                         port=int(self.port))
         elif message.startswith('DN'):
              m = re.search('DN\s+(\/.*)\s+successfully', message)
              event = dict(event=ns('conn.auth.dn'), DN=m.group(1))
