--- netlogger/parsers/modules/gridftp_auth.py	2010-04-29 15:01:18.000000000 -0500
+++ netlogger/parsers/modules/gridftp_auth.py	2012-08-03 10:34:40.510405000 -0500
@@ -60,8 +60,11 @@
         elif message.startswith('Configuration'):
             event = dict(event=ns('config'))
         elif message.startswith('New'):
-            m = re.search('from:\s*(\S+)', message)
-            host, port =  m.group(1).split(':')
+            #m = re.search('from:\s*(\S+)', message)
+            #host, port =  m.group(1).split(':')
+            m = message.split(":", 1)[1]
+            m = m.translate(None, " []")
+            host,port = m.rsplit(':', 1)
             event = dict(event=ns('conn.start'), host=host, 
                          port=int(port))
         elif message.startswith('DN'):
