--- netlogger/parsers/modules/gridftp_auth.py	2012-11-20 11:21:41.347221469 -0600
+++ netlogger/parsers/modules/gridftp_auth.py	2012-11-20 11:22:07.742279369 -0600
@@ -15,6 +15,12 @@
 from netlogger.nldate import parseSyslogDate
 from netlogger import util
 
+logger = logging.getLogger('netlogger.parsers.modules.gridftp_auth')
+filehdlr = logging.FileHandler('/var/log/gratia/netlogger.log','w')
+formatter = logging.Formatter('%(asctime)s %(levelname)s %(message)s')
+filehdlr.setFormatter(formatter)
+logger.addHandler(filehdlr)
+
 def ns(s):
     return 'gridftp_auth.%s' % s
 
@@ -53,7 +59,7 @@
         try:
            self.port = int(message[rcolon+1:])
         except ValueError:
-           raise InvalidURL("Nonnumeric port: '%s'" % message[rcolon+1:])
+           raise ValueError("Nonnumeric port: '%s'" % message[rcolon+1:])
         rsqbrace=message.rfind(']')
         if rcolon-rsqbrace==1:
            lsqbrace=message.rfind('[')
