Index: seg/seg_sge_module.c
===================================================================
--- seg/seg_sge_module.c.orig	2011-08-26 14:20:09.000000002 -0500
+++ seg/seg_sge_module.c	2013-05-28 18:38:55.000000002 -0500
@@ -1199,7 +1199,13 @@
 	}
 
 	/* Batch accounting: resources consumed by the job  */
-	if (strstr(fields[1], "acct") == fields[1]) 
+ 	/* 
+ 	 * UGE reporting file notes that an "extra" acct record is sent to the reporting file at
+ 	 * midnight for long running jobs. These will be uniquely identified with 
+ 	 * exit_status (fields[14]) = -1 and should be skipped.  rjporter 05/2013
+ 	 *
+ 	 */
+ 	if ((strstr(fields[1], "acct") == fields[1]) && !(strstr(fields[14], "-1") == fields[14]))
 	{
             char * job_id;
 	    int failed;
