--- lcmaps-plugins-voms-1.7.1/src/voms/lcmaps_voms_localaccount.c.O	2018-05-09 11:11:35.154401672 -0500
+++ lcmaps-plugins-voms-1.7.1/src/voms/lcmaps_voms_localaccount.c	2018-05-09 11:27:01.129041276 -0500
@@ -542,6 +542,7 @@
     struct passwd       *user_info   = NULL;
     int                 cnt_sec_gid  = 0;
     gid_t *             sec_gid      = NULL;
+    int                 voms_verification = 1;
 
     /* Set suitable logstr */
     if (lcmaps_mode == PLUGIN_RUN)
@@ -554,6 +555,26 @@
         goto fail_plugin;
     }
 
+#ifdef RTLD_DEFAULT
+    int (*lcmaps_verifies_voms)(void);
+    dlerror();
+    lcmaps_verifies_voms=dlsym(RTLD_DEFAULT,"lcmaps_is_set_to_verify_voms_attributes");
+    if (dlerror()==NULL)    {
+        voms_verification=lcmaps_verifies_voms();
+        lcmaps_log(LOG_DEBUG,"%s: lcmaps_is_set_to_verify_voms_attributes() returned %d\n", logstr, voms_verification);
+    }
+#else
+    /* No RTLD_DEFAULT, just hope the symbol exists in LCMAPS */
+    voms_verification=lcmaps_is_set_to_verify_voms_attributes();
+#endif
+    /* The security of this plugin depends on the fact that the VOMS attributes
+     * are valid.  Fail the plugin if VOMS verification has been disabled */
+    if (!voms_verification) {
+        lcmaps_log(LOG_ERR, "%s: VOMS plugin requested but VOMS validation "
+            "is not enabled.  This is an insecure combination; failing plugin\n", logstr);
+        goto fail_plugin;
+    }
+
     /* Try to get FQANs from LCMAPS values: */
     /* First try to obtain FQANs from the credential data (i.e. stored by other
      * plugins */
