--- gratia-probe-1.07.02e.back/common/ProbeConfigTemplate	2011-07-21 11:58:24.926880274 -0500
+++ gratia-probe-1.07.02e.orig/common/ProbeConfigTemplate	2011-07-21 14:41:00.052005261 -0500
@@ -60,8 +60,8 @@
     CertInfoLogPattern="MAGIC_VDT_LOCATION/var/log/accounting/blahp.log-*"
         Comments51="Location and wildcard pattern of log files that contains certificate information about the jobs in the format followed by the 'blah demons'."
 
-    DataFolder="MAGIC_VDT_LOCATION/gratia/var/data/"
-    WorkingFolder="MAGIC_VDT_LOCATION/gratia/var/tmp"
-    LogFolder="MAGIC_VDT_LOCATION/gratia/var/logs/"
+    DataFolder="/var/lib/gratia/data/"
+    WorkingFolder="/var/lib/gratia/tmp"
+    LogFolder="/var/lib/gratia/logs/"
 
 />
