diff -ru gratia-probe-1.07.02e.back/common/DebugPrint.py gratia-probe-1.07.02e.orig/common/DebugPrint.py
--- gratia-probe-1.07.02e.back/common/DebugPrint.py	2011-07-21 11:58:24.922880241 -0500
+++ gratia-probe-1.07.02e.orig/common/DebugPrint.py	2011-07-21 12:00:30.659036573 -0500
@@ -10,7 +10,7 @@
 """
 
 import sys
-import GratiaCore
+from gratia.common import GratiaCore
 import getopt
 import string
 
diff -ru gratia-probe-1.07.02e.back/common/GetProbeConfigAttribute.py gratia-probe-1.07.02e.orig/common/GetProbeConfigAttribute.py
--- gratia-probe-1.07.02e.back/common/GetProbeConfigAttribute.py	2011-07-21 11:58:24.926880274 -0500
+++ gratia-probe-1.07.02e.orig/common/GetProbeConfigAttribute.py	2011-07-21 12:01:11.146052273 -0500
@@ -1,7 +1,7 @@
 #!/usr/bin/env python
 
 import sys
-import GratiaCore
+from gratia.common import GratiaCore
 import getopt
 import xml.sax.saxutils
 
diff -ru gratia-probe-1.07.02e.back/common/GratiaPing.py gratia-probe-1.07.02e.orig/common/GratiaPing.py
--- gratia-probe-1.07.02e.back/common/GratiaPing.py	2011-07-21 11:58:24.926880274 -0500
+++ gratia-probe-1.07.02e.orig/common/GratiaPing.py	2011-07-21 12:00:15.061036058 -0500
@@ -3,7 +3,7 @@
 #@(#)gratia/probe/common:$HeadURL: https://gratia.svn.sourceforge.net/svnroot/gratia/trunk/probe/common/GratiaPing.py $:$Id: GratiaPing.py 4068 2010-06-07 19:32:45Z pcanal $
 
 import getopt,sys
-import GratiaCore
+from gratia.common import GratiaCore
 
 class UsageError(Exception):
     def __init__(self, msg):
diff -ru gratia-probe-1.07.02e.back/dCache-storage/dCache-storage_meter.cron.sh gratia-probe-1.07.02e.orig/dCache-storage/dCache-storage_meter.cron.sh
--- gratia-probe-1.07.02e.back/dCache-storage/dCache-storage_meter.cron.sh	2011-07-21 11:58:24.932880377 -0500
+++ gratia-probe-1.07.02e.orig/dCache-storage/dCache-storage_meter.cron.sh	2011-07-21 14:18:10.273067986 -0500
@@ -10,8 +10,9 @@
 Logger="/usr/bin/logger -s -t $PGM"
 
 Meter_BinDir=$(dirname $0)
+probeconfig_loc = /etc/gratia/dcache-storage/ProbeConfig
 
-eval `grep VDTSetupFile ${Meter_BinDir}/ProbeConfig`
+eval `grep VDTSetupFile $probeconfig_loc`
 for Setupsh in ${VDTSetupFile} '/opt/vdt/setup.sh' '/opt/osg-ce/setup.sh'
 do
   if [[ -f ${Setupsh} && -r ${Setupsh} ]]; then
@@ -32,7 +33,7 @@
 
 # Need to be sure there is not one of these running already
 NCMeter=`ps -ef | grep dCache_storage_probe.py | grep -v grep | wc -l`
-eval `grep WorkingFolder ./ProbeConfig`
+eval `grep WorkingFolder $probeconfig_loc`
 if [ ${NCMeter} -ne 0 -a -e ${WorkingFolder}/dCache-storage_meter.cron.pid ]; then
   # We might have a condor_meter.pl running, let's verify that we 
   # started it.
diff -ru gratia-probe-1.07.02e.back/dCache-storage/GratiaConnector.py gratia-probe-1.07.02e.orig/dCache-storage/GratiaConnector.py
--- gratia-probe-1.07.02e.back/dCache-storage/GratiaConnector.py	2011-07-21 11:58:24.932880377 -0500
+++ gratia-probe-1.07.02e.orig/dCache-storage/GratiaConnector.py	2011-07-21 14:28:54.792052462 -0500
@@ -13,8 +13,8 @@
 has_gratia = True
 try:
     import Gratia
-    import StorageElement
-    import StorageElementRecord
+    import gratia.services.StorageElement
+    import gratia.services.StorageElementRecord
 except:
     has_gratia = False
     Gratia = None
diff -ru gratia-probe-1.07.02e.back/glexec/glexec_meter.cron.sh gratia-probe-1.07.02e.orig/glexec/glexec_meter.cron.sh
--- gratia-probe-1.07.02e.back/glexec/glexec_meter.cron.sh	2011-07-21 11:58:24.930880519 -0500
+++ gratia-probe-1.07.02e.orig/glexec/glexec_meter.cron.sh	2011-07-21 14:25:44.133051834 -0500
@@ -10,8 +10,9 @@
 Logger="/usr/bin/logger -s -t $PGM"
 
 Meter_BinDir=$(dirname $0)
+probeconfig_loc=/etc/gratia/glexec/ProbeConfig
 
-eval `grep VDTSetupFile ${Meter_BinDir}/ProbeConfig`
+eval `grep VDTSetupFile $probeconfig_loc`
 for Setupsh in ${VDTSetupFile} '/opt/vdt/setup.sh' '/opt/osg-ce/setup.sh'
 do
   if [[ -f ${Setupsh} && -r ${Setupsh} ]]; then
diff -ru gratia-probe-1.07.02e.back/gridftp-transfer/gridftp-transfer_meter.cron.sh gratia-probe-1.07.02e.orig/gridftp-transfer/gridftp-transfer_meter.cron.sh
--- gratia-probe-1.07.02e.back/gridftp-transfer/gridftp-transfer_meter.cron.sh	2011-07-21 11:58:24.933880554 -0500
+++ gratia-probe-1.07.02e.orig/gridftp-transfer/gridftp-transfer_meter.cron.sh	2011-07-21 14:27:44.081051568 -0500
@@ -10,8 +10,9 @@
 Logger="/usr/bin/logger -s -t $PGM"
 
 Meter_BinDir=$(dirname $0)
+probeconfig_loc=/etc/gratia/gridftp-transfer/ProbeConfig
 
-eval `grep VDTSetupFile ${Meter_BinDir}/ProbeConfig`
+eval `grep VDTSetupFile $probeconfig_loc`
 for Setupsh in ${VDTSetupFile} '/opt/vdt/setup.sh' '/opt/osg-ce/setup.sh'
 do
   if [[ -f ${Setupsh} && -r ${Setupsh} ]]; then
@@ -32,7 +33,7 @@
 
 # Need to be sure there is not one of these running already
 NCMeter=`ps -ef | grep GridftpTransferProbeDriver.py | grep -v grep | wc -l`
-eval `grep WorkingFolder ./ProbeConfig`
+eval `grep WorkingFolder $probeconfig_loc`
 if [ ${NCMeter} -ne 0 -a -e ${WorkingFolder}/gridftp-transfer_meter.cron.pid ]; then
   # We might have a condor_meter.pl running, let's verify that we 
   # started it.
diff -ru gratia-probe-1.07.02e.back/hadoop-storage/hadoop_storage_probe gratia-probe-1.07.02e.orig/hadoop-storage/hadoop_storage_probe
--- gratia-probe-1.07.02e.back/hadoop-storage/hadoop_storage_probe	2011-07-21 11:58:24.937880517 -0500
+++ gratia-probe-1.07.02e.orig/hadoop-storage/hadoop_storage_probe	2011-07-21 14:28:44.330053946 -0500
@@ -33,8 +33,8 @@
 has_gratia = True
 try:
     import Gratia
-    import StorageElement
-    import StorageElementRecord
+    import gratia.services.StorageElement
+    import gratia.services.StorageElementRecord
 except:
     has_gratia = False
     Gratia = None
diff -ru gratia-probe-1.07.02e.back/metric/Metric.py gratia-probe-1.07.02e.orig/metric/Metric.py
--- gratia-probe-1.07.02e.back/metric/Metric.py	2011-07-21 11:58:24.931880432 -0500
+++ gratia-probe-1.07.02e.orig/metric/Metric.py	2011-07-21 14:29:52.667880054 -0500
@@ -3,7 +3,7 @@
 ## Updated by Arvind Gopu, Indiana University (http://peart.ucs.indiana.edu)
 ## More Updates by Arvind Gopu 2007-10-19
 
-import GratiaCore
+import gratia.common.GratiaCore
 from GratiaCore import *
 
 class MetricRecord(GratiaCore.Record):
diff -ru gratia-probe-1.07.02e.back/services/ComputeElement.py gratia-probe-1.07.02e.orig/services/ComputeElement.py
--- gratia-probe-1.07.02e.back/services/ComputeElement.py	2011-07-21 11:58:24.937880517 -0500
+++ gratia-probe-1.07.02e.orig/services/ComputeElement.py	2011-07-21 14:30:43.942005359 -0500
@@ -4,8 +4,8 @@
 
 import types
 
-import GratiaCore
-from GratiaCore import *
+import gratia.common.GratiaCore
+from gratia.common.GratiaCore import *
 
 class ComputeElement(GratiaCore.Record):
     "Base class for the Gratia ComputeElement"
diff -ru gratia-probe-1.07.02e.back/services/ComputeElementRecord.py gratia-probe-1.07.02e.orig/services/ComputeElementRecord.py
--- gratia-probe-1.07.02e.back/services/ComputeElementRecord.py	2011-07-21 11:58:24.936880762 -0500
+++ gratia-probe-1.07.02e.orig/services/ComputeElementRecord.py	2011-07-21 14:31:02.979958199 -0500
@@ -4,8 +4,8 @@
 
 import types
 
-import GratiaCore
-from GratiaCore import *
+import gratia.common.GratiaCore
+from gratia.common.GratiaCore import *
 
 class ComputeElementRecord(GratiaCore.Record):
     "Base class for the Gratia ComputeElementRecord"
diff -ru gratia-probe-1.07.02e.back/services/StorageElement.py gratia-probe-1.07.02e.orig/services/StorageElement.py
--- gratia-probe-1.07.02e.back/services/StorageElement.py	2011-07-21 11:58:24.937880517 -0500
+++ gratia-probe-1.07.02e.orig/services/StorageElement.py	2011-07-21 14:31:20.346083178 -0500
@@ -4,8 +4,8 @@
 
 import types
 
-import GratiaCore
-from GratiaCore import *
+import gratia.common.GratiaCore
+from gratia.common.GratiaCore import *
 
 class StorageElement(GratiaCore.Record):
     "Base class for the Gratia StorageElement"
diff -ru gratia-probe-1.07.02e.back/services/StorageElementRecord.py gratia-probe-1.07.02e.orig/services/StorageElementRecord.py
--- gratia-probe-1.07.02e.back/services/StorageElementRecord.py	2011-07-21 11:58:24.937880517 -0500
+++ gratia-probe-1.07.02e.orig/services/StorageElementRecord.py	2011-07-21 14:31:39.906098883 -0500
@@ -4,8 +4,8 @@
 
 import types
 
-import GratiaCore
-from GratiaCore import *
+import gratia.common.GratiaCore
+from gratia.common.GratiaCore import *
 
 class StorageElementRecord(GratiaCore.Record):
     "Base class for the Gratia StorageElementRecord"
diff -ru gratia-probe-1.07.02e.back/services/storageReport gratia-probe-1.07.02e.orig/services/storageReport
--- gratia-probe-1.07.02e.back/services/storageReport	2011-07-21 11:58:24.936880762 -0500
+++ gratia-probe-1.07.02e.orig/services/storageReport	2011-07-21 14:38:02.630911164 -0500
@@ -27,8 +27,8 @@
         sys.path.append(gratia_path)
 
 import Gratia
-import StorageElement
-import StorageElementRecord
+import gratia.services.StorageElement
+import gratia.services.StorageElementRecord
 
 log = logging.basicConfig(level=logging.INFO)
 
diff -ru gratia-probe-1.07.02e.back/services/Subcluster.py gratia-probe-1.07.02e.orig/services/Subcluster.py
--- gratia-probe-1.07.02e.back/services/Subcluster.py	2011-07-21 11:58:24.937880517 -0500
+++ gratia-probe-1.07.02e.orig/services/Subcluster.py	2011-07-21 14:31:54.177934171 -0500
@@ -4,8 +4,8 @@
 
 import types
 
-import GratiaCore
-from GratiaCore import *
+import gratia.common.GratiaCore
+from gratia.common.GratiaCore import *
 
 class Subcluster(GratiaCore.Record):
     "Base class for the Gratia Subcluster"
diff -ru gratia-probe-1.07.02e.back/sge/sge_meter.cron.sh gratia-probe-1.07.02e.orig/sge/sge_meter.cron.sh
--- gratia-probe-1.07.02e.back/sge/sge_meter.cron.sh	2011-07-21 11:58:24.930880519 -0500
+++ gratia-probe-1.07.02e.orig/sge/sge_meter.cron.sh	2011-07-21 14:33:25.999958288 -0500
@@ -24,12 +24,13 @@
 #sgeLog_Dir='/usr/local/vdt-1.3.9/globus/tmp/gram_job_state'  # on fermigrid1
 
 Meter_BinDir=$(dirname $0)
+probeconfig_loc=/etc/gratia/sge/ProbeConfig
 
 [[ -n "$1" ]] && sge_log_file="$1"
 
 # Setup the proper Grid environment
 # I need to know a default location to find this setup.sh script ??? ###
-eval `grep VDTSetupFile ${Meter_BinDir}/ProbeConfig`
+eval `grep VDTSetupFile $probeconfig_loc`
 for Setupsh in ${VDTSetupFile} '/root/setup.sh'
 do
   if [[ -f ${Setupsh} && -r ${Setupsh} ]]; then
@@ -50,7 +51,7 @@
   
 # Need to be sure there is not one of these running already
 NCMeter=`ps -ef | grep sge_meter.py | grep -v grep | wc -l`
-eval `grep WorkingFolder ./ProbeConfig`
+eval `grep WorkingFolder $probeconfig_loc`
 if [ ${NCMeter} -ne 0 -a -e ${WorkingFolder}/sge_meter.cron.pid ]; then
   # We might have a condor_meter.pl running, let's verify that we 
   # started it.
