--- _build_results/BUILD/gratia-probe-1.07.02e/dCache-transfer/gratia-dcache-transfer	2011-05-19 08:28:53.000000000 -0500
+++ dCache-transfer/gratia-dcache-transfer	2011-08-22 08:44:47.000000000 -0500
@@ -9,19 +9,8 @@
   HOME='/'; export HOME
 fi
 
-PROBE_DIR=/opt/d-cache/gratia/probe/dCache-transfer
-arch_spec_dir=`echo "${PROBE_DIR}/../lib."*`
-if test -n "$PYTHONPATH" ; then
-  if echo "$PYTHONPATH" | grep -e ':$' >/dev/null 2>&1; then
-    PYTHONPATH="${PYTHONPATH}${PROBE_DIR}/../common:${PYTHONPATH}${PROBE_DIR}/../lib.linux-i686-2.4:${arch_spec_dir}:"
-  else
-    PYTHONPATH="${PYTHONPATH}:${PROBE_DIR}/../common:${PYTHONPATH}${PROBE_DIR}/../lib.linux-i686-2.4:${arch_spec_dir}"
-  fi
-else
-  PYTHONPATH="${PROBE_DIR}/../common:${PYTHONPATH}${PROBE_DIR}/../lib.linux-i686-2.4:${arch_spec_dir}"
-fi
-export PYTHONPATH
-PROBE_CONFIG=$PROBE_DIR/ProbeConfig
+PROBE_DIR=/usr/share/gratia/probe/dCache-transfer
+PROBE_CONFIG=/etc/gratia/dCache-transfer/ProbeConfig
 
 # See if the probe is installed.
 [ -d $PROBE_DIR ] || exit 0
