commit b88ffb91978bb1a5bcd4cf1ac2c88eb02c7e5aa0
Author: Carl Edquist <edquist@cs.wisc.edu>
Date:   Fri Sep 23 16:43:49 2016 -0500

    provide default for CondorCEHistoryFolder in code (SOFTWARE-2463)

diff --git a/common/gratia/common/probe_config.py b/common/gratia/common/probe_config.py
index d16276b..0823ce6 100644
--- a/common/gratia/common/probe_config.py
+++ b/common/gratia/common/probe_config.py
@@ -349,7 +349,8 @@ class ProbeConfiguration:
         return self.__getConfigAttribute('GratiaExtension')
 
     def get_CondorCEHistoryFolder(self):
-        return self.__getConfigAttribute('CondorCEHistoryFolder')
+        _default = "/var/lib/gratia/condorce_data"
+        return self.__getConfigAttribute('CondorCEHistoryFolder') or _default
 
     def get_CertificateFile(self):
         return self.__getConfigAttribute('CertificateFile')
