--- gridftp-transfer/GftpLogParserCorrelator.py.orig     2011-08-29 17:53:02.000000000 -0500
+++ gridftp-transfer/GftpLogParserCorrelator.py  2011-08-29 17:53:32.000000000 -0500
@@ -149,7 +149,7 @@
                     break 
                 else:
                     binId = self.binInt(l['ts'])
-                    authEventPlaceHolder = self.findAndCreateTimeBin(binId,correlationBins,l['id'])
+                    authEventPlaceHolder = self.findAndCreateTimeBin(binId,correlationBins,l['PID'])
                     if ( authEventPlaceHolder != None):
                         authEventPlaceHolder.append(l)

--- gridftp-transfer/GridftpToGratiaEventTransformer.py.orig     2011-08-30 10:50:39.000000000 -0500
+++ gridftp-transfer/GridftpToGratiaEventTransformer.py  2011-08-30 10:53:40.000000000 -0500
@@ -1,4 +1,4 @@
-
+import netlogger
 import time
 import Gratia
 import Logger
@@ -81,7 +81,7 @@
         # Gratia will make a best effort to map this to the VO name.
 
         r.LocalUserId( localUserName )
-        r.LocalJobId( ftpEvent['guid'] )
+        r.LocalJobId(netlogger.parsers.base.getGuid(repr(ftpEvent["ts"]),ftpEvent["end"],localUserName,dstHost))
 
         return r


