--- gridftp-transfer/GftpLogParserCorrelator.py	2011-08-22 10:56:39.000000000 -0500
+++ gridftp_transfer/GftpLogParserCorrelator.py	2011-08-22 10:56:47.000000000 -0500
@@ -81,8 +81,8 @@
 class GftpLogParserCorrelator:
 
     def __init__(self,gftpTransferFile,gftpAuthFile):
-        self.authParser = netlogger.parsers.modules.gridftp_auth.Parser(gftpAuthFile,True, True )
-        self.transferParser  = netlogger.parsers.modules.gridftp.Parser(gftpTransferFile,True, True )
+        self.authParser = netlogger.parsers.modules.gridftp_auth.Parser(gftpAuthFile,True)
+        self.transferParser  = netlogger.parsers.modules.gridftp.Parser(gftpTransferFile,True)
         self.buffer = []
         self.i = 0
 
