--- gridftp-transfer/GridftpTransferProbeDriver.py	2011-05-19 08:28:52.000000000 -0500
+++ gridftp-transfer/GridftpTransferProbeDriver.py	2011-08-22 10:53:00.000000000 -0500
@@ -1,11 +1,12 @@
 
 import Gratia
-import GratiaConnector
-import GftpLogParserCorrelator
-import FileDigest
-import GridftpToGratiaEventTransformer
+import gratia.gridftp_transfer.GratiaConnector as GratiaConnector
+import gratia.gridftp_transfer.GftpLogParserCorrelator as GftpLogParserCorrelator
+import gratia.gridftp_transfer.FileDigest as FileDigest
+import gratia.gridftp_transfer.GridftpToGratiaEventTransformer as GridftpToGratiaEventTransformer
+from gratia.gridftp_transfer.ContextTransaction import ContextTransaction
 
-import Logger
+import gratia.gridftp_transfer.Logger as Logger
 import time
 import sys
 import os
@@ -61,7 +62,6 @@
          self.gridftplogFile = file(self.gridftplogFilePath)
          self.gridftpAuthlogFile = file(self.gridftpAuthlogFilePath)
        
-         from ContextTransaction import ContextTransaction
          self.txn = ContextTransaction(cfg.WorkingFolder+"/GridftpAccountingProbeState")
 
          self.context = self.txn.context()
@@ -121,7 +121,6 @@
    Logger.configureLogger(logger,cfg)
 
    workingDir = cfg.WorkingFolder
-   import os
    if not os.path.isdir( workingDir ):
             os.mkdir( workingDir , 0755 )
 
