--- /tmp/JobManagerGratia.pm	2011-08-18 20:35:34.000000000 -0500
+++ common/GRAM/JobManagerGratia.pm	2011-08-18 20:51:59.000000000 -0500
@@ -39,20 +39,7 @@
   my $jobmanager_name = ref $jobmanager;
   $jobmanager_name =~ s&^.*::&&;
   return if $jobmanager_name eq 'fork'; # Don't save info for non-managed fork jobs.
-  my $env = "$ENV{VDT_LOCATION}";
-  my $tmp_env;
-  unless ($env) {
-    $env = "$ENV{GLOBUS_LOCATION}";
-    if ($env) {
-      $env = "$env/..";
-    } else {
-      $env = ($ENV{TMPDIR} || "/var/tmp");
-      $tmp_env = 1;
-    }
-  }
-  my $log_dir = sprintf("%s/%s",
-                        $env,
-                        $tmp_env?'/':"gratia/var/data/");
+  my $log_dir = "/var/lib/gratia/data/";
   my $gratia_filename = sprintf("gratia_certinfo_%s_%s",
                                 $jobmanager_name,
                                 $job_id);
@@ -101,8 +88,7 @@
   close(GRATIA_CERTINFO);
   if ($DEBUG) {
     my $debug_log = sprintf("%s/%s%s",
-                            $env,
-                            $tmp_env?'/':"gratia/var/logs/",
+                            "/var/lib/gratia/var/logs/",
                             "certinfo_debug.log");
     if (open(DEBUG_OUT, ">>$debug_log")) {
       print DEBUG_OUT scalar localtime(),
