--- pbs-lsf/pbs-lsf_meter.cron.sh	2011-05-19 08:28:53.000000000 -0500
+++ pbs-lsf/pbs-lsf_meter.cron.sh	2011-08-22 11:21:47.000000000 -0500
@@ -49,16 +49,6 @@
 
 Meter_BinDir=$(dirname $0)
 
-eval `grep VDTSetupFile ${Meter_BinDir}/ProbeConfig`
-for Setupsh in ${VDTSetupFile} '/opt/vdt/setup.sh' '/opt/osg-ce/setup.sh'
-do
-  if [[ -f ${Setupsh} && -r ${Setupsh} ]]; then
-    # Should the output of this be directed to /dev/null?
-    . ${Setupsh} >/dev/null
-    break
-  fi
-done
-
 # Set the working directory, where we expect to find the following
 #    necessary files.
 if [ -d ${Meter_BinDir} ]; then
@@ -85,16 +75,6 @@
 done
 
 pp_dir=$(cd "$Meter_BinDir/../common"; pwd)
-if test -n "$PYTHONPATH" ; then
-  if echo "$PYTHONPATH" | grep -e ':$' >/dev/null 2>&1; then
-    PYTHONPATH="${PYTHONPATH}${pp_dir}:"
-  else
-    PYTHONPATH="${PYTHONPATH}:${pp_dir}"
-  fi
-else
-  PYTHONPATH="${pp_dir}"
-fi
-export PYTHONPATH
 
 export URCOLLECTOR_LOCATION=`pwd`  
 
@@ -105,24 +85,21 @@
   exit 0
 fi
 
-enabled=`${pp_dir}/GetProbeConfigAttribute.py EnableProbe`
+enabled=`${pp_dir}/GetProbeConfigAttribute EnableProbe`
 (( status = $? ))
 if (( $status != 0 )); then
   echo "ERROR checking probe configuration!" 1>&2
   exit $status
 fi
 if [[ -n "$enabled" ]] && [[ "$enabled" == "0" ]]; then
-  ${pp_dir}/DebugPrint.py -l -1 "Probe is not enabled: check $Meter_BinDir/ProbeConfig."
+  ${pp_dir}/DebugPrint -l -1 "Probe is not enabled: check $Meter_BinDir/ProbeConfig."
   exit 1
 fi
 
 log_file="`date +'%Y-%m-%d'`.log"
 
-# Remove erroneous log files from probe main area (should be in gratia/var/logs/)
-rm -f 2[0-9][0-9][0-9]-[0-1][0-9]-[0-3][0-9].log
-
 #--- run the probes ----
-./urCollector.pl --nodaemon 2>&1 | ${pp_dir}/DebugPrint.py -l 1
+./urCollector.pl --nodaemon 2>&1 | ${pp_dir}/DebugPrint -l 1
 # Now invoked by urCollector.pl
 #./pbs-lsf_meter.pl 2>&1
 
@@ -130,7 +107,7 @@
 
 # If the probe ended in error, report this in Syslog and exit
 if [ $ExitCode != 0 ]; then
-  ${pp_dir}/DebugPrint.py -l -1 "ALERT: $0 exited abnormally with [$ExitCode]"
+  ${pp_dir}/DebugPrint -l -1 "ALERT: $0 exited abnormally with [$ExitCode]"
   exit $ExitCode
 fi
   
--- pbs-lsf/pbs-lsf_meter.pl	2011-05-19 08:28:53.000000000 -0500
+++ pbs-lsf/pbs-lsf_meter.pl	2011-08-22 11:23:27.000000000 -0500
@@ -20,7 +20,7 @@
 use urCollector::Configuration;
 
 sub local_error {
-  system("$URCOLLECTOR_LOC/../common/DebugPrint.py", "-l", "-1", @_);
+  system("$URCOLLECTOR_LOC/../common/DebugPrint", "-l", "-1", @_);
 }
 
 ####################################
@@ -68,12 +68,12 @@
 
 my $status;
 if ($lrms_version) {
-  $status = system("$URCOLLECTOR_LOC/pbs-lsf.py",
+  $status = system("$URCOLLECTOR_LOC/pbs-lsf",
                    "$configValues{URBox}",
                    $lrms,
                    $lrms_version);
 } else {
-  $status = system("$URCOLLECTOR_LOC/pbs-lsf.py",
+  $status = system("$URCOLLECTOR_LOC/pbs-lsf",
                    "$configValues{URBox}", $lrms);
 }
 
