--- _build_results/BUILD/gratia-probe-1.07.02e/pbs-lsf/urCollector-src/urCollector.conf-template	2011-05-19 08:28:53.000000000 -0500
+++ pbs-lsf/urCollector-src/urCollector.conf-template	2011-08-22 08:54:02.000000000 -0500
@@ -9,19 +9,19 @@
 
 # This is the location of the directory where the urCollector puts the files
 # with the usage records.
-URBox = "/opt/urCollector/var/URBox/"
+URBox = "/var/lib/gratia/tmp/urCollector"
 
 # the lock file for urCollector.pl 
 collectorLockFileName = "/opt/urCollector/var/urCollector.lock"
 
 # the log file for urCollector.pl
-collectorLogFileName = "/opt/urCollector/var/log/urCollector.log"
+collectorLogFileName = "/var/lib/gratia/logs/urCollector.log"
 
 # the buffer file that keeps track of the last processed LRMS job_id and
 # log timestamp for urCollector.pl:
 # (Note: the LRMS type specified in lrmsType will automatically be appended
 #  as suffix, for example: "/opt/urCollector/var/urCollectorBuffer.pbs")
-collectorBufferFileName = "/opt/urCollector/var/urCollectorBuffer"
+collectorBufferFileName = "/var/lib/gratia/tmp/urCollectorBuffer"
 
 # if all jobs have been processed: seconds to wait before looking for new jobs in the LRMS log files (next iteration):
 mainPollInterval = "60"
@@ -29,11 +29,11 @@
 # if still jobs to process: number of jobs to process at each processing step
 # (several steps per mainPollInterval, depending on the number of jobs found
 # in the LRMS logs):
-jobPerTimeInterval = "10"
+jobPerTimeInterval = "1000"
 
 # time in seconds to sleep after each processing step (if there are still
 # jobs to process, otherwise start a new mainPollInterval):
-timeInterval = "5"
+timeInterval = "0"
 
 #siteName parameter can be used to assign to the UsageRecord a tag identifying
 # the site.
