Index: globus_gridftp_server-10.4/modules/file/globus_gridftp_server_file.c
===================================================================
--- globus_gridftp_server-10.4.orig/modules/file/globus_gridftp_server_file.c	2016-08-10 15:31:50.752818277 -0500
+++ globus_gridftp_server-10.4/modules/file/globus_gridftp_server_file.c	2016-08-10 15:51:58.472135252 -0500
@@ -17,10 +17,11 @@
 #include "globus_common.h"
 #include "globus_gridftp_server.h"
 #include "globus_xio.h"
 #include "globus_xio_file_driver.h"
 #include <openssl/md5.h>
+#include <zlib.h>
 #include "version.h"
 
 #include <utime.h>
 #ifndef TARGET_ARCH_WIN32
 #include <grp.h>
@@ -94,10 +95,13 @@ typedef void
 (*globus_l_gfs_file_cksm_cb_t)(
     globus_result_t                     result,
     char *                              cksm,
     void *                              user_arg);
 
+#define POSIX_CKSM_TYPE_ADLER32 1
+#define POSIX_CKSM_TYPE_MD5     2
+
 typedef struct globus_l_gfs_file_cksm_monitor_s
 {
     globus_gfs_operation_t              op;
     globus_off_t                        offset;
     globus_off_t                        length;
@@ -109,12 +113,15 @@ typedef struct globus_l_gfs_file_cksm_mo
     
     globus_callback_handle_t            marker_handle;
     int                                 marker_freq;
     globus_bool_t                       send_marker;
     globus_off_t                        total_bytes;
-    
+
+    unsigned char                       cksum_type;
     MD5_CTX                             mdctx;
+    uint32_t                            adler32ctx;
+
     globus_byte_t                       buffer[1];
 } globus_l_gfs_file_cksm_monitor_t;
 
 typedef struct 
 {
@@ -1713,13 +1720,15 @@ globus_l_gfs_file_cksm_read_cb(
     globus_xio_data_descriptor_t        data_desc,
     void *                              user_arg)
 {
     globus_l_gfs_file_cksm_monitor_t *  monitor;
     globus_bool_t                       eof = GLOBUS_FALSE;
+    char *                              cksmptr;
     char *                              md5ptr;
     unsigned char                       md[MD5_DIGEST_LENGTH];
     char                                md5sum[MD5_DIGEST_LENGTH * 2 + 1] = {0};
+    char                                adler32_human[2*sizeof(uint32_t)+1];
     int                                 i;    
     GlobusGFSName(globus_l_gfs_file_cksm_read_cb);
     GlobusGFSFileDebugEnter();
     
     monitor = (globus_l_gfs_file_cksm_monitor_t *) user_arg;
@@ -1746,12 +1755,19 @@ globus_l_gfs_file_cksm_read_cb(
         {
             eof = GLOBUS_TRUE;
         }
     }
     monitor->total_bytes += nbytes;
-    
-    MD5_Update(&monitor->mdctx, buffer, nbytes);
+
+    if((monitor->cksum_type & POSIX_CKSM_TYPE_MD5) == POSIX_CKSM_TYPE_MD5)
+    {
+        MD5_Update(&monitor->mdctx, buffer, nbytes);
+    }
+    if((monitor->cksum_type & POSIX_CKSM_TYPE_ADLER32) == POSIX_CKSM_TYPE_ADLER32)
+    {
+        monitor->adler32ctx = adler32(monitor->adler32ctx, buffer, nbytes);
+    }
 
     if(!eof)
     {
         if(monitor->send_marker)
         {
@@ -1789,35 +1805,51 @@ globus_l_gfs_file_cksm_read_cb(
                 NULL,
                 NULL);
             monitor->marker_handle = GLOBUS_NULL_HANDLE;
         }
         
-        MD5_Final(md, &monitor->mdctx);
-    
         globus_xio_register_close(
             handle,
             NULL,
             globus_l_gfs_file_close_cb,
             NULL);
-            
-        md5ptr = md5sum;
-        for(i = 0; i < MD5_DIGEST_LENGTH; i++)
-        {
-           sprintf(md5ptr, "%02x", md[i]);
-           md5ptr++;
-           md5ptr++;
+
+        if((monitor->cksum_type & POSIX_CKSM_TYPE_MD5) == POSIX_CKSM_TYPE_MD5) {
+            MD5_Final(md, &monitor->mdctx);
+            md5ptr = md5sum;
+            for(i = 0; i < MD5_DIGEST_LENGTH; i++)
+            {
+                sprintf(md5ptr, "%02x", md[i]);
+                md5ptr++;
+                md5ptr++;
+            }
+            md5ptr = '\0';
+            cksmptr = md5sum;
+        }
+        if((monitor->cksum_type & POSIX_CKSM_TYPE_ADLER32) == POSIX_CKSM_TYPE_ADLER32)
+        {
+            unsigned char * adler32_char = (unsigned char*)&monitor->adler32ctx;
+            char * adler32_ptr = adler32_human;
+            for (i = 0; i < sizeof(uint32_t); i++) {
+                sprintf(adler32_ptr, "%02x", adler32_char[sizeof(uint32_t)-1-i]);
+                adler32_ptr++;
+                adler32_ptr++;
+            }
+            adler32_ptr = '\0';
+            cksmptr = adler32_human;
         }
 
+
         if(monitor->internal_cb)
         {
             monitor->internal_cb(
-                GLOBUS_SUCCESS, md5sum, monitor->internal_cb_arg);
+                GLOBUS_SUCCESS, cksmptr, monitor->internal_cb_arg);
         }
         else
         {
             globus_gridftp_server_finished_command(
-                monitor->op, GLOBUS_SUCCESS, md5sum);
+                monitor->op, GLOBUS_SUCCESS, cksmptr);
         }   
         
         globus_free(monitor);
             
     }        
@@ -1916,11 +1948,12 @@ globus_l_gfs_file_open_cksm_cb(
                 "globus_xio_handle_cntl", result);
             goto error_seek;
         }
     }
     
-    MD5_Init(&monitor->mdctx);  
+    MD5_Init(&monitor->mdctx);
+    monitor->adler32ctx = adler32(0, NULL, 0);
     
     result = globus_xio_register_read(
         handle,
         monitor->buffer,
         monitor->count,
@@ -1981,11 +2014,17 @@ globus_l_gfs_file_cksm(
     if(offset < 0)
     {
         result = GlobusGFSErrorGeneric("Invalid offset.");
         goto param_error;
     }
-        
+
+    if (strcasecmp(algorithm, "md5") && strcasecmp(algorithm, "adler32"))
+    {
+        result = GlobusGFSErrorGeneric("Unknown checksum algorithm requested.");
+        goto alg_error;
+    }
+
     result = globus_xio_attr_init(&attr);
     if(result != GLOBUS_SUCCESS)
     {
         result = GlobusGFSErrorWrapFailed("globus_xio_attr_init", result);
         goto error_attr;
@@ -2061,10 +2100,20 @@ globus_l_gfs_file_cksm(
     monitor->length = length;
     monitor->block_size = block_size;
     monitor->internal_cb = internal_cb;
     monitor->internal_cb_arg = internal_cb_arg;
 
+    monitor->cksum_type = 0;
+    if(!strcasecmp(algorithm, "md5"))
+    {
+        monitor->cksum_type |= POSIX_CKSM_TYPE_MD5;
+    }
+    if(!strcasecmp(algorithm, "adler32"))
+    {
+        monitor->cksum_type |= POSIX_CKSM_TYPE_ADLER32;
+    }
+
     result = globus_xio_register_open(
         file_handle,
         pathname,
         attr,
         globus_l_gfs_file_open_cksm_cb,
@@ -2094,10 +2143,11 @@ error_push:
 error_stack:
 error_cntl:    
     globus_xio_attr_destroy(attr);
     
 error_attr:
+alg_error:
 param_error:
     GlobusGFSFileDebugExitWithError();
     return result;
 }     
 
