Index: gridftp/server/src/globus_i_gfs_control.c
===================================================================
RCS file: /home/globdev/CVS/globus-packages/gridftp/server/src/globus_i_gfs_control.c,v
retrieving revision 1.54.2.3
diff -u -r1.54.2.3 globus_i_gfs_control.c
--- globus_i_gfs_control.c	23 Mar 2012 00:37:49 -0000	1.54.2.3
+++ globus_i_gfs_control.c	17 May 2012 20:47:14 -0000
@@ -348,7 +348,7 @@
                 MAXPATHLEN,
                 &res_pwd);
             free(hd_name);
-            if(sc != 0)
+            if(sc != 0 || res_pwd == NULL)
             {
                 /* XXX expand other usernames here */
                 result = GlobusGFSErrorGeneric(
Index: gridftp/server/src/globus_i_gfs_data.c
===================================================================
RCS file: /home/globdev/CVS/globus-packages/gridftp/server/src/globus_i_gfs_data.c,v
retrieving revision 1.106.2.12
diff -u -r1.106.2.12 globus_i_gfs_data.c
--- globus_i_gfs_data.c	26 Mar 2012 20:53:59 -0000	1.106.2.12
+++ globus_i_gfs_data.c	17 May 2012 20:47:14 -0000
@@ -1058,13 +1058,13 @@
 
     rc = globus_libc_getpwuid_r(getuid(), &pwent_mem, pw_buffer,
                                 pw_buflen, &pw_result);
-    if(rc != 0)
+    if(rc != 0 || pw_result == NULL)
     {
         globus_free(pw_buffer);
         return NULL;
     }
 
-    pwent = globus_l_gfs_pw_copy(&pwent_mem);
+    pwent = globus_l_gfs_pw_copy(pw_result);
     globus_free(pw_buffer);
 
     return pwent;
@@ -1099,13 +1099,13 @@
 
     rc = globus_libc_getpwnam_r(
         (char *)name, &pwent_mem, pw_buffer, pw_buflen, &pw_result);
-    if(rc != 0)
+    if(rc != 0 || pw_result == NULL)
     {
         globus_free(pw_buffer);
         return NULL;
     }
 
-    pwent = globus_l_gfs_pw_copy(&pwent_mem);
+    pwent = globus_l_gfs_pw_copy(pw_result);
     globus_free(pw_buffer);
 
     return pwent;
