Index: globus_gridftp_server-6.0/init/globus-gridftp-server.in
===================================================================
--- globus_gridftp_server-6.0.orig/init/globus-gridftp-server.in	2011-09-16 16:01:49.000000000 -0500
+++ globus_gridftp_server-6.0/init/globus-gridftp-server.in	2011-09-16 16:03:35.000000000 -0500
@@ -21,6 +21,7 @@
     lsb=_lsb
     . /lib/lsb/init-functions
 fi
+. /etc/init.d/functions
 
 rc=0
 conf=${sysconfdir}/gridftp.conf
@@ -61,7 +62,7 @@
       cp $conf.default $conf;
     fi
 
-    start_daemon -p "$pidfile" "$gridftpd" -S -c $conf \
+    daemon --pidfile "$pidfile" "$gridftpd" -S -c $conf \
             -pidfile "${pidfile}"
     rc=$?
      
