diff -ru globus_gridftp_server-5.4-orig/gridftp.conf.default.in globus_gridftp_server-5.4/gridftp.conf.default.in
--- globus_gridftp_server-5.4-orig/gridftp.conf.default.in	2011-08-30 15:54:05.000000000 -0500
+++ globus_gridftp_server-5.4/gridftp.conf.default.in	2011-08-30 15:56:32.000000000 -0500
@@ -2,4 +2,7 @@
 # todo: add common parameters
 
 port 2811
+log_level ERROR,WARN,INFO
+log_single /var/log/gridftp-auth.log
+log_transfer /var/log/gridftp.log
 
diff -ru globus_gridftp_server-5.4-orig/init/globus-gridftp-server.in globus_gridftp_server-5.4/init/globus-gridftp-server.in
--- globus_gridftp_server-5.4-orig/init/globus-gridftp-server.in	2011-08-30 15:54:05.000000000 -0500
+++ globus_gridftp_server-5.4/init/globus-gridftp-server.in	2011-08-30 16:09:54.000000000 -0500
@@ -13,6 +13,8 @@
 pidfile=${localstatedir}/run/globus-gridftp-server.pid
 gridftpd=${sbindir}/globus-gridftp-server
 
+[ -e ${sysconfdir}/sysconfig/$prog ] && . ${sysconfdir}/sysconfig/$prog
+
 start() {
 	echo -n "Starting globus-gridftp-server: "
 	if [ ! -f $conf ]; then
