Index: globus_gridftp_server-6.1/gridftp.conf.default.in
===================================================================
--- globus_gridftp_server-6.1.orig/gridftp.conf.default.in	2011-08-31 21:12:07.000000000 -0500
+++ globus_gridftp_server-6.1/gridftp.conf.default.in	2011-09-23 19:05:55.000000000 -0500
@@ -8,4 +8,8 @@
 
 # port
 port 2811
+log_level ERROR,WARN,INFO
+log_single /var/log/gridftp-auth.log
+log_transfer /var/log/gridftp.log
+#load_dsi_module posix
 
Index: globus_gridftp_server-6.1/init/globus-gridftp-server.in
===================================================================
--- globus_gridftp_server-6.1.orig/init/globus-gridftp-server.in	2011-09-23 08:41:24.000000000 -0500
+++ globus_gridftp_server-6.1/init/globus-gridftp-server.in	2011-09-23 19:05:55.000000000 -0500
@@ -31,6 +31,8 @@
 pidfile=${localstatedir}/run/globus-gridftp-server.pid
 gridftpd=${sbindir}/globus-gridftp-server
 
+[ -e ${sysconfdir}/sysconfig/globus-gridftp-server ] && . ${sysconfdir}/sysconfig/globus-gridftp-server
+
 start()
 {
     printf "Starting globus-gridftp-server: "
