Index: osg-configure-1.0.55/config/20-condor.ini
===================================================================
--- osg-configure-1.0.55.orig/config/20-condor.ini	2014-05-22 07:26:54.000000000 -0500
+++ osg-configure-1.0.55/config/20-condor.ini	2014-05-30 10:11:14.847335555 -0500
@@ -11,18 +11,18 @@
 ; valid answers are True or False
 enabled = FALSE
 
-; The condor_location setting should give the location of condor install directory
-; Set this to DEFAULT if you are using an RPM installation of HTCondor.
-; Otherwise it should be set to the root directory of the HTCondor install,
+; The condor_location setting should give the location of condor install directory.
+; If you are using an RPM installation of HTCondor, set this to DEFAULT.
+; Otherwise, set this to the root directory of the HTCondor install,
 ; such that $CONDOR_LOCATION/bin contains the condor binaries.
 condor_location = DEFAULT
 
-; The condor_location setting should give the location of condor config file,
-; This is typically  /etc/condor/condor_config with RPM installations of HTCondor
-; or $CONDOR_LOCATION/etc/condor_config if you are using a tarball install of 
-; HTCondor.
-; If you leave this set to DEFAULT, osg-configure will attempt to
-; determine the correct value.
+; The condor_config setting should give the location of condor config file.
+; If you are using an RPM installation of HTCondor, or wish osg-configure to
+; detect the correct value, set this to DEFAULT.
+; On an RPM install of HTCondor, this file is located at /etc/condor/condor_config.
+; On a tarball install of HTCondor, this file is typically located at
+; $CONDOR_LOCATION/etc/condor_config.
 condor_config = DEFAULT
 
 ; The job_contact setting should give the contact string for the jobmanager 
@@ -41,4 +41,4 @@
 ; By default this is set to False
 ;
 ; Most sites will not need to change this 
-; accept_limited = False
\ No newline at end of file
+; accept_limited = False
