Index: osg-configure-1.0.57/scripts/osg-configure
===================================================================
--- osg-configure-1.0.57.orig/scripts/osg-configure	2014-07-30 14:29:38.000000000 -0500
+++ osg-configure-1.0.57/scripts/osg-configure	2014-07-30 14:32:44.897623342 -0500
@@ -138,11 +138,14 @@
         else:
           errmsg = "Missing job environment key (%s), exiting." % key
           if key in attribute_to_option_map:
-            section, name = attribute_to_option_map[key]
-            if section:
-              errmsg += " The job environment key is specified in config section %r, option %r." % (section, name)
-            else:
-              errmsg += " The job environment key is specified as option %r." % name
+#            section, name = attribute_to_option_map[key]
+#            if section:
+#              errmsg += " The job environment key is specified in config section %r, option %r." % (section, name)
+#            else:
+#              errmsg += " The job environment key is specified as option %r." % name
+            # Don't print the section -- the option may be in multiple sections
+            # and it would be misleading to only print one.
+            errmsg += " The job environment key is specified as option %r." % name
           error_exit(errmsg, exception)
     utilities.write_attribute_file(filename, temp)
   except IOError, exception:
