Index: osg-configure-1.0.57/scripts/osg-configure
===================================================================
--- osg-configure-1.0.57.orig/scripts/osg-configure	2014-07-29 16:53:28.000000000 -0500
+++ osg-configure-1.0.57/scripts/osg-configure	2014-07-30 14:29:38.830264203 -0500
@@ -133,7 +133,7 @@
       try:
         temp[key] = attributes[key]
       except KeyError, exception:
-        if key == 'OSG_SQUID_LOCATION':
+        if key in ['OSG_SQUID_LOCATION', 'OSG_JOB_CONTACT']:
           continue
         else:
           errmsg = "Missing job environment key (%s), exiting." % key
