From e0fd39bd8ed0cba15a05c1a1305e2aaa8225626a Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Wed, 14 Oct 2015 15:38:01 -0500
Subject: [PATCH] Create PID file directory /var/run/myproxy-server on startup
 (SOFTWARE-2071)

---
 systemd/myproxy-server.service | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/systemd/myproxy-server.service b/systemd/myproxy-server.service
index 930f181..e8aad62 100644
--- a/systemd/myproxy-server.service
+++ b/systemd/myproxy-server.service
@@ -9,10 +9,17 @@ ConditionPathExists=/etc/grid-security/myproxy/hostkey.pem
 [Service]
 Type=forking
 User=myproxy
 Environment=X509_USER_CERT=/etc/grid-security/myproxy/hostcert.pem
 Environment=X509_USER_KEY=/etc/grid-security/myproxy/hostkey.pem
+
+# Create PID file directory (needs to be run as root to set perms)
+PermissionsStartOnly=true
+ExecStartPre=/bin/mkdir -p /var/run/myproxy-server ; \
+             /bin/chown -R myproxy:myproxy /var/run/myproxy-server ; \
+             /bin/chmod -R 0755 /var/run/myproxy-server
+
 ExecStart=/usr/sbin/myproxy-server --pidfile /var/run/myproxy-server/myproxy.pid
 ExecReload=/bin/kill -HUP $MAINPID
 PIDFile=/var/run/myproxy-server/myproxy.pid
 
 [Install]
-- 
2.4.6

