From daa3291f25433da379b99bd487339e2b0d83b2fc Mon Sep 17 00:00:00 2001
From: Jaime Frey <jfrey@cs.wisc.edu>
Date: Wed, 3 Aug 2016 10:58:53 -0500
Subject: [PATCH 4/4] Fix proper build of cream_gahp on rhel7.

---
 externals/bundles/cream/1.15.4/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/externals/bundles/cream/1.15.4/CMakeLists.txt b/externals/bundles/cream/1.15.4/CMakeLists.txt
index 2817480..0843835 100644
--- a/externals/bundles/cream/1.15.4/CMakeLists.txt
+++ b/externals/bundles/cream/1.15.4/CMakeLists.txt
@@ -250,11 +250,13 @@ if ( WITH_CREAM )
 
 		condor_post_external( cream include OFF )
 
 	else( NOT PROPER )
 
-		if ( ${SYSTEM_NAME} MATCHES "rhel6" OR ${SYSTEM_NAME} MATCHES "centos6" OR ${SYSTEM_NAME} MATCHES "sl6")
+		if ( ${SYSTEM_NAME} MATCHES "rhel7" OR ${SYSTEM_NAME} MATCHES "centos7" OR ${SYSTEM_NAME} MATCHES "sl7")
+			find_multiple("glite_ce_cream_client_soap;glite_ce_cream_client_util;glite_security_gsoap_plugin_2816_cxx;glite_security_gss;gsoapssl++;globus_gssapi_gsi;globus_common;log4cpp;gridsite;boost_thread;boost_regex" CREAM_FOUND )
+		elseif ( ${SYSTEM_NAME} MATCHES "rhel6" OR ${SYSTEM_NAME} MATCHES "centos6" OR ${SYSTEM_NAME} MATCHES "sl6")
 			find_multiple("glite_ce_cream_client_soap;glite_ce_cream_client_util;glite_security_gsoap_plugin_2716_cxx;glite_security_gss;log4cpp;gridsite;boost_thread;boost_regex" CREAM_FOUND )
 		else()
 			find_multiple("glite_ce_cream_client_soap;glite_ce_cream_client_util;glite_security_gsoap_plugin_2713_cxx;glite_security_gss;log4cpp;gridsite;boost_thread;boost_regex" CREAM_FOUND )
 		endif()
 
-- 
2.6.3

