diff --git a/src/condor_daemon_client/dc_startd.cpp b/src/condor_daemon_client/dc_startd.cpp
index f88b34626eac516949cfee3a733b5c1e89201541..60a60045b169b9b61a77bfa7da523800323ddb49 100644
--- a/src/condor_daemon_client/dc_startd.cpp
+++ b/src/condor_daemon_client/dc_startd.cpp
@@ -154,6 +154,17 @@ ClaimStartdMsg::writeMsg( DCMessenger * /*messenger*/, Sock *sock ) {
 bool
 ClaimStartdMsg::putExtraClaims(Sock *sock) {
 
+	const CondorVersionInfo *cvi = sock->get_peer_version();
+
+	if (!cvi) {
+		return true;
+	}
+
+		// Older versions of condor don't know about this
+	if (!cvi->built_since_version(8,2,2)) {
+		return true;
+	}
+
 	if (m_extra_claims.length() == 0) {
 		return sock->put(0);
 	}
