diff --git a/build/cmake/CondorConfigure.cmake b/build/cmake/CondorConfigure.cmake
index 58b377b..50b98d4 100644
--- a/build/cmake/CondorConfigure.cmake
+++ b/build/cmake/CondorConfigure.cmake
@@ -712,10 +712,12 @@ if (LINUX
     # I've seen a reference to '-z bind_now', but all the
     # versions I can find actually use just '-z now':
     set(cxx_full_relro_arg "-Wl,-z,now")
-    # compiling everything with -fPIC is important for PIE
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
 endif()
 
+# compiling everything with -fPIC is needed to dynamically load libraries
+# linked against libstdc++
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
+
 
 #####################################
 # Do we want to link in libssl and kerberos or dlopen() them at runtime?
