commit 6e22f57c7c455f2eb1ccae3ca2e5dac715b39436
Author: Carl Edquist <edquist@cs.wisc.edu>
Date:   Fri Nov 8 12:39:58 2019 -0600

    osg/bosco_https.patch

diff --git a/src/condor_contrib/bosco/bosco_cluster b/src/condor_contrib/bosco/bosco_cluster
index e2d2e7ae00..82a040132e 100755
--- a/src/condor_contrib/bosco/bosco_cluster
+++ b/src/condor_contrib/bosco/bosco_cluster
@@ -965,6 +965,9 @@ else
 fi
 rsync -aq  $archive_dir/$libdir/libclassad.so* $remote_glite_dir/lib/ 2>/dev/null
 rsync -aq  $archive_dir/$libdir/libcondor_utils* $remote_glite_dir/lib/ 2>/dev/null
+if [ -d $archive_dir/libexec/glite/libexec ] ; then
+    rsync -aq $archive_dir/libexec/glite/libexec $remote_glite_dir/
+fi
 show_progress "Sending libraries to $remote_host" rsync -aq  $archive_dir/$libdir/condor $remote_glite_dir/lib/ 2>/dev/null
 rsync -aq  $archive_dir/sbin/condor_ft-gahp $remote_glite_dir/bin 2>/dev/null
 
diff --git a/src/condor_contrib/bosco/bosco_findplatform b/src/condor_contrib/bosco/bosco_findplatform
index bf6267819e..c8f64f1bc4 100755
--- a/src/condor_contrib/bosco/bosco_findplatform
+++ b/src/condor_contrib/bosco/bosco_findplatform
@@ -26,15 +26,14 @@ SUPPORTED_PLATFORMS = [ RH6, RH7, RH8, OSX, DEB8, DEB9, DEB10, UBUNTU16, UBUNTU1
 
 # download URLs for the different platforms
 URL_DICT={
-  DEB8: "https://research.cs.wisc.edu/htcondor/bosco/1.2/bosco-1.2-x86_64_Debian8.tar.gz",
-  DEB9: "https://research.cs.wisc.edu/htcondor/bosco/1.2/bosco-1.2-x86_64_Debian9.tar.gz",
-  DEB10: "https://research.cs.wisc.edu/htcondor/bosco/1.2/bosco-1.2-x86_64_Debian10.tar.gz",
-  OSX: "https://research.cs.wisc.edu/htcondor/bosco/1.2/bosco-1.2-x86_64_MacOSX.tar.gz",
-  RH6: "https://research.cs.wisc.edu/htcondor/bosco/1.2/bosco-1.2-x86_64_RedHat6.tar.gz",
-  RH7: "https://research.cs.wisc.edu/htcondor/bosco/1.2/bosco-1.2-x86_64_RedHat7.tar.gz",
-  RH8: "https://research.cs.wisc.edu/htcondor/bosco/1.2/bosco-1.2-x86_64_RedHat8.tar.gz",
-  UBUNTU16: "https://research.cs.wisc.edu/htcondor/bosco/1.2/bosco-1.2-x86_64_Ubuntu16.tar.gz",
-  UBUNTU18: "https://research.cs.wisc.edu/htcondor/bosco/1.2/bosco-1.2-x86_64_Ubuntu18.tar.gz",
+  DEB8: "https://research.cs.wisc.edu/htcondor/bosco/1.3/bosco-1.3-x86_64_Debian8.tar.gz",
+  DEB9: "https://research.cs.wisc.edu/htcondor/bosco/1.3/bosco-1.3-x86_64_Debian9.tar.gz",
+  OSX: "https://research.cs.wisc.edu/htcondor/bosco/1.3/bosco-1.3-x86_64_MacOSX.tar.gz",
+  RH6: "https://research.cs.wisc.edu/htcondor/bosco/1.3/bosco-1.3-x86_64_RedHat6.tar.gz",
+  RH7: "https://research.cs.wisc.edu/htcondor/bosco/1.3/bosco-1.3-x86_64_RedHat7.tar.gz",
+  # UBUNTU14: "https://research.cs.wisc.edu/htcondor/bosco/1.3/bosco-1.3-x86_64_Ubuntu14.tar.gz",
+  UBUNTU16: "https://research.cs.wisc.edu/htcondor/bosco/1.3/bosco-1.3-x86_64_Ubuntu16.tar.gz",
+  UBUNTU18: "https://research.cs.wisc.edu/htcondor/bosco/1.3/bosco-1.3-x86_64_Ubuntu18.tar.gz",
 }
 
 BIT32="32bit"
