Index: condor-7.8.1/externals/bundles/cream/1.12.1_14/CMakeLists.txt
===================================================================
--- condor-7.8.1.orig/externals/bundles/cream/1.12.1_14/CMakeLists.txt	2012-06-27 14:40:14.000000003 -0500
+++ condor-7.8.1/externals/bundles/cream/1.12.1_14/CMakeLists.txt	2012-06-27 14:40:42.000000003 -0500
@@ -245,7 +245,7 @@
 else( NOT PROPER )
 
 	find_multiple(
-"glite_ce_cream_client_soap;glite_ce_cream_client_util;glite_security_gsoap_plugin_2713_cxx;glite_security_gss;log4cpp;gridsite;boost_thread;boost_regex"
+"glite_ce_cream_client_soap;glite_ce_cream_client_util;glite_security_gsoap_plugin_2716_cxx;glite_security_gss;log4cpp;gridsite;boost_thread;boost_regex"
 CREAM_FOUND )
 
 endif( NOT PROPER )
