commit 73727e74c5946ac1acd4e3b63358941a550ac437
Author: Tim Theisen <tim@cs.wisc.edu>
Date:   Fri Apr 23 14:19:36 2021 -0500

    HTCONDOR-438: Make bosco_cluster work with new style tarballs

diff --git a/nmi_tools/glue/build/make-tarball-links b/nmi_tools/glue/build/make-tarball-links
index 6255be8..fad79c4 100755
--- a/nmi_tools/glue/build/make-tarball-links
+++ b/nmi_tools/glue/build/make-tarball-links
@@ -108,8 +108,10 @@ if [ -d usr/lib/condor/libexec/glite ]; then
     (cd usr/lib/condor/libexec/glite/bin; ln -s ../../../../../bin/bl* .)
     (cd usr/lib/condor/libexec/glite/bin; ln -s ../../../../../sbin/bl* .)
     (cd usr/lib/condor/libexec/glite/bin; ln -s ../../../../../libexec/blahp/* .)
-    ln -s ../../../../lib usr/lib/condor/libexec/glite
+    ln -s ../../.. usr/lib/condor/libexec/glite
     ln -s bin usr/lib/condor/libexec/glite/libexec
 fi
 # Comment out default blahp location
 sed -i -e s/blah_libexec_directory/#blah_libexec_directory/ etc/blah.config
+# Make absolute links relative
+(cd usr/libexec/blahp; ln -sf ../../../etc/blahp/*_local_submit_attributes.sh .)
diff --git a/src/condor_contrib/bosco/bosco_cluster b/src/condor_contrib/bosco/bosco_cluster
index a6c0231..298fe85 100755
--- a/src/condor_contrib/bosco/bosco_cluster
+++ b/src/condor_contrib/bosco/bosco_cluster
@@ -377,6 +377,7 @@ add_cluster_to_list () {
         mv $tmpfile $CLUSTERLIST_LOCATION
     fi
 
+    mkdir -p `dirname $CLUSTERLIST_LOCATION`
     echo "entry=$remote_host max_queued=$max_queued cluster_type=$cluster_type" >> $CLUSTERLIST_LOCATION
 
     # Always after add/remove
@@ -1010,22 +1011,30 @@ mkdir -p $remote_glite_dir/lib; mkdir -p $remote_glite_dir/log; mkdir -p $local_
 # TODO: uncomment to enable subdirs
 #ssh $remote_host "ln -sfn $remote_base_dir_host $remote_base_dir_default"
 
-
-# Copy over blahp
-rsync -aq $archive_dir/libexec/glite/[^l]* $remote_glite_dir/
-if [ -d $archive_dir/lib64 ] ; then
+if [ -d $archive_dir/usr/lib64 ] ; then
     libdir=lib64
 else
     libdir=lib
 fi
-rsync -aq  $archive_dir/$libdir/libclassad.so* $remote_glite_dir/lib/ 2>/dev/null
-rsync -aq  $archive_dir/$libdir/libcondor_utils* $remote_glite_dir/lib/ 2>/dev/null
+
+# Copy over Manifest
+rsync -aq $archive_dir/Manifest.txt $remote_glite_dir/
+# Record the tarball name in the Manifest
+echo $(basename $archive_dir) >> $remote_glite_dir/Manifest.txt
+
+# Copy over blahp (location change: must follow symbolic links)
+rsync -rLptgoq $archive_dir/libexec/glite/[^l]* $remote_glite_dir/ 2>/dev/null
+# Copy over libraries preserving symlinks and directory name
+rsync -aq  $archive_dir/$libdir/lib*.so* $remote_glite_dir/$libdir/ 2>/dev/null
 if [ -d $archive_dir/libexec/glite/libexec ] ; then
+    # This is actually a symbolic link
     rsync -aq $archive_dir/libexec/glite/libexec $remote_glite_dir/
 fi
-show_progress "Sending libraries to $remote_host" rsync -aq  $archive_dir/$libdir/condor $remote_glite_dir/lib/ 2>/dev/null
 rsync -aq  $archive_dir/sbin/condor_ft-gahp $remote_glite_dir/bin 2>/dev/null
 
+# link in local submit attributes
+(cd $remote_glite_dir/bin; ln -sf ../etc/blahp/*_local_submit_attributes.sh .)
+
 # create the glidein
 if [ -n "$PRIVATE_ON_MU" ]; then
     echo "You are not running as the factory user. BOSCO Glideins disabled."
diff --git a/src/condor_contrib/bosco/bosco_findplatform b/src/condor_contrib/bosco/bosco_findplatform
index b0999c8..0e1f8b3 100755
--- a/src/condor_contrib/bosco/bosco_findplatform
+++ b/src/condor_contrib/bosco/bosco_findplatform
@@ -23,15 +23,13 @@ MINT_TO_UBUNTU_VERSION = {'18': '16',
 
 # download URLs for the different platforms
 URL_DICT={
-  "DEB9": "https://research.cs.wisc.edu/htcondor/bosco/1.2/bosco-1.2-x86_64_Debian9.tar.gz",
-  "DEB10": "https://research.cs.wisc.edu/htcondor/bosco/1.2/bosco-1.2-x86_64_Debian10.tar.gz",
-  "MAC": "https://research.cs.wisc.edu/htcondor/bosco/1.2/bosco-1.2-x86_64_MacOSX.tar.gz",
-  "RH6": "https://research.cs.wisc.edu/htcondor/bosco/1.2/bosco-1.2-x86_64_RedHat6.tar.gz",
-  "RH7": "https://research.cs.wisc.edu/htcondor/bosco/1.3/bosco-1.3-x86_64_RedHat7.tar.gz",
-  "RH8": "https://research.cs.wisc.edu/htcondor/bosco/1.3/bosco-1.3-x86_64_RedHat8.tar.gz",
-  "UBUNTU16": "https://research.cs.wisc.edu/htcondor/bosco/1.2/bosco-1.2-x86_64_Ubuntu16.tar.gz",
-  "UBUNTU18": "https://research.cs.wisc.edu/htcondor/bosco/1.3/bosco-1.3-x86_64_Ubuntu18.tar.gz",
-  "UBUNTU20": "https://research.cs.wisc.edu/htcondor/bosco/1.2/bosco-1.2-x86_64_Ubuntu20.tar.gz",
+  "DEB9": "https://research.cs.wisc.edu/htcondor/tarball/9.0/current/condor-current-x86_64_Debian9-stripped.tar.gz",
+  "DEB10": "https://research.cs.wisc.edu/htcondor/tarball/9.0/current/condor-current-x86_64_Debian10-stripped.tar.gz",
+  "MAC": "https://research.cs.wisc.edu/htcondor/tarball/9.0/current/condor-current-x86_64_MacOSX-stripped.tar.gz",
+  "RH7": "https://research.cs.wisc.edu/htcondor/tarball/9.0/current/condor-current-x86_64_CentOS7-stripped.tar.gz",
+  "RH8": "https://research.cs.wisc.edu/htcondor/tarball/9.0/current/condor-current-x86_64_CentOS8-stripped.tar.gz",
+  "UBUNTU18": "https://research.cs.wisc.edu/htcondor/tarball/9.0/current/condor-current-x86_64_Ubuntu18-stripped.tar.gz",
+  "UBUNTU20": "https://research.cs.wisc.edu/htcondor/tarball/9.0/current/condor-current-x86_64_Ubuntu20-stripped.tar.gz",
 }
 
 BIT32="32bit"
