diff --git a/src/condor_utils/ipv6_hostname.cpp b/src/condor_utils/ipv6_hostname.cpp
index 06fa79c..02cb7f4 100644
--- a/src/condor_utils/ipv6_hostname.cpp
+++ b/src/condor_utils/ipv6_hostname.cpp
@@ -35,16 +35,25 @@ void init_local_hostname()
 	char hostname[MAXHOSTNAMELEN];
 	int ret;
 
+	std::string network_hostname;
+	if (param(network_hostname, "NETWORK_HOSTNAME")) {
+		strncpy(hostname, network_hostname.c_str(), sizeof(hostname));
+		hostname[sizeof(hostname)-1] = '\0';
+
+		dprintf(D_HOSTNAME, "NETWORK_HOSTNAME says we are %s\n", hostname);
+	} else {
+
 		// [TODO:IPV6] condor_gethostname is not IPv6 safe.
 		// reimplement it.
-	ret = condor_gethostname(hostname, sizeof(hostname));
-	if (ret) {
-		dprintf(D_ALWAYS, "condor_gethostname() failed. Cannot initialize "
-				"local hostname, ip address, FQDN.\n");
-		return;
-	}
+		ret = condor_gethostname(hostname, sizeof(hostname));
+		if (ret) {
+			dprintf(D_ALWAYS, "condor_gethostname() failed. Cannot initialize "
+					"local hostname, ip address, FQDN.\n");
+			return;
+		}
 
-	dprintf(D_HOSTNAME, "condor_gethostname() claims we are %s\n", hostname);
+		dprintf(D_HOSTNAME, "condor_gethostname() claims we are %s\n", hostname);
+	}
 
 	// Fallback case.
 	local_hostname = hostname;
	Modified   src/condor_utils/param_info.in
diff --git a/src/condor_utils/param_info.in b/src/condor_utils/param_info.in
index 0889e77..c62bd83 100644
--- a/src/condor_utils/param_info.in
+++ b/src/condor_utils/param_info.in
@@ -4501,7 +4501,7 @@
 tags=,
 id=258
 
-[NETWORK_INTERFACE]
+[NETWORK_HOSTNAME]
 aliases=
 default=
 version=0.0.0
@@ -4511,6 +4511,21 @@
 is_macro=false
 reconfig=true
 customization=seldom
+friendly_name=Network Hostname
+usage=
+url=
+tags=,
+
+[NETWORK_INTERFACE]
+aliases=
+default=
+version=0.0.0
+range=.*
+state=default
+type=string
+is_macro=false
+reconfig=false
+customization=seldom
 friendly_name=Network Interface
 usage=
 url=http://cs.wisc.edu/condor/manual/v7.3/3_3Configuration.html#SECTION

--- a/doc/admin-man/configure.tex
+++ b/doc/admin-man/configure.tex
@@ -824,6 +824,7 @@ Reconfiguration using \Condor{reconfig} will not be enough.
   \item \verb@MAX_NUM_CPUS@
   \item \verb@MAX_TRACKING_GID@
   \item \verb@MIN_TRACKING_GID@
+  \item \verb@NETWORK_HOSTNAME@
   \item \verb@NETWORK_INTERFACE@
   \item \verb@NUM_CPUS@
   \item \verb@PREEMPTION_REQUIREMENTS_STABLE@
@@ -3091,6 +3092,17 @@ variable require a restart of HTCondor in order to take effect.  Also note
 that only if HTCondor is running as root will it be able to increase the
 limit above the hard limit (on maximum open files) that it inherits.
 
+\label{param:NetworkHostname}
+\item[\Macro{NETWORK\_HOSTNAME}]
+Normally, HTCondor uses the gethostname() function to query the OS about
+the hostname of the local machine. This parameter can be used to override
+the result of that query.
+Among other things, the hostname is used in the to identify daemons in
+an HTCondor pool, via the Machine and Name attributes of daemon ads.
+This parameter can be used when a machine has multiple network interfaces
+with different hostnames, and you want HTCondor to use a hostname that's
+not the primary one.
+
 \label{param:NetworkInterface}
 \item[\Macro{NETWORK\_INTERFACE}]
   An IP address of the form \Expr{123.123.123.123} or the name
	Modified   doc/version-history/8-2.history.tex
--- a/doc/admin-man/configure.tex
+++ b/doc/admin-man/configure.tex
@@ -3094,14 +3094,16 @@ limit above the hard limit (on maximum open files) that it inherits.
 
 \label{param:NetworkHostname}
 \item[\Macro{NETWORK\_HOSTNAME}]
-Normally, HTCondor uses the gethostname() function to query the OS about
-the hostname of the local machine. This parameter can be used to override
-the result of that query.
-Among other things, the hostname is used in the to identify daemons in
-an HTCondor pool, via the Machine and Name attributes of daemon ads.
-This parameter can be used when a machine has multiple network interfaces
-with different hostnames, and you want HTCondor to use a hostname that's
-not the primary one.
+The host name to use,
+overriding the value returned by
+\Procedure{gethostname}, which will be invoked by default
+to query the operating system to obtain the host name of the local machine. 
+Among other things, the host name is used to identify daemons in
+an HTCondor pool, via the \Attr{Machine} and \Attr{Name} attributes of 
+daemon ClassAds.
+This variable can be used when a machine has multiple network interfaces
+with different host names, 
+to use a host name that is not the primary one.
 
 \label{param:NetworkInterface}
 \item[\Macro{NETWORK\_INTERFACE}]
	Modified   doc/version-history/8-2.history.tex
