commit 20aeaf61427961de5880e712868d4f89117d0401
Author: Mark Coatsworth <coatsworth@cs.wisc.edu>
Date:   Wed Jan 24 12:05:44 2018 -0600

    Fixed segfault + core dump bug in preen stable (#6521)

diff --git a/src/condor_tools/preen.cpp b/src/condor_tools/preen.cpp
index 06dcd5f..8ba7514 100644
--- a/src/condor_tools/preen.cpp
+++ b/src/condor_tools/preen.cpp
@@ -417,19 +417,14 @@ check_spool_dir()
 			continue;
 		}
 
-		if( IsDirectory( dir.GetFullPath() ) && !IsSymlink( dir.GetFullPath() ) ) {
-			if( check_job_spool_hierarchy( Spool, f, bad_spool_files ) ) {
-				good_file( Spool, f );
-				continue;
-			}
-		}
-
 			// If none of the previous checks succeeded, we can try a couple
 			// other checks which require an active connection to the schedd. 
 			// Establish a connection (if not already connected). If this fails
 			// for any reason, abort and don't delete any files.
 		if ( !qmgr ) {
 			if ( !( qmgr = ConnectQ (0,0,false) ) ) {
+				dprintf( D_ALWAYS, "Unable to connect to job queue, not "
+						"cleaning spool directory.\n" );
 				return;
 			}
 			last_connection_time = _condor_debug_get_time_double();
@@ -449,6 +444,13 @@ check_spool_dir()
 			continue;
 		}
 
+		if( IsDirectory( dir.GetFullPath() ) && !IsSymlink( dir.GetFullPath() ) ) {
+			if( check_job_spool_hierarchy( Spool, f, bad_spool_files ) ) {
+				good_file( Spool, f );
+				continue;
+			}
+		}
+
 			// We think it's bad.  For now, just append it to a
 			// StringList, instead of actually deleting it.  This way,
 			// if DisconnectQ() fails, we can abort and not actually
