From f6413c14817f20ca65ef56ba618b1860b44948ce Mon Sep 17 00:00:00 2001
From: Colm Talbot <talbotcolm@gmail.com>
Date: Fri, 22 Jan 2021 17:25:01 -0500
Subject: [PATCH] fix last_nodes_total_check_time

---
 src/condor_scripts/condor_watch_q | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/condor_scripts/condor_watch_q b/src/condor_scripts/condor_watch_q
index 9a4660d26d..a95aa82f2d 100755
--- a/src/condor_scripts/condor_watch_q
+++ b/src/condor_scripts/condor_watch_q
@@ -490,7 +490,7 @@ def watch_q(
                     )
                     and time.time() - last_nodes_total_check_time > 60
                 ):
-                    last_nodes_total_check_time = datetime.datetime.now()
+                    last_nodes_total_check_time = time.time()
                     (
                         batch_name_to_nodes_total,
                         cluster_ids_with_nodes_total,
