commit 9b8bc33dd64ed78bac9a6ce8ee22aa8a4cdb3a2d
Author: Mark Coatsworth <coatsworth@cs.wisc.edu>
Date:   Wed Jan 24 15:16:21 2018 -0600

    Fixed preen segfault + overzealous deletions in 8.7.6 (#6521) (#6475)

diff --git a/src/condor_tools/preen.cpp b/src/condor_tools/preen.cpp
index 135c05f..6da3c00 100644
--- a/src/condor_tools/preen.cpp
+++ b/src/condor_tools/preen.cpp
@@ -430,12 +430,6 @@ check_spool_dir()
 			continue;
 		}
 
-		if( IsDirectory( dir.GetFullPath() ) && !IsSymlink( dir.GetFullPath() ) ) {
-			if( check_job_spool_hierarchy( Spool, f, bad_spool_files ) ) {
-				good_file( Spool, f );
-				continue;
-			}
-		}
 			// If none of the previous checks succeeded, we can try a couple
 			// other checks which require an active connection to the schedd. 
 			// Establish a connection (if not already connected). If this fails
@@ -461,6 +455,13 @@ check_spool_dir()
 			continue;
 		}
 
+		if( IsDirectory( dir.GetFullPath() ) && !IsSymlink( dir.GetFullPath() ) ) {
+			if( check_job_spool_hierarchy( Spool, f, bad_spool_files ) ) {
+				good_file( Spool, f );
+				continue;
+			}
+		}
+
 			// We think it's bad.  For now, just append it to a
 			// StringList, instead of actually deleting it.  This way,
 			// if DisconnectQ() fails, we can abort and not actually
@@ -692,7 +693,7 @@ proc_exists( int cluster, int proc )
 		return false;
 	}
 
-	return false;
+	return true;
 }
 
 /*
