Index: condor-8.2.3/src/cream_gahp/cream_gahp_server.cpp
===================================================================
--- condor-8.2.3.orig/src/cream_gahp/cream_gahp_server.cpp	2014-09-30 17:19:27.000000000 -0500
+++ condor-8.2.3/src/cream_gahp/cream_gahp_server.cpp	2014-10-16 18:17:32.990449208 -0500
@@ -2341,7 +2341,7 @@
 		// include in UW builds of Condor.
 		// If we load the library with dlopen() first, then lt_dlopen()
 		// will find it.
-	void *dl_ptr = dlopen( "libglobus_thread_pthread.so", RTLD_LAZY);
+	void *dl_ptr = dlopen( "libglobus_thread_pthread.so.0", RTLD_LAZY);
 	if ( dl_ptr == NULL ) {
 		fprintf( stderr, "Failed to open globus_thread_pthread.\n" );
 		return 1;
