diff --git a/src/condor_schedd.V6/qmgmt_send_stubs.cpp b/src/condor_schedd.V6/qmgmt_send_stubs.cpp
index fd981daf0812370ed5c312c03afbd530a51407ac..0e2abdaa313baf1e62f5ccbd0053937af974155d 100644
--- a/src/condor_schedd.V6/qmgmt_send_stubs.cpp
+++ b/src/condor_schedd.V6/qmgmt_send_stubs.cpp
@@ -439,7 +439,7 @@ RemoteCommitTransaction(SetAttributeFlags_t flags, CondorError *errstack)
 			neg_on_error( getClassAd( qmgmt_sock, reply ) );
 
 			std::string errmsg;
-			if( reply.LookupString( "ErrorReason", errmsg ) ) {
+			if( errstack && reply.LookupString( "ErrorReason", errmsg ) ) {
 				int errCode = terrno;
 				reply.LookupInteger( "ErrorCode", errCode );
 				errstack->push( "SCHEDD", errCode, errmsg.c_str() );
