diff --git a/src/condor_utils/LoadPlugins.cpp b/src/condor_utils/LoadPlugins.cpp
index 6310bc9..b76121a 100644
--- a/src/condor_utils/LoadPlugins.cpp
+++ b/src/condor_utils/LoadPlugins.cpp
@@ -124,7 +124,7 @@ LoadPlugins()
 #ifdef WIN32
         if( NULL == LoadLibrary(plugin_file) ) {
 #else
-		if (!dlopen(plugin_file, RTLD_NOW)) {
+		if (!dlopen(plugin_file, RTLD_NOW | RTLD_GLOBAL)) {
 #endif
 			error = getErrorString();
 			if (error) {
