--- a/src/condor_includes/match_prefix.h
+++ b/src/condor_includes/match_prefix.h
@@ -46,5 +46,11 @@ bool is_arg_prefix(const char * parg, const char * pval, int must_match_length =
 // use this when you want to tack qualifiers onto arguments rather than supplying them as separate arguments.
 bool is_arg_colon_prefix(const char * parg, const char * pval, const char ** ppcolon, int must_match_length = 0);
 
-// return true of parg begins with '-' and is_arg_prefix(parg,pval,mml) is true
+// return true of parg begins with '-' or '--' and is_arg_prefix(parg,pval,mml) is true
+// if parg begins with '--' then parg and pval must match exactly (i.e. must_match_length is set to -1)
 bool is_dash_arg_prefix(const char * parg, const char * pval, int must_match_length = 0);
+
+// return true if parg with '-' or '--' and is_arg_colon_prefix(parg,pval,ppc,mml) is true
+// if parg begins with '--' then parg and pval must match exactly (i.e. must_match_length is set to -1)
+bool is_dash_arg_colon_prefix(const char * parg, const char * pval, const char ** ppcolon, int must_match_length = 0);
+
--- a/src/condor_utils/match_prefix.cpp
+++ b/src/condor_utils/match_prefix.cpp
@@ -92,3 +92,14 @@ is_dash_arg_prefix(const char * parg, const char * pval, int must_match_length /
 	return is_arg_prefix(parg, pval, must_match_length);
 }
 
+bool
+is_dash_arg_colon_prefix(const char * parg, const char * pval, const char ** ppcolon, int must_match_length /*= 0*/)
+{
+	if (*parg != '-') return false;
+	++parg;
+	// if arg begins with --, then we require an exact match for pval.
+	if (*parg == '-') { ++parg; must_match_length = -1; }
+	return is_arg_colon_prefix(parg, pval, ppcolon, must_match_length);
+}
+
+

