From 3b3e4e609b07c9349a6aa8d90cf152e56f1c5f96 Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Thu, 6 Oct 2016 13:30:59 -0500
Subject: [PATCH] disable-sslv3

---
 library/gsi.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/library/gsi.conf b/library/gsi.conf
index b7ce83c..0dd79a6 100644
--- a/library/gsi.conf
+++ b/library/gsi.conf
@@ -1,10 +1,10 @@
 # Flag to choose SSL or TLS for GSSAPI protocol.
 # Set to true to disable SSLv3, or false to allow either to be 
 # negotiated. The value of the GLOBUS_GSSAPI_FORCE_TLS environment
 # variable overrides this
-FORCE_TLS=false
+FORCE_TLS=true
 # GSSAPI Name compatibility mode when trying to determine
 # if a host certificate is legitimate. GSI predates RFC2818,
 # so there are some old, less-secure, practices by default.
 # The different modes are:
 # STRICT_GT2:
-- 
2.6.3

