--- library/export_cred.c.orig	2012-03-09 08:19:41.000000000 -0600
+++ library/export_cred.c	2012-03-09 08:20:13.000000000 -0600
@@ -204,7 +204,8 @@
         export_buffer->value = globus_common_create_string(
             "X509_USER_PROXY=%s",
             proxy_filename);
-        export_buffer->length = strlen((char *) export_buffer->value);
+        /* Add 1 to string length to include null-termination character */
+        export_buffer->length = strlen((char *) export_buffer->value)+1;
     }
     else
     {
