From d0314c7df129c4f7a59f819c0c88dfc0c535e3d4 Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Tue, 3 May 2016 16:13:58 -0500
Subject: [PATCH] Auto-create server keys (SOFTWARE-2288)

---
 gsisshd.init | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

diff --git a/gsisshd.init b/gsisshd.init
index bdbb94c..3fbcd4d 100755
--- a/gsisshd.init
+++ b/gsisshd.init
@@ -120,18 +120,17 @@ do_restart_sanity_check()
 start()
 {
 	[ -x $SSHD ] || exit 5
 	[ -f /etc/gsissh/sshd_config ] || exit 6
 	# Create keys if necessary
-	# - Not necessary if running with keyex only
-	# if [ "x${AUTOCREATE_SERVER_KEYS}" != xNO ]; then
-	#	do_rsa_keygen
-	#	if [ "x${AUTOCREATE_SERVER_KEYS}" != xRSAONLY ]; then
-	#		do_rsa1_keygen
-	#		do_dsa_keygen
-	#	fi
-	# fi
+	if [ "x${AUTOCREATE_SERVER_KEYS}" != xNO ]; then
+		do_rsa_keygen
+		if [ "x${AUTOCREATE_SERVER_KEYS}" != xRSAONLY ]; then
+			do_rsa1_keygen
+			do_dsa_keygen
+		fi
+	fi
 
 	echo -n $"Starting $prog: "
 	$SSHD $OPTIONS && success || failure
 	RETVAL=$?
 	[ $RETVAL -eq 0 ] && touch $lockfile
-- 
2.6.3

