--- _build_results/BUILD/openssh-4.3p2/gss-serv-gsi.c	2011-08-03 12:45:12.000000000 -0500
+++ gss-serv-gsi.c	2011-08-03 12:20:53.000000000 -0500
@@ -53,6 +53,22 @@
 	&ssh_gssapi_gsi_storecreds
 };
 
+static
+void
+undo_globus_signal_blocking()
+{
+	sigset_t mysigset;
+	sigemptyset(&mysigset);
+	sigaddset(&mysigset, SIGCHLD);
+	sigaddset(&mysigset, SIGTERM);
+	sigaddset(&mysigset, SIGALRM);
+	sigaddset(&mysigset, SIGHUP);
+	sigaddset(&mysigset, SIGTERM);
+	sigaddset(&mysigset, SIGQUIT);
+	sigaddset(&mysigset, SIGINT);
+	sigprocmask(SIG_UNBLOCK, &mysigset, NULL);
+}
+
 /*
  * Check if this user is OK to login under GSI. User has been authenticated
  * as identity in global 'client_name.value' and is trying to log in as passed
@@ -71,8 +87,9 @@
     
 #ifdef GLOBUS_GSI_GSS_ASSIST_MODULE
     if (globus_module_activate(GLOBUS_GSI_GSS_ASSIST_MODULE) != 0) {
-	return 0;
+        return 0;
     }
+    undo_globus_signal_blocking();
 #endif
 
 /* use new globus_gss_assist_map_and_authorize() interface if available */
@@ -119,6 +136,7 @@
     if (globus_module_activate(GLOBUS_GSI_GSS_ASSIST_MODULE) != 0) {
 	return 0;
     }
+    undo_globus_signal_blocking();
 #endif
 
 /* use new globus_gss_assist_map_and_authorize() interface if available */
