Index: globus_gram_job_manager.c
===================================================================
--- globus_gram_job_manager.c.orig	2012-04-18 16:48:08.000000000 -0500
+++ globus_gram_job_manager.c	2012-04-18 16:56:15.671939316 -0500
@@ -570,6 +570,10 @@
  *     Success.
  * @retval GLOBUS_GRAM_PROTOCOL_ERROR_MALLOC_FAILED
  *     Malloc failed.
+ * @retval GLOBUS_GRAM_PROTOCOL_ERROR_JM_STOPPED
+ *     Job manager is already stopped.
+ * @retval GLOBUS_GRAM_PROTOCOL_ERROR_OLD_JM_ALIVE
+ *     Old job manager is still alove.
  *
  */
 int
Index: globus_gram_job_manager_query.c
===================================================================
--- globus_gram_job_manager_query.c.orig	2012-04-11 07:53:59.000000000 -0500
+++ globus_gram_job_manager_query.c	2012-04-18 16:52:56.453833614 -0500
@@ -1431,7 +1431,7 @@
 
             if (out_size >= 0 && out_size != local_size_stdout)
             {
-                rc = GLOBUS_GRAM_PROTOCOL_ERROR_STDIO_SIZE;
+                //rc = GLOBUS_GRAM_PROTOCOL_ERROR_STDIO_SIZE;
                 globus_gram_job_manager_request_log(
                         request,
                         GLOBUS_GRAM_JOB_MANAGER_LOG_WARN,
@@ -1458,7 +1458,7 @@
             }
             else if (err_size >= 0 && err_size != local_size_stderr)
             {
-                rc = GLOBUS_GRAM_PROTOCOL_ERROR_STDIO_SIZE;
+                //rc = GLOBUS_GRAM_PROTOCOL_ERROR_STDIO_SIZE;
                 globus_gram_job_manager_request_log(
                         request,
                         GLOBUS_GRAM_JOB_MANAGER_LOG_WARN,
