--- /usr/src/debug/globus_gram_job_manager-10.70/main.c	2011-03-08 13:48:33.000000000 -0600
+++ main.c	2011-08-01 07:46:19.000000000 -0500
@@ -551,16 +551,24 @@
         remove(manager.cred_path);
         remove(manager.socket_path);
         remove(manager.lock_path);
-    }
-    globus_gram_job_manager_destroy(&manager);
-    globus_gram_job_manager_config_destroy(&config);
+        /* Leak the following objects for non-active JM.
+         * This is because 2 threads are needed to successfully call
+         * globus_io_close, and the non-active JM has no threads as
+         * it previously forked (and will deadlock).
+         */
+        globus_gram_job_manager_destroy(&manager);
+        globus_gram_job_manager_config_destroy(&config);
 
-    rc = globus_l_gram_deactivate();
-    if (rc != GLOBUS_SUCCESS)
-    {
-        fprintf(stderr, "deactivation failed with rc=%d\n",
-                rc);
-        exit(1);
+        /* gram_deactivate has the same issue for fork children
+         * as job_manager_destroy above.
+         */
+        rc = globus_l_gram_deactivate();
+        if (rc != GLOBUS_SUCCESS)
+        {
+            fprintf(stderr, "deactivation failed with rc=%d\n",
+                    rc);
+            exit(1);
+        }
     }
 
 /*
