Index: globus_gram_job_manager_script.c
===================================================================
--- globus_gram_job_manager_script.c.orig	2015-02-11 14:13:56.750586967 -0600
+++ globus_gram_job_manager_script.c	2015-02-11 14:14:39.329088485 -0600
@@ -1848,10 +1848,15 @@ globus_l_gram_enqueue_description(
         rc = globus_l_gram_enqueue_string(
                 fifo,
                 ",\n    'jobdir' => [ '%s' ]",
                 request->job_dir);
     }
+    rc = globus_l_gram_enqueue_string(
+            fifo,
+            ",\n    'servicetag' => [ '%s' ]",
+            request->config->service_tag == 0 ? "untagged" :
+                request->config->service_tag);
 
     rc = globus_l_gram_enqueue_string(
             fifo,
             ",\n    'streamingdisabled' => [ %d ]",
             request->config->streaming_disabled);
