Index: globus_gram_job_manager_script.c
===================================================================
--- globus_gram_job_manager_script.c.orig	2011-10-20 13:06:36.000000000 -0500
+++ globus_gram_job_manager_script.c	2011-10-27 15:21:26.129686755 -0500
@@ -1774,6 +1774,11 @@
                 ",\n    'jobdir' => [ '%s' ]",
                 request->job_dir);
     }
+    rc = globus_l_gram_enqueue_string(
+            fifo,
+            ",\n    'servicetag' => [ '%s' ]",
+            request->config->service_tag == 0 ? "untagged" : 
+                request->config->service_tag);
 
     rc = globus_l_gram_enqueue_string(
             fifo,
