Index: globus_gram_job_manager_script.c
===================================================================
--- globus_gram_job_manager_script.c.orig	2013-04-10 10:27:45.000000000 -0500
+++ globus_gram_job_manager_script.c	2013-12-11 16:50:31.272107287 -0600
@@ -1846,6 +1846,11 @@
                 ",\n    'jobdir' => [ '%s' ]",
                 request->job_dir);
     }
+    rc = globus_l_gram_enqueue_string(
+            fifo,
+            ",\n    'servicetag' => [ '%s' ]",
+            request->config->service_tag == 0 ? "untagged" : 
+                request->config->service_tag);
 
     rc = globus_l_gram_enqueue_string(
             fifo,
