--- /usr/src/debug/globus_gram_job_manager-10.70/globus_gram_job_manager_staging.c	2010-06-30 09:16:31.000000000 -0500
+++ globus_gram_job_manager_staging.c	2011-08-01 22:45:27.000000000 -0500
@@ -328,6 +328,8 @@
             rc = GLOBUS_GRAM_PROTOCOL_ERROR_OPENING_STDOUT;
             free(evaled_to);
             goto bad_value;
+        } else {
+            close(tmpfd);
         }
     }
     if (evaled_to)
