--- globus_gram_job_manager-10.70/globus_gram_job_manager_state.c	2010-05-14 08:01:22.000000000 -0500
+++ globus_gram_job_manager_state.c	2011-08-15 20:45:54.000000000 -0500
@@ -2174,6 +2174,17 @@
 
         if (path)
         {
+            globus_gram_job_manager_request_log(
+                    request,
+                    GLOBUS_GRAM_JOB_MANAGER_LOG_TRACE,
+                    "event=gram.file_cleanup.info "
+                    "level=TRACE "
+                    "gramid=%s "
+                    "path=\"%s\" "
+                    "msg=\"About to unlink\" "
+                    "\n",
+                    request->job_contact_path,
+                    path);
             remove(path);
         }
     }
--- globus_gram_job_manager-10.70/globus_gram_job_manager_request.c	2011-08-08 09:09:06.000000000 -0500
+++ globus_gram_job_manager_request.c	2011-08-15 20:37:16.000000000 -0500
@@ -3131,6 +3131,17 @@
             }
             else
             {
+                globus_gram_job_manager_request_log(
+                        request,
+                        GLOBUS_GRAM_JOB_MANAGER_LOG_TRACE,
+                        "event=gram.directory_destroy.info "
+                        "level=TRACE "
+                        "gramid=%s "
+                        "path=\"%s\" "
+                        "msg=\"About to unlink\" "
+                        "\n",
+                        request->job_contact_path,
+                        new_path);
                 rc = unlink(new_path);
                 if (rc < 0)
                 {
