Index: globus_gram_job_manager_state.c
===================================================================
--- globus_gram_job_manager_state.c.orig	2011-09-06 15:04:52.000000000 -0500
+++ globus_gram_job_manager_state.c	2011-09-06 15:20:03.000000000 -0500
@@ -2214,6 +2214,17 @@
 
         if (path)
         {
+            globus_gram_job_manager_request_log(
+                    request,
+                    GLOBUS_GRAM_JOB_MANAGER_LOG_TRACE,
+                    "event=gram.file_cleanup.info "
+                    "level=TRACE "
+                    "gramid=%s "
+                    "path=\"%s\" "
+                    "msg=\"About to unlink\" "
+                    "\n",
+                    request->job_contact_path,
+                    path);
             remove(path);
         }
     }
Index: globus_gram_job_manager_request.c
===================================================================
--- globus_gram_job_manager_request.c.orig	2011-09-06 15:09:02.000000000 -0500
+++ globus_gram_job_manager_request.c	2011-09-06 15:20:03.000000000 -0500
@@ -3230,6 +3230,17 @@
             }
             else
             {
+                globus_gram_job_manager_request_log(
+                        request,
+                        GLOBUS_GRAM_JOB_MANAGER_LOG_TRACE,
+                        "event=gram.directory_destroy.info "
+                        "level=TRACE "
+                        "gramid=%s "
+                        "path=\"%s\" "
+                        "msg=\"About to unlink\" "
+                        "\n",
+                        request->job_contact_path,
+                        new_path);
                 rc = unlink(new_path);
                 if (rc < 0)
                 {
