Index: globus_gram_job_manager.h
===================================================================
--- globus_gram_job_manager.h.orig	2016-08-10 13:31:42.798604232 -0500
+++ globus_gram_job_manager.h	2016-08-10 13:32:33.950258900 -0500
@@ -346,13 +346,19 @@ typedef struct
     char *                              service_tag;
     /**
      * This is the prefix to directory which will contain host+lrm specific
      * state files. By default, it is $HOME/.globus/job, but can be 
      * modified by -globus-job-dir command-line option
+     * For many job managers, this needs to be on a shared file system
      */
     char *                              job_dir_home;
     /**
+     * This is the location to place the host+lrm specific lock files.
+     * This should not be on a shared file system.
+     */
+    char *                              job_dir_locks;
+    /**
      * Boolean flag indicating whether to enable GSI callouts
      * on GRAM operations or not. Default to no.
      */
     globus_bool_t                       enable_callout;
 }
Index: globus_gram_job_manager_config.c
===================================================================
--- globus_gram_job_manager_config.c.orig	2016-08-10 13:31:42.798604232 -0500
+++ globus_gram_job_manager_config.c	2016-08-10 13:32:33.951258913 -0500
@@ -557,17 +557,32 @@ globus_gram_job_manager_config_init(
 
             goto out;
         }
     }
 
-    if (config->job_dir_home == NULL)
+    if (config->job_dir_locks == NULL)
     {
-        config->job_dir_home = 
+        config->job_dir_locks =
             globus_common_create_string("%s/%s",
                     config->job_state_file_dir,
                     config->logname);
 
+        if (config->job_dir_locks == NULL)
+        {
+            rc = GLOBUS_GRAM_PROTOCOL_ERROR_MALLOC_FAILED;
+
+            goto out;
+        }
+    }
+
+    if (config->job_dir_home == NULL)
+    {
+        config->job_dir_home = 
+            globus_common_create_string("%s/.globus/%s",
+                    config->home,
+                    config->logname);
+
         if (config->job_dir_home == NULL)
         {
             rc = GLOBUS_GRAM_PROTOCOL_ERROR_MALLOC_FAILED;
 
             goto out;
Index: globus_gram_job_manager.c
===================================================================
--- globus_gram_job_manager.c.orig	2016-08-10 13:32:00.404829566 -0500
+++ globus_gram_job_manager.c	2016-08-10 13:32:33.951258913 -0500
@@ -206,11 +206,11 @@ globus_gram_job_manager_init(
 
         goto job_id_hashtable_init_failed;
     }
     dir_prefix = globus_common_create_string(
             "%s/%s",
-            manager->config->job_dir_home,
+            manager->config->job_dir_locks,
             manager->config->short_hostname);
     if (dir_prefix == NULL)
     {
         rc = GLOBUS_GRAM_PROTOCOL_ERROR_MALLOC_FAILED;
         goto malloc_dir_prefix_failed;
