Index: globus_gram_job_manager_request.c
===================================================================
--- globus_gram_job_manager_request.c.orig	2011-12-19 14:22:50.000000003 -0600
+++ globus_gram_job_manager_request.c	2011-12-19 14:26:29.000000003 -0600
@@ -3827,7 +3827,7 @@
 
     out_file = globus_common_create_string(
                 "%s/.globus/job/%s/%s",
-                request->config->home,
+                request->config->job_dir_home,
                 request->config->short_hostname,
                 request->uniq_id);
     if (out_file == NULL)
