Index: globus_gram_job_manager_request.c
===================================================================
--- globus_gram_job_manager_request.c.orig	2012-04-18 16:58:47.166096473 -0500
+++ globus_gram_job_manager_request.c	2012-04-18 16:59:24.305973691 -0500
@@ -3810,7 +3810,7 @@
 
     out_file = globus_common_create_string(
                 "%s/.globus/job/%s/%s",
-                request->config->home,
+                request->config->job_dir_home,
                 request->config->short_hostname,
                 request->uniq_id);
     if (out_file == NULL)
