Index: globus_gram_job_manager.h
===================================================================
--- globus_gram_job_manager.h.orig	2012-06-18 11:44:17.000000000 -0500
+++ globus_gram_job_manager.h	2013-12-11 16:51:44.784013969 -0600
@@ -355,9 +355,15 @@
      * This is the prefix to directory which will contain host+lrm specific
      * state files. By default, it is $HOME/.globus/job, but can be 
      * modified by -globus-job-dir command-line option
+     * For many job managers, this needs to be on a shared file system
      */
     char *                              job_dir_home;
     /**
+     * This is the location to place the host+lrm specific lock files.
+     * This should not be on a shared file system.
+     */
+    char *                              job_dir_locks;
+    /**
      * Boolean flag indicating whether to enable GSI callouts
      * on GRAM operations or not. Default to no.
      */
Index: globus_gram_job_manager_config.c
===================================================================
--- globus_gram_job_manager_config.c.orig	2012-08-17 09:37:15.000000000 -0500
+++ globus_gram_job_manager_config.c	2013-12-11 16:51:44.785013982 -0600
@@ -564,11 +564,26 @@
         }
     }
 
+    if (config->job_dir_locks == NULL)
+    {               
+        config->job_dir_locks = 
+            globus_common_create_string("%s/%s",
+                    config->job_state_file_dir,
+                    config->logname);
+                    
+        if (config->job_dir_locks == NULL)
+        {           
+            rc = GLOBUS_GRAM_PROTOCOL_ERROR_MALLOC_FAILED;
+                    
+            goto out;
+        }           
+    }
+
     if (config->job_dir_home == NULL)
     {
         config->job_dir_home = 
-            globus_common_create_string("%s/%s",
-                    config->job_state_file_dir,
+            globus_common_create_string("%s/.globus/%s",
+                    config->home,
                     config->logname);
 
         if (config->job_dir_home == NULL)
Index: globus_gram_job_manager.c
===================================================================
--- globus_gram_job_manager.c.orig	2013-12-11 16:49:18.998216030 -0600
+++ globus_gram_job_manager.c	2013-12-11 16:51:44.786013994 -0600
@@ -212,7 +212,7 @@
     }
     dir_prefix = globus_common_create_string(
             "%s/%s",
-            manager->config->job_dir_home,
+            manager->config->job_dir_locks,
             manager->config->short_hostname);
     if (dir_prefix == NULL)
     {
