Index: globus_gram_job_manager_request.c
===================================================================
--- globus_gram_job_manager_request.c.orig	2016-08-10 13:32:13.035991227 -0500
+++ globus_gram_job_manager_request.c	2016-08-10 13:32:30.187210738 -0500
@@ -3826,11 +3826,11 @@ globus_l_gram_make_job_dir(
             "\n",
             request->job_contact_path);
 
     out_file = globus_common_create_string(
                 "%s/.globus/job/%s/%s",
-                request->config->home,
+                request->config->job_dir_home,
                 request->config->short_hostname,
                 request->uniq_id);
     if (out_file == NULL)
     {
         rc = GLOBUS_GRAM_PROTOCOL_ERROR_MALLOC_FAILED;
