--- main.c	2011-10-20 13:11:48.000000000 -0500
+++ main.c.new	2011-11-29 20:19:33.530000000 -0600
@@ -547,6 +547,16 @@
                 &forked_starter);
     }
 
+    /* Enable log rotation via SIGUSR1 */
+    {
+        struct sigaction act;
+        act.sa_handler = globus_i_job_manager_log_rotate;
+        sigemptyset(&act.sa_mask);
+        sigaddset(&act.sa_mask, SIGUSR1);
+        act.sa_flags = 0;
+        sigaction(SIGUSR1, &act, NULL);
+    }
+
     GlobusGramJobManagerLock(&manager);
     if (manager.socket_fd != -1 &&
         globus_hashtable_empty(&manager.request_hash) &&
--- globus_gram_job_manager.h.old	2011-10-20 10:18:33.000000000 -0500
+++ globus_gram_job_manager.h	2011-11-29 20:20:33.990000000 -0600
@@ -1580,7 +1580,11 @@
 extern globus_logging_handle_t          globus_i_gram_job_manager_log_sys;
 
 extern
+void
+globus_i_job_manager_log_rotate(int sig);
+
 int
+extern
 globus_gram_job_manager_logging_init(
     globus_gram_job_manager_config_t *  config);
 
--- logging.c	2011-07-27 06:27:08.000000000 -0500
+++ logging.c	2011-11-29 20:22:24.881000000 -0600
@@ -28,6 +28,7 @@
 static globus_symboltable_t             globus_l_gram_log_symboltable;
 static FILE *                           globus_l_gram_log_fp = NULL;
 globus_thread_key_t                     globus_i_gram_request_key;
+static globus_bool_t                    globus_l_gram_reopen_log;
 
 
 static
@@ -188,6 +189,12 @@
 }
 /* globus_gram_prepare_log_string() */
 
+void
+globus_i_job_manager_log_rotate(int sig)
+{
+    globus_l_gram_reopen_log = GLOBUS_TRUE;
+}
+
 static
 void
 globus_l_gram_logging_write(
@@ -264,12 +271,16 @@
         return;
     }
 
-    if (last_path == NULL || strcmp(path, last_path) != 0)
+    if (last_path == NULL ||
+        globus_l_gram_reopen_log ||
+        strcmp(path, last_path) != 0)
     {
+        globus_l_gram_reopen_log = GLOBUS_FALSE;
         if (last_path)
         {
             free(last_path);
         }
+
         last_path = path;
         path = NULL;
 
