Index: startup_socket.c
===================================================================
RCS file: /home/globdev/CVS/globus-packages/gram/jobmanager/source/startup_socket.c,v
retrieving revision 1.19.2.8
diff -u -r1.19.2.8 startup_socket.c
--- startup_socket.c	19 Oct 2012 14:54:20 -0000	1.19.2.8
+++ startup_socket.c	16 May 2013 14:59:53 -0000
@@ -1999,6 +1999,14 @@
                     &cred);
             cred = GSS_C_NO_CREDENTIAL;
         }
+        if (peername != NULL)
+        {
+            free(peername);
+        }
+        if (peer_cred_handle != NULL)
+        {
+            globus_gsi_cred_handle_destroy(peer_cred_handle);
+        }
 ackfailed:
 failed_import_cred:
         if (http_body_fd != -1)
@@ -2262,8 +2270,8 @@
     const char * endp = p + len;
     const char * cert_pem;
     X509 * cert;
-    BIO * b;
-    STACK_OF(X509) * chain;
+    BIO * b = NULL;
+    STACK_OF(X509) * chain = NULL;
 
     chain = sk_X509_new_null();
 
@@ -2301,11 +2309,14 @@
             cert = PEM_read_bio_X509(b, NULL, 0, NULL);
 
             globus_gsi_cred_set_cert(*cred_handle, cert);
+            X509_free(cert);
+            cert = NULL;
         }
         p += strlen(p) + 1;
     }
     globus_gsi_cred_set_cert_chain(*cred_handle, chain);
     sk_X509_pop_free(chain, X509_free);
+    BIO_free(b);
 
     return GLOBUS_SUCCESS;
 }
Index: tg_gateway.c
===================================================================
RCS file: /home/globdev/CVS/globus-packages/gram/jobmanager/source/tg_gateway.c,v
retrieving revision 1.5
diff -u -r1.5 tg_gateway.c
--- tg_gateway.c	7 Nov 2011 23:38:16 -0000	1.5
+++ tg_gateway.c	16 May 2013 14:59:53 -0000
@@ -54,14 +54,14 @@
     xmlXPathContextPtr                  xpath_ctx;
     xmlXPathObjectPtr                   xresult;
     int                                 rc;
-    ASN1_OBJECT *                       asn1_desired_object;
+    ASN1_OBJECT *                       asn1_desired_object = NULL;
     int                                 cert_count;
     int                                 found_index;
     int                                 chain_index;
     X509                               *cert;
     X509_EXTENSION *                    extension;
     ASN1_OCTET_STRING                  *asn1_oct_string;
-    STACK_OF(X509)                     *chain;
+    STACK_OF(X509)                     *chain = NULL;
 
     *gateway_user = NULL;
 
@@ -95,7 +95,6 @@
             free(msg);
             rc = GLOBUS_GRAM_PROTOCOL_ERROR_AUTHORIZATION;
 
-            ASN1_OBJECT_free(asn1_desired_object);
             goto no_extension_in_cred_chain;
         }
 
@@ -113,7 +112,6 @@
                     rc = GLOBUS_GRAM_PROTOCOL_ERROR_AUTHORIZATION;
                     globus_gram_protocol_error_7_hack_replace_message(
                         "Unable to extract SAML assertion extension from certificate chain");
-                    ASN1_OBJECT_free(asn1_desired_object);
                     goto no_extension_in_cred_chain;
                 }
                 asn1_oct_string = X509_EXTENSION_get_data(extension);
@@ -122,7 +120,6 @@
                     rc = GLOBUS_GRAM_PROTOCOL_ERROR_AUTHORIZATION;
                     globus_gram_protocol_error_7_hack_replace_message(
                         "Unable to extract SAML assertion extension from certificate chain");
-                    ASN1_OBJECT_free(asn1_desired_object);
                     goto no_extension_in_cred_chain;
                 }
                 p = asn1_oct_string->data;
@@ -133,14 +130,12 @@
                     rc = GLOBUS_GRAM_PROTOCOL_ERROR_AUTHORIZATION;
                     globus_gram_protocol_error_7_hack_replace_message(
                         "Unable to convert SAML assertion text from DER to UTF8");
-                    ASN1_OBJECT_free(asn1_desired_object);
                     goto no_extension_in_cred_chain;
                 }
                 assertion_string = malloc(asn1_str->length + 1);
                 if (assertion_string == NULL)
                 {
                     rc = GLOBUS_GRAM_PROTOCOL_ERROR_MALLOC_FAILED;
-                    ASN1_OBJECT_free(asn1_desired_object);
                     goto no_extension_in_cred_chain;
                 }
                 memcpy(assertion_string, asn1_str->data, asn1_str->length);
@@ -311,6 +306,14 @@
 inquire_failed:
 no_extension_in_cred_chain:
 no_context:
+    if (asn1_desired_object != NULL)
+    {
+        ASN1_OBJECT_free(asn1_desired_object);
+    }
+    if (chain != NULL)
+    {
+        sk_X509_free(chain);
+    }
     return rc;
 #else
     *gateway_user = NULL;
