--- globus_gram_job_manager-10.70/globus_gram_job_manager_script.c	Tue Mar 25 15:12:38 2008
+++ globus_gram_job_manager-10.70/globus_gram_job_manager_script.c	Tue Mar 25 15:21:08 2008
@@ -850,6 +850,7 @@
     {
         size_t len = 0;
         char * line_bit = line;
+        int new_status = 0;
         if( ! fgets(line, sizeof(line), grid_monitor_file) )
         {
             /* end of file (or error), job isn't in file.  It might just not
@@ -896,7 +897,19 @@
             goto FAST_POLL_EXIT_FAILURE;
         }
 
-        local_globus_set_status(request, atoi(line_bit));
+        new_status = atoi(line_bit);
+        if ( new_status == GLOBUS_GRAM_PROTOCOL_JOB_STATE_FAILED ) {
+            /* Condor's grid monitor doesn't report the GRAM failure code
+             * that usually accompanies a FAILED status. Let's call the
+             * poll function ourselves to get that status.
+             */
+            globus_gram_job_manager_request_log(request,
+                GLOBUS_GRAM_JOB_MANAGER_LOG_ERROR,
+                "JMI: poll_fast: Monitoring file reports FAILED, "
+                "Reverting to normal polling to get failure code.\n");
+            goto FAST_POLL_EXIT_FAILURE;
+        }
+        local_globus_set_status(request, new_status);
 
         return_val = GLOBUS_SUCCESS;
         goto FAST_POLL_EXIT;
--- globus_gram_job_manager-10.70/globus_gram_job_manager_state_file.c	Mon Mar 12 09:03:17 2007
+++ globus_gram_job_manager-10.70/globus_gram_job_manager_state_file.c	Tue Mar 25 15:20:30 2008
@@ -838,8 +838,11 @@
     {
         goto error_exit;
     }
+    /* We don't know when the job status in the state file was obtained.
+     * Set the time to 0 so that any new status will be used.
+     */
     globus_gram_job_manager_request_set_status_time(request,
-                atoi( buffer ), statbuf.st_mtime);
+                atoi( buffer ), 0);
 
     if (fgets( buffer, file_len, fp ) == NULL)
     {
