Index: gram.logrotate.in
===================================================================
--- gram.logrotate.in.orig	2011-11-28 11:44:46.000000002 -0600
+++ gram.logrotate.in	2013-09-11 17:12:12.000000002 -0500
@@ -7,7 +7,5 @@
     nocreate
     rotate 4
     sharedscripts
-    postrotate
-        /usr/bin/killall -USR1 globus-job-manager 2> /dev/null || true
-    endscript
+    copytruncate
 }
