--- globus_gram_job_manager_staging.c.orig	2012-03-12 12:29:46.000000000 -0500
+++ globus_gram_job_manager_staging.c	2012-03-12 12:30:25.000000000 -0500
@@ -94,11 +94,6 @@
         0
     };
 
-    if(request->jm_restart)
-    {
-        return GLOBUS_SUCCESS;
-    }
-
     for(i = 0; can_stage_list[i] != NULL; i++)
     {
         list = globus_rsl_param_get_values(request->rsl, can_stage_list[i]);
--- globus_gram_job_manager_request.c	2012-03-12 19:25:29.000000000 -0500
+++ globus_gram_job_manager_request.c	2012-03-12 19:25:22.000000000 -0500
@@ -673,6 +673,12 @@
         rc = GLOBUS_GRAM_PROTOCOL_ERROR_NO_RESOURCES;
         goto cond_init_failed;
     }
+
+    // Redo the staging list.  TODO: if restarting, leaks memory.
+    r->stage_in_todo = NULL;
+    r->stage_in_shared_todo = NULL;
+    r->stage_out_todo = NULL;
+    r->stage_stream_todo = NULL;
     rc = globus_gram_job_manager_staging_create_list(r);
     if(rc != GLOBUS_SUCCESS)
     {
@@ -2432,7 +2438,6 @@
         rc = globus_i_gram_remote_io_url_update(request);
     }
 
-    /*
     globus_gram_job_manager_request_log(
             request,
             GLOBUS_GRAM_JOB_MANAGER_LOG_TRACE,
@@ -2444,7 +2449,6 @@
                 request->job_contact_path,
                 globus_rsl_unparse(original_rsl),
                 globus_rsl_unparse(request->rsl));
-    */
 
     /* Now that we've recreated the stdio, redo the staging list. */
     request->stage_in_todo = NULL;
