--- globus_gram_job_manager_state_file.c.orig	2012-03-08 18:04:09.000000000 -0600
+++ globus_gram_job_manager_state_file.c	2012-03-08 18:35:29.000000000 -0600
@@ -726,28 +726,52 @@
                                            O_RDWR );
         if ( request->job_state_lock_fd < 0 )
         {
-            rc = GLOBUS_GRAM_PROTOCOL_ERROR_LOCKING_STATE_LOCK_FILE;
+            if (errno == ENOENT)
+            {
+                // Someone deleted our lock file.
+                rc = globus_l_gram_state_file_create_lock(request);
 
-            globus_gram_job_manager_request_log(
-                    request,
-                    GLOBUS_GRAM_JOB_MANAGER_LOG_ERROR,
-                    "event=gram.state_file_read.end "
-                    "level=ERROR "
-                    "gramid=%s "
-                    "status=%d "
-                    "path=%s "
-                    "msg=\"%s\" "
-                    "errno=%d "
-                    "reason=\"%s\" "
-                    "\n",
-                    request->job_contact_path,
-                    -rc,
-                    request->job_state_file,
-                    "Error opening job lock file",
-                    errno,
-                    strerror(errno));
+                if (rc != GLOBUS_SUCCESS) {
+                    globus_gram_job_manager_request_log(
+                            request,
+                            GLOBUS_GRAM_JOB_MANAGER_LOG_ERROR,
+                            "event=gram.state_file_read.end "
+                            "level=ERROR "
+                            "gramid=%s "
+                            "status=%d "
+                            "path=%s "
+                            "msg=\"%s\" "
+                            "\n",
+                            request->job_contact_path,
+                            -rc,
+                            request->job_state_file,
+                            "Error re-creating job lock file");
+                goto free_buffer_exit;
+                }
+            } else
+            {
+                rc = GLOBUS_GRAM_PROTOCOL_ERROR_LOCKING_STATE_LOCK_FILE;
 
-            goto free_buffer_exit;
+                globus_gram_job_manager_request_log(
+                        request,
+                        GLOBUS_GRAM_JOB_MANAGER_LOG_ERROR,
+                        "event=gram.state_file_read.end "
+                        "level=ERROR "
+                        "gramid=%s "
+                        "status=%d "
+                        "path=%s "
+                        "msg=\"%s\" "
+                        "errno=%d "
+                        "reason=\"%s\" "
+                        "\n",
+                        request->job_contact_path,
+                        -rc,
+                        request->job_state_file,
+                        "Error opening job lock file",
+                        errno,
+                        strerror(errno));
+                goto free_buffer_exit;
+            }
         }
 
         rc = globus_gram_job_manager_file_lock(request->job_state_lock_fd);
