Index: startup_socket.c
===================================================================
--- startup_socket.c.orig	2011-11-07 17:38:16.000000003 -0600
+++ startup_socket.c	2011-12-19 14:23:36.000000003 -0600
@@ -1303,7 +1303,7 @@
             if (rc <= 0 && errno == EAGAIN)
             {
                 tries--;
-                globus_libc_usleep(10000);
+                globus_libc_usleep(1000000);
             }
             else
             {
