--- globus_gram_job_manager_request.c	7 Nov 2011 23:38:15 -0000	1.35
+++ globus_gram_job_manager_request.c	21 Nov 2011 16:20:50 -0000
@@ -809,6 +809,11 @@
         free(r->cache_tag);
 cache_tag_alloc_failed:
 failed_restart:
+        if (r->job_state_lock_fd >= 0 &&
+            r->job_state_lock_fd != r->manager->lock_fd)
+        {
+            close(r->job_state_lock_fd);
+        }
         free(r->job_state_lock_file);
         free(r->job_state_file);
 failed_state_file_set:
