Index: globus_gram_job_manager.c
===================================================================
--- globus_gram_job_manager.c.orig	2011-11-22 11:55:33.000000003 -0600
+++ globus_gram_job_manager.c	2011-12-19 14:19:30.000000003 -0600
@@ -2867,6 +2867,7 @@
         goto malloc_restart_rsl_failed;
     }
 
+    GlobusGramJobManagerUnlock(manager);
     rc = globus_gram_job_manager_request_init(
             request,
             manager,
@@ -2878,6 +2879,7 @@
             NULL,
             NULL,
             NULL);
+    GlobusGramJobManagerLock(manager);
     free(restart_rsl);
 malloc_restart_rsl_failed:
     return rc;
