Index: globus_gram_job_manager.c
===================================================================
--- globus_gram_job_manager.c.orig	2011-09-06 15:07:44.000000000 -0500
+++ globus_gram_job_manager.c	2011-09-06 15:08:27.000000000 -0500
@@ -2336,10 +2336,13 @@
                  request->restart_state ==
                         GLOBUS_GRAM_JOB_MANAGER_STATE_POLL_QUERY2))
             {
+                /* register_job_id wants to hold the lock itself */
+                GlobusGramJobManagerUnlock(manager);
                 rc = globus_gram_job_manager_register_job_id(
                         request->manager,
                         request->job_id_string,
                         request);
+                GlobusGramJobManagerLock(manager);
                 if (rc != GLOBUS_SUCCESS)
                 {
                     globus_gram_job_manager_request_log(
