Index: JobManager.pm
===================================================================
RCS file: /home/globdev/CVS/globus-packages/gram/jobmanager/scripts/JobManager.pm,v
retrieving revision 1.16
diff -u -r1.16 JobManager.pm
--- JobManager.pm	9 Aug 2011 16:32:04 -0000	1.16
+++ JobManager.pm	15 Nov 2011 18:01:34 -0000
@@ -113,6 +113,23 @@
     return $self;
 }
 
+sub getenv
+{
+    my $self = shift;
+    my $varname = shift;
+    my $description = $self->{JobDescription};
+    my @result;
+    my @environment = $description->environment();
+
+    @result = grep { $_->[0] eq $varname } @environment;
+
+    if (exists($result[0]))
+    {
+        return $result[0]->[1];
+    }
+    return undef;
+}
+
 =item $manager->log($string)
 
 Log a message to the job manager log file. The message is preceded by
@@ -478,10 +495,14 @@
     my $self = shift;
     my $description = $self->{JobDescription};
     my $tag = $description->cache_tag() || $ENV{'GLOBUS_GRAM_JOB_CONTACT'};
+    my $cache_location = $self->getenv('GLOBUS_GASS_CACHE_DEFAULT');
     my $url;
     my $filename;
     my $filestreamout = [];
 
+    local %ENV = %ENV;
+    $ENV{GLOBUS_GASS_CACHE_DEFAULT} = $cache_location if ($cache_location);
+
     foreach ('stdin', 'executable')
     {
 	chomp($url = $description->$_());
@@ -587,10 +608,14 @@
     my $self = shift;
     my $description = $self->{JobDescription};
     my $tag = $description->cache_tag() || $ENV{'GLOBUS_GRAM_JOB_CONTACT'};
+    my $cache_location = $self->getenv('GLOBUS_GASS_CACHE_DEFAULT');
     my ($remote, $local, $local_resolved, $cached, $stderr, $rc, @arg);
 
     $self->log("stage_in(enter)");
 
+    local %ENV = %ENV;
+    $ENV{GLOBUS_GASS_CACHE_DEFAULT} = $cache_location if ($cache_location);
+
     if($description->executable() =~ m|^[a-zA-Z]+://|)
     {
         @arg = ($cache_pgm, '-add', '-t', $tag, $description->executable());
@@ -747,6 +772,7 @@
     my $description = $self->{JobDescription};
     my $url_copy = "$Globus::Core::Paths::bindir/globus-url-copy";
     my $tag = $description->cache_tag() || $ENV{'GLOBUS_GRAM_JOB_CONTACT'};
+    my $cache_location = $self->getenv('GLOBUS_GASS_CACHE_DEFAULT');
     my ($local, $remote);
     my ($local_path, $remote_path);
     my ($stderr, $rc);
@@ -754,6 +780,9 @@
 
     $self->log("stage_out(enter)");
 
+    local %ENV = %ENV;
+    $ENV{GLOBUS_GASS_CACHE_DEFAULT} = $cache_location if ($cache_location);
+
     if (exists($self->{STDIO_MERGER}) && ref($self->{STDIO_MERGER}))
     {
         my $merger = $self->{STDIO_MERGER};
@@ -898,12 +927,16 @@
 {
     my $self = shift;
     my $description = $self->{JobDescription};
+    my $cache_location = $self->getenv('GLOBUS_GASS_CACHE_DEFAULT');
     my $tag = $description->cache_tag() || $ENV{'GLOBUS_GRAM_JOB_CONTACT'};
     my $job_path = $self->job_dir();
     my ($stderr, $rc);
 
     $self->log("cache_cleanup(enter)");
 
+    local %ENV = %ENV;
+    $ENV{GLOBUS_GASS_CACHE_DEFAULT} = $cache_location if ($cache_location);
+
     if ( defined $tag )
     {
          ($stderr, $rc) = $self->pipe_err_cmd($cache_pgm,
