Index: globus-job-manager-script.pl.in
===================================================================
--- globus-job-manager-script.pl.in.orig	2013-09-06 14:58:46.000000000 -0500
+++ globus-job-manager-script.pl.in	2013-12-12 18:34:19.845612570 -0600
@@ -43,6 +43,9 @@
 use Globus::GRAM::Error;
 use Globus::GRAM::JobDescription;
 
+eval "require Globus::GRAM::JobManagerGratia"; # Gratia patch to save DN/FQAN on submit
+my $gratia_callout = $@?0:1; # Don't make the callout if the file is not there.
+
 my($manager_name, $argument_file, $command, $help);
 
 GetOptions('manager-name|m=s' => \$manager_name,
@@ -189,6 +192,11 @@
     else
     {
         $manager->respond($result);
+        # Save cert information for Gratia accounting if appropriate.
+        if ($gratia_callout and ($cmd eq 'submit') and (exists $result->{JOB_ID})) {
+            Globus::GRAM::JobManagerGratia::gratia_save_cert_info($manager, $result->{JOB_ID});
+        }
+
     }
 FILTERED:
     if ($command eq 'interactive')
