Index: JobDescription.pm.in
===================================================================
--- JobDescription.pm.in.orig	2013-12-12 18:34:40.719872233 -0600
+++ JobDescription.pm.in	2013-12-12 18:35:01.834134894 -0600
@@ -396,6 +396,9 @@
     } else {
         my %preset = ( %ENV ); # as meager as it may be
 
+        # PATH is no longer in the present environment, now that we don't use xinetd
+        $result{"PATH"} = "/bin:/usr/bin";
+
         # no previous knowledge, need to read the file
         my $fn = "/var/lib/osg/osg-job-environment.conf";
         if ( open( INFO, "<$fn" ) ) {
