--- /usr/lib/perl5/vendor_perl/5.8.8/Globus/GRAM/JobDescription.pm	2011-07-23 10:27:29.000000000 -0500
+++ JobDescription.pm	2011-08-07 13:48:34.000000000 -0500
@@ -381,6 +381,9 @@
     } else {
         my %preset = ( %ENV ); # as meager as it may be
 
+        # PATH is no longer in the present environment, now that we don't use xinetd
+        $result{"PATH"} = "/bin:/usr/bin"
+
         # no previous knowledge, need to read the file
         my $fn = "/etc/osg/osg-job-environment.conf";
         if ( open( INFO, "<$fn" ) ) {
