Index: JobDescription.pm.in
===================================================================
--- JobDescription.pm.in.orig	2011-09-07 11:52:44.000000000 -0500
+++ JobDescription.pm.in	2011-09-07 11:53:10.000000000 -0500
@@ -395,6 +395,9 @@
     } else {
         my %preset = ( %ENV ); # as meager as it may be
 
+        # PATH is no longer in the present environment, now that we don't use xinetd
+        $result{"PATH"} = "/bin:/usr/bin";
+
         # no previous knowledge, need to read the file
         my $fn = "/etc/osg/osg-job-environment.conf";
         if ( open( INFO, "<$fn" ) ) {
