--- /creation/reconfig_frontend	2011-11-03 15:58:41.000000000 -0500
+++ /creation/reconfig_frontend	2011-11-03 15:59:08.000000000 -0500
@@ -33,17 +33,10 @@
 ################################################################################
 
 def main(params, old_params, update_scripts):
-    # load old files
-    old_frontend_dicts_obj=cvWParamDict.frontendDicts(old_params)
-    old_frontend_dicts_obj.load()
-
     # create dictionaries for new params
     frontend_dicts_obj=cvWParamDict.frontendDicts(params)
     frontend_dicts_obj.populate()
 
-    # merge them together
-    frontend_dicts_obj.reuse(old_frontend_dicts_obj)
-
     # write to disk
     frontend_dicts_obj.save()
     frontend_dicts_obj.set_readonly(True)
