Index: _quilt/privilege-xacml-2.6.2/src/test/XACMLClientTest.sh
===================================================================
--- _quilt.orig/privilege-xacml-2.6.2/src/test/XACMLClientTest.sh
+++ _quilt/privilege-xacml-2.6.2/src/test/XACMLClientTest.sh
@@ -31,7 +31,7 @@ export	    RESOURCE_X509_ISSUER="/DC=org
 unset	    ACTION_RSL_STRING
 
 #MAVEN_REPOSITORY=/home/tdh/maven/repository
-MAVEN_REPOSITORY=/tmp/m2-repository
+#MAVEN_REPOSITORY=/tmp/m2-repository
 
 java \
 -DsslKey=/etc/grid-security/hostkey.pem \
@@ -58,21 +58,7 @@ $bindir:\
 `build-classpath axis/jaxrpc.jar`:\
 `build-classpath wsdl4j` \
 org.opensciencegrid.authz.xacml.client.XACMLClientTest
-: << 'END'
-$MAVEN_REPOSITORY/org/opensaml/opensaml/2.4.1/opensaml-2.4.1.jar:\
-$MAVEN_REPOSITORY/org/opensaml/xmltooling/1.3.1/xmltooling-1.3.1.jar:\
-$MAVEN_REPOSITORY/org/apache/santuario/xmlsec/1.4.1/xmlsec-1.4.1.jar:\
-$MAVEN_REPOSITORY/org/codehaus/fabric3/fabric3-db-exist/sunxacml/1.0/sunxacml-1.0.jar:\
-$MAVEN_REPOSITORY/velocity/velocity/1.5/velocity-1.5.jar:\
-$MAVEN_REPOSITORY/commons-collections/commons-collections/3.1/commons-collections-3.1.jar:\
-$MAVEN_REPOSITORY/org/opensaml/openws/1.4.1/openws-1.4.1.jar:\
-$MAVEN_REPOSITORY/junit/junit/4.8.1/junit-4.8.1.jar:\
-$MAVEN_REPOSITORY/commons-cli/commons-cli/1.2/commons-cli-1.2.jar:\
-$MAVEN_REPOSITORY/javax/servlet/servlet-api/2.5/servlet-api-2.5.jar:\
-$MAVEN_REPOSITORY/commons-httpclient/commons-httpclient/3.1/commons-httpclient-3.1.jar:\
-$MAVEN_REPOSITORY/org/apache/xmlbeans/xmlbeans/2.4.0/xmlbeans-2.4.0.jar \
-org.opensciencegrid.authz.xacml.client.XACMLClientTest
-END
+
 
 
 
