diff --git a/src/main/java/org/opensciencegrid/authz/xacml/service/BasicMappingXACMLAuthZService.java b/src/main/java/org/opensciencegrid/authz/xacml/service/BasicMappingXACMLAuthZService.java
index d5a9e19..164572c 100644
--- a/src/main/java/org/opensciencegrid/authz/xacml/service/BasicMappingXACMLAuthZService.java
+++ b/src/main/java/org/opensciencegrid/authz/xacml/service/BasicMappingXACMLAuthZService.java
@@ -114,7 +114,7 @@ public class BasicMappingXACMLAuthZService extends XACMLAuthorizationPortTypeSOA
         try {
             xacmlAuthz = mapService.mapCredentials(queryRequest);
         } catch (Exception e) {
-            logger.error(e);
+            logger.error("Mapping service failed to map credentials", e);
 
             // Build StatusMessageType
             StatusMessageTypeImplBuilder statusMessageBuilder = (StatusMessageTypeImplBuilder)Configuration.getBuilderFactory().getBuilder(StatusMessageType.DEFAULT_ELEMENT_NAME);
