--- cloudera/install_hadoop.sh.orig	2010-10-11 09:39:27.000000000 -0700
+++ cloudera/install_hadoop.sh	2011-10-06 12:11:13.000000000 -0700
@@ -231,7 +231,7 @@
 fi
 
 if [ "\$LD_LIBRARY_PATH" = "" ]
-	then JVM_LIB=\`find \${JAVA_HOME}/jre/lib -name libjvm.so |tail -n 1\`
+	then JVM_LIB=\`find -L \${JAVA_HOME}/jre/lib -name libjvm.so |tail -n 1\`
         export LD_LIBRARY_PATH=\`dirname \$JVM_LIB\`:/usr/lib/
 
 fi
