Index: pakiti-2.1.4/install/pakiti2.apache2
===================================================================
--- pakiti-2.1.4.orig/install/pakiti2.apache2	2009-11-27 01:21:56.000000003 -0600
+++ pakiti-2.1.4/install/pakiti2.apache2	2012-12-03 15:28:50.000000003 -0600
@@ -2,18 +2,25 @@
 
 	SSLEngine on
 	SSLCipherSuite ALL:!ADH:!EXPORT56:RC4+RSA:+HIGH:+MEDIUM:+LOW:+SSLv2:+EXP
-	
-	SSLCertificateKeyFile "/etc/grid-security/hostkey.pem"
-	SSLCertificateFile "/etc/grid-security/hostcert.pem" 
+	SSLVerifyClient optional
+	SSLVerifyDepth 2
+	SSLOptions +StdEnvVars
+
+	SSLCertificateKeyFile "/etc/grid-security/http/httpkey.pem"
+	SSLCertificateFile "/etc/grid-security/http/httpcert.pem"
 	SSLCACertificatePath "/etc/grid-security/certificates/"
 
-	ServerName pakiti.server.com:443
-        ServerAdmin admin@admin.com	
+	# Configure servername and admin contact here if not already in httpd.conf
+	#ServerName pakiti.server.com:443
+        #ServerAdmin admin@admin.com
 
 	DocumentRoot /var/lib/pakiti2/www/pakiti
-	ErrorLog /var/log/apache2/pakiti-error
-	CustomLog /var/log/apache2/pakiti-access common
-	CustomLog /var/log/apache2/ssl_request_pakiti_log ssl_combined
+	Alias /pakiti /var/lib/pakiti2/www/pakiti
+
+	# Uncomment to have seperate log files for pakiti
+	#ErrorLog /var/log/apache2/pakiti-error
+	#CustomLog /var/log/apache2/pakiti-access common
+	#CustomLog /var/log/apache2/ssl_request_pakiti_log ssl_combined
 	
 	Alias /feed "/var/lib/pakiti2/www/feed"
 	<Directory "/var/lib/pakiti2/www/feed">
Index: pakiti-2.1.4/client/pakiti2-client.conf
===================================================================
--- pakiti-2.1.4.orig/client/pakiti2-client.conf	2010-07-29 09:27:44.000000003 -0500
+++ pakiti-2.1.4/client/pakiti2-client.conf	2012-12-03 15:07:08.000000003 -0600
@@ -1,18 +1,17 @@
 # Configuration file for Pakiti.
 
 # Pakiti servers, use space as delimiter
-# Exmaple: servers_name = pakiti1.server.com:443 pakiti2.server.com:8443
+# Example: servers_name = pakiti1.server.com:443 pakiti2.server.com:8443
 servers_name = pakiti.server.com:443
 
 # URL of Pakiti script on the server, default /feed/
 server_url = /feed/
 
 # CA Path, where is located certificate of the CA which issued SSL certificate for the Pakiti server
-#ca_certificate = /etc/ssl/certs
-#ca_certificate = /etc/grid-security/certificates
+ca_certificate = /etc/grid-security/certificates
 
 # The client certificate and the key
-#host_cert = /etc/ssl/host.pem
+host_cert = /etc/grid-security/hostcert.pem
 
 # Connection method: 'autodetect' (default) or 'curl' or 'openssl' or 'stdout'
 #connection_method = autodetect
