--- globus_gram_protocol.c	2011-08-09 11:32:11.000000000 -0500
+++ globus_gram_protocol.c	2012-05-10 13:16:55.788354000 -0500
@@ -32,7 +32,7 @@
 int					globus_i_gram_protocol_num_connects;
 int                                     globus_i_gram_protocol_max_concurrency;
 globus_gram_protocol_handle_t		globus_i_gram_protocol_handle;
-const int GLOBUS_GRAM_PROTOCOL_DEFAULT_MAX_CONCURRENCY = 10;
+const int GLOBUS_GRAM_PROTOCOL_DEFAULT_MAX_CONCURRENCY = 50;
 
 static int globus_l_gram_protocol_activate(void);
 static int globus_l_gram_protocol_deactivate(void);
