diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/native/libhdfs/exception.c b/hadoop-hdfs-project/hadoop-hdfs/src/main/native/libhdfs/exception.c
index a0a60a6..18180e1 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/native/libhdfs/exception.c
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/native/libhdfs/exception.c
@@ -85,6 +85,11 @@ static const struct ExceptionInfo gExceptionInfo[] = {
         ENOTSUP,
     },
     {
+        .name = "org.apache.hadoop.hdfs.protocol.DSQuotaExceededException",
+        .noPrintFlag = 0,
+        .excErrno = EDQUOT,
+    },
+    {
         "org.apache.hadoop.hdfs.server.namenode.LeaseExpiredException",
         0,
         ESTALE,
