commit 7e7054d1abe10de4e5cc4153bc8910180dfdc535
Author: Carl Edquist <edquist@cs.wisc.edu>
Date:   Mon Oct 23 17:07:12 2017 -0500

    port 2.0.0 dcb patch

diff --git a/do-component-build b/do-component-build
index c8981ef..745335f 100755
--- a/do-component-build
+++ b/do-component-build
@@ -17,8 +17,10 @@
 set -ex
 
 if [ -z "$GIT_REPO" ]; then
-    export HADOOP_PROTOC_CDH5_PATH=/opt/toolchain/protobuf-2.5.0/bin/protoc
+    export HADOOP_PROTOC_CDH5_PATH=/usr/bin/protoc
     export HADOOP_OPENSSL_101J_PATH=/opt/toolchain/openssl-1.0.1j
+    export PATH=$PATH:/usr/share/apache-maven-3.0.4/bin
+
 
     if [ -z "${DO_MAVEN_DEPLOY}" ]; then
         MAVEN_INST_DEPLOY=install
@@ -82,8 +84,8 @@ if [ -z "$GIT_REPO" ]; then
       MAVEN_SKIP_TESTS="-DskipTests -DskipTest -DskipITs"
       JAVA_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" \
           MAVEN_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" \
-          mvn $BUNDLE_SNAPPY $P_NATIVE \
-          -Pdist -Pfuse -Psrc -Dtar -Drequire.fuse=true \
+          mvn -Dmaven.wagon.http.ssl.allowall=true -Dmaven.wagon.http.ssl.insecure=true -B $BUNDLE_SNAPPY $P_NATIVE \
+          -e -Pdist -Pfuse -Psrc -Dtar -Drequire.fuse=true \
           ${MAVEN_SSL_FLAGS} ${MAVEN_SKIP_TESTS} ${MAVEN_SKIP_JAVADOC} \
           $MAVEN_INST_DEPLOY \
           "$@"
@@ -98,14 +100,14 @@ if [ -z "$GIT_REPO" ]; then
 
     if [ -z "${MAVEN_ONLY}" ]; then
 
-        JAVA_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" MAVEN_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" mvn $CAULDRON_MAVEN_EXTRA_ARGS site site:stage ${MAVEN_SKIP_TESTS} $@
+        JAVA_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" MAVEN_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" mvn -Dmaven.wagon.http.ssl.allowall=true -Dmaven.wagon.http.ssl.insecure=true -B $CAULDRON_MAVEN_EXTRA_ARGS site site:stage ${MAVEN_SKIP_TESTS} $@
 # Copy generated docs
         mkdir -p build/hadoop-${FULL_VERSION}/share/doc/
         cp -r target/staging/hadoop-project/* build/hadoop-${FULL_VERSION}/share/doc/
 
 # Rebuild and now mvn deploy the tarball - note the '-pl hadoop-dist' - if we chdir into hadoop-dist and build, it fails.
 
-        JAVA_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" MAVEN_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" mvn $CAULDRON_MAVEN_EXTRA_ARGS -pl hadoop-dist -Pdist -Psrc -Dtar ${MAVEN_SKIP_TESTS} -Dmaven.javadoc.skip=true -Dmaven.deploy.skip=false $MAVEN_INST_DEPLOY "$@"
+        JAVA_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" MAVEN_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" mvn -Dmaven.wagon.http.ssl.allowall=true -Dmaven.wagon.http.ssl.insecure=true -B $CAULDRON_MAVEN_EXTRA_ARGS -pl hadoop-dist -Pdist -Psrc -Dtar ${MAVEN_SKIP_TESTS} -Dmaven.javadoc.skip=true -Dmaven.deploy.skip=false $MAVEN_INST_DEPLOY "$@"
 
         (cd build/hadoop-${FULL_VERSION} ; tar --strip-components=1 -xzvf  ../../hadoop-dist/target/hadoop-${FULL_VERSION}.tar.gz)
 # (cd build/hadoop-${FULL_VERSION}/src ; tar --strip-components=1 -xzvf  ../../../hadoop-dist/target/hadoop-${FULL_VERSION}.tar.gz)
@@ -127,7 +129,8 @@ if [ -z "$GIT_REPO" ]; then
       IVYSETTINGS_OVERRIDE=""
     fi
 
-    IVY_MIRROR_PROP=${IVY_MIRROR_PROP:-http://repo1.maven.org/maven2/}
+    IVY_MIRROR_PROP=${IVY_MIRROR_PROP:-https://repo1.maven.org/maven2}
+
     IVY_EXTRA_ARGS="-Dmvnrepo=$IVY_MIRROR_PROP -Drepo.maven.org=$IVY_MIRROR_PROP ${IVYSETTINGS_OVERRIDE}"
 # Copy them into the main build directory to be included in the tarball
     rm -rf hadoop-mapreduce1-project/build
@@ -155,15 +158,9 @@ if [ -z "$GIT_REPO" ]; then
 
     mv hadoop-mapreduce1-project/build/hadoop-${MR1_VERSION} hadoop-mapreduce1-project/build/hadoop-mr1
 
-    if [ -n "$DO_MAVEN_DEPLOY" ]; then
-# Change to cloudera/maven directory, and install
-# (and if called from CDH nightly build, deploy) artifacts into Maven repository
-        (cd hadoop-mapreduce1-project/cloudera/maven-packaging ; mvn $CAULDRON_MAVEN_EXTRA_ARGS -Dnot.cdh.release.build=false $DO_MAVEN_DEPLOY)
-    fi
-
     if [ -z "${MAVEN_ONLY}" ]; then
 # Create a manifest for the hadoop-0.20-client
-        (cd hadoop-mapreduce1-project/cloudera/maven-packaging/hadoop-client ; mvn $CAULDRON_MAVEN_EXTRA_ARGS assembly:single)
+        (cd hadoop-mapreduce1-project/cloudera/maven-packaging/hadoop-client ; mvn -Dmaven.wagon.http.ssl.allowall=true -Dmaven.wagon.http.ssl.insecure=true -B $CAULDRON_MAVEN_EXTRA_ARGS assembly:single)
         (cd hadoop-mapreduce1-project/cloudera/maven-packaging/hadoop-client/target/hadoop-client*/client ; ls) > hadoop-mapreduce1-project/build/hadoop-mr1/hadoop-client.list
 
 # Move MR1 bits into the final destination
