diff --git a/do-component-build b/do-component-build
index 21d58bf..32e9d61 100755
--- a/do-component-build
+++ b/do-component-build
@@ -16,7 +16,8 @@
 
 set -ex
 
-export HADOOP_PROTOC_CDH4_PATH=/opt/toolchain/protobuf-2.4.1/bin/protoc
+export HADOOP_PROTOC_CDH4_PATH=/usr/bin/protoc
+export PATH=$PATH:/usr/share/apache-maven-3.0.4/bin
 
 # FIXME: this needs to go away once we start using an actuall source tarball
 [ -f pom.xml ] || cd src
@@ -36,22 +37,22 @@ fi
 MAVEN_SKIP_TESTS="-DskipTests -DskipTest -DskipITs"
 JAVA_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" \
   MAVEN_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" \
-  mvn $BUNDLE_SNAPPY \
-    -Pdist -Pnative -Pfuse -Psrc -Dtar -Drequire.fuse=true \
+  mvn -B $BUNDLE_SNAPPY \
+    -e -Pdist -Pnative -Pfuse -Psrc -Dtar -Drequire.fuse=true \
     ${MAVEN_SKIP_TESTS} \
     install \
     "$@" \
     $DO_MAVEN_DEPLOY
 
 if [ -z "${MAVEN_ONLY}" ]; then
-    JAVA_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" MAVEN_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" mvn site site:stage ${MAVEN_SKIP_TESTS} $@
+    JAVA_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" MAVEN_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" mvn -B site site:stage ${MAVEN_SKIP_TESTS} $@
 
 # Copy generated docs
     mkdir -p build/hadoop-${FULL_VERSION}/share/doc/
     cp -r target/staging/hadoop-project/* build/hadoop-${FULL_VERSION}/share/doc/
     
 # Rebuild and now mvn deploy the tarball - note the '-pl hadoop-dist' - if we chdir into hadoop-dist and build, it fails.
-    JAVA_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" MAVEN_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" mvn -pl hadoop-dist -Pdist -Psrc -Dtar ${MAVEN_SKIP_TESTS} -Dmaven.javadoc.skip=true -Dmaven.deploy.skip=false install "$@" $DO_MAVEN_DEPLOY
+    JAVA_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" MAVEN_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" mvn -B -pl hadoop-dist -Pdist -Psrc -Dtar ${MAVEN_SKIP_TESTS} -Dmaven.javadoc.skip=true -Dmaven.deploy.skip=false install "$@" $DO_MAVEN_DEPLOY
 
     (cd build/hadoop-${FULL_VERSION} ; tar --strip-components=1 -xzvf  ../../hadoop-dist/target/hadoop-${FULL_VERSION}.tar.gz)
 # (cd build/hadoop-${FULL_VERSION}/src ; tar --strip-components=1 -xzvf  ../../../hadoop-dist/target/hadoop-${FULL_VERSION}-src.tar.gz)
@@ -59,7 +60,7 @@ if [ -z "${MAVEN_ONLY}" ]; then
 fi
 
 # Build MR1
-IVY_MIRROR_PROP=${IVY_MIRROR_PROP:-http://repo1.maven.org/maven2/}
+IVY_MIRROR_PROP=${IVY_MIRROR_PROP:-http://repo1.maven.org/maven2}
 IVY_EXTRA_ARGS="-Dmvnrepo=$IVY_MIRROR_PROP -Drepo.maven.org=$IVY_MIRROR_PROP"
 # Copy them into the main build directory to be included in the tarball
 rm -rf hadoop-mapreduce1-project/build
@@ -87,14 +88,8 @@ mkdir -p hadoop-mapreduce1-project/build/hadoop-${MR1_VERSION}/docs/
 
 mv hadoop-mapreduce1-project/build/hadoop-${MR1_VERSION} hadoop-mapreduce1-project/build/hadoop-mr1
 
-if [ -n "$DO_MAVEN_DEPLOY" ]; then
-# Change to cloudera/maven directory, and install
-# (and if called from CDH nightly build, deploy) artifacts into Maven repository
-(cd hadoop-mapreduce1-project/cloudera/maven-packaging ; mvn -Dnot.cdh.release.build=false install $DO_MAVEN_DEPLOY)
-fi
-
 # Create a manifest for the hadoop-0.20-client
-(cd hadoop-mapreduce1-project/cloudera/maven-packaging/hadoop-client ; mvn assembly:single)
+(cd hadoop-mapreduce1-project/cloudera/maven-packaging/hadoop-client ; mvn -B assembly:single)
 (cd hadoop-mapreduce1-project/cloudera/maven-packaging/hadoop-client/target/hadoop-client*/client ; ls) > hadoop-mapreduce1-project/build/hadoop-mr1/hadoop-client.list
 
 if [ -z "${MAVEN_ONLY}" ]; then
@@ -165,6 +160,6 @@ cat > hadoop-mr1-client/pom.xml <<__EOT__
   </dependencies>
 </project>
 __EOT__
-mvn -f hadoop-mr1-client/pom.xml dependency:copy-dependencies -DincludeScope=runtime
+mvn -B -f hadoop-mr1-client/pom.xml dependency:copy-dependencies -DincludeScope=runtime
 (cd hadoop-mr1-client/target/dependency ; ls | grep -v hadoop-client) > build/hadoop-${FULL_VERSION}/hadoop-mr1-client.list
 rm -rf hadoop-mr1-client
