diff -u -r 1/install_hadoop.sh 2/install_hadoop.sh
--- 1/install_hadoop.sh
+++ 2/install_hadoop.sh
@@ -160,7 +160,7 @@
   fi
 done
 
-. ${DISTRO_DIR}/packaging_functions.sh
+#. ${DISTRO_DIR}/packaging_functions.sh
 
 HADOOP_DIR=${HADOOP_DIR:-$PREFIX/usr/lib/hadoop}
 HDFS_DIR=${HDFS_DIR:-$PREFIX/usr/lib/hadoop-hdfs}
@@ -603,6 +603,9 @@
   echo "cloudera.pkg.name=${map%%_*}" >> $dir/cdh_version.properties
 done
 
+internal_versionless_symlinks () { :; }
+external_versionless_symlinks () { :; }
+
 internal_versionless_symlinks \
     ${HADOOP_DIR}/hadoop-*.jar \
     ${HDFS_DIR}/hadoop-*.jar \
