--- _build_results/install_hadoop.sh.orig	2012-11-26 13:39:10.847508512 -0600
+++ _build_results/install_hadoop.sh	2012-11-26 13:39:48.021509403 -0600
@@ -258,7 +258,7 @@
 # native libs
 install -d -m 0755 ${SYSTEM_LIB_DIR}
 install -d -m 0755 ${HADOOP_NATIVE_LIB_DIR}
-for library in libhdfs.so.0.0.0; do
+for library in libhdfs.so.0; do
   cp ${BUILD_DIR}/lib/native/${library} ${SYSTEM_LIB_DIR}/
   ldconfig -vlN ${SYSTEM_LIB_DIR}/${library}
   ln -s ${library} ${SYSTEM_LIB_DIR}/${library/.so.*/}.so
