diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/CMakeLists.txt b/hadoop-hdfs-project/hadoop-hdfs/src/CMakeLists.txt
index 5518d39..26663cc 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/CMakeLists.txt
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/CMakeLists.txt
@@ -142,7 +142,7 @@ target_link_dual_libraries(hdfs
 )
 
 dual_output_directory(hdfs ${OUT_DIR})
-set(LIBHDFS_VERSION "0.0.0")
+set(LIBHDFS_VERSION "0")
 set_target_properties(hdfs PROPERTIES
     SOVERSION ${LIBHDFS_VERSION})
 
