--- src/hadoop-hdfs-project/hadoop-hdfs/src/CMakeLists.txt.orig	2012-11-26 11:46:45.367642095 -0600
+++ src/hadoop-hdfs-project/hadoop-hdfs/src/CMakeLists.txt	2012-11-26 11:46:50.605501896 -0600
@@ -92,7 +92,7 @@
     ${JAVA_JVM_LIBRARY}
 )
 dual_output_directory(hdfs target/usr/local/lib)
-set(LIBHDFS_VERSION "0.0.0")
+set(LIBHDFS_VERSION "0")
 set_target_properties(hdfs PROPERTIES
     SOVERSION ${LIBHDFS_VERSION})
 
