diff -ru orig/do-component-build new/do-component-build
--- orig/do-component-build	2012-10-18 15:36:54.000000000 -0500
+++ new/do-component-build	2012-10-18 15:36:31.000000000 -0500
@@ -33,9 +33,9 @@
 MAVEN_SKIP_TESTS="-DskipTests -DskipTest -DskipITs"
 JAVA_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" \
   MAVEN_OPTS="-Xmx1536m -Xms256m -XX:MaxPermSize=256m" \
   mvn $BUNDLE_SNAPPY \
-    -Pdist -Pnative -Pfuse -Psrc -Dtar -Drequire.fuse=true \
+    -e -Pdist -Pnative -Pfuse -Psrc -Dtar -Drequire.fuse=true \
     ${MAVEN_SKIP_TESTS} \
     install \
     "$@" \
     $DO_MAVEN_DEPLOY
