Index: bootstrap
===================================================================
RCS file: /cvs/cd_read_only/privilege/glite-security-lcmaps-plugins-tracking/bootstrap,v
retrieving revision 1.1
diff -u -r1.1 bootstrap
--- bootstrap	7 Mar 2011 22:14:41 -0000	1.1
+++ bootstrap	15 Jul 2011 12:59:21 -0000
@@ -52,7 +52,7 @@
 
 
 mkdir -p src/autogen
-aclocal -I project -I /opt/glite-build-common-cpp/m4
+aclocal -I project -I /usr/share/glite-build-common-cpp/m4
 $LIBTOOLIZE --force
 autoheader
 automake --foreign --add-missing --copy
Index: src/tracking/Makefile.am
===================================================================
RCS file: /cvs/cd_read_only/privilege/glite-security-lcmaps-plugins-tracking/src/tracking/Makefile.am,v
retrieving revision 1.1
diff -u -r1.1 Makefile.am
--- src/tracking/Makefile.am	7 Mar 2011 22:14:41 -0000	1.1
+++ src/tracking/Makefile.am	15 Jul 2011 12:59:21 -0000
@@ -11,7 +11,7 @@
 
 
 AM_CPPFLAGS = \
-	$(GLITE_CFLAGS)/glite/security/lcmaps \
+	$(GLITE_CFLAGS)/lcmaps \
 	$(ADD_THESE_DEFINES)
 

