Index: glexec_monitor
===================================================================
RCS file: /cvs/cd_read_only/privilege/glexec-osg/contrib/glexec_monitor/glexec_monitor,v
retrieving revision 1.20
diff -u -r1.20 glexec_monitor
--- glexec_monitor	13 Apr 2011 18:30:47 -0000	1.20
+++ glexec_monitor	15 Jul 2011 20:21:41 -0000
@@ -464,10 +464,10 @@
     sys.argv.pop(1)
   else:
     # load config file
-    config_data=load_config("/etc/glexec/contrib/glexec_monitor/glexec_monitor.cfg")
+    config_data=load_config("/etc/glexec_monitor/glexec_monitor.cfg")
     if 'log_destination' not in config_data:
       # if it is in regular glexec config, take it's value from there
-      glexec_config_data=load_config("/etc/glexec/glexec.conf")
+      glexec_config_data=load_config("/etc/glexec.conf")
       if 'log_destination' in glexec_config_data:
 	config_data['log_destination']=glexec_config_data['log_destination']
 
