diff --git a/src/scitokens_internal.cpp b/src/scitokens_internal.cpp
index 4715d96..d6ee42b 100644
--- a/src/scitokens_internal.cpp
+++ b/src/scitokens_internal.cpp
@@ -598,7 +598,8 @@ scitokens::Enforcer::scope_validator(const jwt::claim &claim, void *myself) {
                 authz = "write";
                 alt_authz = "modify";
             } else if (authz == "compute.read") {
-                authz = "condor:/READ";
+                authz = "condor";
+                path = "READ";
             } else if (authz == "compute.modify") {
                 compat_modify = true;
             } else if (authz == "compute.create") {
