Index: axis2-bin-1.6.2/bin/axis2server.sh
===================================================================
--- axis2-bin-1.6.2.orig/bin/axis2server.sh	2012-04-17 17:27:28.000000003 -0500
+++ axis2-bin-1.6.2/bin/axis2server.sh	2012-07-12 17:01:34.000000003 -0500
@@ -29,8 +29,7 @@
 #
 # -----------------------------------------------------------------------------
 
-# Get the context and from that find the location of setenv.sh
-. `dirname $0`/setenv.sh
+. /usr/share/axis2/setenv.sh
 
 JAVA_OPTS=""
 while [ $# -ge 1 ]; do
@@ -44,7 +43,7 @@
             shift
         ;;
         -h)
-            echo "Usage: axis2server.sh"
+            echo "Usage: axis2-server"
             echo "commands:"
             echo "  -xdebug    Start Axis2 Server under JPDA debugger"
             echo "  -security  Enable Java 2 security"
@@ -54,7 +53,7 @@
         ;;
         *)
             echo "Error: unknown command:$1"
-            echo "For help: axis2server.sh -h"
+            echo "For help: axis2-server -h"
             shift
             exit 1
     esac
Index: axis2-bin-1.6.2/bin/axis2.sh
===================================================================
--- axis2-bin-1.6.2.orig/bin/axis2.sh	2012-04-17 17:27:28.000000003 -0500
+++ axis2-bin-1.6.2/bin/axis2.sh	2012-07-12 17:00:48.000000003 -0500
@@ -31,8 +31,7 @@
 # -----------------------------------------------------------------------------
 
 # Get the context and from that find the location of setenv.sh
-dir=`dirname "$0"`
-. "${dir}"/setenv.sh
+. /usr/share/axis2/setenv.sh
 
 #add any user given classpath's
 USER_COMMANDS=""
Index: axis2-bin-1.6.2/bin/java2wsdl.sh
===================================================================
--- axis2-bin-1.6.2.orig/bin/java2wsdl.sh	2012-04-17 17:27:28.000000003 -0500
+++ axis2-bin-1.6.2/bin/java2wsdl.sh	2012-07-12 17:02:15.000000003 -0500
@@ -17,5 +17,4 @@
 # specific language governing permissions and limitations
 # under the License.
 
-dir=`dirname "$0"`
-sh "$dir"/axis2.sh org.apache.ws.java2wsdl.Java2WSDL $*
+sh /usr/bin/axis2 org.apache.ws.java2wsdl.Java2WSDL $*
Index: axis2-bin-1.6.2/bin/wsdl2java.sh
===================================================================
--- axis2-bin-1.6.2.orig/bin/wsdl2java.sh	2012-04-17 17:27:28.000000003 -0500
+++ axis2-bin-1.6.2/bin/wsdl2java.sh	2012-07-12 17:02:29.000000003 -0500
@@ -17,5 +17,4 @@
 # specific language governing permissions and limitations
 # under the License.
 
-dir=`dirname "$0"`
-sh "$dir"/axis2.sh org.apache.axis2.wsdl.WSDL2Java $*
+sh /usr/bin/axis2 org.apache.axis2.wsdl.WSDL2Java $*
