From a5d6cd54ebc28a83084e4fec4262c6d565a2d91c Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Wed, 28 Dec 2022 19:38:23 -0600
Subject: [PATCH] SOFTWARE-5425: add gpg_digest_algo

---
 sign.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sign.py b/sign.py
index d34defc..cc66823 100644
--- a/sign.py
+++ b/sign.py
@@ -36,6 +36,10 @@ def sign(cbtype, *args, **kws):
     gpg_name = config.get(tag_name, 'gpg_name')
     gpg_pass = config.get(tag_name, 'gpg_pass')
     try:
+        gpg_digest_algo = config.get(tag_name, 'gpg_digest_algo')
+    except NoOptionError:
+        gpg_digest_algo = None
+    try:
         enabled = config.getboolean(tag_name, 'enabled')
     except NoOptionError:
         # Note that signing is _enabled_ by default
@@ -62,6 +66,8 @@ def sign(cbtype, *args, **kws):
     rpm_cmd = "%s --resign --define '_signature gpg'" % rpm
     rpm_cmd += " --define '_gpgbin %s'" % gpgbin
     rpm_cmd += " --define '_gpg_path %s'" % gpg_path
+    if gpg_digest_algo:
+        rpm_cmd += " --define '_gpg_digest_algo %s'" % gpg_digest_algo
     rpm_cmd += " --define '_gpg_name %s' %s" % (gpg_name, rpms)
     pex = pexpect.spawn(rpm_cmd, timeout=1000)
     # Add rpm output to a temporary file
-- 
2.6.3

