--- init/globus-gatekeeper-lsb.in.orig	2011-11-14 12:18:33.000000000 -0600
+++ init/globus-gatekeeper-lsb.in	2011-11-14 12:21:56.000000000 -0600
@@ -4,8 +4,7 @@
 # Provides:          globus-gatekeeper
 # Required-Start:    $network $time
 # Required-Stop:     $network
-# Default-Start:     2 3 4 5
-# Default-Stop:      0 1 6
+# Default-Stop:      0 1 2 3 4 5 6
 # Short-Description: Globus Gatekeeper
 # Description:       The Globus Gatekeeper service authenticates network
 #                    connections using an SSL-based protocol and then
