Index: init/globus-gatekeeper-lsb.in
===================================================================
--- init/globus-gatekeeper-lsb.in.orig	2015-02-10 18:27:45.971398866 -0600
+++ init/globus-gatekeeper-lsb.in	2015-02-10 18:28:00.472580892 -0600
@@ -95,9 +95,14 @@
         kflag=""
     fi
 
+    unset openssl_is_1_0_1_or_higher
+    if openssl version | grep -q "OpenSSL [1-9]\.[0-9]\.[1-9]"; then
+        openssl_is_1_0_1_or_higher=1
+    fi
+
     start_daemon ${lsb_ok:+-p "${GLOBUS_GATEKEEPER_PIDFILE}"} \
         ${GLOBUS_GATEKEEPER_NICE_LEVEL:+-n "${GLOBUS_GATEKEEPER_NICE_LEVEL}"} \
-        "$prog" \
+        "$prog" ${openssl_is_1_0_1_or_higher:+-launch_method fork_and_proxy} \
         -pidfile ${GLOBUS_GATEKEEPER_PIDFILE} \
         ${GLOBUS_GATEKEEPER_LOG_FACILITY:+-lf "$GLOBUS_GATEKEEPER_LOG_FACILITY"} \
         ${GLOBUS_GATEKEEPER_PORT:+-p ${GLOBUS_GATEKEEPER_PORT}} \
