--- globus_gatekeeper.c.orig	2011-11-10 18:08:59.000000000 -0600
+++ globus_gatekeeper.c	2011-11-10 18:09:06.000000000 -0600
@@ -1003,7 +1003,7 @@
         {
             logging_phase2(); /* now set stderr to logfile after gss prompts */
 
-            net_setup_listener(2, &daemon_port, &listener_fd);
+            net_setup_listener(500, &daemon_port, &listener_fd);
         }
       
         /* ajr,vs --changed printf to sprintf, and added setenv
