Index: init/globus-gatekeeper-lsb.in
===================================================================
--- init/globus-gatekeeper-lsb.in.orig	2014-11-04 07:48:09.270503378 -0600
+++ init/globus-gatekeeper-lsb.in	2015-02-10 18:22:09.319169131 -0600
@@ -52,6 +52,7 @@
     exit 1
 fi
 
+. /usr/share/osg/sysconfig/globus-gatekeeper
 GLOBUS_GATEKEEPER_CONFIG=${GLOBUS_GATEKEEPER_CONFIG:-@GATEKEEPER_CONFIGDIR@/@GATEKEEPER_CONFIGFILE@}
 test -f "${GLOBUS_GATEKEEPER_CONFIG}" && . "${GLOBUS_GATEKEEPER_CONFIG}"
 prog="${sbindir}/globus-gatekeeper"
