Index: gatekeeper.logrotate.in
===================================================================
--- gatekeeper.logrotate.in.orig	2011-11-28 14:19:02.000000002 -0600
+++ gatekeeper.logrotate.in	2013-09-11 16:56:07.000000002 -0500
@@ -6,7 +6,5 @@
     nocreate
     rotate 4
     sharedscripts
-    postrotate
-        /usr/bin/killall -USR1 globus-gatekeeper 2> /dev/null || true
-    endscript
+    copytruncate
 }
