diff --git a/gums-service/src/main/java/gov/bnl/gums/service/GUMSXACMLMappingServiceImpl.java b/gums-service/src/main/java/gov/bnl/gums/service/GUMSXACMLMappingServiceImpl.java
index aef1057..52ed123 100644
--- a/gums-service/src/main/java/gov/bnl/gums/service/GUMSXACMLMappingServiceImpl.java
+++ b/gums-service/src/main/java/gov/bnl/gums/service/GUMSXACMLMappingServiceImpl.java
@@ -167,7 +167,7 @@ public class GUMSXACMLMappingServiceImpl implements XACMLMappingService {
 				obligationgid.getAttributeAssignments().add(attributeAssignmentGid);
 				obligations.getObligations().add(obligationgid);
 			}
-			if (supportsAccount)
+			if (supportsAccount && hasGroup)
 			{
 				ObligationType obligationAccount = obligationBuilder.buildObject();
 				obligationAccount.setFulfillOn(EffectType.Permit);
