Index: gums-1.4.0/gums-core/src/main/scripts/gums-create-config
===================================================================
--- gums-1.4.0.orig/gums-core/src/main/scripts/gums-create-config	2009-08-20 13:02:55.000000003 -0500
+++ gums-1.4.0/gums-core/src/main/scripts/gums-create-config	2012-01-06 15:48:33.000000003 -0600
@@ -1,35 +1,5 @@
 #!/bin/sh
 
-# Detects where gums is installed and puts the directory
-# name in ScriptPath. Then cd to that directory.
-
-ScriptPath=""
-CommandPath=`dirname $0`
-if [ $CommandPath = "." ]
-then
-
-    # The command path is the current directory:
-    # check if file is really here
-
-    if [ ! -e "$0" ]
-    then
-        # The command is not in the current directory:
-        # must have used the path.
-        # Locating with which.
-
-        WhichFull=`which $0`
-        WhichPath=`dirname $WhichFull`
-        ScriptPath=$WhichPath
-    else
-        # It was really the current directory
-        ScriptPath=$CommandPath
-    fi
-else
-    ScriptPath=$CommandPath
-fi
-
-cd $ScriptPath
-
 usage() {
 	echo "Copies a template configuration to the config directory, thus making "
 	echo "it the current configuration, and will also replace user, host, password "
@@ -38,7 +8,7 @@
 	echo "will be made to extract one or more of them from the current configuration."
 	echo "Usage: $0"
 	echo "   --template your_template (relative to script location)"
-	echo "        (default - ../config/gums.config.template) [optional]> "
+	echo "        (default - /usr/lib/gums/config/gums.config.template) [optional]> "
 	echo "   --user <database user for GUMS> [optional]"
 	echo "   --password <GUMS database password> [optional]"
 	echo "   --host <GUMS database server host> [optional]"
@@ -46,8 +16,8 @@
 	echo "   --help [Print this message]"
 	echo
 	echo "Examples:"
-	echo "  $0 --template ../mytemplate"
-	echo "  $0 --user gums --host gums.mysite.com:49151 --password secret --template ../mytemplate"
+	echo "  $0 --template mytemplate"
+	echo "  $0 --user gums --host gums.mysite.com:49151 --password secret --template mytemplate"
 	echo
 
 	exit -1;
@@ -58,7 +28,7 @@
 # Setup global variables
 #
 #-------------------------------------------
-TEMPLATE="../gums.config.template"
+TEMPLATE="/usr/lib/gums/gums.config.template"
 USER=""
 SERVER=""
 PASSWORD=""
@@ -89,10 +59,10 @@
 done
 
 parseConfig() {
-	if [ -e ../config/gums.config ]
+	if [ -e /etc/gums/gums.config ]
 	then
 		unset done
-        exec 3< ../config/gums.config
+        exec 3< /etc/gums/gums.config
         until [ $done ]
         do
 			read <&3 myline
@@ -188,7 +158,7 @@
   exit 1
 fi
 
-if [ -e ../config/gums.config ]
+if [ -e /etc/gums/gums.config ]
 then
 	if [ "$PROMPT" -eq 1 ]
 	then
@@ -203,7 +173,7 @@
 	fi
 
   echo "Backing up your gums.config as gums.config.old"
-  cp ../config/gums.config ../config/gums.config.old
+  cp /etc/gums/gums.config /etc/gums/gums.config.old
 fi
 
 cat $TEMPLATE \
@@ -212,8 +182,8 @@
              -e s/@PASSWORD@/$PASSWORD/g            \
              -e s/@DOMAINNAME@/$DOMAINNAME/g        \
              -e s/GUMS_[0-9]_[0-9]/GUMS_1_3/g		\
-  > ../config/gums.config
+  > /etc/gums/gums.config
 
 echo "New gums.config has been created successfully"
-ls -l ../config/gums.config
-exit 0
\ No newline at end of file
+ls -l /etc/gums/gums.config
+exit 0
