Index: gums-core/src/main/scripts/gums-add-mysql-admin
===================================================================
--- gums-core/src/main/scripts/gums-add-mysql-admin.orig	2008-01-07 09:58:16.000000003 -0600
+++ gums-core/src/main/scripts/gums-add-mysql-admin	2012-01-06 15:33:22.000000003 -0600
@@ -1,36 +1,5 @@
 #!/bin/sh
 
-# Detects where gums is installed and puts the directory
-# name in ScriptPath. Then cd to that directory.
-
-ScriptPath=""
-CommandPath=`dirname $0`
-if [ $CommandPath = "." ]
-then
-
-    # The command path is the current directory:
-    # check if file is really here
-
-    if [ ! -e "$0" ]
-    then
-        # The command is not in the current directory:
-        # must have used the path.
-        # Locating with which.
-
-        WhichFull=`which $0`
-        WhichPath=`dirname $WhichFull`
-        ScriptPath=$WhichPath
-    else
-        # It was really the current directory
-        ScriptPath=$CommandPath
-    fi
-else
-    ScriptPath=$CommandPath
-fi
-
-cd $ScriptPath
-
-
 if [ "$1" == "" ]
 then
   echo Adds an admin in the GUMS database on localhost
@@ -62,6 +31,6 @@
 echo
 echo Adding the admin:
 echo Enter the root mysql password
-cat ../sql/addAdmin.mysql | /bin/sed -e "s%@ADMINDN@%$ADMINDN%g" | mysql $MYSQLPARAM
+cat /usr/lib/gums/sql/addAdmin.mysql | /bin/sed -e "s%@ADMINDN@%$ADMINDN%g" | mysql $MYSQLPARAM
 echo
 
