Index: gums-core/src/main/scripts/gums-add-mysql-admin
===================================================================
--- gums-core/src/main/scripts/gums-add-mysql-admin.orig	2010-07-27 08:53:26.000000003 -0500
+++ gums-core/src/main/scripts/gums-add-mysql-admin	2012-02-17 14:06:03.000000003 -0600
@@ -1,36 +1,5 @@
 #!/bin/sh
 
-# Detects where gums is installed and puts the directory
-# name in ScriptPath. Then cd to that directory.
-
-ScriptPath=""
-CommandPath=`dirname $0`
-if [ $CommandPath = "." ]
-then
-
-    # The command path is the current directory:
-    # check if file is really here
-
-    if [ ! -e "$0" ]
-    then
-        # The command is not in the current directory:
-        # must have used the path.
-        # Locating with which.
-
-        WhichFull=`which $0`
-        WhichPath=`dirname $WhichFull`
-        ScriptPath=$WhichPath
-    else
-        # It was really the current directory
-        ScriptPath=$CommandPath
-    fi
-else
-    ScriptPath=$CommandPath
-fi
-
-cd $ScriptPath
-
-
 if [ "$1" == "" ]
 then
   echo Adds an admin in the GUMS database on localhost
@@ -62,7 +31,7 @@
 echo
 echo Adding the admin:
 echo Enter the gums mysql password
-echo ../sql/addAdmin.mysql pipe sed -e "s%@ADMINDN@%$ADMINDN%g" pipe mysql $MYSQLPARAM
-cat ../sql/addAdmin.mysql | sed -e "s%@ADMINDN@%$ADMINDN%g" | mysql $MYSQLPARAM
+echo "/usr/lib/gums/sql/addAdmin.mysql | sed -e "s%@ADMINDN@%$ADMINDN%g" | mysql $MYSQLPARAM"
+cat /usr/lib/gums/sql/addAdmin.mysql | sed -e "s%@ADMINDN@%$ADMINDN%g" | mysql $MYSQLPARAM
 echo
 
