Index: gums-core/src/main/scripts/gums-create-config
===================================================================
--- gums-core/src/main/scripts/gums-create-config.orig	2012-04-21 20:51:03.000000000 -0500
+++ gums-core/src/main/scripts/gums-create-config	2012-04-23 15:18:43.680837413 -0500
@@ -44,7 +44,7 @@
 # Setup global variables
 #
 #-------------------------------------------
-TEMPLATE="/usr/lib/gums/gums.config.template"
+TEMPLATE="/usr/lib/gums/config/gums.config.template"
 OSGTEMPLATE=0
 USER=""
 SERVER=""
Index: gums-service/src/main/resources/scripts/gums-create-config
===================================================================
--- gums-service/src/main/resources/scripts/gums-create-config.orig	2012-04-21 20:51:03.000000000 -0500
+++ gums-service/src/main/resources/scripts/gums-create-config	2012-04-23 15:19:21.330835386 -0500
@@ -40,7 +40,7 @@
 # Setup global variables
 #
 #-------------------------------------------
-TEMPLATE="/usr/lib/gums/gums.config.template"
+TEMPLATE="/usr/lib/gums/config/gums.config.template"
 OSGTEMPLATE=0
 USER=""
 SERVER=""
