--- _build_results/BUILD/globus_gram_job_manager_setup_condor-4.4/condor.in	2011-08-08 07:39:23.000000000 -0500
+++ condor.in	2011-08-08 07:39:01.000000000 -0500
@@ -28,6 +28,9 @@
 
 use Globus::Core::Paths;
 
+eval { require Globus::GRAM::JobManager::condor_accounting_groups };
+our $accounting_groups_callout = ! $@; # Don't make callout if we couldn't load module
+
 # NOTE: This package name must match the name of the .pm file!!
 package Globus::GRAM::JobManager::condor;
 
@@ -319,6 +322,11 @@
 	$submit_attrs_string = '';
     }
 
+    my $group;
+    if ($accounting_groups_callout) {
+       $group = Globus::GRAM::JobManager::condor_accounting_groups::obtain_condor_group(\@environment, $self);
+    }
+
     # Create script for condor submission
     $script_filename = $self->job_dir() . '/scheduler_condor_submit_script';
 
@@ -377,6 +385,18 @@
             "print: $script_filename: $!",
             Globus::GRAM::Error::TEMP_SCRIPT_FILE_FAILED());
     }
+
+    if ($accounting_groups_callout && $group) {
+        $name = getpwuid($>);
+        print SCRIPT_FILE "+AccountingGroup = \"$group.$name\"\n" if $group;
+        if (!$rc)
+        {
+            return $self->respond_with_failure_extension(
+                "print: $script_filename: $!",
+                Globus::GRAM::Error::TEMP_SCRIPT_FILE_FAILED());
+        }
+    }
+
     if($rank ne '')
     {
 	$rc = print SCRIPT_FILE "$rank\n";
