--- condor-7.9.6/externals/bundles/glibc/CMakeLists.txt.orig	2013-05-08 16:14:37.000000000 -0500
+++ condor-7.9.6/externals/bundles/glibc/CMakeLists.txt	2013-07-15 18:56:32.337671433 -0500
@@ -47,7 +47,7 @@
 
 		set (GLIBC_CONFIGURE ./configure --enable-static-nss --enable-add-ons=linuxthreads,c_stubs,glibc-compat --disable-shared --without-cvs)
 
-		set (GLIBC_MAKE make )
+		set (GLIBC_MAKE $(MAKE) )
 
 	elseif(${GLIBC_VERSION} STREQUAL "25")
 
@@ -58,7 +58,7 @@
 
 			set (GLIBC_CONFIGURE ./configure --enable-add-ons=nptl,rtkaio,c_stubs,libidn --without-cvs --enable-kernel=2.6.9 --with-headers=$HOME/junk/glibc-2.5-20061008T1257/override_headers:/usr/include --enable-bind-now --with-tls --with-__thread --build x86_64-redhat-linux --host x86_64-redhat-linux --disable-profile --enable-static-nss)
 
-			set (GLIBC_MAKE make -j2 -r 'CFLAGS=-mtune=generic -DNDEBUG=1 -g -O3 ' PARALLELMFLAGS=-s)
+			set (GLIBC_MAKE $(MAKE) -r 'CFLAGS=-mtune=generic -DNDEBUG=1 -g -O3 ' PARALLELMFLAGS=-s)
 
 		else()
 			set (VER 2.5-20061008T1257-p0)
@@ -66,7 +66,7 @@
 
 			set (GLIBC_CONFIGURE ./configure --enable-add-ons=nptl,rtkaio,c_stubs,libidn --without-cvs --enable-kernel=2.6.9 --with-headers=$HOME/junk/glibc-2.5-20061008T1257/override_headers:/usr/include --enable-bind-now --with-tls --with-__thread --build i486-redhat-linux --host i486-redhat-linux --disable-profile --enable-static-nss)
 
-			set (GLIBC_MAKE make -j2 -r 'CFLAGS=-march=i386 -mtune=generic -mno-tls-direct-seg-refs -DNDEBUG=1 -g -O3 ' PARALLELMFLAGS=-s )
+			set (GLIBC_MAKE $(MAKE) -r 'CFLAGS=-march=i386 -mtune=generic -mno-tls-direct-seg-refs -DNDEBUG=1 -g -O3 ' PARALLELMFLAGS=-s )
 		endif()
 
 		set (GLIBC_PATCH patch -N --binary -p1 <
@@ -85,7 +85,7 @@
 
             set (GLIBC_CONFIGURE ./configure --host=x86_64-linux-gnu --build=x86_64-linux-gnu --without-cvs --enable-add-ons=libidn,nptl --without-selinux --enable-kernel=2.6.8 --disable-profile --with-headers=${GLIBC_BUILD_PREFIX}/glibc-${VER}/debian/include --enable-bind-now --with-tls --with-__thread --enable-static-nss)
 
-			set (GLIBC_MAKE make -j2 -r)
+			set (GLIBC_MAKE $(MAKE) -r)
 		else()
 			set (VER 2.7-18)
 			set (GLIBC_TARNAME glibc-2.7-18)
@@ -94,7 +94,7 @@
 
 			set (GLIBC_CONFIGURE ./configure --host=i686-linux --build=i486-linux-gnu --without-cvs --enable-add-ons=libidn,nptl --without-selinux --enable-kernel=2.6.8 --disable-profile --with-headers=${GLIBC_BUILD_PREFIX}/glibc-${VER}/debian/include --enable-bind-now --with-tls --with-__thread --enable-static-nss)
 
-			set (GLIBC_MAKE make -j2 -r 'CFLAGS=-pipe -O2 -fstrict-aliasing -g -march=i686 -mtune=i686 -O3 ' PARALLELMFLAGS=-s)
+			set (GLIBC_MAKE $(MAKE) -r 'CFLAGS=-pipe -O2 -fstrict-aliasing -g -march=i686 -mtune=i686 -O3 ' PARALLELMFLAGS=-s)
 		endif()
 
 		set (GLIBC_PATCH patch -N --binary -p1 < ${CMAKE_CURRENT_SOURCE_DIR}/${VER}/enable-static-nss.patch && patch -N --binary -p1 < ${CMAKE_CURRENT_SOURCE_DIR}/${VER}/disable-nscd.patch )
@@ -109,7 +109,7 @@
 
 			set (GLIBC_CONFIGURE ./configure CC="gcc-4.4" CXX="g++-4.4" 'CFLAGS=-pipe -O2 -fstrict-aliasing -g ' --host=x86_64-linux-gnu --build=x86_64-linux-gnu --without-cvs --enable-add-ons=libidn,nptl --disable-profile --without-selinux --with-pkgversion="Debian EGLIBC 2.11.2-10" --with-bugurl="http://www.debian.org/Bugs/" --enable-kernel=2.6.18 --enable-multi-arch --enable-static-nss )
 
-			set (GLIBC_MAKE make -j2 -r)
+			set (GLIBC_MAKE $(MAKE) -r)
 		else()
 			set (VER 2.11.2-10)
 			set (GLIBC_TARNAME glibc-2.11.2-10)
@@ -121,7 +121,7 @@
 			set (GLIBC_CONFIGURE ./configure CC="gcc-4.4" CXX="g++-4.4" 'CFLAGS=-pipe -O2 -fstrict-aliasing -g ' --host=x86_64-linux-gnu --build=x86_64-linux-gnu --without-cvs --enable-add-ons=libidn,nptl --disable-profile --without-selinux --with-pkgversion="Debian EGLIBC 2.11.2-10" --with-bugurl="http://www.debian.org/Bugs/" --with-headers=${GLIBC_BUILD_PREFIX}/glibc-${VER}/debian/include --enable-kernel=2.6.18 --enable-multi-arch --enable-static-nss )
 
 			# XXX FIXME
-			set (GLIBC_MAKE make -j2 -r 'CFLAGS=-pipe -O2 -fstrict-aliasing -g -march=i686 -mtune=i686 -O3 ' PARALLELMFLAGS=-s)
+			set (GLIBC_MAKE $(MAKE) -r 'CFLAGS=-pipe -O2 -fstrict-aliasing -g -march=i686 -mtune=i686 -O3 ' PARALLELMFLAGS=-s)
 
 		endif()
 
@@ -138,7 +138,7 @@
 
             set (GLIBC_CONFIGURE ./configure CC=gcc CXX=g++ 'CFLAGS=-mtune=generic -fasynchronous-unwind-tables -DNDEBUG -g -O3 ' --enable-add-ons=nptl,rtkaio,c_stubs,libidn --without-cvs --enable-kernel=2.6.18 --with-headers=/usr/include --enable-bind-now --with-tls --with-__thread --build x86_64-redhat-linux --host x86_64-redhat-linux --enable-multi-arch --disable-profile --enable-experimental-malloc --enable-nss-crypt --enable-static-nss )
 
-			set (GLIBC_MAKE make -j2 -r)
+			set (GLIBC_MAKE $(MAKE) -r)
 		else()
 			set (VER 2.12-2)
 			set (GLIBC_TARNAME glibc-2.12-2)
@@ -148,7 +148,7 @@
 			# XXX FIXME
 			set (GLIBC_CONFIGURE ./configure --host=i686-linux --build=i486-linux-gnu --without-cvs --enable-add-ons=libidn,nptl --without-selinux --enable-kernel=2.6.8 --disable-profile --with-headers=${GLIBC_BUILD_PREFIX}/glibc-${VER}/debian/include --enable-bind-now --with-tls --with-__thread --enable-static-nss)
 
-			set (GLIBC_MAKE make -j2 -r 'CFLAGS=-pipe -O2 -fstrict-aliasing -g -march=i686 -mtune=i686 -O3 ' PARALLELMFLAGS=-s)
+			set (GLIBC_MAKE $(MAKE) -r 'CFLAGS=-pipe -O2 -fstrict-aliasing -g -march=i686 -mtune=i686 -O3 ' PARALLELMFLAGS=-s)
 		endif()
 
 		# The enable-static-nss patch works, and disable link warning about dynamic nss
